diff -urN sge030809/Makefile sge030809-haiku/Makefile
--- sge030809/Makefile	2003-08-11 22:40:41.015728640 +0000
+++ sge030809-haiku/Makefile	2010-05-31 17:12:42.000000000 +0000
@@ -71,13 +71,13 @@
 endif
 
 install:	shared
-	@mkdir -p $(PREFIX_H)
-	install -c -m 644 sge*.h $(PREFIX_H)
-	@mkdir -p $(PREFIX)/lib
-	install -c -m 644 libSGE.a $(PREFIX)/lib
-	install -c libSGE.so $(PREFIX)/lib/libSGE.so.$(API_VER).$(SGE_VER)
-	@cd $(PREFIX)/lib;\
+	@mkdir -p $(DESTDIR)$(PREFIX_H)
+	install -c -m 644 sge*.h $(DESTDIR)$(PREFIX_H)
+	@mkdir -p $(DESTDIR)$(PREFIX)/lib
+	install -c -m 644 libSGE.a $(DESTDIR)$(PREFIX)/lib
+	install -c libSGE.so $(DESTDIR)$(PREFIX)/lib/libSGE.so.$(API_VER).$(SGE_VER)
+	@cd $(DESTDIR)$(PREFIX)/lib;\
 	ln -sf libSGE.so.$(API_VER).$(SGE_VER) libSGE.so.$(API_VER);\
 	ln -sf libSGE.so.$(API_VER) libSGE.so
-	@echo "** Headerfiles installed in $(PREFIX_H)"
-	@echo "** Library files installed in $(PREFIX)/lib"
+	@echo "** Headerfiles installed in $(DESTDIR)$(PREFIX_H)"
+	@echo "** Library files installed in $(DESTDIR)$(PREFIX)/lib"
diff -urN sge030809/Makefile.conf sge030809-haiku/Makefile.conf
--- sge030809/Makefile.conf	2003-08-11 22:43:35.015728640 +0000
+++ sge030809-haiku/Makefile.conf	2010-05-31 17:14:25.189792256 +0000
@@ -25,14 +25,14 @@
 PREFIX_H =$(shell sdl-config --prefix)/include/SDL
 
 # Flags passed to the compiler
-CFLAGS =-Wall -O3 -ffast-math
+CFLAGS =-Wall -O0 -ffast-math
 SGE_CFLAGS =$(shell sdl-config --cflags)
 # Uncomment to make some more optimizations
 #CFLAGS =-Wall -O9 -ffast-math -march=i686
 
 
 # Libs config
-SGE_LIBS =$(shell sdl-config --libs) -lstdc++
+SGE_LIBS =$(shell sdl-config --libs)
 
 
 # Is freetype-config available?
diff -urN sge030809/sge_tt_text.cpp sge030809-haiku/sge_tt_text.cpp
--- sge030809/sge_tt_text.cpp	2003-08-08 12:48:08.016252928 +0000
+++ sge030809-haiku/sge_tt_text.cpp	2010-05-31 17:12:42.000000000 +0000
@@ -36,10 +36,9 @@
 #include "sge_textpp.h"
 
 #ifndef _SGE_NOTTF
-#include <freetype/freetype.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
-#include <freetype/internal/ftobjs.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_OUTLINE_H
 
 /* The structure used to hold glyph information (cached) */
 struct glyph {
@@ -294,7 +293,7 @@
 	}
 	memset(stream, 0, sizeof(*stream));
 
-	stream->memory = _sge_library->memory;
+	stream->memory = NULL;
 	stream->read = RWread;
 	stream->descriptor.pointer = src;
 	stream->pos = (unsigned long)position;
--- sge030809/Makefile.conf       2013-10-27 04:16:27.931659776 -0700
+++ sge030809-haiku/Makefile.conf      2013-10-27 04:45:27.596639744 -0700
@@ -3,8 +3,8 @@
 # Comment/uncomment the following line to disable/enable build options
 # (See README for more info)
 C_COMP = y
-#USE_FT = n
-#USE_IMG = n
+USE_FT = y
+USE_IMG = y
 #QUIET = y
 
 
