Index: objfw-1.1.1/src/OFFileIRIHandler.m
==================================================================
--- objfw-1.1.1/src/OFFileIRIHandler.m
+++ objfw-1.1.1/src/OFFileIRIHandler.m
@@ -1996,10 +1996,11 @@
 			  attributes: [OFDictionary dictionary]
 		     failedAttribute: @""
 			       errNo: errNo];
 	}
 # elif defined(OF_HAIKU)
+	const char *cName = [name cStringWithEncoding: encoding];
 	int fd;
 
 	if ((fd = open(cPath, O_WRONLY)) == -1) {
 		int errNo = errno;
 

