diff -urN cmake-2.8.3/Modules/FindGLUT.cmake cmake-2.8.3-haiku/Modules/FindGLUT.cmake
--- cmake-2.8.3/Modules/FindGLUT.cmake	2010-11-03 19:58:27.020971520 +0000
+++ cmake-2.8.3-haiku/Modules/FindGLUT.cmake	2011-01-29 10:22:22.032505856 +0000
@@ -46,12 +46,15 @@
       /usr/openwin/include
       /opt/graphics/OpenGL/include
       /opt/graphics/OpenGL/contrib/libglut
+      /boot/develop/headers/os/opengl
       )
   
     FIND_LIBRARY( GLUT_glut_LIBRARY glut
       /usr/openwin/lib
+      /boot/develop/lib/x86/ 
       )
-    
+  
+  IF(NOT BEOS)
     FIND_LIBRARY( GLUT_Xi_LIBRARY Xi
       /usr/openwin/lib
       )
@@ -59,6 +62,7 @@
     FIND_LIBRARY( GLUT_Xmu_LIBRARY Xmu
       /usr/openwin/lib
       )
+  ENDIF(NOT BEOS)
     
   ENDIF (APPLE)
   
@@ -69,12 +73,18 @@
   IF(GLUT_glut_LIBRARY)
     # Is -lXi and -lXmu required on all platforms that have it?
     # If not, we need some way to figure out what platform we are on.
+    IF(BEOS)
+    SET( GLUT_LIBRARIES
+      ${GLUT_glut_LIBRARY}
+      )
+    ELSE(BEOS)
     SET( GLUT_LIBRARIES
       ${GLUT_glut_LIBRARY}
       ${GLUT_Xmu_LIBRARY}
       ${GLUT_Xi_LIBRARY} 
       ${GLUT_cocoa_LIBRARY}
       )
+  	ENDIF(BEOS)
     SET( GLUT_FOUND "YES" )
     
     #The following deprecated settings are for backwards compatibility with CMake1.4
@@ -84,9 +94,16 @@
   ENDIF(GLUT_glut_LIBRARY)
 ENDIF(GLUT_INCLUDE_DIR)
 
+IF(BEOS)
+MARK_AS_ADVANCED(
+  GLUT_INCLUDE_DIR
+  GLUT_glut_LIBRARY
+  )
+ELSE(BEOS)
 MARK_AS_ADVANCED(
   GLUT_INCLUDE_DIR
   GLUT_glut_LIBRARY
   GLUT_Xmu_LIBRARY
   GLUT_Xi_LIBRARY
   )
+ENDIF(BEOS)
diff -urN cmake-2.8.3/Modules/FindLua51.cmake cmake-2.8.3-haiku/Modules/FindLua51.cmake
--- cmake-2.8.3/Modules/FindLua51.cmake	2010-11-03 19:58:27.023330816 +0000
+++ cmake-2.8.3-haiku/Modules/FindLua51.cmake	2011-01-29 10:22:22.035127296 +0000
@@ -57,13 +57,13 @@
 
 IF(LUA_LIBRARY)
   # include the math library for Unix
-  IF(UNIX AND NOT APPLE)
+  IF(UNIX AND NOT APPLE AND NOT BEOS)
     FIND_LIBRARY(LUA_MATH_LIBRARY m)
     SET( LUA_LIBRARIES "${LUA_LIBRARY};${LUA_MATH_LIBRARY}" CACHE STRING "Lua Libraries")
   # For Windows and Mac, don't need to explicitly include the math library
   ELSE(UNIX AND NOT APPLE)
     SET( LUA_LIBRARIES "${LUA_LIBRARY}" CACHE STRING "Lua Libraries")
-  ENDIF(UNIX AND NOT APPLE)
+  ENDIF(UNIX AND NOT APPLE AND NOT BEOS)
 ENDIF(LUA_LIBRARY)
 
 INCLUDE("${CMAKE_CURRENT_LIST_DIR}/FindPackageHandleStandardArgs.cmake")
diff -urN cmake-2.8.3/Modules/FindOpenGL.cmake cmake-2.8.3-haiku/Modules/FindOpenGL.cmake
--- cmake-2.8.3/Modules/FindOpenGL.cmake	2010-11-03 19:58:27.024117248 +0000
+++ cmake-2.8.3-haiku/Modules/FindOpenGL.cmake	2011-01-29 10:22:22.035913728 +0000
@@ -80,6 +80,7 @@
       /usr/share/doc/NVIDIA_GLX-1.0/include
       /usr/openwin/share/include
       /opt/graphics/OpenGL/include /usr/X11R6/include
+      /boot/develop/headers/os/opengl
     )
 
     FIND_PATH(OPENGL_xmesa_INCLUDE_DIR GL/xmesa.h
@@ -94,6 +95,7 @@
             /usr/openwin/lib
             /usr/shlib /usr/X11R6/lib
             ${HPUX_IA_OPENGL_LIB_PATH}
+            /boot/develop/lib/x86/
     )
 
     # On Unix OpenGL most certainly always requires X11.
diff -urN cmake-2.8.3/bootstrap cmake-2.8.3-haiku/bootstrap
--- cmake-2.8.3/bootstrap	2010-11-03 19:58:26.058720256 +0000
+++ cmake-2.8.3-haiku/bootstrap	2011-01-29 10:22:22.031981568 +0000
@@ -137,7 +137,9 @@
     cmake_default_prefix="c:/Program Files/CMake"
   fi
 elif ${cmake_system_haiku}; then
-  cmake_default_prefix=`/bin/finddir B_COMMON_DIRECTORY`
+  cmake_default_prefix=`finddir B_COMMON_DIRECTORY`
+  cmake_man_dir="/documentation/man"
+  cmake_doc_dir="/documentation/doc/cmake-${cmake_version}"
 else
   cmake_default_prefix="/usr/local"
 fi
