diff -urN cmake-2.8.2/bootstrap cmake-2.8.2-haiku/bootstrap
--- cmake-2.8.2/bootstrap	2010-04-06 14:45:29.040370176 +0000
+++ cmake-2.8.2-haiku/bootstrap	2010-08-31 05:39:55.518520832 +0000
@@ -132,7 +132,9 @@
     cmake_default_prefix="c:/Program Files/CMake"
   fi
 elif ${cmake_system_haiku}; then
-  cmake_default_prefix=`/bin/finddir B_COMMON_DIRECTORY`
+  cmake_default_prefix=`finddir B_COMMON_DIRECTORY`
+  cmake_man_dir="/documentation/man"
+  cmake_doc_dir="/documentation/doc/cmake-${cmake_version}"
 else
   cmake_default_prefix="/usr/local"
 fi
