diff --git ubertuber-0.9.5/Makefile ubertuber-0.9.5/Makefile
index b0463ca..7795fd0 100644
--- ubertuber-0.9.5/Makefile
+++ ubertuber-0.9.5/Makefile
@@ -4,9 +4,9 @@ TYPE= APP
 SRCS= App.cpp MainWindow.cpp Settings.cpp
 RSRCS= UberTuber.rsrc
 ifeq ($(gccver),2)
-	LIBS= /boot/develop/lib/x86/libroot.so /boot/develop/lib/x86/libbe.so /boot/develop/lib/x86/libtracker.so
+	LIBS= be tracker stdc++.r4
 else
-	LIBS= /boot/develop/lib/gcc4/libroot.so /boot/develop/lib/gcc4/libbe.so /boot/develop/lib/gcc4/libtracker.so
+	LIBS= be tracker stdc++
 endif
 LIBPATHS=
 SYSTEM_INCLUDE_PATHS= /boot/develop/headers/be /boot/develop/headers/cpp /boot/develop/headers/posix /boot/home/config/include
