diff --git a/libtiff/tiff.h b/libtiff/tiff.h
index 5c32d3a..370827d 100644
--- a/libtiff/tiff.h
+++ b/libtiff/tiff.h
@@ -65,6 +65,12 @@
  * strings              unsigned char*
  */
 
+#ifdef __HAIKU__
+
+#include <SupportDefs.h>
+
+#else
+
 typedef TIFF_INT8_T   int8;
 typedef TIFF_UINT8_T  uint8;
 
@@ -77,6 +83,8 @@ typedef TIFF_UINT32_T uint32;
 typedef TIFF_INT64_T  int64;
 typedef TIFF_UINT64_T uint64;
 
+#endif
+
 /*
  * Some types as promoted in a variable argument list
  * We use uint16_vap rather then directly using int, because this way
