From: Stefano Rivera <stefanor@debian.org>
Date: Thu, 8 Oct 2015 10:01:59 -0700
Subject: Don't even think about using distribute_setup to download things
 during build

Forwarded: not-needed
Last-Update: 2012-02-03
---
 setup.py | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/setup.py b/setup.py
index a710b30..765b7ae 100644
--- a/setup.py
+++ b/setup.py
@@ -7,11 +7,7 @@ py_vers_tag = '-%s.%s' % sys.version_info[:2]
 test_dirs = ['functional_tests', 'unit_tests', os.path.join('doc','doc_tests'), 'nose']
 
 if sys.version_info >= (3,):
-    try:
-        import setuptools
-    except ImportError:
-        from distribute_setup import use_setuptools
-        use_setuptools()
+    import setuptools
 
     extra = {'use_2to3': True,
              'test_dirs': test_dirs,
