From: Dmitry Shachnev <mitya57@debian.org>
Date: Sat, 30 Oct 2021 20:44:42 +0300
Subject: Fix syntax in doc/conf.py for Python 3

---
 doc/conf.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/conf.py b/doc/conf.py
index cccfd95..dab07bf 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -217,8 +217,8 @@ html_theme = 'default'
 # Grouping the document tree into LaTeX files. List of tuples
 # (source start file, target name, title, author, document class [howto/manual]).
 latex_documents = [
-  ('index', 'nose.tex', ur'nose Documentation',
-   ur'Jason Pellerin', 'manual'),
+  ('index', 'nose.tex', u'nose Documentation',
+   u'Jason Pellerin', 'manual'),
 ]
 
 # The name of an image file (relative to this directory) to place at the top of
