--- parser.c	2008-03-10 02:37:21.000000000 -0700
+++ parser-haiku.c	2008-08-29 07:56:04.000000000 -0700
@@ -1381,12 +1381,13 @@
 	       enum xval xv)
 {
   const char *minutes;
+  struct timespec origin;
+  struct time_val tval;
 
   if (collect_arg(argv, arg_ptr, &minutes))
     {
-      struct time_val tval;
       tval.xval = xv;
-      struct timespec origin = options.cur_day_start;
+      origin = options.cur_day_start;
       origin.tv_sec += DAYSECS;
       if (get_relative_timestamp(minutes, &tval, origin, 60,
 				 "arithmetic overflow while converting %s "
