diff --git a/CMakeLists.txt b/CMakeLists.txt
index a738a35..83fe951 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -531,7 +531,7 @@ ADD_PUBLIC_HEADER(libMultiMarkdown Sources/libMultiMarkdown/include/version.h)
 	)
 # 
 #	Link the library to the app?
-	target_link_libraries(multimarkdown libMultiMarkdown)
+	target_link_libraries(multimarkdown libMultiMarkdown bsd)
 # endif()
 
 # Xcode settings for fat binaries
