diff -urN freedroid-1.0.2/configure.ac freedroid-1.0.2-haiku/configure.ac
--- freedroid-1.0.2/configure.ac	2003-08-09 10:04:33.031981568 -0700
+++ freedroid-1.0.2-haiku/configure.ac	2012-12-12 22:59:04.623116288 -0800
@@ -20,9 +20,9 @@
 AC_PROG_MAKE_SET
 
 # Checks for libraries.
-AC_CHECK_LIB([m], [sin],, 
-	AC_MSG_ERROR([libm not found!! 
-No maths library?? What kinda crazy system is that??]))
+
+AC_CHECK_LIB(m, sin)
+AC_SUBST(LIBM)
 
 ##AC_CHECK_LIB([c], [printf],, AC_MSG_ERROR([libc not found!! ]))
 
