diff -ur aiksaurus-1.2.1/base/aiksaurus.t.cpp aiksaurus-1.2.1-haiku/base/aiksaurus.t.cpp
--- aiksaurus-1.2.1/base/aiksaurus.t.cpp	2003-06-27 01:44:38.003407872 +0000
+++ aiksaurus-1.2.1-haiku/base/aiksaurus.t.cpp	2013-12-30 08:47:08.949747712 +0000
@@ -24,6 +24,7 @@
 #include "config.h"
 #include <iostream>
 #include <string>
+#include <stdlib.h>
 using namespace std;
 using AiksaurusImpl::AsciiCompare;
 
diff -ur aiksaurus-1.2.1/configure.ac aiksaurus-1.2.1-haiku/configure.ac
--- aiksaurus-1.2.1/configure.ac	2004-06-12 04:34:05.017563648 +0000
+++ aiksaurus-1.2.1-haiku/configure.ac	2013-12-29 23:53:53.963903488 +0000
@@ -1,5 +1,6 @@
 AC_INIT([aiksaurus],[1.2],[http://aiksaurus.sourceforge.net/])
 AC_CONFIG_SRCDIR([base/Aiksaurus.h])
+AC_CONFIG_MACRO_DIR([m4])
 
 # Aiksaurus, version 1.2.0
 
@@ -47,19 +48,19 @@
 
 AC_SUBST(LT_RELEASE)
 
-LT_CURRENT=`expr $AIK_MICRO_VERSION - $AIK_INTERFACE_AGE`
-LT_REVISION=$AIK_INTERFACE_AGE
-LT_AGE=`expr $AIK_BINARY_AGE - $AIK_INTERFACE_AGE`
+m4_define(LT_CURRENT,`expr $AIK_MICRO_VERSION - $AIK_INTERFACE_AGE`)
+m4_define(LT_REVISION,$AIK_INTERFACE_AGE)
+m4_define(LT_AGE,`expr $AIK_BINARY_AGE - $AIK_INTERFACE_AGE`)
 
-LT_AIK_VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
+LT_AIK_VERSION_INFO="LT_CURRENT():LT_REVISION():LT_AGE()"
 
 AC_SUBST(LT_AIK_VERSION_INFO)
 
-LT_CURRENT=`expr $AIK_MICRO_VERSION - $AIKGTK_INTERFACE_AGE`
-LT_REVISION=$AIKGTK_INTERFACE_AGE
-LT_AGE=`expr $AIKGTK_BINARY_AGE - $AIKGTK_INTERFACE_AGE`
+m4_define(LT_CURRENT,`expr $AIK_MICRO_VERSION - $AIKGTK_INTERFACE_AGE`)
+m4_define(LT_REVISION,$AIKGTK_INTERFACE_AGE)
+m4_define(LT_AGE,`expr $AIKGTK_BINARY_AGE - $AIKGTK_INTERFACE_AGE`)
 
-LT_AIKGTK_VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
+LT_AIKGTK_VERSION_INFO="LT_CURRENT():LT_REVISION():LT_AGE()"
 
 AC_SUBST(LT_AIKGTK_VERSION_INFO)
 
@@ -69,7 +70,7 @@
 
 # Specify a configuration file
 # 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AH_TOP([
 #ifndef AIKSAURUS_CONFIG_H
diff -ur aiksaurus-1.2.1/Makefile.am aiksaurus-1.2.1-haiku/Makefile.am
--- aiksaurus-1.2.1/Makefile.am	2003-06-15 10:17:42.018350080 +0000
+++ aiksaurus-1.2.1-haiku/Makefile.am	2013-12-29 23:53:53.966000640 +0000
@@ -27,3 +27,5 @@
 	ac-helpers/pkg-config/pkg.m4	\
 	aiksaurus.spec.in		\
 	aiksaurus.spec
+
+ACLOCAL_AMFLAGS = -I m4
