diff -urN m4-1.4.15/lib/pipe2.c m4-1.4.15-haiku/lib/pipe2.c
--- m4-1.4.15/lib/pipe2.c	2010-01-28 13:04:07.051118080 +0000
+++ m4-1.4.15-haiku/lib/pipe2.c	2010-02-27 20:00:33.547094528 +0000
@@ -78,7 +78,11 @@
 /* Unix API.  */
 
   /* Check the supported flags.  */
+#if (defined __BEOS__ || defined __HAIKU__)
+  if ((flags & ~(O_CLOEXEC | O_NONBLOCK)) != 0)
+#else
   if ((flags & ~(O_CLOEXEC | O_NONBLOCK | O_TEXT | O_BINARY)) != 0)
+#endif
     {
       errno = EINVAL;
       return -1;
