diff -urN m4-1.4.15/lib/pipe2.c m4-1.4.15-haiku/lib/pipe2.c
--- m4-1.4.15/lib/pipe2.c	2010-03-08 21:47:53.064225280 +0000
+++ m4-1.4.15-haiku/lib/pipe2.c	2010-09-24 04:13:33.514064384 +0000
@@ -78,8 +78,11 @@
 /* Unix API.  */
 
   /* Check the supported flags.  */
-  if ((flags & ~(O_CLOEXEC | O_NONBLOCK | O_TEXT | O_BINARY)) != 0)
-    {
+#if (defined __BEOS__ || defined __HAIKU__)
+   if ((flags & ~(O_CLOEXEC | O_NONBLOCK)) != 0)
+#else
+   if ((flags & ~(O_CLOEXEC | O_NONBLOCK | O_TEXT | O_BINARY)) != 0)
+#endif    {
       errno = EINVAL;
       return -1;
     }
diff -urN m4-1.4.15/src/m4.c m4-1.4.15-haiku/src/m4.c
--- m4-1.4.15/src/m4.c	2010-06-04 17:33:48.024379392 +0000
+++ m4-1.4.15-haiku/src/m4.c	2010-09-24 04:18:51.275513344 +0000
@@ -408,7 +408,8 @@
   sigaction (SIGABRT, &act, NULL);
   sigaction (SIGILL, &act, NULL);
   sigaction (SIGFPE, &act, NULL);
-  sigaction (SIGBUS, &act, NULL);
+  if (SIGBUS != SIGSEGV)
+    sigaction (SIGBUS, &act, NULL);
   if (c_stack_action (fault_handler) == 0)
     nesting_limit = 0;
 
