diff -Naur AVLDupTree/configure.in AVLDupTree-haiku/configure.in
--- AVLDupTree/configure.in	1970-01-01 00:00:00.000000000 +0000
+++ AVLDupTree-haiku/configure.in	2015-01-11 18:54:32.475791360 +0000
@@ -0,0 +1,25 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.69])
+AC_INIT(AVLDupTree, 1.0.0, https://github.com/HaikuArchives/AVLDupTree/issues)
+AC_CONFIG_SRCDIR([Source/AVLDupTree.c])
+# AC_CONFIG_HEADERS([config.h])
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+
+# Checks for libraries.
+
+# Checks for header files.
+AC_CHECK_HEADERS([OS.h stdlib.h string.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+
+# Checks for library functions.
+AC_FUNC_MALLOC
+AC_CHECK_FUNCS([memset strerror])
+
+AC_OUTPUT
diff -Naur AVLDupTree/Makefile AVLDupTree-haiku/Makefile
--- AVLDupTree/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ AVLDupTree-haiku/Makefile	2015-01-11 23:01:41.289406976 +0000
@@ -0,0 +1,32 @@
+CC=gcc
+CFLAGS=-g -O2 -Wall -ISource $(OPTFLAGS)
+
+SOURCES=$(wildcard Source/*.c)
+OBJECTS=$(patsubst %.c,%.o,$(SOURCES))
+GUISOURCES=Source/Main.cpp
+GUIEXE=bin/AGMSAVLTest
+
+TARGET=build/libavlduptree.so
+
+all: $(TARGET) $(SO_TARGET)
+
+$(TARGET): build $(OBJECTS)
+	$(CC) -shared -o $(TARGET) $(OBJECTS)
+
+build:
+	@mkdir -p build
+	@mkdir -p bin
+
+clean:
+	@rm -rf build $(OBJECTS) $(TESTS)
+
+install: all $(GUIEXE)
+	install $(TARGET) $(PREFIX)
+
+$(GUIEXE): CFLAGS += -Wno-multichar
+$(GUIEXE): CFLAGS+= -lbe
+$(GUIEXE): $(SOURCES) $(GUISOURCES)
+	$(CC) $(CFLAGS) $^ -o $@
+	install $(GUIEXE) $(BINDIR)/
+
+.PHONY: all clean install
