diff -urN lame-398-2/configure.in lame-398-2-haiku/configure.in
--- lame-398-2/configure.in	2008-08-05 14:16:05.000000000 +0000
+++ lame-398-2-haiku/configure.in	2009-07-18 07:33:31.000000000 +0000
@@ -343,6 +343,9 @@
 		*solaris*)
 			LIBS="$LIBS -lnsl"
 			;;
+		*haiku*)
+			LIBS="$LIBS -lnetwork"
+			;;
 		esac
 	fi
 fi
@@ -385,14 +388,13 @@
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+dnl	AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
 AC_ARG_ENABLE(efence,
   [  --enable-efence            Use ElectricFence for malloc debugging],
   CONFIG_EFENCE="${enableval}", CONFIG_EFENCE="no")
-
 case "${CONFIG_EFENCE}" in
 yes)
 	AC_CHECK_LIB(efence, EF_Print, HAVE_EFENCE="-lefence")
diff -urN lame-398-2/frontend/Makefile.am lame-398-2-haiku/frontend/Makefile.am
--- lame-398-2/frontend/Makefile.am	2006-09-30 09:17:05.000000000 +0000
+++ lame-398-2-haiku/frontend/Makefile.am	2009-07-18 07:42:04.000000000 +0000
@@ -60,7 +60,7 @@
 	$(top_builddir)/libmp3lame/libmp3lame.la \
 	@FRONTEND_LDADD@
 
-mp3x_LDADD = $(LDADD) @GTK_LIBS@
+mp3x_LDADD = $(LDADD)
 
 CLEANFILES = lclint.txt $(EXTRA_PROGRAMS)
 
