--- Pachi/Makefile.am.org	2014-12-13 14:43:32.076021760 +0100
+++ Pachi/Makefile.am	2014-12-13 14:41:07.499646464 +0100
@@ -1 +1,2 @@
 SUBDIRS = src Tgfx data fonts music sounds docs
+ACLOCAL_AMFLAGS = -I m4
--- Pachi/data/Makefile.am.org	2002-12-08 14:30:17.000000000 +0100
+++ Pachi/data/Makefile.am	2014-12-13 14:46:22.102236160 +0100
@@ -14,9 +14,3 @@
 scores.dat
 
 EXTRA_DIST = $(dat_DATA) $(score_DATA)
-
-install-data-local:
-	for file in $(score_DATA); do\
-	    chgrp $(group) $(scoredir)/$$file; \
-	    chmod $(perms) $(scoredir)/$$file; \
-	done
--- Pachi/configure.in.org	2014-12-14 10:34:54.227278848 +0100
+++ Pachi/configure.in	2014-12-14 10:31:37.060555264 +0100
@@ -1,7 +1,7 @@
 # Process this file with autoconf to produce a configure script.
 AC_INIT(src/faes.cpp, 0.1, T-1000@Bigfoot.com)
 AM_INIT_AUTOMAKE(pachi, 0.1)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 # Checks for programs.
 AC_PROG_CC
@@ -44,7 +44,7 @@
 	SCOREDIR="."
 else
 	DATADIR="$datadir/$PACKAGE"
-	SCOREDIR="/var/lib/games/$PACKAGE"
+	SCOREDIR="$datadir /$PACKAGE"
 fi
 
 eval DATAPATH=`eval echo "$DATADIR"`
@@ -55,3 +55,5 @@
 AC_SUBST(SCOREPATH)
 
 AC_OUTPUT(Makefile src/Makefile Tgfx/Makefile data/Makefile fonts/Makefile music/Makefile sounds/Makefile docs/Makefile)
+
+AC_CONFIG_MACRO_DIR([m4])
