diff -urN libsigc++-2.2.7/Makefile.am libsigc++-2.2.7-haiku/Makefile.am
--- libsigc++-2.2.7/Makefile.am	2009-09-02 19:37:49.029360128 +0000
+++ libsigc++-2.2.7-haiku/Makefile.am	2010-05-29 17:20:38.000000000 +0000
@@ -22,7 +22,7 @@
 else
 doc_subdirs =
 endif
-SUBDIRS = sigc++ tests examples $(doc_subdirs)
+SUBDIRS = sigc++ $(doc_subdirs)
 
 sigc_configdir = $(libdir)/$(SIGCXX_MODULE_NAME)/include
 nodist_sigc_config_HEADERS = sigc++config.h
diff -urN libsigc++-2.2.7/Makefile.in libsigc++-2.2.7-haiku/Makefile.in
--- libsigc++-2.2.7/Makefile.in	2010-05-04 19:44:19.045875200 +0000
+++ libsigc++-2.2.7-haiku/Makefile.in	2010-05-29 17:20:43.000000000 +0000
@@ -114,7 +114,7 @@
 	distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = sigc++ tests examples docs
+DIST_SUBDIRS = sigc++ docs
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -291,7 +291,7 @@
 ACLOCAL_AMFLAGS = -I build ${ACLOCAL_FLAGS}
 @ENABLE_DOCUMENTATION_FALSE@doc_subdirs = 
 @ENABLE_DOCUMENTATION_TRUE@doc_subdirs = docs
-SUBDIRS = sigc++ tests examples $(doc_subdirs)
+SUBDIRS = sigc++ $(doc_subdirs)
 sigc_configdir = $(libdir)/$(SIGCXX_MODULE_NAME)/include
 nodist_sigc_config_HEADERS = sigc++config.h
 pkgconfigdir = $(libdir)/pkgconfig
diff -urN libsigc++-2.2.7/configure.ac libsigc++-2.2.7-haiku/configure.ac
--- libsigc++-2.2.7/configure.ac	2010-05-04 19:41:29.054788096 +0000
+++ libsigc++-2.2.7-haiku/configure.ac	2010-05-29 17:19:57.000000000 +0000
@@ -66,8 +66,6 @@
                  ${SIGCXX_MODULE_NAME}.pc:sigc++.pc.in
                  ${SIGCXX_MODULE_NAME}-uninstalled.pc:sigc++-uninstalled.pc.in
                  sigc++/Makefile
-                 examples/Makefile
-                 tests/Makefile
                  docs/Makefile
                  docs/reference/Doxyfile
                  MSVC_Net2005/sigc.rc
