diff -urN libsigc++-1.2.7/Makefile.in libsigc++-1.2.7-haiku/Makefile.in
--- libsigc++-1.2.7/Makefile.in	2005-04-13 06:32:20.035127296 +0000
+++ libsigc++-1.2.7-haiku/Makefile.in	2010-05-29 18:05:57.633077760 +0000
@@ -196,8 +196,8 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = sigc++ tests doc MSVC_6 MSVC_Net2003 eVC4
-DIST_SUBDIRS = $(SUBDIRS) examples scripts
+SUBDIRS = sigc++ doc MSVC_6 MSVC_Net2003 eVC4
+DIST_SUBDIRS = $(SUBDIRS) scripts
 ACLOCAL_AMFLAGS = -I $(srcdir)/scripts
 EXTRA_DIST = \
 	FEATURES IDEAS			\
diff -urN libsigc++-1.2.7/configure.in libsigc++-1.2.7-haiku/configure.in
--- libsigc++-1.2.7/configure.in	2005-04-13 06:12:02.034603008 +0000
+++ libsigc++-1.2.7-haiku/configure.in	2010-05-29 18:04:52.096468992 +0000
@@ -659,8 +659,6 @@
         sigc++/macros/Makefile  
         sigc++/config/Makefile  
         scripts/Makefile
-        examples/Makefile
-        tests/Makefile
         MSVC_6/Makefile
         MSVC_Net2003/Makefile
           MSVC_Net2003/tests/Makefile
