diff -urN neon-0.28.3/macros/neon.m4 neon-0.28.3-haiku/macros/neon.m4
--- neon-0.28.3/macros/neon.m4	2008-12-08 08:30:42.000000000 -0800
+++ neon-0.28.3-haiku/macros/neon.m4	2008-12-08 08:12:43.000000000 -0800
@@ -626,7 +626,7 @@
 
 # Unixware 7 can only link gethostbyname with -lnsl -lsocket
 # Pick up -lsocket first, then the gethostbyname check will work.
-NE_SEARCH_LIBS(socket, socket inet ws2_32)
+NE_SEARCH_LIBS(socket, socket inet ws2_32 network)
 
 # Enable getaddrinfo support if it, gai_strerror and inet_ntop are
 # all available.
@@ -659,7 +659,7 @@
    # Checks for non-getaddrinfo() based resolver interfaces.
    # QNX has gethostbyname in -lsocket. BeOS only has it in -lbind.
    # CygWin/Winsock2 has it in -lws2_32, allegedly.
-   NE_SEARCH_LIBS(gethostbyname, socket nsl bind ws2_32)
+   NE_SEARCH_LIBS(gethostbyname, socket nsl bind ws2_32 network)
    NE_SEARCH_LIBS(hstrerror, resolv,,[:])
    NE_CHECK_FUNCS(hstrerror)
    # Older Unixes don't declare h_errno.
