diff -Nur freesci-0.6.4/src/config.l freesci-0.6.4_beos/src/config.l
--- freesci-0.6.4/src/config.l	Fri May  2 17:59:51 2008
+++ freesci-0.6.4_beos/src/config.l	Sat Feb 21 08:53:45 2009
@@ -59,9 +59,6 @@
 	void *(*check_driver)(char *name);
 } freesci_subsystems[FREESCI_DRIVER_SUBSYSTEMS_NR] = {
 	{"gfx", NULL},
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
         {"pcm", parse_pcmout_driver},
 	{"midiout", parse_midiout_driver}
@@ -261,9 +258,6 @@
 	OPT_INT("alpha_threshold", alpha_threshold, 0, 255),
 	OPT_INT("animation_delay", animation_delay, 0, 1000000),
 	OPT_INT("animation_granularity", animation_granularity, 1, 160),
-#ifdef __GNUC__
-#warning "Re-enable sound server config"
-#endif
 #if 0
 	OPT_STATICREF("midiout_driver", midiout_driver, parse_midiout_driver),
 	OPT_STATICREF("midi_device", midi_device, parse_midi_device),
@@ -613,9 +607,6 @@
 
 	conf->gfx_driver_name = NULL;
 
-#ifdef __GNUC__
-#warning "Re-enable sound stuff"
-#endif
 #if 0
 	conf->pcmout_driver = pcmout_find_driver(NULL);
 	conf->pcmout_rate = 22050;
@@ -809,9 +800,6 @@
 }
 
 
-#ifdef __GNUC__
-#warning "Re-enable sound stuff"
-#endif
 #if 0
 void *
 parse_sound_server(char *driver_name)
diff -Nur freesci-0.6.4/src/engine/game.c freesci-0.6.4_beos/src/engine/game.c
--- freesci-0.6.4/src/engine/game.c	Fri May  2 16:40:30 2008
+++ freesci-0.6.4_beos/src/engine/game.c	Sat Feb 21 08:48:38 2009
@@ -666,9 +666,6 @@
 int
 game_init(state_t *s)
 {
-#ifdef __GNUC__
-#  warning "Fixme: Use new VM instantiation code all over the place"
-#endif
 	reg_t game_obj; /* Address of the game object */
 	dstack_t *stack;
 
@@ -769,17 +766,11 @@
 
 	sciprintf("Freeing miscellaneous data...\n");
 
-#ifdef __GNUC__
-#warning "Free parser segment here"
-#endif
 	if (send_calls_allocated) {
 		sci_free(send_calls);
 		send_calls_allocated = 0;
 	}
 
-#ifdef __GNUC__
-#warning "Free scripts here"
-#endif
 
 	menubar_free(s->menubar);
 
diff -Nur freesci-0.6.4/src/engine/scriptdebug.c freesci-0.6.4_beos/src/engine/scriptdebug.c
--- freesci-0.6.4/src/engine/scriptdebug.c	Sat Mar 22 12:17:23 2008
+++ freesci-0.6.4_beos/src/engine/scriptdebug.c	Sat Feb 21 08:51:38 2009
@@ -799,9 +799,6 @@
 	return 0;
 }
 
-#ifdef __GNUC__
-#warning "Re-implement con:so"
-#endif
 #if 0
 int
 c_stepover(state_t *s)
@@ -2279,9 +2276,6 @@
 static int
 c_gfx_draw_viewobj(state_t *s)
 {
-#ifdef __GNUC__
-#warning "Re-implement con:gfx_draw_viewobj"
-#endif
 #if 0
 	heap_ptr pos = (heap_ptr) (cmd_params[0].val);
 	int is_view;
@@ -2918,9 +2912,6 @@
 bb = GET_SELECTOR(pos, bb);
 
 #if 0
-#ifdef __GNUC__
-#warning "Re-implement viewobjinfo"
-#endif
 static void
 viewobjinfo(state_t *s, heap_ptr pos)
 {
@@ -3524,9 +3515,6 @@
 			con_hook_command(c_vmvarlist, "vmvarlist", "!", "Displays the addresses of variables in the VM");
 			con_hook_command(c_step, "s", "i*", "Executes one or several operations\n\nEXAMPLES\n\n"
 					 "    s 4\n\n  Execute 4 commands\n\n    s\n\n  Execute next command");
-#ifdef __GNUC__
-#warning "Re-enable con:so hook"
-#endif
 #if 0
 			con_hook_command(c_stepover, "so", "", "Executes one operation skipping over sends");
 #endif
@@ -3669,9 +3657,6 @@
 			con_hook_command(c_gfx_update_zone, "gfx_update_zone", "iiii", "Propagates a rectangular area from\n  the back buffer to the front buffer"
 					 "\n\nUSAGE\n\n"
 					 "  gfx_update_zone <x> <y> <xl> <yl>");
-#ifdef __GNUC__
-#warning "Re-enable con:draw_viewobj"
-#endif
 #if 0
 			con_hook_command(c_gfx_draw_viewobj, "draw_viewobj", "i", "Draws the nsRect and brRect of a\n  dynview object.\n\n  nsRect is green, brRect\n"
 					 "  is blue.\n");
diff -Nur freesci-0.6.4/src/include/kernel_compat.h freesci-0.6.4_beos/src/include/kernel_compat.h
--- freesci-0.6.4/src/include/kernel_compat.h	Sun Aug 19 00:59:26 2007
+++ freesci-0.6.4_beos/src/include/kernel_compat.h	Sat Feb 21 08:50:14 2009
@@ -29,9 +29,6 @@
 #ifndef _SCI_KERNEL_COMPAT_
 #define _SCI_KERNEL_COMPAT_
 
-#ifdef __GNUC__
-#warning "Old kernel compatibility crap"
-#endif
 
 /* Minimal heap position */
 #define HEAP_MIN 800
diff -Nur freesci-0.6.4/src/include/sci_conf.h freesci-0.6.4_beos/src/include/sci_conf.h
--- freesci-0.6.4/src/include/sci_conf.h	Sun Aug 19 00:59:28 2007
+++ freesci-0.6.4_beos/src/include/sci_conf.h	Sat Feb 21 08:48:43 2009
@@ -85,9 +85,6 @@
 	int reverse_stereo;
 	int res_version;
 	
-#ifdef __GNUC__
-#  warning "Re-enable sound stuff"
-#endif
 #if 0
 	midiout_driver_t *midiout_driver; /* the midiout method to use */
 	midi_device_t *midi_device; /* the midi device to use */
diff -Nur freesci-0.6.4/src/main.c freesci-0.6.4_beos/src/main.c
--- freesci-0.6.4/src/main.c	Fri Mar 21 10:32:54 2008
+++ freesci-0.6.4_beos/src/main.c	Sat Feb 21 08:55:23 2009
@@ -308,9 +308,6 @@
 	printf("\n");
 }
 
-#ifdef __GNUC__
-#warning "Re-enable sound stuff"
-#endif
 #if 0
 static void
 list_pcmout_drivers()
@@ -549,9 +546,6 @@
 			printf("Supported graphics drivers: ");
 			list_graphics_drivers();
 
-#ifdef __GNUC__
-#warning "Re-enable sound stuff"
-#endif
 #if 0
 			printf("Supported sound servers: ");
 			list_sound_servers();
@@ -1115,9 +1109,6 @@
 
 	chdir(startdir);
 
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	printf("Mapping instruments to General Midi\n");
 
@@ -1171,9 +1162,6 @@
 		free(cl_options.gfx_driver_name);
 	} /* else it's still NULL */
 
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	if (cl_options.pcmout_driver_name)
 		pcmout_driver = old_lookup_driver((old_lookup_funct_t *)pcmout_find_driver,
@@ -1209,9 +1197,6 @@
 		memcpy(gfx_options, &(active_conf->gfx_options), sizeof(gfx_options_t)); /* memcpy so that console works */
 		if (!gfx_driver_name)
 			gfx_driver_name = active_conf->gfx_driver_name;
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 		if (!sound_server)
 			sound_server = active_conf->sound_server;
@@ -1297,9 +1282,6 @@
 		}
 	}
 
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	/* Configure the pcmout driver */
 	{
@@ -1349,9 +1331,6 @@
 		exit(1);
 	}
 
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	if (!sound_server)
 		sound_server = sound_server_find_driver(NULL);
@@ -1360,9 +1339,6 @@
 	if (cl_options.show_rooms)
 	  set_debug_mode(gamestate, 1, "r");
 
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	gamestate->sound_server = sound_server;
 
@@ -1405,9 +1381,6 @@
 
 	printf("Graphics: Using the %s driver %s\n",
 	       gfx_driver->name, gfx_driver->version);
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	printf("MIDI-out: Using the %s driver %s\n",
 	       midiout_driver->name, midiout_driver->version);
@@ -1430,9 +1403,6 @@
 		game_restore(&gamestate, savegame_name);
 	else
 		game_run(&gamestate); /* Run the game */
-#ifdef __GNUC__
-#warning "sound"
-#endif
 #if 0
 	if (gamestate->sound_server)
 		gamestate->sound_server->exit(gamestate); /* Shutdown sound daemon first */
