--- lines-0.6/main.c.org	2016-05-22 15:04:32.000000000 +0200
+++ lines-0.6/main.c	2016-05-22 15:26:13.000000000 +0200
@@ -7,6 +7,11 @@
 #include <SDL_thread.h>
 // #include <pthread.h>
 
+#ifdef __HAIKU__
+#include <unistd.h>
+#include <libgen.h>
+#endif
+
 #ifndef WINDOWS
 #include <unistd.h>
 #include <pwd.h>
@@ -1559,6 +1564,7 @@
 #ifdef MAEMO
 	putenv("SDL_VIDEO_X11_WMCLASS=lines");
 #endif
+	chdir(dirname(argv[0]));
 	game_mutex = SDL_CreateMutex();
 	if (gfx_init())
 		return 1;
