diff -Nbaur bullet-2.78/Demos/MultiThreadedDemo/CMakeLists.txt bullet-2.78-haiku/Demos/MultiThreadedDemo/CMakeLists.txt
--- bullet-2.78/Demos/MultiThreadedDemo/CMakeLists.txt	2010-07-08 14:02:38.033554432 -0300
+++ bullet-2.78-haiku/Demos/MultiThreadedDemo/CMakeLists.txt	2011-04-25 19:30:57.000000000 -0300
@@ -25,9 +25,9 @@
 	MultiThreadedDemo.cpp
 	MultiThreadedDemo.h
 )
-IF (UNIX)
+IF (UNIX AND NOT BEOS)
   TARGET_LINK_LIBRARIES(AppMultiThreadedDemo pthread)
-ENDIF(UNIX)
+ENDIF(UNIX AND NOT BEOS)
 
 IF(WIN32)
 IF (CMAKE_CL_64)
diff -Nbaur bullet-2.78/Demos/OpenCLClothDemo/MiniCL/CMakeLists.txt bullet-2.78-haiku/Demos/OpenCLClothDemo/MiniCL/CMakeLists.txt
--- bullet-2.78/Demos/OpenCLClothDemo/MiniCL/CMakeLists.txt	2010-09-21 20:09:18.033292288 -0300
+++ bullet-2.78-haiku/Demos/OpenCLClothDemo/MiniCL/CMakeLists.txt	2011-04-25 19:33:03.000000000 -0300
@@ -81,9 +81,9 @@
 			COMMAND ${CMAKE_COMMAND} ARGS -E copy_if_different ${BULLET_PHYSICS_SOURCE_DIR}/Demos/OpenCLClothDemo/atiFlag.bmp ${CMAKE_CURRENT_BINARY_DIR}
 			)
 ENDIF()
-IF (UNIX)
+IF (UNIX AND NOT BEOS)
   TARGET_LINK_LIBRARIES(AppOpenCLClothDemo_Mini pthread)
-ENDIF(UNIX)
+ENDIF(UNIX AND NOT BEOS)
 
 
 IF (INTERNAL_ADD_POSTFIX_EXECUTABLE_NAMES)
diff -Nbaur bullet-2.78/Demos/ThreadingDemo/CMakeLists.txt bullet-2.78-haiku/Demos/ThreadingDemo/CMakeLists.txt
--- bullet-2.78/Demos/ThreadingDemo/CMakeLists.txt	2010-10-06 20:07:00.014155776 -0300
+++ bullet-2.78-haiku/Demos/ThreadingDemo/CMakeLists.txt	2011-04-25 19:32:00.000000000 -0300
@@ -34,9 +34,9 @@
 	)
 ENDIF()
 
-IF (UNIX)
+IF (UNIX AND NOT BEOS)
   TARGET_LINK_LIBRARIES(AppThreadingDemo pthread)
-ENDIF(UNIX)
+ENDIF(UNIX AND NOT BEOS)
 
 IF (INTERNAL_ADD_POSTFIX_EXECUTABLE_NAMES)
 			SET_TARGET_PROPERTIES(AppThreadingDemo PROPERTIES  DEBUG_POSTFIX "_Debug")
