--- libdvdnav-4.2.0/configure.ac	2011-10-07 20:20:59.001572864 +0000
+++ libdvdnav-4.2.0-haiku/configure.ac	2012-04-09 16:41:25.801112064 +0000
@@ -143,7 +143,7 @@
     THREAD_CFLAGS="-I/usr/local/include -D_THREAD_SAFE"
     CFLAGS="$THREAD_CFLAGS $CFLAGS"
     ;;
-  *mingw32* | *cygwin*)
+  *mingw32* | *cygwin* | *haiku*)
     ;;
   *)
     AC_CHECK_LIB(pthread, pthread_create,
@@ -168,6 +168,8 @@
   *os2*)
     LDFLAGS="-no-undefined -Zbin-files $LDFLAGS"
     ;;
+  *haiku*)
+  	;;
   *)
     AC_CHECK_LIB(c, dlopen,
 	         DYNAMIC_LD_LIBS="",
