--- alienblaster/src/main.cc.org	2004-08-19 18:24:20.060555264 +0200
+++ alienblaster/src/main.cc	2014-09-03 18:00:48.988282880 +0200
@@ -21,9 +21,17 @@
 #include "SDL.h"
 #include <stdlib.h>
 
+#ifdef __HAIKU__
+#include <unistd.h>
+#include <libgen.h>
+#endif
+
 using namespace std;
 
 int main(int argc, char *argv[]) {
+  #ifdef __HAIKU__
+  chdir(dirname(argv[0]));
+  #endif
   SDL_Init(0);
   srand(0);
   Game game;
