diff -Naur mp4v2-2.0.0/configure.ac mp4v2-2.0.0-haiku/configure.ac
--- mp4v2-2.0.0/configure.ac	2012-05-20 17:11:55.039845888 -0500
+++ mp4v2-2.0.0-haiku/configure.ac	2013-11-26 23:36:14.129761280 -0600
@@ -4,23 +4,23 @@
 
 m4_include([project/project.m4sugar])
 
-m4_define([SVNINFO],m4_esyscmd([LANG=en svn info 2>/dev/null]))
-
-changequote(<<,>>)dnl
-    m4_define(<<PRJ_repo_url>>,m4_bregexp(SVNINFO,<<^URL: *\(.+\)>>,<<\1>>))
-    m4_define(<<PRJ_repo_branch>>,m4_bregexp(PRJ_repo_url,<<\([^/]+\)$>>,<<\1>>))
-    m4_define(<<PRJ_repo_root>>,m4_bregexp(SVNINFO,<<^Repository Root: *\(.+\)>>,<<\1>>))
-    m4_define(<<PRJ_repo_uuid>>,m4_bregexp(SVNINFO,<<^Repository UUID: *\(.+\)>>,<<\1>>))
-    m4_define(<<PRJ_repo_rev>>,m4_bregexp(SVNINFO,<<^Last Changed Rev: *\(.+\)>>,<<\1>>))
-    m4_define(<<PRJ_repo_date>>,m4_bregexp(SVNINFO,<<^Last Changed Date: *\(.+\)>>,<<\1>>))
-    m4_define(<<PRJ_repo_type>>,ifelse(m4_bregexp(PRJ_repo_url,<</releases/>>),<<-1>>,<<developer>>,<<stable>>))
-changequote([,])dnl
-
-m4_define([PRJ_version_hex],m4_format([0x%04x%02x%02x],PRJ_version_major,PRJ_version_minor,PRJ_version_point))
-
-m4_define([PRJ_version],ifelse(
-    PRJ_repo_type,[stable],m4_format([%s],PRJ_repo_branch),
-    m4_format([%s-r%d],PRJ_repo_branch,PRJ_repo_rev)))
+#m4_define([SVNINFO],m4_esyscmd([LANG=en svn info 2>/dev/null]))
+#
+#changequote(<<,>>)dnl
+#    m4_define(<<PRJ_repo_url>>,m4_bregexp(SVNINFO,<<^URL: *\(.+\)>>,<<\1>>))
+#    m4_define(<<PRJ_repo_branch>>,m4_bregexp(PRJ_repo_url,<<\([^/]+\)$>>,<<\1>>))
+#    m4_define(<<PRJ_repo_root>>,m4_bregexp(SVNINFO,<<^Repository Root: *\(.+\)>>,<<\1>>))
+#    m4_define(<<PRJ_repo_uuid>>,m4_bregexp(SVNINFO,<<^Repository UUID: *\(.+\)>>,<<\1>>))
+#    m4_define(<<PRJ_repo_rev>>,m4_bregexp(SVNINFO,<<^Last Changed Rev: *\(.+\)>>,<<\1>>))
+#    m4_define(<<PRJ_repo_date>>,m4_bregexp(SVNINFO,<<^Last Changed Date: *\(.+\)>>,<<\1>>))
+#    m4_define(<<PRJ_repo_type>>,ifelse(m4_bregexp(PRJ_repo_url,<</releases/>>),<<-1>>,<<developer>>,<<stable>>))
+#changequote([,])dnl
+#
+#m4_define([PRJ_version_hex],m4_format([0x%04x%02x%02x],PRJ_version_major,PRJ_version_minor,PRJ_version_point))
+#
+#m4_define([PRJ_version],ifelse(
+#    PRJ_repo_type,[stable],m4_format([%s],PRJ_repo_branch),
+#    m4_format([%s-r%d],PRJ_repo_branch,PRJ_repo_rev)))
 
 ###############################################################################
 # initialization
diff -Naur mp4v2-2.0.0/libplatform/platform_base.h mp4v2-2.0.0-haiku/libplatform/platform_base.h
--- mp4v2-2.0.0/libplatform/platform_base.h	2012-05-20 17:11:55.021495808 -0500
+++ mp4v2-2.0.0-haiku/libplatform/platform_base.h	2013-11-26 23:49:07.969670656 -0600
@@ -6,9 +6,7 @@
 #include <fstream>
 #include <iomanip>
 #include <iostream>
-#include <limits>
 #include <list>
-#include <locale>
 #include <map>
 #include <set>
 #include <sstream>
diff -Naur mp4v2-2.0.0/libutil/Utility.cpp mp4v2-2.0.0-haiku/libutil/Utility.cpp
--- mp4v2-2.0.0/libutil/Utility.cpp	2012-05-20 17:11:53.025690112 -0500
+++ mp4v2-2.0.0-haiku/libutil/Utility.cpp	2013-11-26 23:36:33.651427840 -0600
@@ -431,7 +431,6 @@
             << '\n' << setw(18) << "repository URL:" << MP4V2_PROJECT_repo_url
             << '\n' << setw(18) << "repository root:" << MP4V2_PROJECT_repo_root
             << '\n' << setw(18) << "repository UUID:" << MP4V2_PROJECT_repo_uuid
-            << '\n' << setw(18) << "repository rev:" << MP4V2_PROJECT_repo_rev
             << '\n' << setw(18) << "repository date:" << MP4V2_PROJECT_repo_date
             << '\n' << setw(18) << "repository type:" << MP4V2_PROJECT_repo_type;
     }
diff -Naur mp4v2-2.0.0/src/itmf/CoverArtBox.h mp4v2-2.0.0-haiku/src/itmf/CoverArtBox.h
--- mp4v2-2.0.0/src/itmf/CoverArtBox.h	2012-05-20 17:11:53.038797312 -0500
+++ mp4v2-2.0.0-haiku/src/itmf/CoverArtBox.h	2013-11-26 23:49:50.736886784 -0600
@@ -111,7 +111,7 @@
     ///
     /// @return <b>true</b> on failure, <b>false</b> on success.
     ///
-    static bool remove( MP4FileHandle hFile, uint32_t index = numeric_limits<uint32_t>::max() );
+    static bool remove( MP4FileHandle hFile, uint32_t index = 2147483647);
 };
 
 ///////////////////////////////////////////////////////////////////////////////
