diff -urN sdl-mixer-1.2-hg/native_midi/native_midi_haiku.cpp sdl-mixer-1.2-hg-haiku/native_midi/native_midi_haiku.cpp
--- sdl-mixer-1.2-hg/native_midi/native_midi_haiku.cpp	2011-04-06 14:55:40.152305664 +0000
+++ sdl-mixer-1.2-hg-haiku/native_midi/native_midi_haiku.cpp	2011-04-06 14:08:15.000000000 +0000
@@ -259,3 +259,23 @@
   currentSong = song;
 }
 void native_midi_stop()
+{
+  if (currentSong == NULL) return;
+  currentSong->store->Stop();
+  currentSong->store->Disconnect(&synth);
+  while (currentSong->store->IsPlaying())
+    usleep(1000);
+  currentSong = NULL;
+}
+int native_midi_active()
+{
+  if (currentSong == NULL) return 0;
+  return currentSong->store->CurrentEvent() < currentSong->store->CountEvents();
+}
+
+const char* native_midi_error(void)
+{
+  return lasterr;
+}
+
+#endif /* __HAIKU__ */
