From 9b1c1c69ffa48e9c4856a3c8903190121be5d58c Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sat, 8 Aug 2026 14:06:49 -0400
Subject: GCC2 fixes.


diff --git a/src/ap/hostapd.h b/src/ap/hostapd.h
index 8d4f346..58f1207 100644
--- a/src/ap/hostapd.h
+++ b/src/ap/hostapd.h
@@ -171,7 +171,7 @@ struct hostapd_sae_commit_queue {
 	struct dl_list list;
 	int rssi;
 	size_t len;
-	u8 msg[];
+	u8 msg[0];
 };
 
 struct mld_link_info {
diff --git a/src/ap/sta_info.h b/src/ap/sta_info.h
index 9810eb2..7c5cdc5 100644
--- a/src/ap/sta_info.h
+++ b/src/ap/sta_info.h
@@ -62,7 +62,7 @@ struct mbo_non_pref_chan_info {
 	u8 pref;
 	u8 reason_code;
 	u8 num_channels;
-	u8 channels[];
+	u8 channels[0];
 };
 
 struct pending_eapol_rx {
diff --git a/src/common/ieee802_11_common.h b/src/common/ieee802_11_common.h
index a7ef918..63f7b9d 100644
--- a/src/common/ieee802_11_common.h
+++ b/src/common/ieee802_11_common.h
@@ -15,7 +15,7 @@
 struct element {
 	u8 id;
 	u8 datalen;
-	u8 data[];
+	u8 data[0];
 } STRUCT_PACKED;
 
 struct hostapd_hw_modes;
diff --git a/src/common/ieee802_11_defs.h b/src/common/ieee802_11_defs.h
index 253b597..6dc5e4a 100644
--- a/src/common/ieee802_11_defs.h
+++ b/src/common/ieee802_11_defs.h
@@ -1121,12 +1121,12 @@ struct ieee80211_mgmt {
 				struct {
 					u8 action;
 					u8 trans_id[WLAN_SA_QUERY_TR_ID_LEN];
-					u8 variable[]; /* OCI element */
+					u8 variable[0]; /* OCI element */
 				} STRUCT_PACKED sa_query_req;
 				struct {
 					u8 action; /* */
 					u8 trans_id[WLAN_SA_QUERY_TR_ID_LEN];
-					u8 variable[]; /* OCI element */
+					u8 variable[0]; /* OCI element */
 				} STRUCT_PACKED sa_query_resp;
 				struct {
 					u8 action;
@@ -1187,7 +1187,7 @@ struct ieee80211_mgmt {
 				struct {
 					u8 action; /* 12 */
 					u8 dialog_token;
-					u8 variable[];
+					u8 variable[0];
 				} STRUCT_PACKED coloc_intf_report;
 				struct {
 					u8 action; /* 15 */
@@ -1207,7 +1207,7 @@ struct ieee80211_mgmt {
 					u8 dialog_token;
 					/* Reconfiguration Multi-Link element,
 					* OCI element (optional) */
-					u8 variable[];
+					u8 variable[0];
 				} STRUCT_PACKED link_reconf_req;
 				struct {
 					u8 action; /* Protected EHT - 12 */
@@ -1217,7 +1217,7 @@ struct ieee80211_mgmt {
 					 * Group Key Data (optional),
 					 * OCI element (optional),
 					 * Basic Multi-Link element (optional) */
-					u8 variable[];
+					u8 variable[0];
 				} STRUCT_PACKED link_reconf_resp;
 				struct {
 					u8 action;
@@ -2908,7 +2908,7 @@ struct eht_ml_basic_common_info {
 	 * AP MLD ID: 1 octet
 	 * Extended MLD Capabilities And Operations: 2 octets
 	 */
-	u8 variable[];
+	u8 variable[0];
 } STRUCT_PACKED;
 
 /* Length of the EML Capabilities field in the Common Info field (in octets) */
@@ -2954,7 +2954,7 @@ struct ieee80211_eht_per_sta_profile {
 	le16 sta_control;
 
 	/* Followed by STA Info and STA Profile fields */
-	u8 variable[];
+	u8 variable[0];
 } STRUCT_PACKED;
 
 /* IEEE Std 802.11be-2024, 9.4.2.322.3 - Probe Request Multi-Link element
@@ -2976,7 +2976,7 @@ struct eht_ml_probe_req_common_info {
 	 * AP MLD ID: 1 octet
 	 * MLD MAC Address: 6 cotets
 	 */
-	u8 variable[];
+	u8 variable[0];
 } STRUCT_PACKED;
 
 /* IEEE Std 802.11be-2024, 9.4.2.322.4 - Reconfiguration Multi-Link element */
@@ -3011,7 +3011,7 @@ struct eht_ml_reconf_common_info {
 	 * MLD Capabilities and Operations: 2 octets
 	 * Extended MLD Capabilities and Operations: 2 octets
 	 */
-	u8 variable[];
+	u8 variable[0];
 } STRUCT_PACKED;
 
 
@@ -3037,7 +3037,7 @@ struct eht_ml_reconf_sta_info {
 	 * Operation Parameters: 3 octets
 	 * NSTR Indication Bitmap: 1 or 2 octets
 	 */
-	u8 variable[];
+	u8 variable[0];
 } STRUCT_PACKED;
 
 /* IEEE Std 802.11be-2024, 9.4.2.322.1 - Multi-Link element / General */
@@ -3046,7 +3046,7 @@ struct ieee80211_eht_ml {
 	le16 ml_control;
 
 	/* Followed by Common Info and Link Info fields */
-	u8 variable[];
+	u8 variable[0];
 } STRUCT_PACKED;
 
 /* IEEE Std 802.11be-2024, 9.4.2.329 (Bandwidth Indication element) defines the
diff --git a/src/drivers/driver_bsd.c b/src/drivers/driver_bsd.c
index 1be7dac..e1d2be1 100644
--- a/src/drivers/driver_bsd.c
+++ b/src/drivers/driver_bsd.c
@@ -36,7 +36,12 @@
 #include <netinet/ether.h>
 #endif /* __GLIBC__ */
 #include <net80211/ieee80211.h>
+/* hack: avoids "incomplete type" errors */
+#undef __HAIKU__
+#define __FreeBSD__
 #include <net80211/ieee80211_ioctl.h>
+#undef __FreeBSD__
+#define __HAIKU__
 #include <net80211/ieee80211_crypto.h>
 #endif /* __DragonFly__ || __GLIBC__ */
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
diff --git a/wpa_supplicant/bss.h b/wpa_supplicant/bss.h
index d9d0401..b73f1ff 100644
--- a/wpa_supplicant/bss.h
+++ b/wpa_supplicant/bss.h
@@ -147,7 +147,7 @@ struct wpa_bss {
 
 	/* followed by ie_len octets of IEs */
 	/* followed by beacon_ie_len octets of IEs */
-	u8 ies[];
+	u8 ies[0];
 };
 
 static inline const u8 * wpa_bss_ie_ptr(const struct wpa_bss *bss)
diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index 4f85da1..41f4bd6 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -2826,6 +2826,7 @@ static int wpas_select_network_from_last_scan(struct wpa_supplicant *wpa_s,
 			if (new_scan)
 				wpa_supplicant_rsn_preauth_scan_results(wpa_s);
 		} else if (own_request) {
+			int timeout_sec, timeout_usec;
 			if (wpa_s->support_6ghz && trigger_6ghz_scan && data &&
 			    wpas_trigger_6ghz_scan(wpa_s, data) > 0)
 				return 1;
@@ -2835,8 +2836,8 @@ static int wpas_select_network_from_last_scan(struct wpa_supplicant *wpa_s,
 			 * own scan request and not due to a scan request on
 			 * another shared interface, try another scan.
 			 */
-			int timeout_sec = wpa_s->scan_interval;
-			int timeout_usec = 0;
+			timeout_sec = wpa_s->scan_interval;
+			timeout_usec = 0;
 #ifdef CONFIG_P2P
 			int res;
 
diff --git a/wpa_supplicant/wpa_gui-haiku/main_haiku.cpp b/wpa_supplicant/wpa_gui-haiku/main_haiku.cpp
index d96f656..4319a0a 100644
--- a/wpa_supplicant/wpa_gui-haiku/main_haiku.cpp
+++ b/wpa_supplicant/wpa_gui-haiku/main_haiku.cpp
@@ -55,6 +55,7 @@ extern "C" {
 #undef B_TRANSLATION_CONTEXT
 #define B_TRANSLATION_CONTEXT "wpa_supplicant"
 
+#undef bool
 
 static const uint32 kMsgJoinTimeout = 'jnto';
 static const char *kWPASupplicantKeyring = "wpa_supplicant";
-- 
2.54.0

