diff --git a/TolmachApp.cpp b/TolmachApp.cpp
index acfacb1..c7c190a 100644
--- a/TolmachApp.cpp
+++ b/TolmachApp.cpp
@@ -421,7 +421,7 @@ void
 TolmachApplication::LoadDictList()
 {
   BPath pathDicts;
-  find_directory(B_COMMON_DATA_DIRECTORY, &pathDicts);
+  find_directory(B_USER_DATA_DIRECTORY, &pathDicts);
   pathDicts.Append(cszDictionariesDir);
   BDirectory dir(pathDicts.Path());
   BEntry entry;
