diff -up mdocml-1.12.2/Makefile.orig mdocml-1.12.2/Makefile
--- mdocml-1.12.2/Makefile.orig	2013-10-16 23:58:37.828637184 -0600
+++ mdocml-1.12.2/Makefile	2013-10-16 23:59:08.458227712 -0600
@@ -32,7 +32,7 @@ STATIC		 = -static
 #STATIC		+= -pthread
 
 CFLAGS		+= -g -DHAVE_CONFIG_H -DVERSION="\"$(VERSION)\""
-CFLAGS     	+= -W -Wall -Wstrict-prototypes -Wno-unused-parameter -Wwrite-strings
+CFLAGS     	+= -W -Wall -Wstrict-prototypes -Wwrite-strings
 PREFIX		 = /usr/local
 WWWPREFIX	 = /var/www
 HTDOCDIR	 = $(WWWPREFIX)/htdocs
@@ -316,7 +316,6 @@ clean:
 
 install: all
 	mkdir -p $(DESTDIR)$(BINDIR)
-	mkdir -p $(DESTDIR)$(EXAMPLEDIR)
 	mkdir -p $(DESTDIR)$(LIBDIR)
 	mkdir -p $(DESTDIR)$(INCLUDEDIR)
 	mkdir -p $(DESTDIR)$(MANDIR)/man1
@@ -328,7 +327,6 @@ install: all
 	$(INSTALL_MAN) mandoc.1 preconv.1 demandoc.1 $(DESTDIR)$(MANDIR)/man1
 	$(INSTALL_MAN) mandoc.3 tbl.3 $(DESTDIR)$(MANDIR)/man3
 	$(INSTALL_MAN) man.7 mdoc.7 roff.7 eqn.7 tbl.7 mandoc_char.7 $(DESTDIR)$(MANDIR)/man7
-	$(INSTALL_DATA) example.style.css $(DESTDIR)$(EXAMPLEDIR)
 
 installcgi: all
 	mkdir -p $(DESTDIR)$(CGIBINDIR)
