diff -ur ncurses-5.6/include/curses.h ncurses-5.6-haiku/include/curses.h
--- ncurses-5.6/include/curses.h	2008-06-04 05:57:46.000000000 -0700
+++ ncurses-5.6-haiku/include/curses.h	2008-06-14 18:12:44.000000000 -0700
@@ -450,7 +450,7 @@
  */
 #if 1
 #ifdef NCURSES_WGETCH_EVENTS
-#if !defined(__BEOS__)		/* Fix _nc_timed_wait() on BEOS... */
+#if !defined(__BEOS__) || defined(__HAIKU__)	/* Fix _nc_timed_wait() on BEOS... */
 #  define NCURSES_EVENT_VERSION	1
 #endif	/* !defined(__BEOS__) */
 
diff -ur ncurses-5.6/ncurses/tty/lib_twait.c ncurses-5.6-haiku/ncurses/tty/lib_twait.c
--- ncurses-5.6/ncurses/tty/lib_twait.c	2006-05-27 14:57:43.000000000 -0700
+++ ncurses-5.6-haiku/ncurses/tty/lib_twait.c	2008-06-14 18:26:31.000000000 -0700
@@ -42,6 +42,10 @@
 
 #include <curses.priv.h>
 
+#if defined __HAIKU__ && defined __BEOS__
+#undef __BEOS__
+#endif
+
 #ifdef __BEOS__
 #undef false
 #undef true
diff -ur ncurses-5.6/ncurses/tty/tty_update.c ncurses-5.6-haiku/ncurses/tty/tty_update.c
--- ncurses-5.6/ncurses/tty/tty_update.c	2006-11-25 14:33:21.000000000 -0800
+++ ncurses-5.6-haiku/ncurses/tty/tty_update.c	2008-06-14 18:27:03.000000000 -0700
@@ -44,6 +44,10 @@
 
 #include <curses.priv.h>
 
+#if defined __HAIKU__ && defined __BEOS__
+#undef __BEOS__
+#endif
+
 #ifdef __BEOS__
 #undef false
 #undef true
diff -ur ncurses-5.6/tack/sysdep.c ncurses-5.6-haiku/tack/sysdep.c
--- ncurses-5.6/tack/sysdep.c	2005-09-17 12:49:16.000000000 -0700
+++ ncurses-5.6-haiku/tack/sysdep.c	2008-06-14 18:26:03.000000000 -0700
@@ -34,6 +34,10 @@
 #include <term.h>
 #include <errno.h>
 
+#if defined __HAIKU__ && defined __BEOS__
+#undef __BEOS__
+#endif
+
 #if defined(__BEOS__)
 #undef false
 #undef true
