diff -Naur backup/BackupView.cpp backup-fix/BackupView.cpp
--- backup/BackupView.cpp	2013-11-18 21:15:38.949223424 +0000
+++ backup-fix/BackupView.cpp	2013-11-18 21:15:33.142344192 +0000
@@ -120,7 +120,7 @@
 
 	// Refresh System Directory
 	BPath sysSettingsDirectory;
-	find_directory(B_COMMON_SETTINGS_DIRECTORY, &sysSettingsDirectory);
+	find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &sysSettingsDirectory);
 	fSysSettingBytes = DirectorySize(&sysSettingsDirectory);
 	size_to_string(fSysSettingBytes, sizeText, 512);
 	fSysSettingSizeText->SetText(sizeText);
