--- OpenTTD-15.3/src/ini.cpp.old	2026-08-19 23:07:18.403177472 +0200
+++ OpenTTD-15.3/src/ini.cpp	2026-08-19 23:07:35.937426944 +0200
@@ -17,10 +17,8 @@
 #	include <emscripten.h>
 #endif
 
-#if (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 199309L) || (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 500)
-# include <unistd.h>
-# include <fcntl.h>
-#endif
+#include <unistd.h>
+#include <fcntl.h>
 
 #include <filesystem>
 
