--- gzip-1.4/inflate.c       2010-01-20 14:15:12.017039360 +0100
+++ gzip-1.4/inflate.c 2011-12-03 00:03:37.421003264 +0100
@@ -588,6 +588,7 @@
       /* do the copy */
       do {
         n -= (e = (e = WSIZE - ((d &= WSIZE-1) > w ? d : w)) > n ? n : e);
+        {
 #if !defined(NOMEMCPY) && !defined(DEBUG)
         unsigned int delta = w > d ? w - d : d - w;
         if (delta >= e)
@@ -602,6 +603,7 @@
             slide[w++] = slide[d++];
            Tracevv((stderr, "%c", slide[w-1]));
           } while (--e);
+        }
         if (w == WSIZE)
         {
           flush_output(w);
