diff -Naur gzip-1.2.4a/Makefile.in gzip-1.2.4a-beos/Makefile.in
--- gzip-1.2.4a/Makefile.in	Tue Aug 17 19:12:40 1993
+++ gzip-1.2.4a-beos/Makefile.in	Thu Jan  3 00:19:57 2008
@@ -102,7 +102,7 @@
 # Where to install the manual pages.
 mandir = $(prefix)/man/man$(manext)
 # Use manlinks=so to use the .so method instead of hard links
-manlinks = ln
+manlinks = ln -s
 
 alldirs = $(bindir) $(scriptdir) $(datadir) $(libdir) $(infodir) $(mandir)
 
@@ -208,7 +208,7 @@
 	for f in $(SCRIPTS); do \
 	  $(INSTALL_PROGRAM) $${f} $(scriptdir)/$${f}; done
 	rm -f $(scriptdir)/$(G)zcmp; \
-	  ln $(scriptdir)/$(G)zdiff $(scriptdir)/$(G)zcmp
+	  ln -s $(scriptdir)/$(G)zdiff $(scriptdir)/$(G)zcmp
 	for f in gunzip$X ungzip$X $(ZCAT)$X ; do \
 	  rm -f $(bindir)/$${f}; done
 	@if echo $(DEFS) | grep GNU_STANDARD > /dev/null; then \
@@ -216,8 +216,8 @@
 	  echo 'exec gzip -dc $${1+"$$@"}' > $(bindir)/$(ZCAT)$X; \
 	  chmod 755 $(bindir)/gunzip$X  $(bindir)/$(ZCAT)$X; \
 	else \
-	  ln $(bindir)/gzip$X $(bindir)/gunzip$X; \
-	  ln $(bindir)/gzip$X $(bindir)/$(ZCAT)$X; \
+	  ln -s $(bindir)/gzip$X $(bindir)/gunzip$X; \
+	  ln -s $(bindir)/gzip$X $(bindir)/$(ZCAT)$X; \
 	fi
 
 installman: gzip.info
@@ -233,9 +233,9 @@
 	  echo .so man$(manext)/gzip.$(manext)      > gunzip.$(manext);\
 	  chmod 644 $(ZCAT).$(manext) $(G)zcmp.$(manext) gunzip.$(manext);\
 	else \
-	  ln gzip.$(manext)  $(ZCAT).$(manext);\
-	  ln $(G)zdiff.$(manext) $(G)zcmp.$(manext);\
-	  ln gzip.$(manext)  gunzip.$(manext);\
+	  ln -s gzip.$(manext)  $(ZCAT).$(manext);\
+	  ln -s $(G)zdiff.$(manext) $(G)zcmp.$(manext);\
+	  ln -s gzip.$(manext)  gunzip.$(manext);\
 	fi
 	-cd $(srcdir); for f in gzip.info* ; do $(INSTALL_DATA) $${f} \
 	  $(infodir)/$${f}; done
@@ -251,9 +251,9 @@
 install_compress: install
 	-test -f $(bindir)/compress.old || \
 	  mv $(bindir)/compress$X $(bindir)/compress.old
-	ln $(bindir)/gzip$X $(bindir)/compress$X
+	ln -s $(bindir)/gzip$X $(bindir)/compress$X
 	rm -f $(bindir)/uncompress$X
-	ln $(bindir)/gzip$X $(bindir)/uncompress$X
+	ln -s $(bindir)/gzip$X $(bindir)/uncompress$X
 
 # Make sure all installation directories, e.g. $(bindir) actually exist by
 # making them if necessary. At most one level is created (except for man).
@@ -329,8 +329,8 @@
 gzip$X:	Makefile $(OBJS)
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS)
 	rm -f gunzip$X $(ZCAT)$X
-	ln gzip$X gunzip$X
-	ln gzip$X $(ZCAT)$X
+	ln -s gzip$X gunzip$X
+	ln -s gzip$X $(ZCAT)$X
 
 gzip$O zip$O deflate$O trees$O bits$O unzip$O inflate$O: gzip.h tailor.h
 util$O lzw$O unlzw$O unpack$O unlzh$O crypt$O: gzip.h tailor.h
