diff -up global-5.9.1/configure.ac.orig global-5.9.1/configure.ac
--- global-5.9.1/configure.ac.orig	2010-07-18 23:28:56.009175040 -0600
+++ global-5.9.1/configure.ac	2010-07-18 23:59:03.300941312 -0600
@@ -57,7 +57,7 @@ LTDL_INIT([recursive])
 dnl Checks for libraries.
 
 dnl Checks for header files.
-AC_CHECK_HEADERS(limits.h string.h unistd.h stdarg.h sys/time.h fcntl.h)
+AC_CHECK_HEADERS(limits.h string.h unistd.h stdarg.h stdint.h sys/time.h fcntl.h)
 AC_CHECK_HEADERS(sys/resource.h)
 AC_HEADER_DIRENT
 if test ${ac_header_dirent} = no; then
diff -up global-5.9.1/libdb/db.h.orig global-5.9.1/libdb/db.h
--- global-5.9.1/libdb/db.h.orig	2010-07-18 23:28:56.033292288 -0600
+++ global-5.9.1/libdb/db.h	2010-07-18 23:59:03.303824896 -0600
@@ -40,6 +40,9 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
 
 #include "compat.h"
 
