diff -up global-6.2/configure.ac.orig global-6.2/configure.ac
--- global-6.2/configure.ac.orig	2012-01-20 22:15:13.034340864 -0700
+++ global-6.2/configure.ac	2012-02-20 23:31:39.196870144 -0700
@@ -58,7 +58,7 @@ LTDL_INIT([recursive])
 dnl Checks for libraries.
 
 dnl Checks for header files.
-AC_CHECK_HEADERS(limits.h string.h unistd.h stdarg.h sys/time.h fcntl.h)
+AC_CHECK_HEADERS(limits.h string.h unistd.h stdarg.h stdint.h sys/time.h fcntl.h)
 AC_CHECK_HEADERS(sys/resource.h)
 AC_HEADER_DIRENT
 if test ${ac_header_dirent} = no; then
diff -up global-6.2/libdb/db.h.orig global-6.2/libdb/db.h
--- global-6.2/libdb/db.h.orig	2012-01-20 22:15:13.058982400 -0700
+++ global-6.2/libdb/db.h	2012-02-20 23:31:39.197918720 -0700
@@ -40,6 +40,9 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
 
 #include "compat.h"
 
diff -up global-6.2/m4/check_curses.m4.orig global-6.2/m4/check_curses.m4
--- global-6.2/m4/check_curses.m4.orig	2012-02-20 23:35:06.375128064 -0700
+++ global-6.2/m4/check_curses.m4	2012-02-20 23:36:47.207618048 -0700
@@ -240,6 +240,9 @@ AC_DEFUN([AC_SEARCH_NCURSES], [
 
     AC_NCURSES(/usr/include/ncurses, curses.h, -lncurses, -I/usr/include/ncurses -DRENAMED_NCURSES, "renamed ncurses on /usr/include/ncurses")
 
+    AC_NCURSES(/boot/develop/headers/3rdparty, ncurses.h, -lncurses, -I/boot/develop/headers/3rdparty, "ncurses on /boot/develop")
+    AC_NCURSES(/boot/common/include/ncurses, ncurses.h, -lncurses, -I/boot/common/include/ncurses, "ncurses on /boot/common")
+
     dnl
     dnl We couldn't find ncurses, try SysV curses
     dnl
