diff -up global-5.8.2/configure.ac.orig global-5.8.2/configure.ac
--- global-5.8.2/configure.ac.orig	2010-05-23 01:40:56.774635520 -0600
+++ global-5.8.2/configure.ac	2010-05-23 01:41:23.901775360 -0600
@@ -57,7 +57,7 @@ LTDL_INIT([recursive])
 dnl Checks for libraries.
 
 dnl Checks for header files.
-AC_CHECK_HEADERS(limits.h string.h unistd.h stdarg.h sys/time.h fcntl.h)
+AC_CHECK_HEADERS(limits.h string.h unistd.h stdarg.h stdint.h sys/time.h fcntl.h)
 AC_CHECK_HEADERS(sys/resource.h)
 AC_HEADER_DIRENT
 if test ${ac_header_dirent} = no; then
diff -up global-5.8.2/libdb/db.h.orig global-5.8.2/libdb/db.h
--- global-5.8.2/libdb/db.h.orig	2010-05-23 01:42:09.494665728 -0600
+++ global-5.8.2/libdb/db.h	2010-05-23 01:42:55.554434560 -0600
@@ -40,6 +40,9 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
 
 #include "compat.h"
 
