diff --git a/Source/BesolWindow.h b/Source/BesolWindow.h
index d703c1c..ad648f8 100644
--- a/Source/BesolWindow.h
+++ b/Source/BesolWindow.h
@@ -27,6 +27,8 @@
 #include <View.h>
 #include <Beep.h>
 #include <SupportKit.h>
+#include <File.h>
+#include <Roster.h>
 
 #ifndef IEWINDOW
 #define IEWINDOW
diff --git a/Source/PILE.c++ b/Source/PILE.c++
index e826b76..3c0f259 100644
--- a/Source/PILE.c++
+++ b/Source/PILE.c++
@@ -238,7 +238,7 @@ void PILE::construction_helper(pile_props* props)
 PILE::~PILE()
 {
 	BMenuItem* item = popmenu->FindItem(POP_EDIT);
-	if ( item->IsMarked() ) peditor->Remove_Target(cast_as(this, BView));
+	if ( item->IsMarked() ) peditor->Remove_Target((BView*)this);
 
 	list< list<TRANSRULES*> >::iterator itert = allrules->begin();
 	list<TRANSRULES*>::iterator iterf;
@@ -685,7 +685,7 @@ void PILE::MessageReceived(BMessage* message)
 		{
 			BMenuItem* item = popmenu->FindItem(POP_EDIT);
 			item->SetMarked( !(item->IsMarked()) );
-			BView* temp = cast_as (this, BView);
+			BView* temp = (BView*)this;
 			isbeingedited = !isbeingedited;
 			Invalidate(Bounds());
 			
