diff --git a/lisp/files.el b/lisp/files.el
index 0d24852358e..8f8af2c492f 100644
--- a/lisp/files.el
+++ b/lisp/files.el
@@ -6357,6 +6357,15 @@ file-equal-p
       (let (f1-attr f2-attr)
         (and (setq f1-attr (file-attributes (file-truename file1)))
 	     (setq f2-attr (file-attributes (file-truename file2)))
+             ;; Haiku systems change the file's last access timestamp
+             ;; every time `stat' is called.  Make sure to not compare
+             ;; the timestamps in that case.
+
+             (when (eq system-type 'haiku)
+               (ignore-errors
+                 (setcar (nthcdr 4 f1-attr) nil)
+                 (setcar (nthcdr 4 f2-attr) nil))
+               t)
 	     (equal f1-attr f2-attr))))))
 
 (defun file-in-directory-p (file dir)
