diff -urN cpio-2.11/configure.ac cpio-2.11-haiku/configure.ac
--- cpio-2.11/configure.ac	2010-03-10 04:59:16.008126464 -0800
+++ cpio-2.11-haiku/configure.ac	2013-03-12 15:18:23.338165760 -0700
@@ -47,6 +47,8 @@
 AC_HEADER_STDC
 AC_HEADER_DIRENT
 
+AC_SEARCH_LIBS(gethostbyname, [socket, network])
+
 AC_CHECK_FUNCS([fchmod fchown])
 # This is needed for mingw build
 AC_CHECK_FUNCS([setmode getpwuid getpwnam getgrgid getgrnam pipe fork getuid geteuid])
diff -urN cpio-2.11/src/makepath.c cpio-2.11-haiku/src/makepath.c
--- cpio-2.11/src/makepath.c	2010-02-12 02:19:23.043778048 -0800
+++ cpio-2.11-haiku/src/makepath.c	2013-03-12 15:18:30.096993280 -0700
@@ -56,6 +56,7 @@
   char *dirpath;		/* A copy we can scribble NULs on.  */
   struct stat stats;
   int retval = 0;
+  char *slash;
   mode_t tmpmode;
   mode_t invert_permissions;
   int we_are_root = getuid () == 0;
@@ -68,7 +69,7 @@
       tmpmode = MODE_RWX & ~ newdir_umask;
       invert_permissions = we_are_root ? 0 : MODE_WXUSR & ~ tmpmode;
 
-      char *slash = dirpath;
+      slash = dirpath;
       while (*slash == '/')
 	slash++;
       while ((slash = strchr (slash, '/')))
