diff -x .git -Naur ../globe0.4/Makefile ./Makefile
--- ../globe0.4/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ ./Makefile	2014-12-02 19:15:53.759918816 +0100
@@ -0,0 +1,27 @@
+NAME= Globe
+TYPE= APP
+SRCS= RHTML_app.cpp RHTML_win.cpp RHTML_numview.cpp RHTML_savewin.cpp RHTML_aboutwin.cpp RHTML_prefwin.cpp RHTML_fmenu.cpp RHTML_buttonmenu.cpp RHTML_bmenu.cpp RHTML_parammenu.cpp RHTML_colorwin.cpp RHTML_amenu.cpp RHTML_textview.cpp RHTML_tabbutton.cpp RHTML_tagmenu.cpp RHTML_filetypes.cpp RHTML_options.cpp ToolTip/TToolTip.cpp Timer/RTimer.cpp URLView/URLView.cpp Tool/RadioBar.cpp Tool/ToolBar.cpp Tool/ToolItem.cpp Tool/ToolView.cpp
+RSRCS= Globe.rsrc
+LIBS= /boot/system/lib/libbe.so /boot/system/lib/libtextencoding.so /boot/system/lib/libtracker.so /boot/system/lib/libroot.so /boot/system/lib/libtranslation.so /boot/system/lib/libgame.so /boot/system/lib/libstdc++.so
+LIBPATHS=
+SYSTEM_INCLUDE_PATHS= /boot/system/develop/headers/be /boot/system/develop/headers/cpp /boot/system/develop/headers/posix /boot/system/develop/headers/os/storage /boot/system/develop/lib /boot/system/beos/system/lib
+LOCAL_INCLUDE_PATHS= ./../../../config/lib ./Genesis ToolTip Timer URLView Tool
+OPTIMIZE=FULL
+#	specify any preprocessor symbols to be defined.  The symbols will not
+#	have their values set automatically; you must supply the value (if any)
+#	to use.  For example, setting DEFINES to "DEBUG=1" will cause the
+#	compiler option "-DDEBUG=1" to be used.  Setting DEFINES to "DEBUG"
+#	would pass "-DDEBUG" on the compiler's command line.
+DEFINES=
+#	specify special warning levels
+#	if unspecified default warnings will be used
+#	NONE = supress all warnings
+#	ALL = enable all warnings
+WARNINGS =
+# Build with debugging symbols if set to TRUE
+SYMBOLS=
+COMPILER_FLAGS=-Woverloaded-virtual -funsigned-bitfields -D_HAIKU_
+LINKER_FLAGS=
+
+## include the makefile-engine
+include /boot/system/develop/etc/makefile-engine
diff -x .git -Naur ../globe0.4/RHTML_filetypes.cpp ./RHTML_filetypes.cpp
--- ../globe0.4/RHTML_filetypes.cpp	2004-10-21 16:18:44.000000000 +0200
+++ ./RHTML_filetypes.cpp	2014-12-02 18:56:43.069920922 +0100
@@ -39,7 +39,7 @@
 // - End - RHTML_File_Types - RHTMLFileTypes -----------------------------------------------------------------
 
 // ---------------------------------------------------------------------- RHTML_File_Types - SetFontAndColor -
-void RHTMLFileTypes::SetFontAndColor(BString *oldtext,BString *currenttext,int mode=0,int offset1=-1,int offset2=-1)
+void RHTMLFileTypes::SetFontAndColor(BString *oldtext,BString *currenttext,int mode,int offset1,int offset2)
 {
  if (fOptions->FileTypes[FileType]->Name->ICompare("HTML")==0)
  {
diff -x .git -Naur ../globe0.4/RHTML_options.cpp ./RHTML_options.cpp
--- ../globe0.4/RHTML_options.cpp	2004-10-21 16:18:57.000000000 +0200
+++ ./RHTML_options.cpp	2014-12-02 19:07:34.499919730 +0100
@@ -34,6 +34,8 @@
 #include <Directory.h>
 #include <fstream>
 
+using std::ifstream;
+
 RHTMLOptions *fOptions;
 
 // ---------------------------------------------------------------------------- RHTML_Options - RHTMLOptions -
@@ -554,7 +556,7 @@
      tmp.CopyInto(val1,tmp.FindFirst('=')+1,tmp.Length()-tmp.FindFirst('=')-1);
      val1=rgetstring(val1);
      font_style fstyle;
-     get_font_style((font_family) val1.String(), 0, &fstyle);
+     get_font_style((char *) val1.String(), 0, &fstyle);
      DefaultFont->SetFamilyAndStyle(val1.String(),fstyle);
     } else
     if (name.ICompare("Default_Font_Style")==0)	// - Default_Font_Style
diff -x .git -Naur ../globe0.4/RHTML_options.h ./RHTML_options.h
--- ../globe0.4/RHTML_options.h	2004-10-21 16:19:01.000000000 +0200
+++ ./RHTML_options.h	2014-12-02 18:31:28.159923695 +0100
@@ -33,7 +33,7 @@
 #include <String.h>
 #include <View.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 
 class RHTMLOptions;
 
diff -x .git -Naur ../globe0.4/RHTML_parammenu.cpp ./RHTML_parammenu.cpp
--- ../globe0.4/RHTML_parammenu.cpp	2004-10-21 16:17:48.000000000 +0200
+++ ./RHTML_parammenu.cpp	2014-12-02 18:53:29.809921276 +0100
@@ -35,7 +35,7 @@
 #include <Clipboard.h>
 #include <StringView.h>
 #include <stdio.h>
-#include <ostream.h>
+#include <ostream>
 #include <math.h>
 
 // ------------------------------------------------------------------------ RHTML_parammenu - RHTMLparammenu -
@@ -602,7 +602,7 @@
 void RHTMLptextview::Paste(BClipboard *clipboard)
 {
 	const char *text;
-	int32 textlen;
+	ssize_t textlen;
 	BMessage *clip = (BMessage *) NULL;
 
 	if (clipboard->Lock())
diff -x .git -Naur ../globe0.4/RHTML_prefwin.cpp ./RHTML_prefwin.cpp
--- ../globe0.4/RHTML_prefwin.cpp	2004-10-21 16:17:14.000000000 +0200
+++ ./RHTML_prefwin.cpp	2014-12-02 18:49:57.739921664 +0100
@@ -386,7 +386,7 @@
 // - End - RHTML_Preferences_Win - FTCreateKitList -----------------------------------------------------------
 
 // ----------------------------------------------------------------------- RHTML_Preferences_Win - FTSetFont -
-void RHTMLPreferencesWin::FTSetFont(const char *font,const char *style ="")
+void RHTMLPreferencesWin::FTSetFont(const char *font,const char *style)
 {
  fFTFontsBox->RemoveChild(fFTDFStyle);
 
@@ -649,7 +649,7 @@
     msg->FindString("font_family",&family);
     FTSetFont(family);
     font_style fStyle;
-    get_font_style((font_family) family, 0, &fStyle);
+    get_font_style((char *) family, 0, &fStyle);
     fOptions->DefaultFont->SetFamilyAndStyle(family,fStyle);
     fOptions->Save();
    }
diff -x .git -Naur ../globe0.4/RHTML_win.cpp ./RHTML_win.cpp
--- ../globe0.4/RHTML_win.cpp	2004-10-21 16:16:28.000000000 +0200
+++ ./RHTML_win.cpp	2014-12-02 18:30:50.089923764 +0100
@@ -288,7 +288,9 @@
  if (textdb>0)
  {
   close->SetEnabled(true);
+  #ifndef _HAIKU_
   fmenu->FindToolItem("CLOSEBUTTON")->SetEnabled(true);
+  #endif
  }
 }
 // - End - RHTML_Win - AddText -------------------------------------------------------------------------------
@@ -310,7 +312,9 @@
 	if (textdb<=0)
 	{
 		close->SetEnabled(false);
+		#ifndef _HAIKU_
 		fmenu->FindToolItem("CLOSEBUTTON")->SetEnabled(false);
+		#endif
 	}
 }
 // - End - RHTML_Win - RemoveText ----------------------------------------------------------------------------
@@ -1122,7 +1126,9 @@
   break;
   case 'RTMM':
   {
+  	#ifndef _HAIKU_
    fmenu->FindToolItem("SAVEBUTTON")->SetEnabled(textview[current]->Modify());
+    #endif
    save->SetEnabled(textview[current]->Modify());
 //   save_as->SetEnabled(textview[current]->Modify());
    bool mod=false;
@@ -1130,7 +1136,9 @@
     if (textview[i]->Modify())
      mod=true;
    save_all->SetEnabled(mod);
+   #ifndef _HAIKU_
    fmenu->FindToolItem("SAVEALLBUTTON")->SetEnabled(mod);
+   #endif
   }
   break;
   case 'ABWQ':
diff -x .git -Naur ../globe0.4/Timer/RTimer.cpp ./Timer/RTimer.cpp
--- ../globe0.4/Timer/RTimer.cpp	2004-10-21 16:22:13.000000000 +0200
+++ ./Timer/RTimer.cpp	2014-12-02 19:09:02.649919568 +0100
@@ -28,7 +28,7 @@
 // - Includes
 #include "RTimer.h"
 #include <Autolock.h>
-#include <ostream.h>
+#include <ostream>
 
 #define SLEEP_TIME 1000
 // ---------------------------------------------------------------------------------------- R_Timer - RTimer -
diff -x .git -Naur ../globe0.4/Tool/ToolItem.cpp ./Tool/ToolItem.cpp
--- ../globe0.4/Tool/ToolItem.cpp	2004-10-21 16:23:12.000000000 +0200
+++ ./Tool/ToolItem.cpp	2014-12-02 19:13:26.509919086 +0100
@@ -31,10 +31,13 @@
 #include <TranslationUtils.h>
 #include <Window.h>
 #include <stdio.h>
-#include <ostream.h>
+#include <iostream>
+#include <ostream>
 
 #include "ToolItem.h"
 
+using namespace std;
+
 const uint32 B_UPDATE_SIZE_MSG = 'BUSM';
 const uint32 B_MOUSE_DOWN_MSG = 'BMDM';
 const uint32 B_MOUSE_UP_MSG = 'BMUM';
diff -x .git -Naur ../globe0.4/URLView/URLView.cpp ./URLView/URLView.cpp
--- ../globe0.4/URLView/URLView.cpp	2002-03-10 01:38:20.000000000 +0100
+++ ./URLView/URLView.cpp	2014-12-02 18:30:50.089923764 +0100
@@ -30,9 +30,13 @@
 
 #include <unistd.h>
 
+#ifndef _HAIKU_
 #include <printf.h>
+#endif
 
-
+#ifdef _HAIKU_
+#include<Clipboard.h>
+#endif
 
 URLView::URLView( BRect frame, const char *name, const char *label,
 				  const char *url, uint32 resizingMode, uint32 flags )
