diff -Naur globe0.4/Makefile globe0.4-haiku/Makefile
--- globe0.4/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ globe0.4-haiku/Makefile	2012-12-30 19:40:59.058720256 +0000
@@ -0,0 +1,27 @@
+NAME= Globe
+TYPE= APP
+SRCS= RHTML_app.cpp RHTML_win.cpp RHTML_numview.cpp RHTML_savewin.cpp RHTML_aboutwin.cpp RHTML_prefwin.cpp RHTML_fmenu.cpp RHTML_buttonmenu.cpp RHTML_bmenu.cpp RHTML_parammenu.cpp RHTML_colorwin.cpp RHTML_amenu.cpp RHTML_textview.cpp RHTML_tabbutton.cpp RHTML_tagmenu.cpp RHTML_filetypes.cpp RHTML_options.cpp ToolTip/TToolTip.cpp Timer/RTimer.cpp URLView/URLView.cpp Tool/RadioBar.cpp Tool/ToolBar.cpp Tool/ToolItem.cpp Tool/ToolView.cpp
+RSRCS= Globe.rsrc
+LIBS= /boot/system/lib/libbe.so /boot/system/lib/libtextencoding.so /boot/system/lib/libtracker.so /boot/system/lib/libroot.so /boot/system/lib/libtranslation.so /boot/system/lib/libgame.so /boot/system/lib/libstdc++.r4.so
+LIBPATHS=
+SYSTEM_INCLUDE_PATHS= /boot/develop/headers/be /boot/develop/headers/cpp /boot/develop/headers/posix /boot/develop/lib /boot/beos/system/lib
+LOCAL_INCLUDE_PATHS= ./../../../config/lib ./Genesis ToolTip Timer URLView Tool
+OPTIMIZE=FULL
+#	specify any preprocessor symbols to be defined.  The symbols will not
+#	have their values set automatically; you must supply the value (if any)
+#	to use.  For example, setting DEFINES to "DEBUG=1" will cause the
+#	compiler option "-DDEBUG=1" to be used.  Setting DEFINES to "DEBUG"
+#	would pass "-DDEBUG" on the compiler's command line.
+DEFINES=
+#	specify special warning levels
+#	if unspecified default warnings will be used
+#	NONE = supress all warnings
+#	ALL = enable all warnings
+WARNINGS =
+# Build with debugging symbols if set to TRUE
+SYMBOLS=
+COMPILER_FLAGS=-Woverloaded-virtual -funsigned-bitfields -D_HAIKU_
+LINKER_FLAGS=
+
+## include the makefile-engine
+include $(BUILDHOME)/etc/makefile-engine
diff -Naur globe0.4/RHTML_win.cpp globe0.4-haiku/RHTML_win.cpp
--- globe0.4/RHTML_win.cpp	2004-10-21 14:16:28.000000000 +0000
+++ globe0.4-haiku/RHTML_win.cpp	2012-12-30 18:28:09.000000000 +0000
@@ -288,7 +288,9 @@
  if (textdb>0)
  {
   close->SetEnabled(true);
+  #ifndef _HAIKU_
   fmenu->FindToolItem("CLOSEBUTTON")->SetEnabled(true);
+  #endif
  }
 }
 // - End - RHTML_Win - AddText -------------------------------------------------------------------------------
@@ -310,7 +312,9 @@
 	if (textdb<=0)
 	{
 		close->SetEnabled(false);
+		#ifndef _HAIKU_
 		fmenu->FindToolItem("CLOSEBUTTON")->SetEnabled(false);
+		#endif
 	}
 }
 // - End - RHTML_Win - RemoveText ----------------------------------------------------------------------------
@@ -1122,7 +1126,9 @@
   break;
   case 'RTMM':
   {
+  	#ifndef _HAIKU_
    fmenu->FindToolItem("SAVEBUTTON")->SetEnabled(textview[current]->Modify());
+    #endif
    save->SetEnabled(textview[current]->Modify());
 //   save_as->SetEnabled(textview[current]->Modify());
    bool mod=false;
@@ -1130,7 +1136,9 @@
     if (textview[i]->Modify())
      mod=true;
    save_all->SetEnabled(mod);
+   #ifndef _HAIKU_
    fmenu->FindToolItem("SAVEALLBUTTON")->SetEnabled(mod);
+   #endif
   }
   break;
   case 'ABWQ':
diff -Naur globe0.4/URLView/URLView.cpp globe0.4-haiku/URLView/URLView.cpp
--- globe0.4/URLView/URLView.cpp	2002-03-10 00:38:20.000000000 +0000
+++ globe0.4-haiku/URLView/URLView.cpp	2012-12-29 14:17:53.000000000 +0000
@@ -30,9 +30,13 @@
 
 #include <unistd.h>
 
+#ifndef _HAIKU_
 #include <printf.h>
+#endif
 
-
+#ifdef _HAIKU_
+#include<Clipboard.h>
+#endif
 
 URLView::URLView( BRect frame, const char *name, const char *label,
 				  const char *url, uint32 resizingMode, uint32 flags )
