--- SDL_sound.c	2008-08-30 21:13:28.000000000 -0700
+++ SDL_sound-haiku.c	2008-08-30 22:41:20.000000000 -0700
@@ -928,8 +928,9 @@
 
 Sint32 Sound_GetDuration(Sound_Sample *sample)
 {
+    Sound_SampleInternal *internal;
     BAIL_IF_MACRO(!initialized, ERR_NOT_INITIALIZED, -1);
-    Sound_SampleInternal *internal = (Sound_SampleInternal *) sample->opaque;
+    internal = (Sound_SampleInternal *) sample->opaque;
     return(internal->total_time);
 } /* Sound_GetDuration */
 
