diff --git a/mlk/src/gui/x11/mlk_x11_cursor.c b/mlk/src/gui/x11/mlk_x11_cursor.c
index 27e3308..c2c8d5d 100644
--- a/mlk/src/gui/x11/mlk_x11_cursor.c
+++ b/mlk/src/gui/x11/mlk_x11_cursor.c
@@ -31,8 +31,10 @@ $*/
 
 void __mX11CursorFree(mCursor cur)
 {
+#ifndef __HAIKU__
 	if(cur)
 		XFreeCursor(MLKX11_DISPLAY, (Cursor)cur);
+#endif
 }
 
 /** デフォルトテーマから指定名のカーソル読み込み
@@ -50,6 +52,7 @@ mCursor __mX11CursorLoad(const char *name)
 
 mCursor __mX11CursorCreate1bit(const uint8_t *buf)
 {
+#ifndef __HAIKU__
 	Display *disp = MLKX11_DISPLAY;
 	int w,h;
 	Pixmap img,mask;
@@ -88,6 +91,9 @@ mCursor __mX11CursorCreate1bit(const uint8_t *buf)
 	if(mask) XFreePixmap(disp, mask);
 
 	return cur;
+#else
+	return 0;
+#endif
 }
 
 /** RGBA イメージからカーソル作成 */
