diff --git a/src/frontend/qt_sdl/CMakeLists.txt b/src/frontend/qt_sdl/CMakeLists.txt
index 6b964f3..ca6b73a 100644
--- a/src/frontend/qt_sdl/CMakeLists.txt
+++ b/src/frontend/qt_sdl/CMakeLists.txt
@@ -111,20 +111,28 @@ elseif (APPLE)
         ../duckstation/gl/context_agl.mm
     )
 else()
-    # we only need ECM for Wayland
-    # so we only require it from here
-    find_package(ECM REQUIRED NO_MODULE)
-    list(APPEND CMAKE_MODULE_PATH "${ECM_MODULE_PATH}")
-
     find_package(X11 REQUIRED)
     find_package(EGL REQUIRED)
-    find_package(Wayland REQUIRED Client)
+
+    if (ENABLE_WAYLAND)
+        # we only need ECM for Wayland
+        # so we only require it from here
+        find_package(ECM REQUIRED NO_MODULE)
+        list(APPEND CMAKE_MODULE_PATH "${ECM_MODULE_PATH}")
+
+        find_package(Wayland REQUIRED Client)
+
+        target_compile_definitions(melonDS PRIVATE WAYLAND_ENABLED)
+
+        target_sources(melonDS PRIVATE
+            ../duckstation/gl/context_egl_wayland.cpp
+        )
+    endif()
 
     target_sources(melonDS PRIVATE
         ../duckstation/gl/context_egl.cpp
         ../duckstation/gl/context_egl_x11.cpp
         ../duckstation/gl/context_glx.cpp
-        ../duckstation/gl/context_egl_wayland.cpp
         ../duckstation/gl/x11_window.cpp
 
         ../glad/glad_egl.c
diff --git a/src/frontend/duckstation/gl/context.cpp b/src/frontend/duckstation/gl/context.cpp
index 98e6bd1..5674e91 100644
--- a/src/frontend/duckstation/gl/context.cpp
+++ b/src/frontend/duckstation/gl/context.cpp
@@ -15,7 +15,9 @@ Log_SetChannel(GL::Context);
 #elif defined(__APPLE__)
 #include "context_agl.h"
 #else
+#ifdef WAYLAND_ENABLED
 #include "context_egl_wayland.h"
+#endif
 #include "context_egl_x11.h"
 #include "context_glx.h"
 #endif
@@ -79,9 +81,10 @@ std::unique_ptr<GL::Context> Context::Create(const WindowInfo& wi, const Version
     else
       context = ContextGLX::Create(wi, versions_to_try, num_versions_to_try);
   }
-
+#ifdef WAYLAND_ENABLED
   if (wi.type == WindowInfo::Type::Wayland)
     context = ContextEGLWayland::Create(wi, versions_to_try, num_versions_to_try);
+#endif
 #endif
 
   if (!context)
