diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0a49406..2b2800e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -129,8 +129,10 @@ endif()
 
 if (WIN32)
     target_link_libraries(core PRIVATE ole32 comctl32 ws2_32)
-elseif(NOT APPLE)
+elseif(NOT APPLE AND NOT HAIKU)
     target_link_libraries(core PRIVATE rt)
+elseif(HAIKU)
+    target_link_libraries(core PRIVATE network)
 endif()
 
 if (ENABLE_JIT_PROFILING)
diff --git a/src/sha1/sha1.c b/src/sha1/sha1.c
index 3729550..85f745c 100644
--- a/src/sha1/sha1.c
+++ b/src/sha1/sha1.c
@@ -27,6 +27,9 @@ A million repetitions of "a"
 #if defined(__sun)
 #include "solarisfixes.h"
 #endif
+#if defined(__HAIKU__)
+#include <ByteOrder.h>
+#endif
 #include "sha1.h"
 
 #ifndef BYTE_ORDER
