--- ../main.c	2006-05-16 22:11:29.042991616 +0200
+++ main-new.c	2013-12-09 16:52:30.000000000 +0100
@@ -16,6 +16,10 @@
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA. */
 
 /* Includes */
+// chdir()
+#include <unistd.h>
+#include <libgen.h>
+
 #include <SDL/SDL_main.h>
 #include <SDL/SDL.h>
 #include <string.h>
@@ -102,6 +106,9 @@
 int main(int argc, char *argv[]) {
 	int count, count2;
 	
+	// For tracker (Thanks PulkoMandy)
+	chdir(dirname(argv[0]));
+	
 	/* Initialise the highscoreboard array because there may not yet
 	   be a saved highscore[board]/rc file */
 	for (count = 0; count < 5; count++) {
