--- moleinvasion-0.4/src/tool.c.org	2014-08-26 21:59:45.137625600 +0200
+++ moleinvasion-0.4/src/tool.c	2014-08-26 21:26:28.667680768 +0200
@@ -89,6 +89,7 @@
 #endif
 	{	mapfailed=1;
 		/* no nmap() ... do it myself */
+		{
 		unsigned char *ptr2,buff[256];
 		int lu;
 		ptr=malloc(statb.st_size);
@@ -98,6 +99,7 @@
 			ptr2+=lu;
 		}
 		memcpy(ptr2,buff,lu);
+		}
 	}
 
 	ret=update_crc(0xffffffffL, ptr, statb.st_size) ^ 0xffffffffL;
--- moleinvasion-0.4/src/Makefile.org	2014-08-29 21:07:37.381157376 +0200
+++ moleinvasion-0.4/src/Makefile	2014-08-29 21:04:24.757596160 +0200
@@ -2,7 +2,7 @@
 
 CFLAGS  = -g -Wall $$(sdl-config --cflags) -DHAVE_OPENGL #--coverage #-fprofile-use #-pg
 LDFLAGS = -O2 #--coverage #-fprofile-use #-pg
-LDLIBS  = $$(sdl-config --libs) -lSDL_image -lSDL_mixer -lSDL_ttf
+LDLIBS  = $$(sdl-config --libs) -lSDL_image -lSDL_mixer -lSDL_ttf -lsmpeg
 
 OBJS =  graphics.o font.o list.o sprite.o events.o editor_texts.o \
 	Splayer.o Sother.o background.o video.o tool.o \
@@ -46,7 +46,7 @@
 	mkdir -p $(FINALDATADIR)/lvl
 	mkdir -p $(FINALDATADIR)/txt
 	mkdir -p $(FINALDATADIR)/font
-	cp -rf ../gfx ../font ../lvl ../snd ../music ../txt $(FINALDATADIR)/
+	cp -rf ../gfx ../font ../lvl ../snd ../txt $(FINALDATADIR)/
 	find $(FINALDATADIR)/ -name "*.*g" -exec chmod -x {} \;
 	-find $(FINALDATADIR)/ -name ".xvpics" -exec rm -rf {} \;
 
