diff --git a/MainView.cpp b/MainView.cpp
index 098027d..dc122bf 100644
--- a/MainView.cpp
+++ b/MainView.cpp
@@ -18,8 +18,7 @@ MainView::MainView(BRect frame)
 	:
 	//BTextView("MainView")
 	BTextView(frame, "textview", frame,
-		B_FOLLOW_ALL, B_FRAME_EVENTS | B_WILL_DRAW),
-	fPythonPlugin(this, "klong")
+		B_FOLLOW_ALL, B_FRAME_EVENTS | B_WILL_DRAW)
 {
 	fHighlighter = new Highlighter(this);
 	SetStylable(true);
@@ -100,7 +99,6 @@ void
 MainView::DeleteText(int32 start, int32 finish)
 {
 	BTextView::DeleteText(start, finish);
-	fPythonPlugin.Func1(Text()/*text*/);
 }
 
 
@@ -113,6 +111,5 @@ MainView::InsertText(const char *text, int32 length, int32 offset,
 
 	BTextView::InsertText(text, length, offset, runs);
 	bigtime_t start = system_time();
-	fPythonPlugin.Func1(Text()/*text*/);
 	printf("script time: %lims\n", (system_time() - start) / 1000);
 }
diff --git a/MainView.h b/MainView.h
index f1c456f..e705a27 100644
--- a/MainView.h
+++ b/MainView.h
@@ -9,8 +9,6 @@
 
 #include <TextView.h>
 
-#include "PythonPlugin.h"
-
 #include <vector>
 
 
@@ -40,7 +38,6 @@ public:
 
 private:
 			Highlighter* 		fHighlighter;
-			PythonPlugin 		fPythonPlugin;
 };
 
 
diff --git a/makefile b/makefile
index e2f41a7..1ac7063 100644
--- a/makefile
+++ b/makefile
@@ -2,17 +2,16 @@ TYPE = APP
 
 NAME = Microbe
 
-SYSTEM_INCLUDE_PATHS = /boot/common/include/python2.6
+SYSTEM_INCLUDE_PATHS = /boot/system/develop/headers/python2.7
 
 SRCS = App.cpp
 SRCS += Highlighter.cpp
 SRCS += MainWindow.cpp
 SRCS += MainView.cpp
-SRCS += PythonPlugin.cpp
 SRCS += TextRange.cpp
-LIBS = be python2.6 tracker #stdc++
+LIBS = be tracker stdc++
 
 RDEFS = Microbe.rdef
 
 # include the makefile-engine
-include $(BUILDHOME)/etc/makefile-engine
+include /boot/system/develop/etc/makefile-engine
