diff -urN libevent-2.0.10-stable/configure.in libevent-2.0.10-stable-haiku/configure.in
--- libevent-2.0.10-stable/configure.in	2010-12-16 17:38:34.004194304 +0000
+++ libevent-2.0.10-stable-haiku/configure.in	2011-01-19 14:00:04.442236928 +0000
@@ -54,8 +54,8 @@
 AC_SUBST(LIBTOOL_DEPS)
 
 dnl Checks for libraries.
-AC_SEARCH_LIBS([inet_ntoa], [nsl])
-AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([inet_ntoa], [nsl network])
+AC_SEARCH_LIBS([socket], [socket network])
 AC_SEARCH_LIBS([inet_aton], [resolv])
 AC_SEARCH_LIBS([clock_gettime], [rt])
 AC_SEARCH_LIBS([sendfile], [sendfile])
