diff -Naur libevent-2.0.10-stable/configure.in libevent-2.0.10-stable-haiku/configure.in
--- libevent-2.0.10-stable/configure.in	2010-12-16 18:38:34.018350080 +0100
+++ libevent-2.0.10-stable-haiku/configure.in	2011-02-21 23:55:55.070254592 +0100
@@ -54,8 +54,8 @@
 AC_SUBST(LIBTOOL_DEPS)
 
 dnl Checks for libraries.
-AC_SEARCH_LIBS([inet_ntoa], [nsl])
-AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([inet_ntoa], [nsl network])
+AC_SEARCH_LIBS([socket], [socket network])
 AC_SEARCH_LIBS([inet_aton], [resolv])
 AC_SEARCH_LIBS([clock_gettime], [rt])
 AC_SEARCH_LIBS([sendfile], [sendfile])
diff -Naur libevent-2.0.10-stable/util-internal.h libevent-2.0.10-stable-haiku/util-internal.h
--- libevent-2.0.10-stable/util-internal.h	2010-12-14 20:07:56.021233664 +0100
+++ libevent-2.0.10-stable-haiku/util-internal.h	2011-02-22 00:00:14.000000000 +0100
@@ -173,7 +173,7 @@
 
 /* Evaluates to the same boolean value as 'p', and hints to the compiler that
  * we expect this value to be false. */
-#ifdef __GNUC__
+#if __GNUC__ >= 3
 #define EVUTIL_UNLIKELY(p) __builtin_expect(!!(p),0)
 #else
 #define EVUTIL_UNLIKELY(p) (p)