diff -Naur libevent-1.4.14b-stable/configure.in libevent-1.4.14b-stable-haiku/configure.in
--- libevent-1.4.14b-stable/configure.in	2009-05-24 13:37:49.000000000 +0000
+++ libevent-1.4.14b-stable-haiku/configure.in	2009-05-28 16:55:17.000000000 +0000
@@ -33,7 +33,7 @@
 AC_SUBST(LIBTOOL_DEPS)

 dnl Checks for libraries.
-AC_CHECK_LIB(socket, socket)
+AC_SEARCH_LIBS(socket, socket network)
 AC_CHECK_LIB(resolv, inet_aton)
 AC_CHECK_LIB(rt, clock_gettime)
 AC_CHECK_LIB(nsl, inet_ntoa)
