--- configure.in	2012-11-17 22:34:13.040894464 -0800
+++ haiku/configure.in	2013-08-28 15:07:32.000000000 -0700
@@ -104,8 +104,8 @@
 AM_CONDITIONAL([BUILD_REGRESS], [test "$enable_libevent_regress" = "yes"])
 
 dnl Checks for libraries.
-AC_SEARCH_LIBS([inet_ntoa], [nsl])
-AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([inet_ntoa], [nsl network])
+AC_SEARCH_LIBS([socket], [socket network])
 AC_SEARCH_LIBS([inet_aton], [resolv])
 AC_SEARCH_LIBS([clock_gettime], [rt])
 AC_SEARCH_LIBS([sendfile], [sendfile])
