diff -Naur xproto-7.0.16/Xos_r.h xproto-7.0.16-haiku/Xos_r.h
--- xproto-7.0.16/Xos_r.h	2009-09-23 23:51:43.007602176 +0000
+++ xproto-7.0.16-haiku/Xos_r.h	2010-04-26 16:25:28.920649728 +0000
@@ -88,6 +88,10 @@
 # endif
 #endif /* _XOS_R_H */
 
+#ifdef __HAIKU__
+#define _POSIX_THREAD_SAFE_FUNCTIONS
+#endif
+
 #ifndef WIN32
 
 #ifdef __cplusplus
diff -Naur xproto-7.0.16/configure.ac xproto-7.0.16-haiku/configure.ac
--- xproto-7.0.16/configure.ac	2009-09-24 00:03:29.005505024 +0000
+++ xproto-7.0.16-haiku/configure.ac	2010-04-26 16:25:11.543686656 +0000
@@ -90,6 +90,28 @@
 fi
 
 if test "x$fds_bits_found" = xfalse ; then
+AC_CHECK_MEMBER(fd_set.bits,
+			[
+			    fds_bits_found=plain_simple
+			    USE_FDS_BITS="bits"
+			],,
+			[
+                               #ifdef HAVE_SYS_PARAM_H
+                               #include <sys/param.h>
+                               #endif
+                               #ifdef HAVE_SYS_TYPES_H
+                               #include <sys/types.h>
+                               #endif
+                               #ifdef HAVE_SYS_TIME_H
+                               #include <sys/time.h>
+                               #endif
+                               #ifdef HAVE_SYS_SELECT_H
+			       #include <sys/select.h>
+                               #endif
+			])
+fi
+
+if test "x$fds_bits_found" = xfalse ; then
     AC_MSG_ERROR([Could not determine how to access the fds_bits or equivalent
                   structure in fd_set on your platform.])
 fi
