diff -urN paragui-1.1.8/Makefile.am paragui-1.1.8-haiku/Makefile.am
--- paragui-1.1.8/Makefile.am	2004-10-18 18:48:10.047710208 +0000
+++ paragui-1.1.8-haiku/Makefile.am	2010-05-29 01:02:52.581435392 +0000
@@ -59,8 +59,8 @@
 	mkdir $(distdir)/lib
 	cp $(srcdir)/lib/README.txt $(distdir)/lib
 
-install-hook:
-	-ldconfig
+#install-hook:
+#	-ldconfig
 
 rpm: $(PACKAGE)-$(VERSION).tar.gz
 	cp $(PACKAGE)-$(VERSION).tar.gz $(RPM_ROOT)
diff -urN paragui-1.1.8/configure.in paragui-1.1.8-haiku/configure.in
--- paragui-1.1.8/configure.in	2004-10-18 18:58:09.047710208 +0000
+++ paragui-1.1.8-haiku/configure.in	2010-05-29 01:01:16.400818176 +0000
@@ -1,8 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
 
+m4_pattern_allow([^PKG_SIGC$])
 AC_INIT(README)
 AM_CONFIG_HEADER(include/paraconfig_gnu.h:include/paraconfig_gnu.h.in)
-
+AC_CONFIG_MACRO_DIR([m4])
 TOP_BUILDDIR=`pwd`
 
 AC_SUBST(TOP_BUILDDIR)
