From 9d0025fb6ed0fc31dd20d800ed3189f8bc82c0b1 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Thu, 17 Jul 2014 12:33:12 -0400
Subject: [PATCH] Don't hang on CMake colored text output. From FreeBSD.

---
 kde4automoc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kde4automoc.cpp b/kde4automoc.cpp
index ae2cff4..a6a5634 100644
--- a/kde4automoc.cpp
+++ b/kde4automoc.cpp
@@ -71,7 +71,7 @@ class AutoMoc
             QStringList args(cmakeEchoColorArgs);
             args << msg;
             cmakeEcho.start(cmakeExecutable, args, QIODevice::NotOpen);
-            cmakeEcho.waitForFinished(-1);
+            cmakeEcho.waitForFinished(2000);
         }
 
         QString builddir;
-- 
1.8.3.4

