diff -Naur djvulibre-3.5.25/libdjvu/miniexp.cpp djvulibre-3.5.25-haiku/libdjvu/miniexp.cpp
--- djvulibre-3.5.25/libdjvu/miniexp.cpp	2012-03-17 23:49:02.048758784 +0100
+++ djvulibre-3.5.25-haiku/libdjvu/miniexp.cpp	2012-03-25 14:42:11.949223424 +0200
@@ -1010,7 +1010,7 @@
   if (io == &miniexp_io && !CompatCounter::count)
     return (*minilisp_getc)(); /* compatibility hack */
   FILE *f = (io->data[0]) ? (FILE*)(io->data[0]) : stdin;
-  return ::getc(f);
+  return getc(f);
 }
 
 static int 
