diff -up p7zip_9.04/install.sh.orig p7zip_9.04/install.sh
--- p7zip_9.04/install.sh.orig	2010-04-23 23:59:47.153092096 -0600
+++ p7zip_9.04/install.sh	2010-04-24 00:00:57.560201728 -0600
@@ -56,10 +56,10 @@ then
     cp bin/7za "${DEST_DIR}${DEST_SHARE}/7za"
     chmod 777 "${DEST_DIR}${DEST_SHARE}/7za"
     strip     "${DEST_DIR}${DEST_SHARE}/7za"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7za"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7za"
     echo "#! /bin/sh" > "${DEST_DIR}${DEST_BIN}/7za"
     echo "\"${DEST_SHARE}/7za\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/7za"
-    chmod 555 "${DEST_DIR}${DEST_BIN}/7za"
+    chmod 755 "${DEST_DIR}${DEST_BIN}/7za"
   fi
 
   if [ -x bin/7zr ]
@@ -68,10 +68,10 @@ then
     cp bin/7zr "${DEST_DIR}${DEST_SHARE}/7zr"
     chmod 777 "${DEST_DIR}${DEST_SHARE}/7zr"
     strip     "${DEST_DIR}${DEST_SHARE}/7zr"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7zr"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7zr"
     echo "#! /bin/sh" > "${DEST_DIR}${DEST_BIN}/7zr"
     echo "\"${DEST_SHARE}/7zr\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/7zr"
-    chmod 555 "${DEST_DIR}${DEST_BIN}/7zr"
+    chmod 755 "${DEST_DIR}${DEST_BIN}/7zr"
   fi
 
   if [ -x bin/7zCon.sfx ]
@@ -80,7 +80,7 @@ then
     cp bin/7zCon.sfx "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
     chmod 777 "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
     strip     "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
   fi
 
   if [ -x bin/7z ]
@@ -89,14 +89,14 @@ then
     cp bin/7z "${DEST_DIR}${DEST_SHARE}/7z"
     chmod 777 "${DEST_DIR}${DEST_SHARE}/7z"
     strip     "${DEST_DIR}${DEST_SHARE}/7z"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7z"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7z"
     cp -r bin/Codecs "${DEST_DIR}${DEST_SHARE}/"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}"/*/*
+    chmod 755 "${DEST_DIR}${DEST_SHARE}"/*/*
     cp bin/7z.so "${DEST_DIR}${DEST_SHARE}/7z.so"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/7z.so"
+    chmod 755 "${DEST_DIR}${DEST_SHARE}/7z.so"
     echo "#! /bin/sh" > "${DEST_DIR}${DEST_BIN}/7z"
     echo "\"${DEST_SHARE}/7z\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/7z"
-    chmod 555 "${DEST_DIR}${DEST_BIN}/7z"
+    chmod 755 "${DEST_DIR}${DEST_BIN}/7z"
   fi
 
 else
@@ -105,7 +105,7 @@ else
     echo "- installing ${DEST_DIR}${DEST_BIN}/7za"
     mkdir -p "${DEST_DIR}${DEST_BIN}"
     cp bin/7za "${DEST_DIR}${DEST_BIN}/7za"
-    chmod 555  "${DEST_DIR}${DEST_BIN}/7za"
+    chmod 755  "${DEST_DIR}${DEST_BIN}/7za"
   fi
 
   if [ -x bin/7zr ]
@@ -113,7 +113,7 @@ else
     echo "- installing ${DEST_DIR}${DEST_BIN}/7zr"
     mkdir -p "${DEST_DIR}${DEST_BIN}"
     cp bin/7zr "${DEST_DIR}${DEST_BIN}/7zr"
-    chmod 555  "${DEST_DIR}${DEST_BIN}/7zr"
+    chmod 755  "${DEST_DIR}${DEST_BIN}/7zr"
   fi
 fi
 
@@ -122,27 +122,27 @@ if [ -d DOCS ]
 then
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7z.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOCS"?g man1/7z.1 > "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7za.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOCS"?g man1/7za.1 > "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7zr.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOCS"?g man1/7zr.1 > "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 else
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7z.1"
    grep -v "{DEST_SHARE_DOC}" man1/7z.1 > "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7za.1"
    grep -v "{DEST_SHARE_DOC}" man1/7za.1 > "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7zr.1"
    grep -v "{DEST_SHARE_DOC}" man1/7zr.1 > "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 fi
 
 if [ -f README ]
@@ -150,7 +150,7 @@ then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/README"
   mkdir -p  "${DEST_DIR}${DEST_SHARE_DOC}"
   cp README "${DEST_DIR}${DEST_SHARE_DOC}/README"
-  chmod 444 "${DEST_DIR}${DEST_SHARE_DOC}/README"
+  chmod 644 "${DEST_DIR}${DEST_SHARE_DOC}/README"
 fi
 
 if [ -f ChangeLog ]
@@ -158,7 +158,7 @@ then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
   mkdir -p     "${DEST_DIR}${DEST_SHARE_DOC}"
   cp ChangeLog "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
-  chmod 444    "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
+  chmod 644    "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
 fi
 
 if [ -d DOCS ]
@@ -166,7 +166,7 @@ then
   echo "- installing HTML help in ${DEST_DIR}${DEST_SHARE_DOC}/DOCS"
   mkdir -p "${DEST_DIR}${DEST_SHARE_DOC}"
   cp -r DOCS "${DEST_DIR}${DEST_SHARE_DOC}/DOCS"
-  find "${DEST_DIR}${DEST_SHARE_DOC}/DOCS" -type d -exec chmod 555 {} \;
-  find "${DEST_DIR}${DEST_SHARE_DOC}/DOCS" -type f -exec chmod 444 {} \;
+  find "${DEST_DIR}${DEST_SHARE_DOC}/DOCS" -type d -exec chmod 755 {} \;
+  find "${DEST_DIR}${DEST_SHARE_DOC}/DOCS" -type f -exec chmod 644 {} \;
 fi
 
