diff -Naur transmission-1.93/libtransmission/JSON_parser.c transmission-1.93-haiku/libtransmission/JSON_parser.c
--- transmission-1.93/libtransmission/JSON_parser.c	2010-05-01 21:35:04.035389440 +0000
+++ transmission-1.93-haiku/libtransmission/JSON_parser.c	2010-05-05 23:28:25.131858432 +0000
@@ -75,6 +75,10 @@
 #   endif
 #endif

+/* This can be removed once http://dev.haiku-os.org/ticket/3408 is fixed */
+#ifdef __HAIKU__
+#   define strtold strtod
+#endif

 #define true  1
 #define false 0
diff -Naur transmission-1.93/third-party/miniupnp/connecthostport.c transmission-1.93-haiku/third-party/miniupnp/connecthostport.c
--- transmission-1.93/third-party/miniupnp/connecthostport.c	2010-05-01 21:35:46.041680896 +0000
+++ transmission-1.93-haiku/third-party/miniupnp/connecthostport.c	2010-05-05 23:24:18.628621312 +0000
@@ -29,6 +29,7 @@
 #ifndef USE_GETHOSTBYNAME
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <sys/select.h>
 #endif /* #ifndef USE_GETHOSTBYNAME */
 #endif /* #else WIN32 */

