diff -Naur transmission-1.93/configure.ac transmission-1.93-haiku/configure.ac
--- transmission-1.93/configure.ac	2010-05-01 21:35:58.020447232 +0000
+++ transmission-1.93-haiku/configure.ac	2010-05-05 23:28:25.129761280 +0000
@@ -205,7 +205,8 @@
 dnl build our copy of libevent whether we use it or not,
 dnl because "make dist" needs its Makefiles to exist, and
 dnl AM_CONDITIONAL + AC_CONFIG_SUBDIRS don't seem to play nice
-AC_CONFIG_SUBDIRS([third-party/libevent])
+dnl (HaikuPorts: require an installed copy)
+dnl AC_CONFIG_SUBDIRS([third-party/libevent])
 libevent_source=bundled
 AC_CHECK_LIB([event],[evutil_vsnprintf],
              [libevent_found=yes],
@@ -221,6 +222,8 @@
     fi
 fi
 if test "x$libevent_source" = "xbundled"; then
+    dnl (HaikuPorts: require an installed copy)
+    AC_MSG_ERROR("libevent not found!")
     AC_MSG_WARN([using our own libevent from third-party/libevent/])
     AC_MSG_WARN([if you are cross-compiling this is probably NOT what you want.])
     LIBEVENT_CFLAGS="-I\$(top_srcdir)/third-party/libevent -I\$(top_builddir)/third-party/libevent"
diff -Naur transmission-1.93/libtransmission/JSON_parser.c transmission-1.93-haiku/libtransmission/JSON_parser.c
--- transmission-1.93/libtransmission/JSON_parser.c	2010-05-01 21:35:04.035389440 +0000
+++ transmission-1.93-haiku/libtransmission/JSON_parser.c	2010-05-05 23:28:25.131858432 +0000
@@ -75,6 +75,10 @@
 #   endif
 #endif

+/* This can be removed once http://dev.haiku-os.org/ticket/3408 is fixed */
+#ifdef __HAIKU__
+#   define strtold strtod
+#endif

 #define true  1
 #define false 0
diff -Naur transmission-1.93/third-party/Makefile.am transmission-1.93-haiku/third-party/Makefile.am
--- transmission-1.93/third-party/Makefile.am	2010-05-01 21:35:49.039059456 +0000
+++ transmission-1.93-haiku/third-party/Makefile.am	2010-05-05 23:28:25.131072000 +0000
@@ -1,4 +1,5 @@
-SUBDIRS = libnatpmp miniupnp libevent dht
+# HaikuPorts: require an installed copy of libevent
+SUBDIRS = libnatpmp miniupnp dht

 EXTRA_DIST = \
   macosx-libevent-config.h \
diff -Naur transmission-1.93/third-party/miniupnp/connecthostport.c transmission-1.93-haiku/third-party/miniupnp/connecthostport.c
--- transmission-1.93/third-party/miniupnp/connecthostport.c	2010-05-01 21:35:46.041680896 +0000
+++ transmission-1.93-haiku/third-party/miniupnp/connecthostport.c	2010-05-05 23:24:18.628621312 +0000
@@ -29,6 +29,7 @@
 #ifndef USE_GETHOSTBYNAME
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <sys/select.h>
 #endif /* #ifndef USE_GETHOSTBYNAME */
 #endif /* #else WIN32 */

