diff -Naur transmission-1.73/configure.ac transmission-1.73-haiku/configure.ac
--- transmission-1.73/configure.ac	2009-07-18 01:21:12.000000000 +0000
+++ transmission-1.73-haiku/configure.ac	2009-07-18 17:57:20.000000000 +0000
@@ -167,7 +167,8 @@
 dnl build our copy of libevent whether we use it or not,
 dnl because "make dist" needs its Makefiles to exist, and
 dnl AM_CONDITIONAL + AC_CONFIG_SUBDIRS don't seem to play nice
-AC_CONFIG_SUBDIRS([third-party/libevent])
+dnl (HaikuPorts: require an installed copy)
+dnl AC_CONFIG_SUBDIRS([third-party/libevent])
 libevent_source=bundled
 AC_CHECK_LIB([event],[evutil_vsnprintf],
              [libevent_found=yes],
@@ -183,6 +184,8 @@
     fi
 fi
 if test "x$libevent_source" = "xbundled"; then
+    dnl (HaikuPorts: require an installed copy)
+    AC_MSG_ERROR("libevent not found!")
     AC_MSG_WARN([using our own libevent from third-party/libevent/])
     AC_MSG_WARN([if you are cross-compiling this is probably NOT what you want.])
     LIBEVENT_CFLAGS="-I\$(top_srcdir)/third-party/libevent -I\$(top_builddir)/third-party/libevent"
@@ -313,17 +316,19 @@
 dnl it should be safe to re-edit 0.40 back down to 0.23
 
 use_nls=no
-if test "x$enable_nls" = "xyes" ; then
-    use_nls=yes
-    IT_PROG_INTLTOOL([0.40.0],[no-xml])
-    AC_CHECK_HEADERS([libintl.h])
-    GETTEXT_PACKAGE=transmission
-    AC_SUBST(GETTEXT_PACKAGE)
-    AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[Gettext package])
-    AM_GLIB_GNU_GETTEXT
-    transmissionlocaledir='${prefix}/${DATADIRNAME}/locale'
-    AC_SUBST(transmissionlocaledir)
-fi
+dnl (HaikuPorts: until intltool is ported, need to comment out this part
+dnl otherwise autoconf won't generate a proper configure)
+dnl if test "x$enable_nls" = "xyes" ; then
+dnl     use_nls=yes
+dnl     IT_PROG_INTLTOOL([0.40.0],[no-xml])
+dnl     AC_CHECK_HEADERS([libintl.h])
+dnl     GETTEXT_PACKAGE=transmission
+dnl     AC_SUBST(GETTEXT_PACKAGE)
+dnl     AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[Gettext package])
+dnl     AM_GLIB_GNU_GETTEXT
+dnl     transmissionlocaledir='${prefix}/${DATADIRNAME}/locale'
+dnl     AC_SUBST(transmissionlocaledir)
+dnl fi
 
 
 dnl ----------------------------------------------------------------------------
diff -Naur transmission-1.73/libtransmission/JSON_parser.c transmission-1.73-haiku/libtransmission/JSON_parser.c
--- transmission-1.73/libtransmission/JSON_parser.c	2009-07-18 01:20:54.000000000 +0000
+++ transmission-1.73-haiku/libtransmission/JSON_parser.c	2009-07-18 17:54:44.000000000 +0000
@@ -75,6 +75,10 @@
 #   endif
 #endif
 
+/* This can be removed once http://dev.haiku-os.org/ticket/3408 is fixed */
+#ifdef __HAIKU__
+#   define strtold strtod
+#endif
 
 #define true  1
 #define false 0
diff -Naur transmission-1.73/libtransmission/net.h transmission-1.73-haiku/libtransmission/net.h
--- transmission-1.73/libtransmission/net.h	2009-07-18 01:20:54.000000000 +0000
+++ transmission-1.73-haiku/libtransmission/net.h	2009-07-18 17:54:44.000000000 +0000
@@ -56,6 +56,32 @@
  #define sockerrno errno
 #endif
 
+/* To be removed once those macros are added to Haiku's headers */
+#ifndef IN6ADDR_ANY_INIT
+ #define IN6ADDR_ANY_INIT {{{0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0}}}
+#endif
+#ifndef INET6_ADDRSTRLEN
+ #define INET6_ADDRSTRLEN 46
+#endif
+#ifndef IN6_IS_ADDR_V4MAPPED
+ #define IN6_IS_ADDR_V4MAPPED(a)                   \
+  ((((const uint32_t *) (a))[0] == 0) &&           \
+   (((const uint32_t *) (a))[1] == 0) &&           \
+   (((const uint32_t *) (a))[2] == htonl(0xFFFF)))
+#endif
+#ifndef IN6_IS_ADDR_V4COMPAT
+ #define IN6_IS_ADDR_V4COMPAT(a)                   \
+  ((((const uint32_t *) (a))[0] == 0) &&           \
+   (((const uint32_t *) (a))[1] == 0) &&           \
+   (((const uint32_t *) (a))[2] == 0) &&           \
+   (ntohl(((const uint32_t *) (a))[2]) > 1))
+#endif
+#ifndef IN6_IS_ADDR_LINKLOCAL
+ #define IN6_IS_ADDR_LINKLOCAL(a)                    \
+  ((((const uint32_t *) (a))[0] & htonl(0xFFC00000)) \
+   == htonl(0xFE800000))
+#endif
+
 struct tr_session;
 
 typedef enum tr_address_type
diff -Naur transmission-1.73/third-party/Makefile.am transmission-1.73-haiku/third-party/Makefile.am
--- transmission-1.73/third-party/Makefile.am	2009-07-18 01:21:07.000000000 +0000
+++ transmission-1.73-haiku/third-party/Makefile.am	2009-07-18 17:58:54.000000000 +0000
@@ -4,7 +4,8 @@
   DHT_DIR =
 endif
 
-SUBDIRS = libnatpmp miniupnp libevent $(DHT_DIR)
+# HaikuPorts: require an installed copy of libevent
+SUBDIRS = libnatpmp miniupnp $(DHT_DIR)
 
 EXTRA_DIST = \
   macosx-libevent-config.h \
