diff -Naur transmission-2.12/libtransmission/JSON_parser.c transmission-2.12-haiku/libtransmission/JSON_parser.c
--- transmission-2.12/libtransmission/JSON_parser.c	2010-06-27 01:28:16.064225280 +0000
+++ transmission-2.12-haiku/libtransmission/JSON_parser.c	2010-07-10 06:08:57.465829888 +0000
@@ -80,6 +80,10 @@
 #   endif
 #endif

+/* This can be removed once http://dev.haiku-os.org/ticket/3408 is fixed */
+#ifdef __HAIKU__
+#   define strtold strtod
+#endif

 #define true  1
 #define false 0
diff -Naur transmission-2.12/third-party/miniupnp/connecthostport.c transmission-2.12-haiku/third-party/miniupnp/connecthostport.c
--- transmission-2.12/third-party/miniupnp/connecthostport.c	2010-06-27 01:28:53.066584576 +0000
+++ transmission-2.12-haiku/third-party/miniupnp/connecthostport.c	2010-07-10 06:08:57.467664896 +0000
@@ -29,6 +29,7 @@
 #ifndef USE_GETHOSTBYNAME
 #include <sys/types.h>
 #include <sys/socket.h>
+#include <sys/select.h>
 #endif /* #ifndef USE_GETHOSTBYNAME */
 #endif /* #else WIN32 */

