diff -up diffutils-2.9/lib/verify.h.orig diffutils-2.9/lib/verify.h
--- diffutils-2.9/lib/verify.h.orig	2010-04-18 13:06:04.590872576 -0600
+++ diffutils-2.9/lib/verify.h	2010-04-18 13:08:52.910163968 -0600
@@ -135,6 +135,10 @@ template <int w>
 /* Verify requirement R at compile-time, as a declaration without a
    trailing ';'.  */
 
+# if __GNUC__ == 2
+# define verify(R)
+# else
 # define verify(R) extern int (* verify_function__ (void)) [verify_true (R)]
+# endif
 
 #endif
