diff -urN diffutils-3.0/gnulib-tests/test-dup2.c diffutils-3.0-haiku/gnulib-tests/test-dup2.c
--- diffutils-3.0/gnulib-tests/test-dup2.c	2010-04-15 13:42:07.036700160 +0000
+++ diffutils-3.0-haiku/gnulib-tests/test-dup2.c	2010-10-06 11:17:38.000000000 +0000
@@ -176,6 +176,7 @@
 
   /* On systems that distinguish between text and binary mode, dup2
      reuses the mode of the source.  */
+#if O_BINARY
   setmode (fd, O_BINARY);
   ASSERT (is_mode (fd, O_BINARY));
   ASSERT (dup2 (fd, fd + 1) == fd + 1);
@@ -190,6 +191,6 @@
   ASSERT (close (fd + 1) == 0);
   ASSERT (close (fd) == 0);
   ASSERT (unlink (file) == 0);
-
+#endif
   return 0;
 }
diff -urN diffutils-3.0/gnulib-tests/test-fcntl.c diffutils-3.0-haiku/gnulib-tests/test-fcntl.c
--- diffutils-3.0/gnulib-tests/test-fcntl.c	2010-04-15 13:42:08.027787264 +0000
+++ diffutils-3.0-haiku/gnulib-tests/test-fcntl.c	2010-10-06 11:21:54.000000000 +0000
@@ -298,6 +298,7 @@
   ASSERT (is_mode (fd + 2, O_BINARY));
   ASSERT (close (fd) == 0);
 
+#if O_TEXT
   setmode (fd + 2, O_TEXT);
   ASSERT (fcntl (fd + 2, F_DUPFD, fd + 1) == fd + 1);
   ASSERT (!is_open (fd));
@@ -318,7 +319,7 @@
   ASSERT (is_mode (fd, O_TEXT));
   ASSERT (is_mode (fd + 2, O_TEXT));
   ASSERT (close (fd + 2) == 0);
-
+#endif
   /* Test F_GETFD.  */
   errno = 0;
   ASSERT (fcntl (-1, F_GETFD) == -1);
diff -urN diffutils-3.0/gnulib-tests/test-signal.c diffutils-3.0-haiku/gnulib-tests/test-signal.c
--- diffutils-3.0/gnulib-tests/test-signal.c	2010-04-15 13:42:09.030932992 +0000
+++ diffutils-3.0-haiku/gnulib-tests/test-signal.c	2010-10-06 11:28:31.000000000 +0000
@@ -58,7 +58,7 @@
 #ifdef SIGALRM
     case SIGALRM:
 #endif
-#ifdef SIGBUS
+#if defined SIGBUS && !defined(__HAIKU__)
     case SIGBUS:
 #endif
 #ifdef SIGCHLD
