diff -urN glib-2.26.1/configure.ac glib-2.26.1-haiku/configure.ac
--- glib-2.26.1/configure.ac	2010-11-14 05:59:52.043778048 +0000
+++ glib-2.26.1-haiku/configure.ac	2011-01-11 12:46:46.976486400 +0000
@@ -111,6 +111,10 @@
   *-*-beos*)
     glib_native_beos="yes"
     ;;
+  *-*-haiku*)
+    glib_native_beos="yes"
+    LIBS="-lnetwork $LIBS"
+    ;;
   *)
     glib_native_beos="no"
     ;;
@@ -3595,6 +3599,9 @@
   *-*-beos*)
     glib_os="#define G_OS_BEOS"
     ;;
+  *-*-haiku*)
+    glib_os="#define G_OS_HAIKU"
+    ;;
   *-*-cygwin*)
     glib_os="#define G_OS_UNIX
 #define G_PLATFORM_WIN32
diff -urN glib-2.26.1/glib/gmain.c glib-2.26.1-haiku/glib/gmain.c
--- glib-2.26.1/glib/gmain.c	2010-11-12 13:52:52.057409536 +0000
+++ glib-2.26.1-haiku/glib/gmain.c	2011-01-11 13:13:26.355991552 +0000
@@ -79,10 +79,11 @@
 #include <windows.h>
 #endif /* G_OS_WIN32 */
 
-#ifdef G_OS_BEOS
+#if defined G_OS_HAIKU
 #include <sys/socket.h>
 #include <sys/wait.h>
-#endif /* G_OS_BEOS */
+#include <fcntl.h>
+#endif /* G_OS_HAIKU */
 
 #ifdef G_OS_UNIX
 #include <fcntl.h>
diff -urN glib-2.26.1/glib/gstdio.c glib-2.26.1-haiku/glib/gstdio.c
--- glib-2.26.1/glib/gstdio.c	2010-09-13 13:40:53.047972352 +0000
+++ glib-2.26.1-haiku/glib/gstdio.c	2011-01-11 13:17:06.033292288 +0000
@@ -45,7 +45,7 @@
 #include "gstdio.h"
 
 
-#if !defined (G_OS_UNIX) && !defined (G_OS_WIN32) && !defined (G_OS_BEOS)
+#if !defined (G_OS_UNIX) && !defined (G_OS_WIN32) && !defined (G_OS_HAIKU)
 #error Please port this to your operating system
 #endif
 
diff -urN glib-2.26.1/glib/gstrfuncs.c glib-2.26.1-haiku/glib/gstrfuncs.c
--- glib-2.26.1/glib/gstrfuncs.c	2010-09-04 01:40:14.015466496 +0000
+++ glib-2.26.1-haiku/glib/gstrfuncs.c	2011-01-11 12:43:29.319815680 +0000
@@ -1423,12 +1423,12 @@
 #ifdef HAVE_STRSIGNAL
   const char *msg_locale;
 
-#if defined(G_OS_BEOS) || defined(G_WITH_CYGWIN)
+#if defined(G_OS_HAIKU) || defined(G_WITH_CYGWIN)
 extern const char *strsignal(int);
 #else
-  /* this is declared differently (const) in string.h on BeOS */
+  /* this is declared differently (const) in string.h on HAIKU */
   extern char *strsignal (int sig);
-#endif /* !G_OS_BEOS && !G_WITH_CYGWIN */
+#endif /* !G_OS_HAIKU && !G_WITH_CYGWIN */
   msg_locale = strsignal (signum);
   if (g_get_charset (NULL))
     return msg_locale;
diff -urN glib-2.26.1/glib/gutils.c glib-2.26.1-haiku/glib/gutils.c
--- glib-2.26.1/glib/gutils.c	2010-11-12 13:52:52.056098816 +0000
+++ glib-2.26.1-haiku/glib/gutils.c	2011-01-11 13:27:36.000000000 +0000
@@ -491,7 +491,7 @@
     }
   
   path = g_getenv ("PATH");
-#if defined(G_OS_UNIX) || defined(G_OS_BEOS)
+#if defined(G_OS_UNIX) || defined(G_OS_HAIKU)
   if (path == NULL)
     {
       /* There is no `PATH' in the environment.  The default
