diff -Naur glib-1.2.10-original/Makefile.am glib-1.2.10/Makefile.am
--- glib-1.2.10-original/Makefile.am	2009-01-22 23:29:09.000000000 +0000
+++ glib-1.2.10/Makefile.am	2009-01-22 23:32:21.000000000 +0000
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I m4
+
 # require automake 1.4
 AUTOMAKE_OPTIONS = 1.4
 
@@ -7,8 +9,8 @@
 
 configincludedir = $(pkglibdir)/include
 
-noinst_SCRIPTS=glib-config
-#BUILT_SOURCES=glib-config
+bin_SCRIPTS=glib-config
+BUILT_SOURCES=glib-config
 glib-config: glib-config.in
 
 INCLUDES = -DG_LOG_DOMAIN=g_log_domain_glib @GLIB_DEBUG_FLAGS@
@@ -30,7 +32,7 @@
 	gthread.pc.in
 
 
-noinst_LTLIBRARIES = libglib.la
+lib_LTLIBRARIES = libglib.la
 
 libglib_la_SOURCES = \
 	garray.c	\
@@ -60,11 +62,11 @@
 	gutils.c
 
 glibincludedir=$(includedir)/glib-1.2
-noinst_HEADERS =   \
+glibinclude_HEADERS =   \
 	glib.h
 
-#noinst_DATA = \
-#	glibconfig.h
+configinclude_DATA = \
+	glibconfig.h
 
 CONFIGURE_DEPENDENCIES = acglib.m4		
 
@@ -86,10 +88,10 @@
 testgdateparser_LDADD = libglib.la
 
 m4datadir = $(datadir)/aclocal
-noinst_DATA = glib.m4
+m4data_DATA = glib.m4
 
 pkgconfigdir = $(libdir)/pkgconfig
-noinst_DATA = glib.pc gmodule.pc gthread.pc
+pkgconfig_DATA = glib.pc gmodule.pc gthread.pc
 
 glibconfig.h.win32: $(top_builddir)/config.status $(top_srcdir)/glibconfig.h.win32.in
 	cd $(top_builddir) && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
diff -Naur glib-1.2.10-original/configure.in glib-1.2.10/configure.in
--- glib-1.2.10-original/configure.in	2009-01-22 23:29:09.000000000 +0000
+++ glib-1.2.10/configure.in	2009-01-22 23:30:28.000000000 +0000
@@ -9,6 +9,8 @@
 # init autoconf (and check for presence of glist.c)
 AC_INIT(glist.c)
 
+AC_CONFIG_MACRO_DIR([m4])
+
 # Save this value here, since automake will set cflags later
 cflags_set=${CFLAGS+set}
 
