diff -urN vlc-0.8.6i/src/input/input.c vlc-0.8.6i-haiku/src/input/input.c
--- vlc-0.8.6i/src/input/input.c	2008-07-08 14:08:36.000000000 -0700
+++ vlc-0.8.6i-haiku/src/input/input.c	2008-08-14 06:52:18.000000000 -0700
@@ -1886,7 +1886,7 @@
 {
     vlc_meta_t *p_meta = p_input->p_meta;
     int i;
-
+    char *psz_cat;
     if( !p_meta || p_meta->i_meta == 0 )
         return VLC_SUCCESS;
 
@@ -1917,7 +1917,6 @@
         if( tk->i_meta > 0 )
         {
             msg_Dbg( p_input, "  - track[%d]:", i );
-            char *psz_cat;
             if( asprintf( &psz_cat, "%s %d", _("Stream"), i ) != -1 )
             {
                 for( j = 0; j < tk->i_meta; j++ )
diff -urN vlc-0.8.6i/src/network/httpd.c vlc-0.8.6i-haiku/src/network/httpd.c
--- vlc-0.8.6i/src/network/httpd.c	2008-07-08 14:08:36.000000000 -0700
+++ vlc-0.8.6i-haiku/src/network/httpd.c	2008-08-14 07:07:11.000000000 -0700
@@ -366,6 +366,7 @@
 {
     httpd_file_t *file = (httpd_file_t*)p_sys;
     uint8_t **pp_body, *p_body;
+    uint8_t *psz_args = 0;
     char *psz_connection = NULL;
     int *pi_body, i_body;
 
@@ -402,7 +403,8 @@
         /* msg_Warn not supported */
     }
 
-    uint8_t *psz_args = query->psz_args;
+    *psz_args = query->psz_args;
+
     file->pf_fill( file->p_sys, file, psz_args, pp_body, pi_body );
 
     if( query->i_type == HTTPD_MSG_HEAD && p_body != NULL )
@@ -487,6 +489,7 @@
 {
     httpd_handler_t *handler = (httpd_handler_t*)p_sys;
     char psz_remote_addr[NI_MAXNUMERICHOST];
+    uint8_t *psz_args = 0;
 
     if( answer == NULL || query == NULL )
     {
@@ -502,7 +505,7 @@
     if( httpd_ClientIP( cl, psz_remote_addr ) == NULL )
         *psz_remote_addr = '\0';
 
-    uint8_t *psz_args = query->psz_args;
+    *psz_args = query->psz_args;
     handler->pf_fill( handler->p_sys, handler, query->psz_url, psz_args,
                       query->i_type, query->p_body, query->i_body,
                       psz_remote_addr, NULL,
