diff -urN subversion-1.6.5/subversion/libsvn_subr/config_impl.h subversion-1.6.5-haiku/subversion/libsvn_subr/config_impl.h
--- subversion-1.6.5/subversion/libsvn_subr/config_impl.h	2009-01-27 13:23:50.000000000 +0000
+++ subversion-1.6.5-haiku/subversion/libsvn_subr/config_impl.h	2009-09-02 13:43:03.000000000 +0000
@@ -109,6 +109,9 @@
    or svn_config_get_user_config_path() instead. */
 #ifdef WIN32
 #  define SVN_CONFIG__SUBDIRECTORY    "Subversion"
+#elif defined __HAIKU__ /* HAIKU */
+#  define SVN_CONFIG__SYS_DIRECTORY   "/boot/common/etc/subversion"
+#  define SVN_CONFIG__USR_DIRECTORY   "config/settings/subversion"
 #else  /* ! WIN32 */
 #  define SVN_CONFIG__SYS_DIRECTORY   "/etc/subversion"
 #  define SVN_CONFIG__USR_DIRECTORY   ".subversion"
