--- penguin-command-1.6.11/configure.org	2016-04-27 21:32:44.663748608 +0200
+++ penguin-command-1.6.11/configure	2016-04-27 21:32:25.360710144 +0200
@@ -4878,7 +4878,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lm  $LIBS"
+LIBS="$LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -4932,7 +4932,7 @@
 #define HAVE_LIBM 1
 _ACEOF
 
-  LIBS="-lm $LIBS"
+  LIBS="$LIBS"
 
 else
   { { echo "$as_me:$LINENO: error: Your system is missing libm!" >&5
@@ -5075,7 +5075,7 @@
 
 ######## Set compiler flags and libraries
 CFLAGS="$CFLAGS $SDL_CFLAGS -Wall -DJOYSTICK -DDATADIR=\"\\\"$datadir/penguin-command\\\"\""
-LIBS="$SDL_LIBS $SOUND -lSDL_image -lSDL -lm"
+LIBS="$SDL_LIBS $SOUND -lSDL_image -lSDL"
 
 ######## Finally create all the generated files
 ac_config_files="$ac_config_files Makefile src/Makefile data/Makefile data/gfx/Makefile data/sound/Makefile penguin-command.spec"
--- penguin-command-1.6.11/src/main.c.org	2016-04-27 21:43:29.272105472 +0200
+++ penguin-command-1.6.11/src/main.c	2016-04-27 21:37:38.985661440 +0200
@@ -32,7 +32,7 @@
 #endif
 
 
-    int fullscreen=1, CityDestroyed[6]={0,0,0,0,0,0},
+    int fullscreen=0, CityDestroyed[6]={0,0,0,0,0,0},
 	CannonDestroyed[3]={0,0,0}, Score=0, HWCursor=0, NoAlpha=0, ArcadeMode=0, MusicVol=10, SoundVol=10;
 //images
     SDL_Surface *ShotPic,*CityPic,*BackPic,*CityHitPic,*Cannon[21],*Font,
