--- penguin-command-1.6.11/src/main.c.org	2016-04-27 21:43:29.272105472 +0200
+++ penguin-command-1.6.11/src/main.c	2016-04-27 21:37:38.985661440 +0200
@@ -32,7 +32,7 @@
 #endif
 
 
-    int fullscreen=1, CityDestroyed[6]={0,0,0,0,0,0},
+    int fullscreen=0, CityDestroyed[6]={0,0,0,0,0,0},
 	CannonDestroyed[3]={0,0,0}, Score=0, HWCursor=0, NoAlpha=0, ArcadeMode=0, MusicVol=10, SoundVol=10;
 //images
     SDL_Surface *ShotPic,*CityPic,*BackPic,*CityHitPic,*Cannon[21],*Font,
