From f26cd828c65cd8d6b521165171c6373db7bd2d79 Mon Sep 17 00:00:00 2001
From: Markus Himmel <markus@himmel-villmar.de>
Date: Sat, 26 Dec 2015 16:57:30 +0100
Subject: [PATCH] Fix build on Haiku

---
 common/socket_utils.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/socket_utils.cc b/common/socket_utils.cc
index 99a4a23..cf43533 100644
--- a/common/socket_utils.cc
+++ b/common/socket_utils.cc
@@ -28,6 +28,7 @@ using namespace std;
 
 #ifndef __WIN32__
 # include <sys/socket.h>
+# include <sys/time.h>
 #else
 # include "safeerrno.h"
 
-- 
2.2.2

