diff --git a/src/test/include.am b/src/test/include.am
index 2765cf2..db3e717 100644
--- a/src/test/include.am
+++ b/src/test/include.am
@@ -13,12 +13,6 @@ TESTS_ENVIRONMENT = \
 TESTSCRIPTS = \
 	src/test/fuzz_static_testcases.sh \
 	src/test/test_zero_length_keys.sh \
-	src/test/test_workqueue_cancel.sh \
-	src/test/test_workqueue_efd.sh \
-	src/test/test_workqueue_efd2.sh \
-	src/test/test_workqueue_pipe.sh \
-	src/test/test_workqueue_pipe2.sh \
-	src/test/test_workqueue_socketpair.sh \
 	src/test/test_switch_id.sh \
 	src/test/test_cmdline.sh \
 	src/test/test_parseconf.sh \
@@ -55,7 +49,6 @@ TESTSCRIPTS += \
 endif
 
 TESTS += src/test/test-slow src/test/test-memwipe \
-	src/test/test_workqueue \
 	src/test/test_keygen.sh \
 	src/test/test_key_expiration.sh \
 	src/test/test-timers \
