diff --git a/configure.in b/configure.in
index 016ea95..f222548 100644
--- a/configure.in
+++ b/configure.in
@@ -121,13 +121,7 @@ d_test.d_type = 0;
 
 
 dnl Check for powf
-AC_MSG_CHECKING(for powf in libm);
-LIBS_BACKUP=$LIBS;
-LIBS="$LIBS -lm";
-AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <math.h>]],[[
-        powf(1.0f, 1.0f);
-]])], [AC_MSG_RESULT(yes)], [AC_DEFINE([DB_HAVE_NO_POWF],[1],[libm doesn't include powf])])
-LIBS=$LIBS_BACKUP
+AC_SEARCH_LIBS([powf],[m], [AC_MSG_RESULT(yes)], [AC_DEFINE([DB_HAVE_NO_POWF],[1],[no powf function avalible])])
 
 
 dnl Checks for libraries.
