diff --git a/configure.in b/configure.in
index 016ea95..b114016 100644
--- a/configure.in
+++ b/configure.in
@@ -120,16 +120,6 @@ d_test.d_type = 0;
 }],[AC_MSG_RESULT(yes);AC_DEFINE(DIRENT_HAS_D_TYPE,1,[struct dirent has d_type])],AC_MSG_RESULT(no))
 
 
-dnl Check for powf
-AC_MSG_CHECKING(for powf in libm);
-LIBS_BACKUP=$LIBS;
-LIBS="$LIBS -lm";
-AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <math.h>]],[[
-        powf(1.0f, 1.0f);
-]])], [AC_MSG_RESULT(yes)], [AC_DEFINE([DB_HAVE_NO_POWF],[1],[libm doesn't include powf])])
-LIBS=$LIBS_BACKUP
-
-
 dnl Checks for libraries.
 
 #Check if the compiler support attributes
