diff -Naur Mesa-9.1.1/src/mesa/main/execmem.c Mesa-9.1.1-haiku/src/mesa/main/execmem.c
--- Mesa-9.1.1/src/mesa/main/execmem.c	2013-03-20 00:14:38.014417920 +0000
+++ Mesa-9.1.1-haiku/src/mesa/main/execmem.c	2013-04-27 00:42:08.707264512 +0000
@@ -36,7 +36,7 @@
 
 
 
-#if defined(__linux__) || defined(__OpenBSD__) || defined(_NetBSD__) || defined(__sun)
+#if defined(__linux__) || defined(__OpenBSD__) || defined(_NetBSD__) || defined(__sun) || defined(__HAIKU__)
 
 /*
  * Allocate a large block of memory which can hold code then dole it out
