diff -Naur Mesa-9.0.2/scons/llvm.py Mesa-9.0.2-haiku/scons/llvm.py
--- Mesa-9.0.2/scons/llvm.py	2013-01-22 18:09:32.066322432 +0000
+++ Mesa-9.0.2-haiku/scons/llvm.py	2013-02-17 21:20:51.626262016 +0000
@@ -183,6 +183,9 @@
             if llvm_version >= distutils.version.LooseVersion('3.1'):
                 components.append('mcjit')
 
+            if llvm_version >= distutils.version.LooseVersion('3.2'):
+                env.Append(CXXFLAGS = ('-fno-rtti',))
+
             env.ParseConfig('llvm-config --libs ' + ' '.join(components))
             env.ParseConfig('llvm-config --ldflags')
         except OSError:
diff -Naur Mesa-9.0.2/src/mesa/SConscript Mesa-9.0.2-haiku/src/mesa/SConscript
--- Mesa-9.0.2/src/mesa/SConscript	2013-01-22 18:09:32.040370176 +0000
+++ Mesa-9.0.2-haiku/src/mesa/SConscript	2013-02-17 21:20:22.510394368 +0000
@@ -371,7 +371,7 @@
 #
 # Assembly sources
 #
-if env['gcc'] and env['platform'] not in ('darwin', 'windows'):
+if env['gcc'] and env['platform'] not in ('darwin', 'windows', 'haiku'):
     if env['machine'] == 'x86':
         env.Append(CPPDEFINES = [
             'USE_X86_ASM',
