diff -Naur Mesa-9.0.1/scons/llvm.py Mesa-9.0.1-haiku/scons/llvm.py
--- Mesa-9.0.1/scons/llvm.py	2012-08-31 18:33:41.041680896 -0500
+++ Mesa-9.0.1-haiku/scons/llvm.py	2012-12-18 23:05:50.677642240 -0600
@@ -183,6 +183,9 @@
             if llvm_version >= distutils.version.LooseVersion('3.1'):
                 components.append('mcjit')
 
+            if llvm_version >= distutils.version.LooseVersion('3.2'):
+                env.Append(CXXFLAGS = ('-fno-rtti',))
+
             env.ParseConfig('llvm-config --libs ' + ' '.join(components))
             env.ParseConfig('llvm-config --ldflags')
         except OSError:
diff -Naur Mesa-9.0.1/src/gallium/auxiliary/Makefile Mesa-9.0.1-haiku/src/gallium/auxiliary/Makefile
--- Mesa-9.0.1/src/gallium/auxiliary/Makefile	2012-08-31 18:33:41.050331648 -0500
+++ Mesa-9.0.1-haiku/src/gallium/auxiliary/Makefile	2012-12-18 23:06:31.179306496 -0600
@@ -15,6 +15,10 @@
 	$(GALLIVM_CPP_SOURCES)
 endif
 
+# LLVM >= 3.2 requires -fno-rtti
+ifeq ($(shell expr `echo $(LLVM_VERSION) | sed -e 's/\([0-9]\)\.\([0-9]\)/\10\2/g'` \>= 302),1)
+CXXFLAGS += -fno-rtti
+endif
 
 include ../Makefile.template
 
diff -Naur Mesa-9.0.1/src/mesa/SConscript Mesa-9.0.1-haiku/src/mesa/SConscript
--- Mesa-9.0.1/src/mesa/SConscript	2012-11-09 12:16:31.015728640 -0600
+++ Mesa-9.0.1-haiku/src/mesa/SConscript	2012-12-19 11:37:28.289144832 -0600
@@ -331,6 +331,14 @@
     statetracker_sources
 )
 
+mesa_gallium_sources = (
+    main_sources +
+    math_sources +
+    vbo_sources +
+    program_sources,
+    'x86/common_x86.c'
+)
+
 GLAPI = '#src/mapi/glapi/'
 
 if env['gles']:
@@ -495,8 +503,15 @@
     source = mesa_sources,
 )
 
+mesagallium = env.ConvenienceLibrary(
+    target = 'mesagallium',
+    source = mesa_gallium_sources,
+)
+
 env.Alias('mesa', mesa)
+env.Alias('mesagallium', mesagallium)
 
 Export('mesa')
+Export('mesagallium')
 
 SConscript('drivers/SConscript')
