diff -up physfs-2.0.1/CMakeLists.txt.orig physfs-2.0.1/CMakeLists.txt
--- physfs-2.0.1/CMakeLists.txt.orig	2010-04-27 21:17:08.770441216 -0600
+++ physfs-2.0.1/CMakeLists.txt	2010-04-27 21:25:09.183238656 -0600
@@ -108,7 +108,9 @@ IF(BEOS)
     # We add this explicitly, since we don't want CMake to think this
     #  is a C++ project unless we're on BeOS.
     SET(PHYSFS_BEOS_SRCS platform/beos.cpp)
-    SET(OPTIONAL_LIBRARY_LIBS ${OPTIONAL_LIBRARY_LIBS} be root)
+    FIND_LIBRARY(BE_LIBRARY be)
+    FIND_LIBRARY(ROOT_LIBRARY root)
+    SET(OPTIONAL_LIBRARY_LIBS ${OPTIONAL_LIBRARY_LIBS} ${BE_LIBRARY} ${ROOT_LIBRARY})
 ENDIF(BEOS)
 
 # Almost everything is "compiled" here, but things that don't apply to the
@@ -298,7 +300,8 @@ IF(PHYSFS_BUILD_TEST)
     FIND_PATH(READLINE_H readline/readline.h)
     FIND_PATH(HISTORY_H readline/history.h)
     IF(READLINE_H AND HISTORY_H)
-        SET(CMAKE_REQUIRED_LIBRARIES curses)
+        FIND_LIBRARY(CURSES_LIBRARY NAMES curses ncurses)
+        SET(CMAKE_REQUIRED_LIBRARIES ${CURSES_LIBRARY})
         FIND_LIBRARY(READLINE_LIBRARY readline)
         FIND_LIBRARY(HISTORY_LIBRARY history)
         IF(READLINE_LIBRARY AND HISTORY_LIBRARY)
