? ylwrap
Index: Makefile.am
===================================================================
RCS file: /sources/dotgnu-pnet/treecc/Makefile.am,v
retrieving revision 1.12
diff -u -p -r1.12 Makefile.am
--- Makefile.am	4 Jun 2007 10:14:29 -0000	1.12
+++ Makefile.am	26 Apr 2008 19:49:11 -0000
@@ -47,8 +47,8 @@ SKELETON_FILES = $(top_srcdir)/etc/c_ske
 				 $(top_srcdir)/etc/cpp_gc_skel.h \
 				 $(top_srcdir)/etc/cpp_gc_skel.cc
 
-skels.c: $(SKELETON_FILES) mkskel-sh
-	$(SHELL) mkskel-sh $(SKELETON_FILES) >skels.c
+skels.c: $(SKELETON_FILES) $(top_srcdir)/mkskel-sh
+	$(SHELL) $(top_srcdir)/mkskel-sh $(SKELETON_FILES) >skels.c
 
 CLEANFILES = skels.c
 
Index: configure.in
===================================================================
RCS file: /sources/dotgnu-pnet/treecc/configure.in,v
retrieving revision 1.52
diff -u -p -r1.52 configure.in
--- configure.in	4 Jun 2007 18:45:48 -0000	1.52
+++ configure.in	26 Apr 2008 19:49:11 -0000
@@ -23,7 +23,7 @@ AC_MSG_RESULT($platform_win32)
 
 AC_MSG_CHECKING([if building for BeOS platform])
 case "$host" in
-	*-*-beos*)
+	*-*-beos*|*-*-haiku*)
 		platform_beos=yes
 		LDADD=""
 		;;
