diff -urN bsdbuild-2.9/configure.in bsdbuild-2.9-haiku/configure.in
--- bsdbuild-2.9/configure.in	2011-06-20 12:46:10.031719424 +0000
+++ bsdbuild-2.9-haiku/configure.in	2011-08-11 21:05:19.596377600 +0000
@@ -4,7 +4,7 @@
 VERSION("2.9")
 RELEASE("Blast of Ancient Atmospheres")
 
-MDEFINE(SHAREDIR, "$PREFIX/share/bsdbuild")
+MDEFINE(SHAREDIR, "$SHAREDIR/bsdbuild")
 
 echo "*"
 echo "* Configuration successful."
diff -urN bsdbuild-2.9/mkconfigure.pl bsdbuild-2.9-haiku/mkconfigure.pl
--- bsdbuild-2.9/mkconfigure.pl	2011-05-17 08:06:19.033554432 +0000
+++ bsdbuild-2.9-haiku/mkconfigure.pl	2011-08-11 22:05:40.015728640 +0000
@@ -865,7 +865,10 @@
 	    datadir=$optarg
 	    ;;
 	--sharedir=*)
-	    datadir=$optarg
+	    sharedir=$optarg
+	    ;;
+	--sbindir=*)
+	    sbindir=$optarg
 	    ;;
 	--localedir=*)
 	    localedir=$optarg
