diff --git a/src/main.cpp b/src/main.cpp
index c8cfdc1..b9d53a7 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -109,6 +109,11 @@ void startUp (int argc, char *argv[]) {
 	firstPath = NULL;
 #endif
 
+#ifdef __HAIKU__
+	firstPath = new Path(firstPath, "/boot/system/data/openjazz/");
+	firstPath = new Path(firstPath, "/boot/home/config/non-packaged/data/openjazz/");
+#endif
+
 #ifdef __SYMBIAN32__
 	#ifdef UIQ3
 	firstPath = new Path(firstPath, createString("c:\\shared\\openjazz\\"));
