diff -ur u-boot-2012.10/common/image.c tmp/u-boot-2012.10/common/image.c
--- u-boot-2012.10/common/image.c	2012-11-08 10:52:02.774111232 +0100
+++ tmp/u-boot-2012.10/common/image.c	2012-10-15 17:14:08.046399488 +0200
@@ -71,7 +71,7 @@
 #include "mkimage.h"
 #include <u-boot/md5.h>
 #include <time.h>
-#include <uimage.h>
+#include <image.h>
 #endif /* !USE_HOSTCC*/
 
 static const table_entry_t uimage_arch[] = {
diff -ur u-boot-2012.10/tools/aisimage.c tmp/u-boot-2012.10/tools/aisimage.c
--- u-boot-2012.10/tools/aisimage.c	2012-11-08 10:48:20.923271168 +0100
+++ tmp/u-boot-2012.10/tools/aisimage.c	2012-10-15 17:14:08.063176704 +0200
@@ -26,7 +26,7 @@
 
 #include "mkimage.h"
 #include "aisimage.h"
-#include <uimage.h>
+#include <image.h>
 
 #define IS_FNC_EXEC(c)	(cmd_table[c].AIS_cmd == AIS_CMD_FNLOAD)
 #define WORD_ALIGN0	4
diff -ur u-boot-2012.10/tools/default_image.c tmp/u-boot-2012.10/tools/default_image.c
--- u-boot-2012.10/tools/default_image.c	2012-11-08 10:48:39.345505792 +0100
+++ tmp/u-boot-2012.10/tools/default_image.c	2012-10-15 17:14:08.066060288 +0200
@@ -28,7 +28,7 @@
  */
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include <u-boot/crc.h>
 
 static image_header_t header;
diff -ur u-boot-2012.10/tools/fit_image.c tmp/u-boot-2012.10/tools/fit_image.c
--- u-boot-2012.10/tools/fit_image.c	2012-11-08 10:48:56.498860032 +0100
+++ tmp/u-boot-2012.10/tools/fit_image.c	2012-10-15 17:14:08.000786432 +0200
@@ -28,7 +28,7 @@
  */
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include <u-boot/crc.h>
 
 static image_header_t header;
diff -ur u-boot-2012.10/tools/imximage.c tmp/u-boot-2012.10/tools/imximage.c
--- u-boot-2012.10/tools/imximage.c	2012-11-08 10:52:34.872677376 +0100
+++ tmp/u-boot-2012.10/tools/imximage.c	2012-10-15 17:14:08.003145728 +0200
@@ -29,7 +29,7 @@
 #define _GNU_SOURCE
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include "imximage.h"
 
 /*
diff -ur u-boot-2012.10/tools/kwbimage.c tmp/u-boot-2012.10/tools/kwbimage.c
--- u-boot-2012.10/tools/kwbimage.c	2012-11-08 10:52:53.073662464 +0100
+++ tmp/u-boot-2012.10/tools/kwbimage.c	2012-10-15 17:14:08.003407872 +0200
@@ -26,7 +26,7 @@
 #define _GNU_SOURCE
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include "kwbimage.h"
 
 /*
diff -ur u-boot-2012.10/tools/mkimage.c tmp/u-boot-2012.10/tools/mkimage.c
--- u-boot-2012.10/tools/mkimage.c	2012-11-08 10:55:06.533463040 +0100
+++ tmp/u-boot-2012.10/tools/mkimage.c	2012-10-15 17:14:08.005505024 +0200
@@ -22,7 +22,7 @@
  */
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include <version.h>
 
 static void copy_file(int, const char *, int);
diff -ur u-boot-2012.10/tools/omapimage.c tmp/u-boot-2012.10/tools/omapimage.c
--- u-boot-2012.10/tools/omapimage.c	2012-11-08 10:55:39.680525824 +0100
+++ tmp/u-boot-2012.10/tools/omapimage.c	2012-10-15 17:14:08.006291456 +0200
@@ -34,7 +34,7 @@
 #define _GNU_SOURCE
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include "omapimage.h"
 
 /* Header size is CH header rounded up to 512 bytes plus GP header */
diff -ur u-boot-2012.10/tools/pblimage.c tmp/u-boot-2012.10/tools/pblimage.c
--- u-boot-2012.10/tools/pblimage.c	2012-11-08 10:53:09.547356672 +0100
+++ tmp/u-boot-2012.10/tools/pblimage.c	2012-10-15 17:14:08.008126464 +0200
@@ -22,7 +22,7 @@
 #define _GNU_SOURCE
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include "pblimage.h"
 
 /*
diff -ur u-boot-2012.10/tools/ublimage.c tmp/u-boot-2012.10/tools/ublimage.c
--- u-boot-2012.10/tools/ublimage.c	2012-11-08 10:55:59.951844864 +0100
+++ tmp/u-boot-2012.10/tools/ublimage.c	2012-10-15 17:14:08.009175040 +0200
@@ -33,7 +33,7 @@
 #define _GNU_SOURCE
 
 #include "mkimage.h"
-#include <uimage.h>
+#include <image.h>
 #include "ublimage.h"
 
 /*
