diff -ur libogg-1.1.3/configure.in libogg-1.1.3-haiku/configure.in
--- libogg-1.1.3/configure.in	2005-11-28 04:26:07.000000000 +0000
+++ libogg-1.1.3-haiku/configure.in	2008-05-30 11:00:35.000000000 +0000
@@ -84,7 +84,7 @@
 AC_MSG_CHECKING(for int16_t)
 AC_CACHE_VAL(has_int16_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
@@ -100,7 +100,7 @@
 AC_MSG_CHECKING(for int32_t)
 AC_CACHE_VAL(has_int32_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
@@ -116,7 +116,7 @@
 AC_MSG_CHECKING(for uint32_t)
 AC_CACHE_VAL(has_uint32_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
@@ -132,7 +132,7 @@
 AC_MSG_CHECKING(for uint16_t)
 AC_CACHE_VAL(has_uint16_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
@@ -148,7 +148,7 @@
 AC_MSG_CHECKING(for u_int32_t)
 AC_CACHE_VAL(has_u_int32_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
@@ -164,7 +164,7 @@
 AC_MSG_CHECKING(for u_int16_t)
 AC_CACHE_VAL(has_u_int16_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
@@ -180,7 +180,7 @@
 AC_MSG_CHECKING(for int64_t)
 AC_CACHE_VAL(has_int64_t,
 [AC_TRY_RUN([
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <inttypes.h>
 #endif
 #include <sys/types.h>
diff -ur libogg-1.1.3/include/ogg/os_types.h libogg-1.1.3-haiku/include/ogg/os_types.h
--- libogg-1.1.3/include/ogg/os_types.h	2005-11-28 04:26:05.000000000 +0000
+++ libogg-1.1.3-haiku/include/ogg/os_types.h	2008-05-30 12:50:50.000000000 +0000
@@ -73,7 +73,17 @@
    typedef u_int32_t ogg_uint32_t;
    typedef int64_t ogg_int64_t;
 
-#elif defined(__BEOS__)
+#elif defined(__HAIKU__)
+
+  /* Haiku */
+#  include <sys/types.h>
+   typedef short ogg_int16_t;
+   typedef unsigned short ogg_uint16_t;
+   typedef int ogg_int32_t;
+   typedef unsigned int ogg_uint32_t;
+   typedef long long ogg_int64_t;   
+
+#elif defined(__BEOS__) 
 
    /* Be */
 #  include <inttypes.h>
