commit c7734598e2ee084c17e1b8a41daa569fb021d593
Author: Florian Loitsch <florian@loitsch.com>
Date:   Sat Mar 8 15:02:13 2014 +0100

    Support shared libraries with Cmake. Improve README.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5bf382e..7b09be0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,12 @@ cmake_minimum_required(VERSION 2.8)
 project(double-conversion)
 
 # pick a version #
-set(double-conversion_VERSION
-  1.1.1)
+set(double-conversion_VERSION 2.0.1)
+set(double-conversion_SOVERSION_MAJOR 1)
+set(double-conversion_SOVERSION_MINOR 0)
+set(double-conversion_SOVERSION_PATCH 0)
+set(double-conversion_SOVERSION
+  ${double-conversion_SOVERSION_MAJOR}.${double-conversion_SOVERSION_MINOR}.${double-conversion_SOVERSION_PATCH})
 
 # set paths for install -- empty initially
 # Offer the user the choice of overriding the installation directories
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7f5c985..0da50b1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -29,6 +29,12 @@ ${headers}
 set_target_properties(double-conversion
     PROPERTIES PUBLIC_HEADER "${headers}")
 
+if (BUILD_SHARED_LIBS)
+  set_target_properties(double-conversion
+    PROPERTIES VERSION ${double-conversion_SOVERSION}
+               SOVERSION ${double-conversion_SOVERSION_MAJOR})
+endif()
+
 #
 # install command to set up library install
 # given the above PUBLIC_HEADER property set, this
