diff -Naur jbigkit-2.0/libjbig/Makefile-libjbig_so jbigkit-2.0-haiku/libjbig/Makefile-libjbig_so
--- jbigkit-2.0/libjbig/Makefile-libjbig_so	1970-01-01 00:00:00.000000000 +0000
+++ jbigkit-2.0-haiku/libjbig/Makefile-libjbig_so	2009-11-03 16:06:37.000000000 +0000
@@ -0,0 +1,37 @@
+
+# This Makefile builds a shared version of the library, 
+# libjbig.so.2.0, with soname libjbig.so.2,
+# Please see the README file for some important info 
+# about building the library like this.
+# ------------------------------------------------------------------
+# Please read the WARNING, DISCLAIMER and PATENTS sections in the 
+# README file.
+#
+# This program is released under the terms of the license contained
+# in the file LICENSE.
+# ------------------------------------------------------------------
+
+
+SHELL=/bin/sh
+CC=gcc
+CFLAGS=-fPIC -g -c -Wall
+
+OBJS=  jbig.o  \
+      jbig_ar.o    \
+      jbig85.o
+
+all: $(OBJS)
+	$(CC) -shared -Wl,-soname,libjbig.so -o libjbig.so.2.0 $(OBJS)
+	rm -f libjbig.so
+	ln -s libjbig.so.2.0 libjbig.so.2
+	ln -s libjbig.so.2.0 libjbig.so
+
+clean: 
+	rm -f $(OBJS) jbig.o libjbig.so.2.0 libjbig.so
+
+jbig85.o: jbig85.c
+	$(CC) $(CFLAGS) -c jbig85.c
+jbig.o: jbig.c
+	$(CC) $(CFLAGS) -c jbig.c
+jbig_ar.o: jbig_ar.c
+	$(CC) $(CFLAGS) -c jbig_ar.c
