diff -Naur cln-1.3.2/src/base/random/cl_random_from.cc cln-1.3.2-haiku/src/base/random/cl_random_from.cc
--- cln-1.3.2/src/base/random/cl_random_from.cc	2009-05-10 22:32:30.056360960 +0200
+++ cln-1.3.2-haiku/src/base/random/cl_random_from.cc	2012-06-14 20:16:49.920387584 +0200
@@ -18,7 +18,7 @@
 #include "base/cl_low.h"
 #include <cstdlib>  // declares rand()
 
-#if defined(unix) || defined(__unix) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__) || defined(_AIX) || defined(sinix) || (defined(__MACH__) && defined(__APPLE__)) || (defined(__CYGWIN__) && defined(__GNUC__)) || defined(__BEOS__)
+#if defined(unix) || defined(__unix) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__) || defined(_AIX) || defined(sinix) || (defined(__MACH__) && defined(__APPLE__)) || (defined(__CYGWIN__) && defined(__GNUC__)) || defined(__BEOS__) || defined(__HAIKU__)
 
 #include <sys/types.h>
 #include <unistd.h> // declares getpid()
@@ -86,7 +86,7 @@
 {
 	var uint32 seed_hi;
 	var uint32 seed_lo;
-#if defined(unix) || defined(__unix) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__) || defined(_AIX) || defined(sinix) || (defined(__MACH__) && defined(__APPLE__)) || (defined(__CYGWIN__) && defined(__GNUC__)) || defined(__BEOS__)
+#if defined(unix) || defined(__unix) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__) || defined(_AIX) || defined(sinix) || (defined(__MACH__) && defined(__APPLE__)) || (defined(__CYGWIN__) && defined(__GNUC__)) || defined(__BEOS__) || defined(__HAIKU__)
 	seed_lo = get_seed();
 	seed_hi = (rand() // zufällige 31 Bit (bei UNIX_BSD) bzw. 16 Bit (bei UNIX_SYSV)
                           << 8) ^ (uintL)(getpid()); // ca. 8 Bit von der Process ID
