diff --git a/clamd/CMakeLists.txt b/clamd/CMakeLists.txt
index cd23ab931..733939cc8 100644
--- a/clamd/CMakeLists.txt
+++ b/clamd/CMakeLists.txt
@@ -41,6 +41,10 @@ target_include_directories( clamd
 )
 set_target_properties( clamd PROPERTIES COMPILE_FLAGS "${WARNCFLAGS}" )
 
+if(HAIKU)
+    set_target_properties( clamd PROPERTIES COMPILE_FLAGS "${WARNCFLAGS} -fPIC" )
+endif()
+
 if (APPLE AND CLAMAV_SIGN_FILE)
     set_target_properties( clamd PROPERTIES
         XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${CODE_SIGN_IDENTITY}
diff --git a/clamdscan/CMakeLists.txt b/clamdscan/CMakeLists.txt
index f30002443..5c4780128 100644
--- a/clamdscan/CMakeLists.txt
+++ b/clamdscan/CMakeLists.txt
@@ -30,6 +30,10 @@ target_include_directories( clamdscan
 )
 set_target_properties( clamdscan PROPERTIES COMPILE_FLAGS "${WARNCFLAGS}" )
 
+if(HAIKU)
+    set_target_properties( clamdscan PROPERTIES COMPILE_FLAGS "${WARNCFLAGS} -fPIC" )
+endif()
+
 if (APPLE AND CLAMAV_SIGN_FILE)
     set_target_properties( clamdscan PROPERTIES
         XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${CODE_SIGN_IDENTITY}
diff --git a/clamscan/CMakeLists.txt b/clamscan/CMakeLists.txt
index ab0863544..ad66607c6 100644
--- a/clamscan/CMakeLists.txt
+++ b/clamscan/CMakeLists.txt
@@ -29,6 +29,10 @@ target_include_directories( clamscan
 )
 set_target_properties( clamscan PROPERTIES COMPILE_FLAGS "${WARNCFLAGS}" )
 
+if(HAIKU)
+    set_target_properties( clamscan PROPERTIES COMPILE_FLAGS "${WARNCFLAGS} -fPIC" )
+endif()
+
 if (APPLE AND CLAMAV_SIGN_FILE)
     set_target_properties( clamscan PROPERTIES
         XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${CODE_SIGN_IDENTITY}
diff --git a/clamsubmit/CMakeLists.txt b/clamsubmit/CMakeLists.txt
index ecbd7014c..ef22c5f2e 100644
--- a/clamsubmit/CMakeLists.txt
+++ b/clamsubmit/CMakeLists.txt
@@ -23,6 +23,10 @@ target_include_directories( clamsubmit
 )
 set_target_properties( clamsubmit PROPERTIES COMPILE_FLAGS "${WARNCFLAGS}" )
 
+if(HAIKU)
+    set_target_properties( clamsubmit PROPERTIES COMPILE_FLAGS "${WARNCFLAGS} -fPIC" )
+endif()
+
 if (APPLE AND CLAMAV_SIGN_FILE)
     set_target_properties( clamscan PROPERTIES
         XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${CODE_SIGN_IDENTITY}
diff --git a/freshclam/CMakeLists.txt b/freshclam/CMakeLists.txt
index ef3616fe3..fd754196b 100644
--- a/freshclam/CMakeLists.txt
+++ b/freshclam/CMakeLists.txt
@@ -27,6 +27,10 @@ if(WIN32)
 endif()
 set_target_properties( freshclam-bin PROPERTIES COMPILE_FLAGS "${WARNCFLAGS}" )
 
+if(HAIKU)
+    set_target_properties( freshclam-bin PROPERTIES COMPILE_FLAGS "${WARNCFLAGS} -fPIC" )
+endif()
+
 if (APPLE AND CLAMAV_SIGN_FILE)
     set_target_properties( freshclam-bin PROPERTIES
         XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY ${CODE_SIGN_IDENTITY}
diff --git a/libfreshclam/CMakeLists.txt b/libfreshclam/CMakeLists.txt
index 01b456c30..218d74c32 100644
--- a/libfreshclam/CMakeLists.txt
+++ b/libfreshclam/CMakeLists.txt
@@ -48,6 +48,10 @@ if(ENABLE_SHARED_LIB)
                 resolv
                 ${APPLE_CORE_FOUNDATION}
                 ${APPLE_SECURITY} )
+    elseif(HAIKU)
+        target_link_libraries( freshclam
+            PUBLIC
+                network )
     elseif(UNIX)
         if(HAVE_RESOLV_H AND NOT C_BSD) # BSD appears to have libresolv inside libc
             target_link_libraries( freshclam
@@ -128,6 +132,10 @@ if(ENABLE_STATIC_LIB)
                 resolv
                 ${APPLE_CORE_FOUNDATION}
                 ${APPLE_SECURITY} )
+    elseif(HAIKU)
+        target_link_libraries( freshclam_static
+            PUBLIC
+                network )
     elseif(UNIX)
         if(HAVE_RESOLV_H AND NOT C_BSD) # BSD appears to have libresolv inside libc
             target_link_libraries( freshclam_static
diff --git a/unit_tests/CMakeLists.txt b/unit_tests/CMakeLists.txt
index dd24de4b5..f6bfe36d8 100644
--- a/unit_tests/CMakeLists.txt
+++ b/unit_tests/CMakeLists.txt
@@ -116,6 +116,9 @@ target_link_libraries(check_clamav
         ClamAV::libclamav
         ClamAV::common
         libcheck::check)
+if(HAIKU)
+    set_target_properties( check_clamav PROPERTIES COMPILE_FLAGS "-fPIC" )
+endif()
 if (ENABLE_UNRAR)
     if(ENABLE_SHARED_LIB)
         target_link_libraries(check_clamav
