--- icebreaker-1.9.7/icebreaker.h.org	2016-06-12 07:50:07.712245248 +0200
+++ icebreaker-1.9.7/icebreaker.h	2016-06-12 07:49:31.050331648 +0200
@@ -64,6 +64,16 @@
   // HISCOREPREFIX and DATAPREFIX should probably be ".". Set in Makefile.w32
 #endif
 
+#ifdef __HAIKU__
+
+  #define NEEDCHANGETOARGV0PATH
+
+  #ifndef OPTIONFILE
+    #define OPTIONFILE "config/settings/icebreaker.cfg"
+  #endif
+
+  // HISCOREPREFIX and DATAPREFIX should probably be ".". Set in Makefile.beos
+#endif
 
 #ifdef __BEOS__
 
