From 0d6475fe39fbf50a489f48b540ca848a499639f7 Mon Sep 17 00:00:00 2001
From: Alessandro Molina <alessandro@molina.fyi>
Date: Sun, 5 Feb 2023 00:03:02 +0000
Subject: [PATCH] Select directories in directory selection dialog

---
 src/kdialog.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/kdialog.cpp b/src/kdialog.cpp
index 310b66b54..d55b1820c 100644
--- a/src/kdialog.cpp
+++ b/src/kdialog.cpp
@@ -890,6 +890,7 @@ QT_WARNING_POP
 
         QFileDialog dlg;
         dlg.setOption(QFileDialog::ShowDirsOnly, true);
+        dlg.setFileMode(QFileDialog::Directory);
         dlg.setDirectoryUrl(initialDirectory(startUrl));
         dlg.selectFile(initialSelection(startUrl));
         Utils::handleXGeometry(&dlg);
-- 
2.37.3

