diff -urN almp3/Makefile almp3-haiku/Makefile
--- almp3/Makefile	2005-07-26 02:36:11.066846720 +0000
+++ almp3-haiku/Makefile	2010-06-08 08:16:15.839909376 +0000
@@ -7,7 +7,7 @@
 #TARGET=MINGW32_STATIC
 #TARGET=MINGW32_DLL
 #TARGET=LINUX_STATIC
-TARGET=SOLARIS_DYNAMIC
+#TARGET=SOLARIS_DYNAMIC
 
 prefix = /opt/BCallegro
 exec_prefix = ${prefix}
@@ -60,6 +60,19 @@
 endif
 
 
+# HAIKU_STATIC
+ifeq ($(TARGET),HAIKU_STATIC)
+TARGETFLAGS=-Wall -O2 -march=pentium -fomit-frame-pointer -finline-functions -ffast-math
+OBJDIR=obj/haiku/static
+LIBDEST=lib/libalmp3.a
+endif
+
+# HAIKU_SHARED
+ifeq ($(TARGET),HAIKU_SHARED)
+TARGETFLAGS=-Wall -O2 -march=pentium -fomit-frame-pointer -finline-functions -ffast-math
+OBJDIR=obj/haiku/dynamic
+LIBDEST=lib/libalmp3.so
+endif
 
 # LINUX_STATIC
 ifeq ($(TARGET),LINUX_STATIC)
@@ -106,7 +119,6 @@
 endif
 endif
 
-
 # compiling of the library
 %.o: %.c
 	$(CC) -c $(CFLAGS) $< -o $(OBJDIR)/$@
@@ -118,7 +130,8 @@
 	$(INSTALL_DATA) -D docs/AUTHORS.txt $(DESTDIR)$(docdir)/almp3/AUTHORS.txt
 	$(INSTALL_DATA) -D docs/COPYING.txt $(DESTDIR)$(docdir)/almp3/COPYING.txt
 	$(INSTALL_DATA) -D include/almp3.h $(DESTDIR)$(includedir)/almp3.h
-	$(INSTALL) -m 755 -D $(LIBDEST) $(DESTDIR)$(libdir)/libalmp3.so
+	$(INSTALL) -m 755 -D lib/libalmp3.so $(DESTDIR)$(libdir)/libalmp3.so
+	$(INSTALL) -m 755 -D lib/libalmp3.a $(DESTDIR)$(libdir)/libalmp3.a
 
 clean:
 	rm -f $(OBJECTS2) $(LIBDEST) $(LIBIMP)
