From 0f122c388ee79b8221b4e3f0d7615437743db577 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sun, 19 Apr 2020 15:03:11 -0400
Subject: GCC2 fixes.


diff --git a/roff.c b/roff.c
index de75a26..cc622cb 100644
--- a/roff.c
+++ b/roff.c
@@ -94,7 +94,7 @@ struct	roffreg {
  */
 struct	roffreq {
 	enum roff_tok	 tok;
-	char		 name[];
+	char		 name[0];
 };
 
 /*
diff --git a/tag.c b/tag.c
index fcaad99..82e0259 100644
--- a/tag.c
+++ b/tag.c
@@ -40,7 +40,7 @@ struct tag_entry {
 	size_t	 maxnodes;
 	size_t	 nnodes;
 	int	 prio;
-	char	 s[];
+	char	 s[0];
 };
 
 static void		 tag_move_href(struct roff_man *,
-- 
2.50.1

