From 0f122c388ee79b8221b4e3f0d7615437743db577 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sun, 19 Apr 2020 15:03:11 -0400
Subject: GCC2 fixes.


diff --git a/roff.c b/roff.c
index de75a26..cc622cb 100644
--- a/roff.c
+++ b/roff.c
@@ -94,7 +94,7 @@ struct	roffreg {
  */
 struct	roffreq {
 	enum roff_tok	 tok;
-	char		 name[];
+	char		 name[0];
 };
 
 /*
diff --git a/tag.c b/tag.c
index fcaad99..82e0259 100644
--- a/tag.c
+++ b/tag.c
@@ -40,7 +40,7 @@ struct tag_entry {
 	size_t	 maxnodes;
 	size_t	 nnodes;
 	int	 prio;
-	char	 s[];
+	char	 s[0];
 };
 
 static void		 tag_move_href(struct roff_man *,
diff --git a/dba.c b/dba.c
index ee43933..1cadf20 100644
--- a/dba.c
+++ b/dba.c
@@ -43,7 +43,7 @@
 
 struct macro_entry {
 	struct dba_array	*pages;
-	char			 value[];
+	char			 value[0];
 };
 
 static void	*prepend(const char *, char);
diff --git a/html.c b/html.c
index 71c9c71..15dbe2a 100644
--- a/html.c
+++ b/html.c
@@ -116,7 +116,7 @@ static	const struct htmldata htmltags[TAG_MAX] = {
 
 struct	id_entry {
 	int	 ord;	/* Ordinal number of the latest occurrence. */
-	char	 id[];	/* The id= attribute without any ordinal suffix. */
+	char	 id[0];	/* The id= attribute without any ordinal suffix. */
 };
 static	struct ohash	 id_unique;
 
diff --git a/mandoc_xr.h b/mandoc_xr.h
index e0c6af0..7e0b619 100644
--- a/mandoc_xr.h
+++ b/mandoc_xr.h
@@ -22,7 +22,7 @@ struct	mandoc_xr {
 	int		  line;  /* Or -1 for this page's own names. */
 	int		  pos;
 	int		  count;
-	char		  hashkey[];
+	char		  hashkey[0];
 };
 
 void		  mandoc_xr_reset(void);
diff --git a/mandocdb.c b/mandocdb.c
index f42c235..4690983 100644
--- a/mandocdb.c
+++ b/mandocdb.c
@@ -74,7 +74,7 @@ enum	op {
 struct	str {
 	const struct mpage *mpage; /* if set, the owning parse */
 	uint64_t	 mask; /* bitmask in sequence */
-	char		 key[]; /* rendered text */
+	char		 key[0]; /* rendered text */
 };
 
 struct	inodev {
-- 
2.50.1

