diff -ur orig/subversion-1.6.18/configure.ac subversion-1.6.18/configure.ac
--- orig/subversion-1.6.18/configure.ac	2009-10-13 18:09:30.043778048 +0200
+++ subversion-1.6.18/configure.ac	2013-04-30 22:10:52.741605376 +0200
@@ -249,7 +249,7 @@
 dnl Check for libraries --------------------
 
 # Berkeley DB on SCO OpenServer needs -lsocket
-AC_CHECK_LIB(socket, socket)
+AC_SEARCH_LIBS(socket, socket network)
 
 # Build the BDB filesystem library only if we have an appropriate
 # version of Berkeley DB.
diff -ur orig/subversion-1.6.18/subversion/libsvn_subr/config_file.c subversion-1.6.18/subversion/libsvn_subr/config_file.c
--- orig/subversion-1.6.18/subversion/libsvn_subr/config_file.c	2009-11-25 18:27:35.015728640 +0100
+++ subversion-1.6.18/subversion/libsvn_subr/config_file.c	2013-04-30 22:20:33.557842432 +0200
@@ -32,6 +32,11 @@
 
 #include "svn_private_config.h"
 
+#ifdef __HAIKU__
+#  include <FindDirectory.h>
+#  include <StorageDefs.h>
+#endif
+
 /* Used to terminate lines in large multi-line string literals. */
 #define NL APR_EOL_STR
 
@@ -1115,8 +1120,21 @@
     *path = svn_path_join_many(pool, folder,
                                SVN_CONFIG__SUBDIRECTORY, fname, NULL);
   }
+  
+#elif defined(__HAIKU__)
+{
+  char folder[B_PATH_NAME_LENGTH];
+
+  status_t error = find_directory (B_USER_SETTINGS_DIRECTORY, -1, false,
+                                   folder, sizeof(folder));
+  if (error)
+    return SVN_NO_ERROR;
+
+  *path = svn_path_join_many(pool, folder,
+                             SVN_CONFIG__USR_DIRECTORY, fname, NULL);
+}
+#else  /* ! WIN32 && !__HAIKU__ */
 
-#else  /* ! WIN32 */
   {
     const char *homedir = svn_user_get_homedir(pool);
     if (! homedir)
diff -ur orig/subversion-1.6.18/subversion/libsvn_subr/config_impl.h subversion-1.6.18/subversion/libsvn_subr/config_impl.h
--- orig/subversion-1.6.18/subversion/libsvn_subr/config_impl.h	2009-01-27 14:23:50.020971520 +0100
+++ subversion-1.6.18/subversion/libsvn_subr/config_impl.h	2013-04-30 22:19:32.338427904 +0200
@@ -109,8 +109,11 @@
    or svn_config_get_user_config_path() instead. */
 #ifdef WIN32
 #  define SVN_CONFIG__SUBDIRECTORY    "Subversion"
-#else  /* ! WIN32 */
-#  define SVN_CONFIG__SYS_DIRECTORY   "/etc/subversion"
+#elif defined __HAIKU__ /* HAIKU */
+#  define SVN_CONFIG__SYS_DIRECTORY   "subversion"
+#  define SVN_CONFIG__USR_DIRECTORY   "subversion"
+#else  /* ! WIN32 && ! __HAIKU__ */
+#  define SVN_CONFIG__SYS_DIRECTORY   "@HAIKU_GLOBAL_SETTINGS_DIR@"
 #  define SVN_CONFIG__USR_DIRECTORY   ".subversion"
 #endif /* WIN32 */
 
