diff -urN subversion-1.5.4/configure.ac subversion-1.5.4-haiku/configure.ac
--- subversion-1.5.4/configure.ac	2008-12-15 00:35:49.000000000 -0800
+++ subversion-1.5.4-haiku/configure.ac	2008-12-15 00:32:30.000000000 -0800
@@ -220,6 +220,7 @@
 
 # Berkeley DB on SCO OpenServer needs -lsocket
 AC_CHECK_LIB(socket, socket)
+AC_CHECK_LIB(network, socket)
 
 # Build the BDB filesystem library only if we have an appropriate
 # version of Berkeley DB.
