From 6c18d36b30e6eefe9a757ca9f6c3505d9218e5f3 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Thu, 18 Dec 2014 12:35:36 -0500
Subject: [PATCH] Fix autoconf script to include the proper header for timeval.

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 76e160d..56c731f 100644
--- a/configure.in
+++ b/configure.in
@@ -159,7 +159,7 @@ AC_CHECK_FUNCS([gettimeofday isascii memchr memmove memset select strcasecmp str
 AC_CHECK_FUNCS([time localtime snprintf vsnprintf usleep select getenv isatty chmod])
 AC_CHECK_FUNCS([tcsetattr tcgetattr tcflush tcdrain cfmakeraw cfsetospeed cfsetispeed])
 AC_CHECK_FUNCS([ioctl open creat close read write unlink fileno])
-AC_CHECK_TYPE([struct timeval],[AC_DEFINE(HAS_STRUCT_TIMEVAL,[1],[Define to 1 if struct timeval is available])])
+AC_CHECK_TYPE([struct timeval],[AC_DEFINE(HAS_STRUCT_TIMEVAL,[1],[Define to 1 if struct timeval is available])],[],[[#include <sys/time.h>]])
 #
 #
 # Check for where X11 could be
-- 
1.8.3.4

