Index: compiler/Makefile
===================================================================
--- compiler/Makefile	(revision 41258)
+++ compiler/Makefile	(working copy)
@@ -12,7 +12,7 @@
 ACPICA_OSL =    $(ACPICA_SRC)/os_specific/service_layers
 
 NOMAN=	YES
-CFLAGS+= -Wall -O2 -Wstrict-prototypes -D_LINUX -DACPI_ASL_COMPILER -I../include -I../compiler
+CFLAGS+= -Wall -O2 -Wstrict-prototypes -DACPI_ASL_COMPILER -I../include -I../compiler
 
 #
 # Bison/Flex configuration
@@ -25,9 +25,16 @@
 
 
 COMPILE = $(CC) -c $(CFLAGS) -o$@ $?
+
+OS ?= Linux
+ifeq ($(OS),Linux)
+CFLAGS += -D_LINUX
 LDFLAGS += -lpthread -lrt
+endif
+ifeq ($(OS),Haiku)
+CFLAGS += -D_LINUX
+endif
 
-
 OBJS = \
 	adfile.o \
 	adisasm.o \
