? gettext-runtime/libasprintf/autom4te.cache
Index: gettext-runtime/libasprintf/configure.ac
===================================================================
RCS file: /sources/gettext/gettext/gettext-runtime/libasprintf/configure.ac,v
retrieving revision 1.24
diff -u -r1.24 configure.ac
--- gettext-runtime/libasprintf/configure.ac	20 Apr 2008 06:48:03 -0000	1.24
+++ gettext-runtime/libasprintf/configure.ac	4 May 2008 17:26:34 -0000
@@ -41,7 +41,7 @@
 case "$host_os" in
   # On Cygwin, without -no-undefined, a warning is emitted and only a static
   # library is built.
-  beos* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
+  beos* | haiku* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
   *) LTNOUNDEF='' ;;
 esac
 AC_SUBST([LTNOUNDEF])
Index: gettext-tools/configure.ac
===================================================================
RCS file: /sources/gettext/gettext/gettext-tools/configure.ac,v
retrieving revision 1.96
diff -u -r1.96 configure.ac
--- gettext-tools/configure.ac	20 Apr 2008 06:48:03 -0000	1.96
+++ gettext-tools/configure.ac	4 May 2008 17:29:23 -0000
@@ -92,7 +92,7 @@
 case "$host_os" in
   # On Cygwin, without -no-undefined, a warning is emitted and only a static
   # library is built.
-  beos* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
+  beos* | haiku* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
   *) LTNOUNDEF='' ;;
 esac
 AC_SUBST([LTNOUNDEF])
