diff --git a/filetype.pro b/filetype.pro
index 47691c4..e4aa9ee 100644
--- a/filetype.pro
+++ b/filetype.pro
@@ -34,9 +34,7 @@ SOURCES += \
 	filetype/file/src/fsmagic.c \
 	filetype/file/src/funcs.c \
 	filetype/file/src/getline.c \
-	filetype/file/src/gmtime_r.c \
 	filetype/file/src/is_tar.c \
-	filetype/file/src/localtime_r.c \
 	filetype/file/src/magic.c \
 	filetype/file/src/pread.c \
 	filetype/file/src/print.c \
diff --git a/filetype/file/src/gmtime_r.c b/filetype/file/src/gmtime_r.c
index 7e27ed6..bc05d6c 100644
--- a/filetype/file/src/gmtime_r.c
+++ b/filetype/file/src/gmtime_r.c
@@ -15,5 +15,5 @@ gmtime_r(const time_t *t, struct tm *tm)
 	if (tmp == NULL)
 		return NULL;
 	memcpy(tm, tmp, sizeof(*tm));
-	return tmp;
+	return tm;
 }
diff --git a/filetype/filetype.pri b/filetype/filetype.pri
index 279a5de..932ef59 100644
--- a/filetype/filetype.pri
+++ b/filetype/filetype.pri
@@ -26,11 +26,9 @@ SOURCES += \
 	filetype/file/src/fsmagic.c \
 	filetype/file/src/funcs.c \
 	filetype/file/src/getline.c \
-	filetype/file/src/gmtime_r.c \
 	filetype/file/src/is_csv.c \
 	filetype/file/src/is_json.c \
 	filetype/file/src/is_tar.c \
-	filetype/file/src/localtime_r.c \
 	filetype/file/src/magic.c \
 	filetype/file/src/pread.c \
 	filetype/file/src/print.c \
