diff -ur cscope-15.7a/configure.in cscope-15.7a-haiku/configure.in
--- cscope-15.7a/configure.in	2009-04-30 13:17:34.000000000 -0600
+++ cscope-15.7a-haiku/configure.in	2009-12-29 02:33:05.000000000 -0700
@@ -264,6 +264,9 @@
 
     AC_NCURSES(/usr/include/ncurses, curses.h, -lncurses, -I/usr/include/ncurses -DRENAMED_NCURSES, "renamed ncurses on /usr/include/ncurses")
 
+	AC_NCURSES(/boot/system/develop/headers, ncurses.h, -lncurses,, "ncurses on /boot/system/develop/headers")
+	AC_NCURSES(/boot/home/config/develop/headers, ncurses.h, -lncurses,, "ncurses on /boot/home/config/develop/headers")
+
     dnl
     dnl We couldn't find ncurses, try SysV curses
     dnl
diff -ur cscope-15.7a/src/main.c cscope-15.7a-haiku/src/main.c
--- cscope-15.7a/src/main.c	2009-04-10 08:36:38.000000000 -0600
+++ cscope-15.7a-haiku/src/main.c	2009-12-29 02:25:46.000000000 -0700
@@ -121,7 +121,7 @@
 void	fixkeypad();
 #endif
 
-#if defined(KEY_RESIZE) && !defined(__DJGPP__)
+#if defined(KEY_RESIZE) && !defined(__DJGPP__) && !defined(__HAIKU__)
 void 
 sigwinch_handler(int sig, siginfo_t *info, void *unused)
 {
@@ -402,7 +402,7 @@
 	signal(SIGINT, SIG_IGN);	/* ignore interrupts */
 	signal(SIGPIPE, SIG_IGN);/* | command can cause pipe signal */
 
-#if defined(KEY_RESIZE) && !defined(__DJGPP__)
+#if defined(KEY_RESIZE) && !defined(__DJGPP__) && !defined(__HAIKU__)
 	winch_action.sa_sigaction = sigwinch_handler;
 	sigemptyset(&winch_action.sa_mask);
 	winch_action.sa_flags = SA_SIGINFO;
