diff --git a/configure.in b/configure.in
index a1c58ab..2d8a3b1 100644
--- a/configure.in
+++ b/configure.in
@@ -1,6 +1,6 @@
 AC_PREREQ(2.50)
 AC_INIT(libiptcdata/iptc-data.h)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE(libiptcdata, 1.0.4)
 AM_MAINTAINER_MODE

@@ -75,7 +75,7 @@ dnl Check for headers (Mac OSX often doesn't have them)
 AC_CHECK_HEADERS([getopt.h wchar.h iconv.h])


-GTK_DOC_CHECK([1.0])
+dnl GTK_DOC_CHECK([1.0])


 dnl ---------------------------------------------------------------------------
diff --git a/Makefile.am b/Makefile.am
index 192529a..504ed54 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,4 +7,4 @@ SUBDIRS = m4 libiptcdata po iptc docs win $(MAYBE_PYTHONLIB)
 EXTRA_DIST = @PACKAGE@.spec

 ACLOCAL_AMFLAGS = -I m4
-DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
+DISTCHECK_CONFIGURE_FLAGS =
diff --git a/docs/reference/Makefile.am b/docs/reference/Makefile.am
index f91793a..1f52449 100644
--- a/docs/reference/Makefile.am
+++ b/docs/reference/Makefile.am
@@ -42,7 +42,7 @@ content_files = \
 	libjpeg.sgml \
 	overview.sgml

-include $(top_srcdir)/gtk-doc.make
+include

-EXTRA_DIST += \
+EXTRA_DIST = \
 	version.xml.in
