diff -urN /develop/projects/matroska/install/libebml-1.0.0/ebml/c/libebml_t.h /develop/projects/matroska/libebml-1.0.0/ebml/c/libebml_t.h
--- /develop/projects/matroska/install/libebml-1.0.0/ebml/c/libebml_t.h	2010-03-23 11:58:36.042729472 +0000
+++ /develop/projects/matroska/libebml-1.0.0/ebml/c/libebml_t.h	2010-06-22 23:34:22.247201792 +0000
@@ -72,7 +72,7 @@
     typedef uint16_t uint16;
     typedef uint8_t uint8;
 # endif // __GNUC__
-#elif defined(__BEOS__)
+#elif defined(__BEOS__) || defined(__HAIKU__) 
 #include <SupportDefs.h>
 #elif defined(DJGPP)				/* SL : DJGPP doesn't support POSIX types ???? */
     typedef signed long long int64;
@@ -98,7 +98,7 @@
     typedef uint32_t uint32;
     typedef uint16_t uint16;
     typedef uint8_t uint8;
-#elif defined(__BEOS__)
+#elif defined(__BEOS__) || defined(__HAIKU__) 
 # include <support/SupportDefs.h>
 #else // anything else (Linux, BSD, ...)
 # include <inttypes.h>
diff -urN /develop/projects/matroska/install/libebml-1.0.0/make/linux/Makefile /develop/projects/matroska/libebml-1.0.0/make/linux/Makefile
--- /develop/projects/matroska/install/libebml-1.0.0/make/linux/Makefile	2010-06-04 08:44:54.048758784 +0000
+++ /develop/projects/matroska/libebml-1.0.0/make/linux/Makefile	2010-06-23 01:31:00.288358400 +0000
@@ -13,6 +13,9 @@
 # BeOS wants the libs and headers in /boot/home/config
 ifeq (BeOS,$(shell uname -s))
 prefix=/boot/home/config
+# Haiku wants the libs and headers in /boot/common
+else ifeq (Haiku,$(shell uname -s))
+prefix=/boot/common
 else
 prefix=/usr/local
 endif
