diff -urN e2fsprogs-1.43/configure.in e2fsprogs-1.43-haiku/configure.in
--- e2fsprogs-1.43/configure.in	2013-03-18 01:46:29.019922944 -0700
+++ e2fsprogs-1.43-haiku/configure.in	2013-03-18 02:02:17.526123008 -0700
@@ -96,8 +96,11 @@
 dnl
 AC_PROG_CC
 if test "$GCC" = yes; then
-   RDYNAMIC="-rdynamic"
-   AC_SUBST(RDYNAMIC)
+  case "$host_os" in 
+    haiku*) RDYNAMIC="-Wl,-export-dynamic" ;; 
+    *) RDYNAMIC="-rdynamic" ;; 
+  esac 
+  AC_SUBST(RDYNAMIC)
 fi
 AC_PROG_CPP
 dnl
@@ -1097,6 +1100,7 @@
 dnl
 SOCKET_LIB=''
 AC_CHECK_LIB(socket, socket, [SOCKET_LIB=-lsocket])
+AC_CHECK_LIB(network, socket, [SOCKET_LIB=-lnetwork])
 AC_SUBST(SOCKET_LIB)
 dnl
 dnl See if optreset exists
