diff -Naur -x .git /boot/home/MrPeeps/src/ListData.h ./src/ListData.h
--- /boot/home/MrPeeps/src/ListData.h	2013-12-01 19:17:49.441188352 +0400
+++ ./src/ListData.h	2013-12-01 20:25:27.129236992 +0400
@@ -2,6 +2,7 @@
 #define LIST_DATA
 
 #include <Entry.h>
+#include <Node.h>
 #include <List.h>
 #include <String.h>
 #include "BStringList.h"
diff -Naur -x .git /boot/home/MrPeeps/src/PeepsWindow.h ./src/PeepsWindow.h
--- /boot/home/MrPeeps/src/PeepsWindow.h	2013-12-01 19:17:49.467927040 +0400
+++ ./src/PeepsWindow.h	2013-12-01 20:24:59.028573696 +0400
@@ -8,6 +8,8 @@
 #include <Button.h>
 #include <Menu.h>
 #include <MenuItem.h>
+#include <ListItem.h>
+#include <Entry.h>
 #include "NewTextControl.h"
 #include <Messenger.h>
 #include "LocaleRoster.h"
@@ -117,4 +119,4 @@
 
 #define TRANSLATE(x) gCurrentLocale->Translate(x).String()
 
-#endif
\ No newline at end of file
+#endif
