diff -Naur -x .git /boot/home/MrPeeps/src/ListData.h ./src/ListData.h
--- /boot/home/MrPeeps/src/ListData.h	2013-12-01 19:17:49.441188352 +0400
+++ ./src/ListData.h	2013-12-09 19:07:33.235667456 +0400
@@ -3,10 +3,11 @@
 
 #include <Entry.h>
 #include <List.h>
+#include <Node.h>
 #include <String.h>
-#include "BStringList.h"
 #include <Bitmap.h>
 #include "BitmapDump.h"
+#include "BStringList.h"
 
 class PeepsListItem;
 class GroupItem;
diff -Naur -x .git /boot/home/MrPeeps/src/PeepsWindow.h ./src/PeepsWindow.h
--- /boot/home/MrPeeps/src/PeepsWindow.h	2013-12-01 19:17:49.467927040 +0400
+++ ./src/PeepsWindow.h	2013-12-09 19:07:59.108789760 +0400
@@ -1,16 +1,18 @@
 #ifndef PEEPSWINDOW_H
 #define PEEPSWINDOW_H
 
-#include <Window.h>
-#include <View.h>
-#include <TextControl.h>
-#include <PopUpMenu.h>
 #include <Button.h>
+#include <Entry.h>
+#include <ListItem.h>
 #include <Menu.h>
 #include <MenuItem.h>
-#include "NewTextControl.h"
 #include <Messenger.h>
+#include <PopUpMenu.h>
+#include <TextControl.h>
+#include <View.h>
+#include <Window.h>
 #include "LocaleRoster.h"
+#include "NewTextControl.h"
 
 class PeopleList;
 class PeepsListItem;
@@ -117,4 +119,4 @@
 
 #define TRANSLATE(x) gCurrentLocale->Translate(x).String()
 
-#endif
\ No newline at end of file
+#endif
