Index: configure
===================================================================
--- configure	(revision 17469)
+++ configure	(working copy)
@@ -1461,16 +1461,21 @@
         esac
         SHFLAGS=-nostart
         # enable BeOS things
-        enable audio_beos
+        # enable audio_beos
         # no need for libm, but the inet stuff
         # Check for BONE
         # XXX: actually should check for NOT net_server
-        if echo $BEINCLUDES | grep -q 'headers/be/bone'; then
-            network_extralibs="-lbind -lsocket"
+        if [ $target_os = "haiku" ]; then
+            network_extralibs="-lnetwork"
         else
-            enable beos_netserver
-            network_extralibs="-lnet"
+            if echo $BEINCLUDES | grep -q 'headers/be/bone'; then
+                network_extralibs="-lbind -lsocket"
+            else
+                enable beos_netserver
+                network_extralibs="-lnet"
+            fi
         fi ;;
+	    
     sunos)
         FFSERVERLDFLAGS=""
         SHFLAGS='-shared -Wl,-h,$$(@F)'
