diff -ur automake-1.10.1-orig/lib/config.guess automake-1.10.1/lib/config.guess
--- automake-1.10.1-orig/lib/config.guess	2008-01-19 09:54:57.000000000 +0000
+++ automake-1.10.1/lib/config.guess	2008-03-20 23:52:43.000000000 +0000
@@ -1216,6 +1216,12 @@
     BePC:BeOS:*:*)	# BeOS running on Intel PC compatible.
 	echo i586-pc-beos
 	exit ;;
+    BeMac:Haiku:*:*)
+	echo powerpc-apple-haiku
+	exit ;;
+    BePC:Haiku:*:*)
+	echo i586-pc-haiku
+	exit ;;
     SX-4:SUPER-UX:*:*)
 	echo sx4-nec-superux${UNAME_RELEASE}
 	exit ;;
Only in automake-1.10.1/lib: config.guess~
