diff -urN libharu-2.1.0/configure.in libharu-2.1.0-haiku/configure.in
--- libharu-2.1.0/configure.in	2008-05-27 18:23:31.013107200 +0000
+++ libharu-2.1.0-haiku/configure.in	2010-09-25 20:01:55.000000000 +0000
@@ -21,10 +21,10 @@
 dnl Check for header files
 AC_CHECK_HEADERS(string.h strings.h unistd.h stdint.h)
 
-AC_CHECK_LIB([m], [floor], [LIBS="$LIBS -lm"], [AC_MSG_ERROR([can't continue without libm])])
+AC_CHECK_LIB([m], [floor], [LIBS="$LIBS -lm"], [LIBS="$LIBS"])
 
 DEFAULT_INSTALL_PREFIX="/usr/local"
-STANDARD_PREFIXES="/usr /usr/local /opt /local"
+STANDARD_PREFIXES="/usr /usr/local /opt /local /boot/common"
 
 HPDF_MAJOR_VERSION=2
 HPDF_MINOR_VERSION=1
diff -urN libharu-2.1.0/src/hpdf_u3d.c libharu-2.1.0-haiku/src/hpdf_u3d.c
--- libharu-2.1.0/src/hpdf_u3d.c	2008-05-27 18:23:31.051904512 +0000
+++ libharu-2.1.0-haiku/src/hpdf_u3d.c	2010-09-25 20:13:02.000000000 +0000
@@ -542,18 +542,19 @@
 	return ret;
 }
 
-#define normalize(x, y, z)		\
-{					\
-	HPDF_REAL modulo;			\
-	modulo = sqrtf(x*x + y*y + z*z);	\
-	if (modulo != 0.0)			\
-	{					\
-		x = x/modulo;			\
-		y = y/modulo;			\
-		z = z/modulo;			\
-	}					\
+void normalize(x, y, z)		
+{					
+	HPDF_REAL modulo;			
+	modulo = sqrtf(x*x + y*y + z*z);	
+	if (modulo != 0.0)			
+	{					
+		x = x/modulo;			
+		y = y/modulo;			
+		z = z/modulo;			
+	}					
 }
 
+
 /* building the transformation matrix*/
 /* #1,#2,#3 centre of orbit coordinates (coo)*/
 /* #4,#5,#6 centre of orbit to camera direction vector (c2c)*/
