diff -up file-5.25/configure.ac.orig file-5.25/configure.ac
--- file-5.25/configure.ac.orig	2015-09-16 13:51:05.000000000 +0000
+++ file-5.25/configure.ac
@@ -128,14 +128,14 @@ else
    WARNINGS="-Wall -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith \
        -Wmissing-declarations -Wredundant-decls -Wnested-externs \
        -Wsign-compare -Wreturn-type -Wswitch -Wshadow \
-       -Wcast-qual -Wwrite-strings -Wextra -Wunused-parameter -Wformat=2"
+       -Wcast-qual -Wwrite-strings"
 fi], [
 if test "$GCC" = yes; then
    AC_MSG_RESULT(yes)
    WARNINGS="-Wall -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith \
        -Wmissing-declarations -Wredundant-decls -Wnested-externs \
        -Wsign-compare -Wreturn-type -Wswitch -Wshadow \
-       -Wcast-qual -Wwrite-strings -Wextra -Wunused-parameter -Wformat=2"
+       -Wcast-qual -Wwrite-strings"
 else
    WARNINGS=
    AC_MSG_RESULT(no)
