diff -urN paragui-1.1.8/configure.in paragui-1.1.8-haiku/configure.in
--- paragui-1.1.8/configure.in	2009-02-09 23:49:26.000000000 -0800
+++ paragui-1.1.8-haiku/configure.in	2009-02-09 19:38:54.000000000 -0800
@@ -1,8 +1,9 @@
 dnl Process this file with autoconf to produce a configure script.
 
+m4_pattern_allow([^PKG_SIGC$])
 AC_INIT(README)
 AM_CONFIG_HEADER(include/paraconfig_gnu.h:include/paraconfig_gnu.h.in)
-
+AC_CONFIG_MACRO_DIR([m4])
 TOP_BUILDDIR=`pwd`
 
 AC_SUBST(TOP_BUILDDIR)
