--- KoboDeluxe-0.5.1/configure.org	2016-04-13 17:19:21.956039168 +0200
+++ KoboDeluxe-0.5.1/configure	2016-04-13 17:19:09.067108864 +0200
@@ -5044,6 +5044,11 @@
 echo "${ECHO_T}BeOS; Un*x style install" >&6; }
 		MATHLIB=""
 		;;
+	haiku)
+		{ echo "$as_me:$LINENO: result: HAIKU; Un*x style install" >&5
+echo "${ECHO_T}HAIKU; Un*x style install" >&6; }
+		MATHLIB=""
+		;;
 	mingw32*)
 		{ echo "$as_me:$LINENO: result: Win32 (mingw32); Simple bundle" >&5
 echo "${ECHO_T}Win32 (mingw32); Simple bundle" >&6; }
