From dadf80f3802ad275f3efa30418490efcac63148e Mon Sep 17 00:00:00 2001
From: pisculichi <pisculichi@openmailbox.com>
Date: Fri, 8 Apr 2016 16:13:24 +0000
Subject: [PATCH] Fix search ncurses in haiku

---
 m4/curses.m4 | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/m4/curses.m4 b/m4/curses.m4
index 41c0e6c..00039d0 100644
--- a/m4/curses.m4
+++ b/m4/curses.m4
@@ -57,7 +57,7 @@ dnl
 dnl /*=== Curses version detection defines ===*/
 dnl /* Found some version of curses that we're going to use */
 dnl #undef HAS_CURSES
-dnl
+dnl
 dnl /* Use SunOS SysV curses? */
 dnl #undef USE_SUNOS_CURSES
 dnl
@@ -246,6 +246,24 @@ AC_DEFUN([AC_SEARCH_NCURSES], [
 	       [renamed ncurses in /usr/include/ncurses])

     dnl
+    dnl Try Haiku ncurses
+    dnl Depending if we are building for the primary or secondary arch
+    dnl it can be installed at different locations
+    dnl
+    if $search_ncurses
+    then
+	if test -d /system/develop/headers
+	then
+	    haiku_arch=`getarch`
+	    for haiku_h in `findpaths -a $haiku_arch B_FIND_PATH_HEADERS_DIRECTORY`
+	    do
+		AC_NCURSES($haiku_h, ncurses.h, -lncurses,,
+		    [ncurses in $haiku_h])
+	    done
+        fi
+    fi
+
+    dnl
     dnl We couldn't find ncurses, try SysV curses
     dnl
     if $search_ncurses
--
2.2.2
