From e9617d64b8b8846dd7c7379cab59fe6ee85ca629 Mon Sep 17 00:00:00 2001
From: pisculichi <pisculichi@openmailbox.org>
Date: Fri, 18 Dec 2015 00:12:26 +0000
Subject: [PATCH] Add support for ncurses in Haiku

---
 m4/curses.m4 | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/m4/curses.m4 b/m4/curses.m4
index 82b110f..1def9e9 100644
--- a/m4/curses.m4
+++ b/m4/curses.m4
@@ -131,7 +131,7 @@ AC_DEFUN([AC_CHECK_CURSES],[
 	  if test x$withval = xno ; then
 		search_ncurses=false
 	  elif test x$withval != xyes ; then
-		AC_NCURSES($withval/include, ncurses.h, -L$withval/lib -lncurses, -I$withval/include, "ncurses on $withval/include")
+		AC_NCURSES($withval/include, ncurses.h, -L$withval/lib -lncurses, -I$withval/include, [ncurses on $withval/include])
 	  fi
 	)

@@ -244,6 +244,24 @@ AC_DEFUN([AC_SEARCH_NCURSES], [
 	       [renamed ncurses in /usr/local/include/ncurses])
     AC_NCURSES(/usr/include/ncurses, curses.h, -lncurses, -I/usr/include/ncurses -DRENAMED_NCURSES,
 	       [renamed ncurses in /usr/include/ncurses])
+
+    dnl
+    dnl Try Haiku ncurses
+    dnl Depending if we are building for the primary or secondary arch
+    dnl it can be installed at different locations
+    dnl
+    if $search_ncurses
+    then
+	if test -d /system/develop/headers
+	then
+	    haiku_arch=`getarch`
+	    for haiku_h in `findpaths -a $haiku_arch B_FIND_PATH_HEADERS_DIRECTORY`
+	    do
+		AC_NCURSES($haiku_h, ncurses.h, -lncurses,,
+		    [ncurses in $haiku_h])
+	    done
+        fi
+    fi

     dnl
     dnl We couldn't find ncurses, try SysV curses
--
2.2.2

From e610ab583827cef07a92b354d07a5bb3f1a2e05d Mon Sep 17 00:00:00 2001
From: pisculichi <pisculichi@openmailbox.org>
Date: Fri, 18 Dec 2015 00:29:23 +0000
Subject: [PATCH] Fix variable declarations to support ANSI C

---
 src/fe-common/core/fe-messages.c |  6 ++++--
 src/fe-common/core/themes.c      | 14 +++++++++-----
 src/fe-text/statusbar.c          | 27 ++++++++++++++++-----------
 3 files changed, 29 insertions(+), 18 deletions(-)

diff --git a/src/fe-common/core/fe-messages.c b/src/fe-common/core/fe-messages.c
index 97f3d84..2121653 100644
--- a/src/fe-common/core/fe-messages.c
+++ b/src/fe-common/core/fe-messages.c
@@ -369,6 +369,8 @@ static void sig_message_quit(SERVER_REC *server, const char *nick,
 	GString *chans;
 	GSList *tmp, *windows;
 	char *print_channel;
+	CHANNEL_REC *rec;
+	QUERY_REC *query;
 	int once, count, level = MSGLEVEL_QUITS;

 	if (ignore_check(server, nick, address, NULL, reason, MSGLEVEL_QUITS))
@@ -384,7 +386,7 @@ static void sig_message_quit(SERVER_REC *server, const char *nick,
 	chans = g_string_new(NULL);
 	for (tmp = server->channels; tmp != NULL; tmp = tmp->next) {
 		level = MSGLEVEL_QUITS;
-		CHANNEL_REC *rec = tmp->data;
+		rec = (CHANNEL_REC *) tmp->data;

 		if (!nicklist_find(rec, nick))
 			continue;
@@ -421,7 +423,7 @@ static void sig_message_quit(SERVER_REC *server, const char *nick,
 	if (!once) {
 		/* check if you had query with the nick and
 		   display the quit there too */
-		QUERY_REC *query = query_find(server, nick);
+		query = (QUERY_REC *) query_find(server, nick);
 		if (query != NULL) {
 			printformat(server, nick, level,
 				    TXT_QUIT, nick, address, reason, "");
diff --git a/src/fe-common/core/themes.c b/src/fe-common/core/themes.c
index c0741ce..c81ab11 100644
--- a/src/fe-common/core/themes.c
+++ b/src/fe-common/core/themes.c
@@ -365,9 +365,10 @@ char *theme_format_expand_get(THEME_REC *theme, const char **format)
 	GString *str;
 	char *ret;
 	theme_rm_col dummy, reset;
+	int braces = 1; /* we start with one brace opened */
 	dummy.m[0] = '\0';
 	strcpy(reset.m, "n");
-	int braces = 1; /* we start with one brace opened */
+

 	str = g_string_new(NULL);
 	while (**format != '\0' && braces != 0) {
@@ -864,9 +865,10 @@ void theme_set_default_abstract(const char *key, const char *value)
 static THEME_REC *theme_find(const char *name)
 {
 	GSList *tmp;
+	THEME_REC *rec;

 	for (tmp = themes; tmp != NULL; tmp = tmp->next) {
-		THEME_REC *rec = tmp->data;
+		rec = (THEME_REC *) tmp->data;

 		if (g_ascii_strcasecmp(rec->name, name) == 0)
 			return rec;
@@ -878,9 +880,10 @@ static THEME_REC *theme_find(const char *name)
 static void window_themes_update(void)
 {
 	GSList *tmp;
-
+	WINDOW_REC *rec;
+
 	for (tmp = windows; tmp != NULL; tmp = tmp->next) {
-		WINDOW_REC *rec = tmp->data;
+		rec = (WINDOW_REC *) tmp->data;

 		if (rec->theme_name != NULL)
                         rec->theme = theme_load(rec->theme_name);
@@ -1122,6 +1125,7 @@ static void cmd_format(const char *data)
 	char *module, *key, *value;
 	void *free_arg;
 	int reset;
+	THEME_SEARCH_REC *rec;

 	if (!cmd_get_params(data, &free_arg, 3 | PARAM_FLAG_GETREST | PARAM_FLAG_OPTIONS,
 			    "format", &optlist, &module, &key, &value))
@@ -1149,7 +1153,7 @@ static void cmd_format(const char *data)
 		value = NULL;

 	for (tmp = modules; tmp != NULL; tmp = tmp->next) {
-		THEME_SEARCH_REC *rec = tmp->data;
+		rec = (THEME_SEARCH_REC *) tmp->data;

 		if (module == NULL || g_ascii_strcasecmp(rec->short_name, module) == 0)
 			theme_show(rec, key, value, reset);
diff --git a/src/fe-text/statusbar.c b/src/fe-text/statusbar.c
index b340553..167043b 100644
--- a/src/fe-text/statusbar.c
+++ b/src/fe-text/statusbar.c
@@ -669,10 +669,11 @@ void statusbar_item_default_handler(SBAR_ITEM_REC *item, int get_size_only,
 {
 	SERVER_REC *server;
 	WI_ITEM_REC *wiitem;
+	GString *out;
 	char *tmpstr, *tmpstr2;
 	theme_rm_col reset;
-	strcpy(reset.m, "n");
 	int len;
+	strcpy(reset.m, "n");

 	if (str == NULL)
 		str = statusbar_item_get_value(item);
@@ -709,7 +710,6 @@ void statusbar_item_default_handler(SBAR_ITEM_REC *item, int get_size_only,
 	if (get_size_only) {
 		item->min_size = item->max_size = format_get_length(tmpstr);
 	} else {
-		GString *out;

 		if (item->size < item->min_size) {
                         /* they're forcing us smaller than minimum size.. */
@@ -757,11 +757,11 @@ static void statusbar_update_server(SERVER_REC *server)
 {
         SERVER_REC *item_server;
 	GSList *items;
-
+	SBAR_ITEM_REC *item;
 	items = g_hash_table_lookup(sbar_signal_items,
 				    GINT_TO_POINTER(signal_get_emitted_id()));
 	while (items != NULL) {
-		SBAR_ITEM_REC *item = items->data;
+		item = (SBAR_ITEM_REC *) items->data;

 		item_server = item->bar->parent_window != NULL ?
 			item->bar->parent_window->active->active_server :
@@ -778,11 +778,11 @@ static void statusbar_update_window(WINDOW_REC *window)
 {
         WINDOW_REC *item_window;
 	GSList *items;
-
+	SBAR_ITEM_REC *item;
 	items = g_hash_table_lookup(sbar_signal_items,
 				    GINT_TO_POINTER(signal_get_emitted_id()));
 	while (items != NULL) {
-		SBAR_ITEM_REC *item = items->data;
+		item = (SBAR_ITEM_REC *) items->data;

 		item_window = item->bar->parent_window != NULL ?
 			item->bar->parent_window->active : active_win;
@@ -798,11 +798,12 @@ static void statusbar_update_window_item(WI_ITEM_REC *wiitem)
 {
         WI_ITEM_REC *item_wi;
 	GSList *items;
+	SBAR_ITEM_REC *item;

 	items = g_hash_table_lookup(sbar_signal_items,
 				    GINT_TO_POINTER(signal_get_emitted_id()));
 	while (items != NULL) {
-		SBAR_ITEM_REC *item = items->data;
+		item = (SBAR_ITEM_REC *) items->data;

 		item_wi = item->bar->parent_window != NULL ?
 			item->bar->parent_window->active->active :
@@ -965,6 +966,7 @@ static void statusbar_redraw_needed_items(STATUSBAR_REC *bar)
         WINDOW_REC *old_active_win;
 	GSList *tmp;
 	char *str;
+	SBAR_ITEM_REC *rec;

 	old_active_win = active_win;
         if (bar->parent_window != NULL)
@@ -977,7 +979,7 @@ static void statusbar_redraw_needed_items(STATUSBAR_REC *bar)
 	}

 	for (tmp = bar->items; tmp != NULL; tmp = tmp->next) {
-		SBAR_ITEM_REC *rec = tmp->data;
+		rec = (SBAR_ITEM_REC *) tmp->data;

 		if (rec->dirty ||
 		    (bar->dirty_xpos != -1 &&
@@ -994,6 +996,7 @@ static void statusbar_redraw_needed_items(STATUSBAR_REC *bar)
 void statusbar_redraw_dirty(void)
 {
 	GSList *tmp;
+	STATUSBAR_REC *rec;

 	if (statusbar_need_recreate_items) {
 		statusbar_need_recreate_items = FALSE;
@@ -1001,7 +1004,7 @@ void statusbar_redraw_dirty(void)
 	}

 	for (tmp = active_statusbar_group->bars; tmp != NULL; tmp = tmp->next) {
-		STATUSBAR_REC *rec = tmp->data;
+		rec = (STATUSBAR_REC *) tmp->data;

 		if (rec->dirty) {
                         statusbar_redraw_needed_items(rec);
@@ -1021,9 +1024,10 @@ void statusbar_redraw_dirty(void)
 static void statusbars_remove_unvisible(MAIN_WINDOW_REC *window)
 {
 	GSList *tmp, *next;
+	STATUSBAR_REC *bar;

 	for (tmp = window->statusbars; tmp != NULL; tmp = next) {
-		STATUSBAR_REC *bar = tmp->data;
+		bar = (STATUSBAR_REC *) tmp->data;

 		next = tmp->next;
                 if (!STATUSBAR_IS_VISIBLE(bar->config, window))
@@ -1036,10 +1040,11 @@ static void statusbars_add_visible(MAIN_WINDOW_REC *window)
 	STATUSBAR_GROUP_REC *group;
         STATUSBAR_REC *bar;
 	GSList *tmp;
+	STATUSBAR_CONFIG_REC *config;

         group = active_statusbar_group;
 	for (tmp = group->config_bars; tmp != NULL; tmp = tmp->next) {
-		STATUSBAR_CONFIG_REC *config = tmp->data;
+		config = (STATUSBAR_CONFIG_REC *) tmp->data;

 		if (config->type == STATUSBAR_TYPE_WINDOW &&
 		    STATUSBAR_IS_VISIBLE(config, window) &&
--
2.2.2

From 8a4b09aaf54e25d693f3c52d27181e38797773f9 Mon Sep 17 00:00:00 2001
From: pisculichi <pisculichi@openmailbox.org>
Date: Fri, 18 Dec 2015 00:10:45 +0000
Subject: [PATCH] Support haiku socket

---
 configure.ac | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index f148885..c534a3e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -196,18 +196,8 @@ case "$host_os" in
     ;;
 esac

-
-AC_CHECK_FUNC(socket, [], [
-	AC_CHECK_LIB(socket, socket, [
-		LIBS="$LIBS -lsocket"
-	])
-])
-
-AC_CHECK_FUNC(inet_addr, [], [
-	AC_CHECK_LIB(nsl, inet_addr, [
-		LIBS="$LIBS -lnsl"
-	])
-])
+AC_SEARCH_LIBS([socket], [socket])
+AC_SEARCH_LIBS([inet_addr], [nsl])

 dnl * gcc specific options
 if test "x$ac_cv_prog_gcc" = "xyes"; then
--
2.2.2

From cfff5e78aba54d0f881f212080c2beb7bc090bea Mon Sep 17 00:00:00 2001
From: pisculichi <pisculichi@openmailbox.com>
Date: Sun, 27 Mar 2016 15:04:26 +0000
Subject: [PATCH] Add links to convert html -> text in autogen.sh

---
 autogen.sh | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index cf4b979..5409569 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -25,14 +25,16 @@ cat docs/help/in/Makefile.am.gen|sed "s/@HELPFILES@/$files/g"|sed 's/?/\\?/g'|tr
 files=`echo $files|sed 's/\.in//g'`
 cat docs/help/Makefile.am.gen|sed "s/@HELPFILES@/$files/g"|sed 's/?/\\?/g'|tr '!?' '\t\n' > docs/help/Makefile.am

-# .html -> .txt with lynx or elinks
+# .html -> .txt with lynx, elinks or links
 echo "Documentation: html -> txt..."
 if type lynx >/dev/null 2>&1 ; then
   lynx -dump -nolist docs/faq.html|perl -pe 's/^ *//; if ($_ eq "\n" && $state eq "Q") { $_ = ""; } elsif (/^([QA]):/) { $state = $1 } elsif ($_ ne "\n") { $_ = "   $_"; };' > docs/faq.txt
 elif type elinks >/dev/null 2>&1 ; then
   elinks -dump docs/faq.html|perl -pe 's/^ *//; if ($_ eq "\n" && $state eq "Q") { $_ = ""; } elsif (/^([QA]):/) { $state = $1 } elsif ($_ ne "\n") { $_ = "   $_"; };' > docs/faq.txt
+elif type links >/dev/null 2>&1 ; then
+  links -dump docs/faq.html|perl -pe 's/^ *//; if ($_ eq "\n" && $state eq "Q") { $_ = ""; } elsif (/^([QA]):/) { $state = $1 } elsif ($_ ne "\n") { $_ = "   $_"; };' > docs/faq.txt
 else
-  echo "**Error**: No lynx or elinks present"
+  echo "**Error**: No lynx, elinks or links present"
   exit 1
 fi

--
2.2.2
