diff -Naur cal3d-0.11.0/src/cal3d/hardwaremodel.cpp cal3d-0.11.0-haiku/src/cal3d/hardwaremodel.cpp
--- cal3d-0.11.0/src/cal3d/hardwaremodel.cpp	2004-12-01 22:42:18.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/hardwaremodel.cpp	2009-11-01 20:11:19.000000000 +0000
@@ -12,6 +12,7 @@
 #include "config.h"
 #endif
 
+#include <cstring>
 
 #include "cal3d/error.h"
 #include "cal3d/hardwaremodel.h"
diff -Naur cal3d-0.11.0/src/cal3d/platform.cpp cal3d-0.11.0-haiku/src/cal3d/platform.cpp
--- cal3d-0.11.0/src/cal3d/platform.cpp	2003-12-07 18:22:39.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/platform.cpp	2009-11-01 20:11:59.000000000 +0000
@@ -15,7 +15,7 @@
 //****************************************************************************//
 // Includes                                                                   //
 //****************************************************************************//
-
+#include <cstring>
 #include "cal3d/platform.h"
 
  /*****************************************************************************/
diff -Naur cal3d-0.11.0/src/cal3d/renderer.cpp cal3d-0.11.0-haiku/src/cal3d/renderer.cpp
--- cal3d-0.11.0/src/cal3d/renderer.cpp	2004-12-30 19:20:54.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/renderer.cpp	2009-11-01 20:12:13.000000000 +0000
@@ -15,7 +15,7 @@
 //****************************************************************************//
 // Includes                                                                   //
 //****************************************************************************//
-
+#include <cstring>
 #include "cal3d/error.h"
 #include "cal3d/renderer.h"
 #include "cal3d/coremodel.h"
diff -Naur cal3d-0.11.0/src/cal3d/submesh.cpp cal3d-0.11.0-haiku/src/cal3d/submesh.cpp
--- cal3d-0.11.0/src/cal3d/submesh.cpp	2004-12-01 22:42:18.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/submesh.cpp	2009-11-01 20:12:36.000000000 +0000
@@ -11,7 +11,7 @@
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
-
+#include <cstring>
 #include "cal3d/submesh.h"
 #include "cal3d/error.h"
 #include "cal3d/coresubmesh.h"
diff -Naur cal3d-0.11.0/src/cal3d_converter.cpp cal3d-0.11.0-haiku/src/cal3d_converter.cpp
--- cal3d-0.11.0/src/cal3d_converter.cpp	2006-05-28 17:35:09.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d_converter.cpp	2009-11-01 20:13:18.000000000 +0000
@@ -10,7 +10,7 @@
 //****************************************************************************//
 
 #include "cal3d/cal3d.h"
-
+#include <cstring>
 #define SKELETON 0
 #define MESH 1
 #define ANIMATION 2
