diff -urN cal3d-0.11.0/src/cal3d/buffersource.h cal3d-0.11.0-haiku/src/cal3d/buffersource.h
--- cal3d-0.11.0/src/cal3d/buffersource.h	2003-08-19 02:53:25.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/buffersource.h	2009-11-02 04:24:16.000000000 +0000
@@ -17,7 +17,7 @@
 
 #include "cal3d/global.h"
 #include "cal3d/datasource.h"
-#include <istream>
+#include <iostream>
 
 /**
  * CalBufferSource class.
diff -urN cal3d-0.11.0/src/cal3d/hardwaremodel.cpp cal3d-0.11.0-haiku/src/cal3d/hardwaremodel.cpp
--- cal3d-0.11.0/src/cal3d/hardwaremodel.cpp	2004-12-01 22:42:18.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/hardwaremodel.cpp	2009-11-02 04:18:44.000000000 +0000
@@ -12,6 +12,7 @@
 #include "config.h"
 #endif
 
+#include <cstring>
 
 #include "cal3d/error.h"
 #include "cal3d/hardwaremodel.h"
diff -urN cal3d-0.11.0/src/cal3d/loader.h cal3d-0.11.0-haiku/src/cal3d/loader.h
--- cal3d-0.11.0/src/cal3d/loader.h	2006-06-08 15:12:13.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/loader.h	2009-11-02 04:25:29.000000000 +0000
@@ -17,7 +17,7 @@
 
 
 #include <string>
-#include <istream>
+#include <iostream>
 #include "cal3d/global.h"
 #include "cal3d/datasource.h"
 #include "cal3d/coreanimation.h"
diff -urN cal3d-0.11.0/src/cal3d/platform.cpp cal3d-0.11.0-haiku/src/cal3d/platform.cpp
--- cal3d-0.11.0/src/cal3d/platform.cpp	2003-12-07 18:22:39.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/platform.cpp	2009-11-02 04:18:44.000000000 +0000
@@ -15,7 +15,7 @@
 //****************************************************************************//
 // Includes                                                                   //
 //****************************************************************************//
-
+#include <cstring>
 #include "cal3d/platform.h"
 
  /*****************************************************************************/
diff -urN cal3d-0.11.0/src/cal3d/renderer.cpp cal3d-0.11.0-haiku/src/cal3d/renderer.cpp
--- cal3d-0.11.0/src/cal3d/renderer.cpp	2004-12-30 19:20:54.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/renderer.cpp	2009-11-02 04:18:44.000000000 +0000
@@ -15,7 +15,7 @@
 //****************************************************************************//
 // Includes                                                                   //
 //****************************************************************************//
-
+#include <cstring>
 #include "cal3d/error.h"
 #include "cal3d/renderer.h"
 #include "cal3d/coremodel.h"
diff -urN cal3d-0.11.0/src/cal3d/streamsource.h cal3d-0.11.0-haiku/src/cal3d/streamsource.h
--- cal3d-0.11.0/src/cal3d/streamsource.h	2003-08-19 02:53:25.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/streamsource.h	2009-11-02 04:25:03.000000000 +0000
@@ -17,7 +17,7 @@
 
 #include "cal3d/global.h"
 #include "cal3d/datasource.h"
-#include <istream>
+#include <iostream>
 
 /**
  * CalStreamSource class.
diff -urN cal3d-0.11.0/src/cal3d/submesh.cpp cal3d-0.11.0-haiku/src/cal3d/submesh.cpp
--- cal3d-0.11.0/src/cal3d/submesh.cpp	2004-12-01 22:42:18.000000000 +0000
+++ cal3d-0.11.0-haiku/src/cal3d/submesh.cpp	2009-11-02 04:18:44.000000000 +0000
@@ -11,7 +11,7 @@
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
-
+#include <cstring>
 #include "cal3d/submesh.h"
 #include "cal3d/error.h"
 #include "cal3d/coresubmesh.h"
