diff -r 4bb6aa3dee15 Jamrules
--- pe-2.4.3_hg602/Jamrules	Wed Dec 26 21:33:51 2012 +0100
+++ pe-2.4.3_hg602/Jamrules	Sun Apr 28 01:12:55 2013 +0200
@@ -11,7 +11,7 @@
 	# /boot/develop/lib/x86 to judge whether this is a BeOS compatible and thus
 	# gcc 2 platform. This is not entirely correct, but should be good enough
 	# for the time being.
-	local hasLibStdC++.R4 = [ Glob /boot/develop/lib/x86 : libstdc++.r4.so ] ;
+	local hasLibStdC++.R4 = [ Glob /boot/system/lib : libstdc++.r4.so ] ;
 	if ! $(hasLibStdC++.R4) {
 		IS_GCC_4_PLATFORM = 1 ;
 	}
diff -r 4bb6aa3dee15 build/BuildSettings
--- pe-2.4.3_hg602/build/BuildSettings	Wed Dec 26 21:33:51 2012 +0100
+++ pe-2.4.3_hg602/build/BuildSettings	Sun Apr 28 01:12:55 2013 +0200
@@ -48,7 +48,7 @@
 	CCFLAGS += -isystem /boot/develop/headers/pcre ;
 	C++FLAGS += -isystem /boot/develop/headers/pcre ;
 } else {
-	haveHeadersPcre = [ GLOB /boot/common/include : pcre.h ] ;
+	haveHeadersPcre = [ GLOB /boot/common/develop/headers : pcre.h ] ;
 	if ! $(haveHeadersPcre) {
 		haveHeadersPcre = [ GLOB /boot/home/config/include : pcre.h ] ;
 		if ! $(haveHeadersPcre) {
