diff --git a/build/BuildSettings b/build/BuildSettings
index 3c10281..e9c0c62 100644
--- a/build/BuildSettings
+++ b/build/BuildSettings
@@ -43,7 +43,7 @@ if $(OS) = HAIKU {
 }
 
 # check for pcre...
-local haveHeadersPcre = [ GLOB /boot/common/develop/headers : pcre.h ] ;
+local haveHeadersPcre = [ GLOB /boot/system/develop/headers : pcre.h ] ;
 if ! $(haveHeadersPcre) {
 	Exit "You need to have pcre installed on your system." ;
 }
