--- atanks-6.4/src/debug.h.org	2016-07-04 10:24:30.937164800 +0200
+++ atanks-6.4/src/debug.h	2016-07-04 10:22:22.313786368 +0200
@@ -22,7 +22,7 @@
 # define ATANKS_IS_BSD
 #endif // BSD
 
-#if defined(__linux__) || defined(__linux) || defined(linux) || defined(ATANKS_IS_BSD)
+#if defined(__linux__) || defined(__linux) || defined(linux) || defined(ATANKS_IS_BSD) || defined(__HAIKU__)
 # define ATANKS_IS_LINUX
 #endif // Linux
 
--- atanks-6.4/Makefile.org	2016-07-05 11:51:41.483655680 +0200
+++ atanks-6.4/Makefile	2016-07-05 11:51:30.470024192 +0200
@@ -162,8 +162,8 @@
     export CPLUS_INCLUDE_PATH
   endif
   CPPFLAGS := ${CPPFLAGS} -DNETWORK $(shell allegro-config --cppflags)
-  CXXFLAGS := ${CXXFLAGS} -pthread
-  LDFLAGS  := ${LDFLAGS} $(shell allegro-config --libs) -lm -lpthread
+  CXXFLAGS := ${CXXFLAGS}
+  LDFLAGS  := ${LDFLAGS} $(shell allegro-config --libs) -lnetwork
 endif
 
 
--- atanks-6.4/src/atanks.cpp.org	2016-07-05 12:05:40.893648896 +0200
+++ atanks-6.4/src/atanks.cpp	2016-07-05 12:04:14.533725184 +0200
@@ -598,6 +598,11 @@
 	if (env.sound_enabled) {
 		int32_t sound_type = DIGI_AUTODETECT;
 
+#		if defined (__HAIKU__)
+		switch ( env.sound_driver ) {
+			default: sound_type = DIGI_AUTODETECT; break;
+		}
+#		else
 #       ifdef ATANKS_IS_LINUX
 		switch ( env.sound_driver ) {
 			case SD_OSS: sound_type = DIGI_OSS; break;
@@ -612,7 +617,7 @@
 				sound_type = DIGI_OSS;
 #         endif // UBUNTU
 #       endif // ATANKS_IS_LINUX
-
+#endif //HAIKU
 		int32_t channels = detect_digi_driver(sound_type);
 
 		if (!channels && (DIGI_AUTODETECT != sound_type)) {
