diff -Naur xar-1.5.2/lib/darwinattr.c xar-1.5.2-haiku/lib/darwinattr.c
--- xar-1.5.2/lib/darwinattr.c	2007-12-29 18:49:57.035651584 +0000
+++ xar-1.5.2-haiku/lib/darwinattr.c	2013-01-09 20:55:26.246415360 +0000
@@ -40,7 +40,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <libgen.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <sys/stat.h>
 #include <arpa/inet.h>
 #include "xar.h"
diff -Naur xar-1.5.2/lib/stat.c xar-1.5.2-haiku/lib/stat.c
--- xar-1.5.2/lib/stat.c	2007-12-29 18:49:57.033554432 +0000
+++ xar-1.5.2-haiku/lib/stat.c	2013-01-09 20:55:34.206307328 +0000
@@ -53,7 +53,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/time.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <time.h>
 #include <pwd.h>
 #include <grp.h>
