diff --git a/Audio/Recorder2.cpp b/Audio/Recorder2.cpp
old mode 100755
new mode 100644
index 9e2587f..bc798f5
--- a/Audio/Recorder2.cpp
+++ b/Audio/Recorder2.cpp
@@ -36,7 +36,7 @@
 
 #include <MediaAddOn.h>
 #include <MediaRoster.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 #include "MediaRecorder.h"
 
diff --git a/Audio/ScopeView.cpp b/Audio/ScopeView.cpp
old mode 100755
new mode 100644
index 46c6647..f36d076
--- a/Audio/ScopeView.cpp
+++ b/Audio/ScopeView.cpp
@@ -6,7 +6,7 @@
 #include <interface/Window.h>
 
 #include <string.h>	// for the memset()
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 
 ScopeView::ScopeView(BRect frame, bool bShrink)
diff --git a/Audio/SoundControlsView.cpp b/Audio/SoundControlsView.cpp
old mode 100755
new mode 100644
index d3117af..4f00ac2
--- a/Audio/SoundControlsView.cpp
+++ b/Audio/SoundControlsView.cpp
@@ -11,7 +11,7 @@
 #endif
 
 #include <interface/Window.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 SoundControlsView::SoundControlsView(BRect AFrame, bool bShrink)
 	: ShrinkView(AFrame, (char *) _T("Sound Controls"), bShrink)
diff --git a/Network/client.cpp b/Network/client.cpp
index 7fb44c5..16ea8b8 100755
--- a/Network/client.cpp
+++ b/Network/client.cpp
@@ -14,7 +14,7 @@
 #include <unistd.h>
 
 #include <Alert.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 TClient* TClient::m_instance = 0;
 
diff --git a/Network/connection.cpp b/Network/connection.cpp
index cafe317..8b8b7ec 100644
--- a/Network/connection.cpp
+++ b/Network/connection.cpp
@@ -293,7 +293,7 @@ int CConnection::RecvRtcpUdp(char* pchBuffer, int nMaxLen, sockaddr_in& lRemoteA
 
 	lenAddr = sizeof(lRemoteAddress);  // OliverESP
 //	
-	int rv = recvfrom(socketRecvRtcpUdp, pchBuffer, nMaxLen, 0, (struct sockaddr *)&lRemoteAddress, &lenAddr);
+	int rv = recvfrom(socketRecvRtcpUdp, pchBuffer, nMaxLen, 0, (struct sockaddr *)&lRemoteAddress, (socklen_t *)&lenAddr);
 //	int rv = recvfrom(socketRecvRtcpUdp, pchBuffer, nMaxLen, 0, (sockaddr*)&sinRemote, &lenAddr);  // OliverESP: following a sure way
 
 //	*lRemoteAddress = (ulong)sinRemote.sin_addr.s_addr;  // it updates? don´t need to be done?
diff --git a/Network/network.cpp b/Network/network.cpp
index 3d5a363..ee8a455 100755
--- a/Network/network.cpp
+++ b/Network/network.cpp
@@ -12,7 +12,7 @@
 #include "Output.h"
 #endif
 
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 #include <support/Beep.h>
 
 
diff --git a/UI/About.cpp b/UI/About.cpp
old mode 100755
new mode 100644
index f4e3203..4dd1ffb
--- a/UI/About.cpp
+++ b/UI/About.cpp
@@ -21,7 +21,7 @@
 
 #include <interface/StringView.h> // experimentals...
 //#include "/boot/yt/src/unstable/headers_p/interface/StringView.h"
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 // BRect(100,100,220,400)
 AboutWindow::AboutWindow() :
diff --git a/UI/BeNetMenu.cpp b/UI/BeNetMenu.cpp
old mode 100755
new mode 100644
index 1ec6922..ba39b8c
--- a/UI/BeNetMenu.cpp
+++ b/UI/BeNetMenu.cpp
@@ -6,11 +6,7 @@
 #include "Def.h"
 #endif
 
-#ifdef ZETA
-#include <interface/IconMenu.h>
-#endif
-
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 BeNetMenu::BeNetMenu(BRect frame)
 	:	BMenuBar(frame, "Menu")
@@ -24,14 +20,8 @@ BeNetMenu::BeNetMenu(BRect frame)
 	m_pFile->AddItem(m_pPreferences = new BMenuItem(_T("Preferences"), new BMessage(MSG_MENU_PREFERENCES), 'P'));
 	//OliverESP: I think here preferences is better
 	m_pFile->AddSeparatorItem();
-	m_pFile->AddItem(m_pExit = new BMenuItem(_T("Exit"), new BMessage(MSG_MENU_EXIT), 'X'));
-		
-#ifdef ZETA
-	BIconMenu* app_menu = new BIconMenu(m_pFile);		// Iconified menu app
-	AddItem(app_menu);
-#else
-	AddItem(m_pFile);									// Standard  menu app
-#endif
+	m_pFile->AddItem(m_pExit = new BMenuItem(_T("Exit"), new BMessage(MSG_MENU_EXIT), 'X'));
+	AddItem(m_pFile);
 
 
 	
diff --git a/UI/BeNetWindow.cpp b/UI/BeNetWindow.cpp
index 63f7943..20b9125 100644
--- a/UI/BeNetWindow.cpp
+++ b/UI/BeNetWindow.cpp
@@ -30,7 +30,7 @@
 //#include <storage/Entry.h> // OliverESP: not needed thanks IRA
 #include <app/Roster.h>
 
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 BeNetWindow::BeNetWindow() :
 	BWindow(BRect(0,0,230,209),_T("Whisper"), B_TITLED_WINDOW, B_NOT_RESIZABLE | B_NOT_ZOOMABLE),
diff --git a/UI/BookmarksShortcut.cpp b/UI/BookmarksShortcut.cpp
old mode 100755
new mode 100644
index 2d9e670..e564b5c
--- a/UI/BookmarksShortcut.cpp
+++ b/UI/BookmarksShortcut.cpp
@@ -1,4 +1,4 @@
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 #ifndef _BookmarksShortcut_h_
 #include "BookmarksShortcut.h"
diff --git a/UI/BookmarksWindow.cpp b/UI/BookmarksWindow.cpp
old mode 100755
new mode 100644
index e36fe47..ff005ea
--- a/UI/BookmarksWindow.cpp
+++ b/UI/BookmarksWindow.cpp
@@ -7,7 +7,7 @@
 #endif
 
 #include <Alert.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 #include "CLVEasyItem.h"
 
diff --git a/UI/ConnectView.cpp b/UI/ConnectView.cpp
old mode 100755
new mode 100644
index d8d5d52..97a4647
--- a/UI/ConnectView.cpp
+++ b/UI/ConnectView.cpp
@@ -38,7 +38,7 @@
 #include <Alert.h>
 #include <Window.h>
 
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 #include <string.h>
 #include <OS.h>
 
diff --git a/UI/InformationView.cpp b/UI/InformationView.cpp
old mode 100755
new mode 100644
index 9078284..e0f4cf1
--- a/UI/InformationView.cpp
+++ b/UI/InformationView.cpp
@@ -1,4 +1,4 @@
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 #ifndef __INFORMATION_VIEW__
 #include "InformationView.h"
diff --git a/UI/OsciloscopeView.cpp b/UI/OsciloscopeView.cpp
old mode 100755
new mode 100644
index 363bfef..7969012
--- a/UI/OsciloscopeView.cpp
+++ b/UI/OsciloscopeView.cpp
@@ -1,4 +1,4 @@
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 #ifndef _PictureView_h
 #include "OsciloscopeView.h"
diff --git a/UI/PictureView.cpp b/UI/PictureView.cpp
old mode 100755
new mode 100644
index fe442de..dea6ac4
--- a/UI/PictureView.cpp
+++ b/UI/PictureView.cpp
@@ -1,4 +1,4 @@
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 #ifndef _PictureView_h
 #include "PictureView.h"
diff --git a/UI/PreferencesViews.cpp b/UI/PreferencesViews.cpp
old mode 100755
new mode 100644
index fa1b510..a291843
--- a/UI/PreferencesViews.cpp
+++ b/UI/PreferencesViews.cpp
@@ -17,7 +17,7 @@
 #endif
 
 #include <stdlib.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 /************************************/
 /* General Preferences              */
diff --git a/UI/PreferencesWindow.cpp b/UI/PreferencesWindow.cpp
old mode 100755
new mode 100644
index d6185ba..4e6057c
--- a/UI/PreferencesWindow.cpp
+++ b/UI/PreferencesWindow.cpp
@@ -12,7 +12,7 @@
 
 //#include <experimental/TabView.h>
 #include <interface/TabView.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 PreferencesWindow::PreferencesWindow() 
 	: BWindow(BRect(0,0,280,175), _T("Preferences"), B_TITLED_WINDOW, B_NOT_RESIZABLE | B_NOT_ZOOMABLE)
diff --git a/UI/RemoteWindow.cpp b/UI/RemoteWindow.cpp
old mode 100755
new mode 100644
index f38514d..e29e1d2
--- a/UI/RemoteWindow.cpp
+++ b/UI/RemoteWindow.cpp
@@ -18,7 +18,7 @@
 #include "network.h"
 #endif
 
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 RemoteWindow::RemoteWindow(BeNetWindow* window) :
 	BWindow(BRect(0,0,0,0), "RemoteUser", B_TITLED_WINDOW_LOOK, B_NORMAL_WINDOW_FEEL, B_NOT_RESIZABLE | B_NOT_ZOOMABLE, B_CURRENT_WORKSPACE),
diff --git a/UI/StatusBar.cpp b/UI/StatusBar.cpp
old mode 100755
new mode 100644
index d26da7a..4b8cacb
--- a/UI/StatusBar.cpp
+++ b/UI/StatusBar.cpp
@@ -15,7 +15,7 @@
 #endif
 
 #include <stdio.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 StatusBar::StatusBar(BRect frame) :
 	BView(frame, "StatusBar", B_FOLLOW_NONE, B_WILL_DRAW)
diff --git a/UI/ToolBarView.cpp b/UI/ToolBarView.cpp
old mode 100755
new mode 100644
index c3ccea7..2160f63
--- a/UI/ToolBarView.cpp
+++ b/UI/ToolBarView.cpp
@@ -14,7 +14,7 @@
 #include "BitmapCatalog.h"
 #endif
 
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 ToolBarView::ToolBarView(BRect frame) :
 	BView(frame, "ToolBarView", 0, B_WILL_DRAW)
diff --git a/UI/WaitWindow.cpp b/UI/WaitWindow.cpp
old mode 100755
new mode 100644
index e9397d7..a499ae3
--- a/UI/WaitWindow.cpp
+++ b/UI/WaitWindow.cpp
@@ -8,7 +8,7 @@
 
 #include <View.h>
 #include <Button.h>
-#include <locale/Locale.h>
+#include "locale/Locale.h"
 
 WaitWindow::WaitWindow() :
 	BWindow(BRect(300,300,500,350), "Connecting...",B_MODAL_WINDOW_LOOK, B_MODAL_APP_WINDOW_FEEL, B_NOT_RESIZABLE | B_NOT_ZOOMABLE, B_CURRENT_WORKSPACE)
diff --git a/UI/email/BeNetMailIPPictureView.cpp b/UI/email/BeNetMailIPPictureView.cpp
old mode 100755
new mode 100644
index 06db74a..699d94a
--- a/UI/email/BeNetMailIPPictureView.cpp
+++ b/UI/email/BeNetMailIPPictureView.cpp
@@ -1,5 +1,3 @@
-#include <locale/Locale.h>
-
 /***********************************************************************************************\
 *	BeNetMailIPInfoView.h																		*
 *************************************************************************************************
@@ -16,6 +14,8 @@
 #include "BitmapCatalog.h"
 #endif
 
+#include "locale/Locale.h"
+
 /*=============================================================================================*\
 |	BeNetMailIPPictureView																		|
 +-----------------------------------------------------------------------------------------------+
diff --git a/UI/email/EMailIP.cpp b/UI/email/EMailIP.cpp
index bbe4c14..444e179 100755
--- a/UI/email/EMailIP.cpp
+++ b/UI/email/EMailIP.cpp
@@ -6,6 +6,9 @@
 #include <Entry.h>
 #include <Directory.h>
 #include <Path.h>
+#include <MailDaemon.h>
+
+#include <mail_encoding.h>
 
 #include <stdio.h>
 #include <string.h>
@@ -46,7 +49,7 @@ bool EMailIP::SendIP(char * pzMailTo, char * pzMyNickName, char * pzMyIPAddress,
 	{
 		return true;
 	}
-	send_queued_mail();
+	BMailDaemon::SendQueuedMail();
 	return false;
 }
 
@@ -60,8 +63,9 @@ PEOPLE_ID_NODE * EMailIP::CheckMail(bool bCheckAll)
 	int32 iResult = 0;
 	m_pFirstID = NULL;
 
-	check_for_mail(&iResult);
-	if(!bCheckAll && iResult == 0)
+	status_t err = BMailDaemon::CheckMail(true);
+	iResult = BMailDaemon::CountNewMessages(true);
+	if(!bCheckAll && (iResult == 0 || err < B_OK))
 	{
 		return NULL;	
 	}
@@ -221,7 +225,7 @@ PEOPLE_ID_NODE* EMailIP::GetInfo(BFile * pFile)
 		
 		size = pFile->Read(pzBuffer, 64000);
 		
-		size = decode_base64(pzBufferOut, pzBuffer, size, false);
+		size = decode_base64(pzBufferOut, pzBuffer, size);
 		filePicture.Write(pzBufferOut, size);
 
 		filePicture.Unset();		
diff --git a/makefile b/makefile
index 329245e..1c55f17 100644
--- a/makefile
+++ b/makefile
@@ -119,7 +119,6 @@ LIBS :=   \
 	translation \
 	media \
 	socket \
-	zeta \
 	bind \
 	be \
 	stdc++.r4
@@ -295,5 +294,5 @@ LANG_FILES		:= $(wildcard Language/Dictionaries/$(NAME).*)
 #       ie: Language/Dictionaries 
 LANG_FILES_DIR  := Language/Dictionaries
 
-include /boot/develop/etc/makefile-engine
+include ${BUILDHOME}/etc/makefile-engine
 
