--- a/configure.ac	2013-10-15 09:50:02.011796480 +0000
+++ b/configure.ac	2013-11-08 21:25:01.681574400 +0000
@@ -43,6 +43,7 @@
 libmikmod_openbsd=no
 libmikmod_os2=no
 libmikmod_djgpp=no
+libmikmod_haiku=no
 
 case $host_os in
 	mingw*) libmikmod_mingw=yes ;;
@@ -53,6 +54,7 @@
 	openbsd*) libmikmod_openbsd=yes ;;
 	emx*) libmikmod_os2=yes;;
 	*djgpp) libmikmod_djgpp=yes ;;
+	haiku*) libmikmod_haiku=yes ;;
 esac
 
 # ==============================================================
@@ -444,9 +446,9 @@
 	)
 	if test $libmikmod_cv_gcc_pthread = yes
 	then
-		dnl There is no need for -pthread on darwin, and gcc
+		dnl There is no need for -pthread on darwin or haiku, and gcc
 		dnl complains about unrecognized option -pthread
-		if test $libmikmod_darwin = no
+		if test $libmikmod_darwin = no && test $libmikmod_haiku = no
 		then
 			libmikmod_threads=""
 			CFLAGS="$CFLAGS -pthread"
