diff -ur ruby-1.9.1-p243/configure.in ruby-1.9.1-p243-haiku/configure.in
--- ruby-1.9.1-p243/configure.in	2009-10-27 21:03:32.000000000 -0600
+++ ruby-1.9.1-p243-haiku/configure.in	2009-10-27 21:03:25.000000000 -0600
@@ -1466,7 +1466,7 @@
                             ;;
 			  i586*)
 			    : ${LDSHARED="ld -shared"}
-			    DLDFLAGS="$DLDFLAGS -L/boot/develop/lib/x86 -lbe -lroot"
+			    DLDFLAGS="$DLDFLAGS -L/boot/develop/lib/x86 -lbe -lroot -L/boot/common/lib"
 			    ;;
 			esac
 			: ${LIBPATHENV=LIBRARY_PATH}
diff -ur ruby-1.9.1-p243/ext/nkf/nkf-utf8/nkf.h ruby-1.9.1-p243-haiku/ext/nkf/nkf-utf8/nkf.h
--- ruby-1.9.1-p243/ext/nkf/nkf-utf8/nkf.h	2008-11-11 01:41:20.000000000 -0700
+++ ruby-1.9.1-p243-haiku/ext/nkf/nkf-utf8/nkf.h	2009-10-26 15:14:01.000000000 -0600
@@ -164,6 +164,11 @@
 # ifndef HAVE_LOCALE_H
 #  define HAVE_LOCALE_H
 # endif
+#elif defined(__HAIKU__)
+# undef HAVE_LANGINFO_H
+# ifndef HAVE_LOCALE_H
+#  define HAVE_LOCALE_H
+# endif
 #else
 # ifndef HAVE_LANGINFO_H
 #  define HAVE_LANGINFO_H
diff -ur ruby-1.9.1-p243/signal.c ruby-1.9.1-p243-haiku/signal.c
--- ruby-1.9.1-p243/signal.c	2009-02-22 02:48:12.000000000 -0700
+++ ruby-1.9.1-p243-haiku/signal.c	2009-10-26 15:03:07.000000000 -0600
@@ -34,7 +34,7 @@
 # define ATOMIC_DEC(var) (--(var))
 #endif
 
-#ifdef __BEOS__
+#if defined __BEOS__ || defined __HAIKU__
 #undef SIGBUS
 #endif
 
@@ -586,7 +586,7 @@
 static RETSIGTYPE
 sigsegv(int sig SIGINFO_ARG)
 {
-#ifdef USE_SIGALTSTACK
+#if defined USE_SIGALTSTACK && defined SA_SIGINFO
     int ruby_stack_overflowed_p(const rb_thread_t *, const void *);
     NORETURN(void ruby_thread_stack_overflow(rb_thread_t *th));
     rb_thread_t *th = GET_THREAD();
