diff -urN mpg123-1.12.1/configure.ac mpg123-1.12.1-haiku/configure.ac
--- mpg123-1.12.1/configure.ac	2010-03-31 08:27:37.061603840 +0000
+++ mpg123-1.12.1-haiku/configure.ac	2010-05-31 20:43:38.656932864 +0000
@@ -545,10 +545,10 @@
       ;;
     esac
   ;;
-  i?86-*-dragonfly* | i?86-*-freebsd* | i?86-*-midnightbsd* | i?86-*-mirbsd* | i?86-*-netbsd* | i?86-*-openbsd*)
+  i?86-*-dragonfly* | i?86-*-freebsd* | i?86-*-midnightbsd* | i?86-*-mirbsd* | i?86-*-netbsd* | i?86-*-openbsd* | i?86-*-haiku*)
     cpu_type="x86"
   ;;
-  x86_64-*-dragonfly* | x86_64-*-freebsd* | x86_64-*-midnightbsd* | x86_64-*-mirbsd* | x86_64-*-netbsd* | x86_64-*-openbsd*)
+  x86_64-*-dragonfly* | x86_64-*-freebsd* | x86_64-*-midnightbsd* | x86_64-*-mirbsd* | x86_64-*-netbsd* | x86_64-*-openbsd* | x86_64-*-haiku*)
     cpu_type="x86-64_all"
   ;;
   *-*-dragonfly* | *-*-freebsd* | *-*-midnightbsd* | *-*-mirbsd* | *-*-netbsd* | *-*-openbsd*)
@@ -963,7 +963,7 @@
 fi
 
 dnl trying to get that socket lib settled in one line
-AC_SEARCH_LIBS(gethostbyname, nsl socket) 
+AC_SEARCH_LIBS(gethostbyname, nsl socket network) 
 dnl OK, two lines... Solaris needs -lnsl -lsocket
 AC_SEARCH_LIBS(socket, socket) 
 
