diff -urN lua-5.1.3/Makefile lua-5.1.3-haiku/Makefile
--- lua-5.1.3/Makefile	2008-01-17 17:53:37.000000000 +0000
+++ lua-5.1.3-haiku/Makefile	2008-07-24 20:31:06.000000000 +0000
@@ -33,7 +33,7 @@
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
 # Convenience platforms targets.
-PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
+PLATS= aix ansi bsd freebsd generic haiku linux macosx mingw posix solaris
 
 # What to install.
 TO_BIN= lua luac
diff -urN lua-5.1.3/src/Makefile lua-5.1.3-haiku/src/Makefile
--- lua-5.1.3/src/Makefile	2008-01-19 19:37:58.000000000 +0000
+++ lua-5.1.3-haiku/src/Makefile	2008-07-24 20:19:12.000000000 +0000
@@ -20,7 +20,7 @@
 
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
-PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
+PLATS= aix ansi bsd freebsd generic haiku linux macosx mingw posix solaris
 
 LUA_A=	liblua.a
 CORE_O=	lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o \
@@ -95,6 +95,9 @@
 generic:
 	$(MAKE) all MYCFLAGS=
 
+haiku:
+	$(MAKE) all MYCFLAGS="-DLUA_ANSI" LIBS=$(MYLIBS)
+
 linux:
 	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl -lreadline -lhistory -lncurses"
 
