--- /boot/home/Desktop/GNUmakefile.in	2014-12-06 15:06:33.557580288 +0000
+++ privoxy-3.0.21-stable/GNUmakefile.in	2013-02-05 14:04:11.051642368 +0000
@@ -889,7 +889,7 @@
 	@# FIXME: group/user validation is overly convoluted.
 	@# If superuser install ... we require a minimum of group ownership
 	@# of those files the daemon writes to, to be non-root owned.
-	@if [ "`$(ID) |sed 's/(.*//' |sed 's/.*=//'`" = "0" ] ;then\
+	@if [ "`$(ID) |sed 's/(.*//' |sed 's/.*=//'`" = "0" ] && [$host_os != haiku] ;then\
 		if [ x$(USER) = x ] || [ $(USER) = root ]; then \
 			if [ x$(GROUP) = x ] || [ $(GROUP) = root ]; then \
 				if [ "`$(ID) privoxy`" ] && \

