--- /boot/home/Desktop/GNUmakefile.in	2014-12-06 15:06:33.557580288 +0000
+++ privoxy-3.0.21-stable/GNUmakefile.in	2013-02-05 14:04:11.051642368 +0000
@@ -219,10 +219,10 @@
 PROJECT_H_DEPS = project.h $(REGEX_HDRS) $(PCRS_HDRS) @STATIC_PCRE_ONLY@pcre/pcre.h
 
 # Socket libraries for platforms that need them explicitly defined
-SOCKET_LIB   = @SOCKET_LIB@
+SOCKET_LIB   = -lnetwork
 
 # PThreads library, if needed.
-PTHREAD_LIB  = @PTHREAD_ONLY@@PTHREAD_LIB@
+PTHREAD_LIB  = 
 
 SRCS         = $(C_SRC)  $(W32_SRC)  $(PCRS_SRC)  $(PCRE_SRC)  $(REGEX_SRC)
 OBJS         = $(C_OBJS) $(W32_OBJS) $(PCRS_OBJS) $(PCRE_OBJS) $(REGEX_OBJS)
@@ -239,7 +239,7 @@
 # Win32 GUI build.
 # The flag "-pthread" is required if using Pthreads under Linux (and
 # possibly other OSs).
-SPECIAL_CFLAGS = @SPECIAL_CFLAGS@
+SPECIAL_CFLAGS = 
 
 # Add your flags here
 OTHER_CFLAGS =
@@ -889,7 +889,7 @@
 	@# FIXME: group/user validation is overly convoluted.
 	@# If superuser install ... we require a minimum of group ownership
 	@# of those files the daemon writes to, to be non-root owned.
-	@if [ "`$(ID) |sed 's/(.*//' |sed 's/.*=//'`" = "0" ] ;then\
+	@if [ "`$(ID) |sed 's/(.*//' |sed 's/.*=//'`" = "0" ] && [$host_os != haiku] ;then\
 		if [ x$(USER) = x ] || [ $(USER) = root ]; then \
 			if [ x$(GROUP) = x ] || [ $(GROUP) = root ]; then \
 				if [ "`$(ID) privoxy`" ] && \

