From 205c757a02136dea9679088c686f902febd9997d Mon Sep 17 00:00:00 2001
From: nilFinx <nulflox@disroot.org>
Date: Sun, 12 Oct 2025 18:05:11 +0900
Subject: [PATCH 2/2] Add Haiku build

---
 makefile           | 6 +++++-
 src/host/premake.h | 3 +++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/makefile b/makefile
index f551fcf..d97884a 100644
--- a/makefile
+++ b/makefile
@@ -4,7 +4,7 @@
 #
 
 UNAME := $(shell uname)
-ifeq ($(UNAME),$(filter $(UNAME),Linux Darwin SunOS FreeBSD GNU/kFreeBSD NetBSD OpenBSD GNU))
+ifeq ($(UNAME),$(filter $(UNAME),Linux Darwin SunOS FreeBSD GNU/kFreeBSD NetBSD OpenBSD GNU Haiku))
 ifeq ($(UNAME),$(filter $(UNAME),Darwin))
 OS=darwin
 else
@@ -14,10 +14,14 @@ else
 ifeq ($(UNAME),$(filter $(UNAME),FreeBSD GNU/kFreeBSD NetBSD OpenBSD))
 OS=bsd
 else
+ifeq ($(UNAME),$(filter $(UNAME),Haiku))
+OS=haiku
+else
 OS=linux
 endif
 endif
 endif
+endif
 else
 OS=windows
 endif
diff --git a/src/host/premake.h b/src/host/premake.h
index f0782db..eb4a12b 100644
--- a/src/host/premake.h
+++ b/src/host/premake.h
@@ -27,6 +27,9 @@
 #elif defined(__OS2__)
 #define PLATFORM_OS2      (1)
 #define PLATFORM_STRING   "os2"
+#elif defined(__HAIKU__)
+#define PLATFORM_HAIKU (1)
+#define PLATFORM_STRING "haiku"
 #else
 #define PLATFORM_WINDOWS  (1)
 #define PLATFORM_STRING   "windows"
-- 
2.51.0

