diff --git a/makefile b/makefile
--- a/makefile
+++ b/makefile
@@ -4,7 +4,7 @@
 #
 
 UNAME := $(shell uname)
-ifeq ($(UNAME),$(filter $(UNAME),Linux Darwin SunOS FreeBSD GNU/kFreeBSD NetBSD OpenBSD GNU))
+ifeq ($(UNAME),$(filter $(UNAME),Linux Darwin SunOS FreeBSD GNU/kFreeBSD NetBSD OpenBSD GNU Haiku))
 ifeq ($(UNAME),$(filter $(UNAME),Darwin))
 OS=darwin
 else
@@ -14,10 +14,14 @@ else
 ifeq ($(UNAME),$(filter $(UNAME),FreeBSD GNU/kFreeBSD NetBSD OpenBSD))
 OS=bsd
 else
+ifeq ($(UNAME),$(filter $(UNAME),Haiku))
+OS=haiku
+else
 OS=linux
 endif
 endif
 endif
+endif
 else
 OS=windows
 endif
@@ -43,6 +47,7 @@
 	$(SILENT) $(GENIE) --to=../build/$(PROJECT_TYPE).linux   --os=linux $(PROJECT_TYPE)
  	$(SILENT) $(GENIE) --to=../build/$(PROJECT_TYPE).darwin  --os=macosx --platform=universal32 $(PROJECT_TYPE)
  	$(SILENT) $(GENIE) --to=../build/$(PROJECT_TYPE).freebsd --os=bsd $(PROJECT_TYPE)
+	$(SILENT) $(GENIE) --to=../build/$(PROJECT_TYPE).haiku   --os=haiku $(PROJECT_TYPE)
 
 rebuild:
        $(SILENT) $(MAKE) -C build/$(PROJECT_TYPE).$(OS) clean all
diff --git a/scripts/genie.lua b/scripts/genie.lua
index 7e64ad9..b20f5c7 100644
--- a/scripts/genie.lua
+++ b/scripts/genie.lua
@@ -86,6 +86,11 @@
                        buildoptions { "-mmacosx-version-min=11.0" }
                        linkoptions  { "-mmacosx-version-min=11.0" }
 
+		configuration "haiku"
+			targetdir   "../bin/haiku"
+			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN" }
+			links       { "root" }
+
 		configuration {}
 
 
diff --git a/scripts/release.lua b/scripts/release.lua
index f320091..15aa073 100644
--- a/scripts/release.lua
+++ b/scripts/release.lua
@@ -40,6 +40,7 @@ function dorelease()
 	exec(_PREMAKE_COMMAND .. " /to=../build/gmake.windows /os=windows gmake")
 	exec(_PREMAKE_COMMAND .. " /to=../build/gmake.linux /os=linux gmake")
 	exec(_PREMAKE_COMMAND .. " /to=../build/gmake.darwin /os=macosx /platform=universal32 gmake")
+	exec(_PREMAKE_COMMAND .. " /to=../build/gmake.haiku /os=haiku gmake")
 
 	print("")
 	print( "Finished.")
diff --git a/src/base/cmdline.lua b/src/base/cmdline.lua
index c3357b7..dcdc637 100644
--- a/src/base/cmdline.lua
+++ b/src/base/cmdline.lua
@@ -57,6 +57,7 @@
 			{ "macosx",   "Apple Mac OS X" },
 			{ "solaris",  "Solaris" },
 			{ "windows",  "Microsoft Windows" },
+			{ "haiku",    "Haiku" },
 		}
 	}
 
diff --git a/src/base/os.lua b/src/base/os.lua
index ae88c0a..b2c1cfa 100644
--- a/src/base/os.lua
+++ b/src/base/os.lua
@@ -66,6 +66,9 @@
 			if os.is("macosx") then
 				formats = { "lib%s.dylib", "%s.dylib" }
 				path = os.getenv("DYLD_LIBRARY_PATH")
+			elseif os.is("haiku") then
+				formats = { "lib%s.so", "%s.so" }
+				path = os.getenv("LIBRARY_PATH")
 			else
 				formats = { "lib%s.so", "%s.so" }
 				path = os.getenv("LD_LIBRARY_PATH") or ""
diff --git a/src/host/premake.h b/src/host/premake.h
index f0782db..5a3a636 100644
--- a/src/host/premake.h
+++ b/src/host/premake.h
@@ -27,6 +27,9 @@
 #elif defined(__OS2__)
 #define PLATFORM_OS2      (1)
 #define PLATFORM_STRING   "os2"
+#elif defined(__HAIKU__)
+#define PLATFORM_HAIKU      (1)
+#define PLATFORM_STRING   "haiku"
 #else
 #define PLATFORM_WINDOWS  (1)
 #define PLATFORM_STRING   "windows"
-- 
2.21.0

