--- Makefile.am	2010-10-11 22:35:32.020447232 +0200
+++ Makefile.am	2010-11-18 11:10:09.946339840 +0100
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = foreign no-dependencies

 SVN_DEF = -D'SVN_REV="$(shell svnversion -n .)"'
-AM_CFLAGS = -g -O3 -std=gnu99 -ffast-math -DPACKAGE_DATA_DIR=\"$(pkgdatadir)\" $(SVN_DEF)
+AM_CFLAGS = -g -O3 -ffast-math -DPACKAGE_DATA_DIR=\"$(pkgdatadir)\" $(SVN_DEF)

 ACLOCAL_AMFLAGS = -I m4

