diff -Naur W6Haiku/InfoView.cpp W6Haiku-haiku/InfoView.cpp
--- W6Haiku/InfoView.cpp	2008-08-16 02:57:03.065798144 +0000
+++ W6Haiku-haiku/InfoView.cpp	2013-01-13 11:59:46.657457152 +0000
@@ -280,7 +280,7 @@
 	sprintf(str2,_jour->month_name);  	strcat(str,str2);
 	sprintf(str2," %i, %i",_jour->month_counter, _jour->year);	strcat(str,str2);
 	strcat(str,")");
-	Window()->SetTitle(str);
+	//Window()->SetTitle(str);
 
 	
 	//A l'ancienne
@@ -567,4 +567,4 @@
    			x->Close();
 			delete upBitmap;
    			delete downBitmap;  			
-}
\ No newline at end of file
+}
diff -Naur W6Haiku/W6.cpp W6Haiku-haiku/W6.cpp
--- W6Haiku/W6.cpp	2008-08-16 04:35:06.002359296 +0000
+++ W6Haiku-haiku/W6.cpp	2013-01-13 12:00:52.285736960 +0000
@@ -89,9 +89,9 @@
  	
  	
  	//classe qui tourne en permanence et qui fait les daypassed (et par conséquent, les battle etc. 
-	run_me = new Running(_country,_group,_politic,_weapon,_jour);
+	//run_me = new Running(_country,_group,_politic,_weapon,_jour);
 	runWin = new BWindow(BRect(0,0,0,0), "W6 time run", B_TITLED_WINDOW,  B_NOT_RESIZABLE); 
-	runWin->AddChild(run_me); //sinon le pulse fonctionne pas...					  
+	//runWin->AddChild(run_me); //sinon le pulse fonctionne pas...
 	runWin->Lock();  
 	runWin->Show(); //faut faire show pour lancer le pulse dans les views
 	runWin->Hide(); //quitte à hider la window ensuite
diff -Naur W6Haiku/makefile W6Haiku/makefil
e-new
--- W6Haiku/makefile    2014-01-09 17:29:39.128974848 +0100
+++ W6Haiku/makefile-new        2014-01-09 17:31:10.101974016 +0100
@@ -3,9 +3,12 @@
 NAME := W6Haiku
 TYPE := APP
 SRCS := BarView.cpp BubbleHelper.cpp BuyWindow.cpp ChooseView.cpp ChooseWindow.cpp ColGrpView.cpp ColorWindow.cpp EditGrpWindow.cpp EditPolWindow.cpp EditWindow.cpp EditorGrpView.cpp EditorPolView.cpp EditorView.cpp FlagView.cpp GroupView.cpp GroupWindow.cpp HistoryView.cpp InfoView.cpp InfoWindow.cpp Init.cpp MapView.cpp OrderWindow.cpp PowerView.cpp Running.cpp SendWindow.cpp SpeedWindow.cpp StatusView.cpp TopView.cpp TxtView.cpp W6.cpp W6Window.cpp WeatherView.cpp attack.cpp classSound.cpp classSoundMaster.cpp country.cpp day.cpp group.cpp politic.cpp weapon.cpp 
-LIBS := be media png tracker root translation
+LIBS := be media tracker root translation
 OPTIMIZE := FULL
 RSRCS := img.rsrc w6.rsrc 
 CC=g++#Additional makerules go here, use '#' to comment out lines
 
-include /boot/develop/etc/makefile-engine
+## include the makefile-engine
+DEVEL_DIRECTORY := \
+       $(shell findpaths -r "makefile_engine" B_FIND_PATH_DEVELOP_DIRECTORY)
+include $(DEVEL_DIRECTORY)/etc/makefile-engine
