diff --git a/bomber.c b/bomber.c
index 6c66afe..78a6fb0 100644
--- a/bomber.c
+++ b/bomber.c
@@ -14,6 +14,9 @@
 #include <fcntl.h>
 #include <stdarg.h>
 
+#ifdef __HAIKU__
+#include <libgen.h>
+#endif
 
 void domode(void);
 int iterate(void);
@@ -1917,6 +1920,10 @@ int main(int argc,char **argv)
 int i;
 char *p;
 
+#ifdef __HAIKU__
+	chdir(dirname(argv[0]));
+#endif
+
 	mname=0;
 
 	for(i=1;i<argc;++i)