diff -urN httpd-2.2.22/server/mpm_common.c httpd-2.2.22-haiku/server/mpm_common.c
--- httpd-2.2.22/server/mpm_common.c	2010-10-07 16:56:54.055836672 +0000
+++ httpd-2.2.22-haiku/server/mpm_common.c	2011-02-05 10:58:12.295436288 +0000
@@ -950,7 +950,7 @@
 
 static const char *dash_k_arg;
 
-static int send_signal(pid_t pid, int sig)
+static int _send_signal(pid_t pid, int sig)
 {
     if (kill(pid, sig) < 0) {
         ap_log_error(APLOG_MARK, APLOG_STARTUP, errno, NULL,
@@ -1007,7 +1007,7 @@
             printf("%s\n", status);
         }
         else {
-            send_signal(otherpid, SIGTERM);
+            _send_signal(otherpid, SIGTERM);
         }
         return 1;
     }
@@ -1017,7 +1017,7 @@
             printf("httpd not running, trying to start\n");
         }
         else {
-            *exit_status = send_signal(otherpid, SIGHUP);
+            *exit_status = _send_signal(otherpid, SIGHUP);
             return 1;
         }
     }
@@ -1027,7 +1027,7 @@
             printf("httpd not running, trying to start\n");
         }
         else {
-            *exit_status = send_signal(otherpid, AP_SIG_GRACEFUL);
+            *exit_status = _send_signal(otherpid, AP_SIG_GRACEFUL);
             return 1;
         }
     }
@@ -1038,7 +1038,7 @@
             printf("%s\n", status);
         }
         else {
-            *exit_status = send_signal(otherpid, AP_SIG_GRACEFUL_STOP);
+            *exit_status = _send_signal(otherpid, AP_SIG_GRACEFUL_STOP);
         }
 #else
         printf("httpd MPM \"" MPM_NAME "\" does not support graceful-stop\n");
