diff -urN gettext-0.18.1.1/gettext-runtime/libasprintf/configure.ac gettext-0.18.1.1-haiku/gettext-runtime/libasprintf/configure.ac
--- gettext-0.18.1.1/gettext-runtime/libasprintf/configure.ac	2010-06-06 12:49:57.044040192 +0000
+++ gettext-0.18.1.1-haiku/gettext-runtime/libasprintf/configure.ac	2011-01-14 15:54:38.003670016 +0000
@@ -44,7 +44,7 @@
 case "$host_os" in
   # On Cygwin, without -no-undefined, a warning is emitted and only a static
   # library is built.
-  beos* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
+  beos* | haiku* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
   *) LTNOUNDEF='' ;;
 esac
 AC_SUBST([LTNOUNDEF])
diff -urN gettext-0.18.1.1/gettext-tools/configure.ac gettext-0.18.1.1-haiku/gettext-tools/configure.ac
--- gettext-0.18.1.1/gettext-tools/configure.ac	2010-06-06 20:04:04.061079552 +0000
+++ gettext-0.18.1.1-haiku/gettext-tools/configure.ac	2011-01-14 15:54:53.031981568 +0000
@@ -100,7 +100,7 @@
 case "$host_os" in
   # On Cygwin, without -no-undefined, a warning is emitted and only a static
   # library is built.
-  beos* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
+  beos* | haiku* | mingw* | cygwin*) LTNOUNDEF='-no-undefined' ;;
   *) LTNOUNDEF='' ;;
 esac
 AC_SUBST([LTNOUNDEF])
