--- lighttpd-1.4.39/configure.ac	2015-12-05 14:17:22.000000000 +0000
+++ lighttpd-1.4.39-haiku/configure.ac
@@ -690,7 +690,7 @@ AM_CONDITIONAL(CHECK_WITH_FASTCGI, test
 dnl check for extra compiler options (warning options)
 if test "${GCC}" = "yes"; then
   TRY_CFLAGS([-Wall -W -Wshadow -pedantic])
-  TRY_CFLAGS([-std=gnu99])
+  TRY_CFLAGS([-std=gnu9x])
 fi
 
 AC_ARG_ENABLE(extra-warnings,
--- lighttpd-1.4.39/configure	2016-01-02 11:47:11.000000000 +0000
+++ lighttpd-1.4.39-haiku/configure
@@ -16726,8 +16726,8 @@ ac_link='$CC -o conftest$ac_exeext $CFLA
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports -std=gnu99" >&5
-$as_echo_n "checking if $CC supports -std=gnu99... " >&6; }
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports -std=gnu9x" >&5
+$as_echo_n "checking if $CC supports -std=gnu9x... " >&6; }
   ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
@@ -16735,7 +16735,7 @@ ac_link='$CC -o conftest$ac_exeext $CFLA
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
   ac_try_cflags_saved_cflags="${CFLAGS}"
-  CFLAGS="${CFLAGS} -std=gnu99"
+  CFLAGS="${CFLAGS} -std=gnu9x"
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
