diff -upr lighttpd-1.4.39/configure.ac lighttpd-1.4.39-haiku/configure.ac
--- lighttpd-1.4.39/configure.ac	2015-12-05 14:17:22.000000000 +0000
+++ lighttpd-1.4.39-haiku/configure.ac
@@ -279,6 +279,9 @@ AC_SEARCH_LIBS(socket,socket)
 AC_SEARCH_LIBS(gethostbyname,nsl socket)
 AC_SEARCH_LIBS(hstrerror,resolv)
 
+dnl On Haiku the accept function is in libnetwork
+AC_SEARCH_LIBS(accept,network)
+
 save_LIBS=$LIBS
 AC_SEARCH_LIBS(dlopen,dl,[
   AC_CHECK_HEADERS([dlfcn.h],[
@@ -690,7 +693,7 @@ AM_CONDITIONAL(CHECK_WITH_FASTCGI, test
 dnl check for extra compiler options (warning options)
 if test "${GCC}" = "yes"; then
   TRY_CFLAGS([-Wall -W -Wshadow -pedantic])
-  TRY_CFLAGS([-std=gnu99])
+  TRY_CFLAGS([-std=gnu9x])
 fi
 
 AC_ARG_ENABLE(extra-warnings,
