diff -Naur zip30-original/beos/beos.c zip30/beos/beos.c
--- zip30-original/beos/beos.c	2009-04-12 17:23:24.000000000 +0000
+++ zip30/beos/beos.c	2009-04-12 15:11:50.000000000 +0000
@@ -31,8 +31,12 @@
 
 #include <kernel/fs_attr.h>
 #include <storage/Mime.h>
+#ifdef __BEOS__
 #include <support/byteorder.h>
-
+#endif
+#ifdef __HAIKU__
+#include <support/ByteOrder.h>
+#endif
 
 #define PAD 0
 #define PATH_END '/'
diff -Naur zip30-original/tailor.h zip30/tailor.h
--- zip30-original/tailor.h	2009-04-12 17:23:24.000000000 +0000
+++ zip30/tailor.h	2009-04-12 15:15:56.000000000 +0000
@@ -52,7 +52,7 @@
 #include "atheos/osdep.h"
 #endif
 
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
 #include "beos/osdep.h"
 #endif
 
@@ -835,7 +835,7 @@
 #ifdef MVS
 #  define OS_CODE  0xf00
 #endif
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
 #  define OS_CODE  0x1000
 #endif
 #ifdef TANDEM
diff -Naur zip30-original/ttyio.c zip30/ttyio.c
--- zip30-original/ttyio.c	2009-04-12 17:23:25.000000000 +0000
+++ zip30/ttyio.c	2009-04-12 14:05:26.000000000 +0000
@@ -52,7 +52,7 @@
 #  define GLOBAL(g) G.g
 #endif
 
-#if (defined(__ATHEOS__) || defined(__BEOS__))  /* why yes, we do */
+#if (defined(__ATHEOS__) || defined(__BEOS__) || defined(__HAIKU__))  /* why yes, we do */
 #  define HAVE_TERMIOS_H
 #endif
 
diff -Naur zip30-original/ttyio.h zip30/ttyio.h
--- zip30-original/ttyio.h	2009-04-12 17:23:25.000000000 +0000
+++ zip30/ttyio.h	2009-04-12 14:06:12.000000000 +0000
@@ -60,7 +60,7 @@
 #  endif
 #endif
 
-#if (defined(__ATHEOS__) || defined(__BEOS__) || defined(UNIX))
+#if (defined(__ATHEOS__) || defined(__BEOS__) || defined(__HAIKU__) || defined(UNIX))
 #  ifndef ATH_BEO_UNX
 #    define ATH_BEO_UNX
 #  endif
diff -Naur zip30-original/zip.c zip30/zip.c
--- zip30-original/zip.c	2009-04-12 17:23:25.000000000 +0000
+++ zip30/zip.c	2009-04-12 14:06:31.000000000 +0000
@@ -1845,7 +1845,7 @@
 int set_filetype(out_path)
   char *out_path;
 {
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
   /* Set the filetype of the zipfile to "application/zip" */
   setfiletype( out_path, "application/zip" );
 #endif
diff -Naur zip30-original/zipcloak.c zip30/zipcloak.c
--- zip30-original/zipcloak.c	2009-04-12 17:23:25.000000000 +0000
+++ zip30/zipcloak.c	2009-04-12 14:06:59.000000000 +0000
@@ -54,7 +54,7 @@
 int set_filetype(out_path)
   char *out_path;
 {
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
   /* Set the filetype of the zipfile to "application/zip" */
   setfiletype( out_path, "application/zip" );
 #endif
diff -Naur zip30-original/zipnote.c zip30/zipnote.c
--- zip30-original/zipnote.c	2009-04-12 17:23:25.000000000 +0000
+++ zip30/zipnote.c	2009-04-12 14:07:19.000000000 +0000
@@ -72,7 +72,7 @@
 int set_filetype(out_path)
   char *out_path;
 {
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
   /* Set the filetype of the zipfile to "application/zip" */
   setfiletype( out_path, "application/zip" );
 #endif
diff -Naur zip30-original/zipsplit.c zip30/zipsplit.c
--- zip30-original/zipsplit.c	2009-04-12 17:23:25.000000000 +0000
+++ zip30/zipsplit.c	2009-04-12 14:07:37.000000000 +0000
@@ -96,7 +96,7 @@
 int set_filetype(out_path)
   char *out_path;
 {
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
   /* Set the filetype of the zipfile to "application/zip" */
   setfiletype( out_path, "application/zip" );
 #endif
diff -Naur zip30-original/zipup.c zip30/zipup.c
--- zip30-original/zipup.c	2009-04-12 17:23:25.000000000 +0000
+++ zip30/zipup.c	2009-04-12 14:07:58.000000000 +0000
@@ -68,7 +68,7 @@
 #  include "atari/zipup.h"
 #endif
 
-#ifdef __BEOS__
+#if (defined(__BEOS__) || defined(__HAIKU__))
 #  include "beos/zipup.h"
 #endif
 
