diff -up openssh-5.5p1/configure.ac.orig openssh-5.5p1/configure.ac
--- openssh-5.5p1/configure.ac.orig	2010-04-10 06:58:01.058458112 -0600
+++ openssh-5.5p1/configure.ac	2010-04-29 17:31:43.944242688 -0600
@@ -1277,7 +1277,7 @@ AC_ARG_WITH(libedit,
 			LIBEDIT=`$PKGCONFIG --libs-only-l libedit`
 			CPPFLAGS="$CPPFLAGS `$PKGCONFIG --cflags libedit`"
 		else
-			LIBEDIT="-ledit -lcurses"
+			AC_CHECK_LIB(curses, tgetent, [LIBEDIT="-ledit -lcurses"], AC_CHECK_LIB(ncurses, tgetent, [LIBEDIT="-ledit -lncurses"]))
 		fi
 		OTHERLIBS=`echo $LIBEDIT | sed 's/-ledit//'`
 		AC_CHECK_LIB(edit, el_init,
