diff -urN openssh-5.4p1/configure.ac openssh-5.4p1-haiku/configure.ac
--- openssh-5.4p1/configure.ac	2010-03-05 04:04:35.015990784 +0000
+++ openssh-5.4p1-haiku/configure.ac	2010-03-16 09:39:10.537133056 +0000
@@ -488,6 +488,11 @@
 *-*-dragonfly*)
 	SSHDLIBS="$SSHDLIBS -lcrypt"
 	;;
+*-*-haiku*) 
+    LIBS="$LIBS -lbsd "
+    AC_CHECK_LIB(network, socket)
+    MANTYPE=man 
+    ;; 
 *-*-hpux*)
 	# first we define all of the options common to all HP-UX releases
 	CPPFLAGS="$CPPFLAGS -D_HPUX_SOURCE -D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED=1"
@@ -2648,6 +2653,19 @@
 	)
 fi
 
+if test -z "$have_u_int64_t" ; then
+    AC_MSG_CHECKING([for u_int64_t type in posix/stdint.h])
+	AC_TRY_COMPILE(
+		[ #include <posix/stdint.h> ],
+		[ u_int64_t a; a = 1],
+		[
+			AC_DEFINE(HAVE_U_INT64_T)
+			AC_MSG_RESULT(yes)
+		],
+		[ AC_MSG_RESULT(no) ]
+	)
+fi
+
 if test -z "$have_u_intxx_t" ; then
 	AC_CACHE_CHECK([for uintXX_t types], ac_cv_have_uintxx_t, [
 		AC_TRY_COMPILE(
