diff --git a/pathnames.h b/pathnames.h
index ec89fc6..6ad1b64 100644
--- a/pathnames.h
+++ b/pathnames.h
@@ -59,7 +59,7 @@
  * The directory in user's home directory in which the files reside. The
  * directory should be world-readable (though not all files are).
  */
-#define _PATH_SSH_USER_DIR		".ssh"
+#define _PATH_SSH_USER_DIR		"config/settings/ssh"
 
 /*
  * Per-user file containing host keys of known hosts.  This file need not be
