From f41470d95341ab803010de2687fe5aa166378f41 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sat, 11 Apr 2020 13:58:14 -0400
Subject: [PATCH] GCC2 build fix.

---
 ssh-pkcs11-client.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ssh-pkcs11-client.c b/ssh-pkcs11-client.c
index 8a0ffef..cd75bf2 100644
--- a/ssh-pkcs11-client.c
+++ b/ssh-pkcs11-client.c
@@ -248,6 +248,7 @@ pkcs11_start_helper_methods(void)
 		return (0);
 
 #ifdef HAVE_EC_KEY_METHOD_NEW
+	{
 	int (*orig_sign)(int, const unsigned char *, int, unsigned char *,
 	    unsigned int *, const BIGNUM *, const BIGNUM *, EC_KEY *) = NULL;
 	if (helper_ecdsa != NULL)
@@ -257,6 +258,7 @@ pkcs11_start_helper_methods(void)
 		return (-1);
 	EC_KEY_METHOD_get_sign(helper_ecdsa, &orig_sign, NULL, NULL);
 	EC_KEY_METHOD_set_sign(helper_ecdsa, orig_sign, NULL, ecdsa_do_sign);
+	}
 #endif /* HAVE_EC_KEY_METHOD_NEW */
 
 	if ((helper_rsa = RSA_meth_dup(RSA_get_default_method())) == NULL)
-- 
2.24.1

