diff -urN echat-0.04beta1/Makefile echat-0.04beta1-haiku/Makefile
--- echat-0.04beta1/Makefile	2003-08-20 08:08:36.000000000 +0000
+++ echat-0.04beta1-haiku/Makefile	2009-06-13 16:43:20.000000000 +0000
@@ -6,8 +6,8 @@
 
 NAME=echat
 VERSION=0.04beta1
-PREFIX=/usr/local
-LIBS=-lcurses
+PREFIX=/boot/common
+LIBS=-lncurses -lnetwork
 STATIC=
 CC=gcc
 CFLAGS=-Wall -g -O2
@@ -27,8 +27,9 @@
 # -DPORTREUSE == run more than one chat on the same port
 # -DTCP == simple tcp server (laggy)
 # -DSHELL == enable `/!' command (may be dangerous!)
-DEFINES=-DFREEBSD -DCHARSET
+# DEFINES=-DFREEBSD -DCHARSET
 #DEFINES=-DFREEBSD -DCHARSET -DPORTREUSE -DTCP -DSHELL
+DEFINES=-DCHARSET
 
 .c.o:
 	@echo Compiling $*.c
diff -urN echat-0.04beta1/chat_cmnd.c echat-0.04beta1-haiku/chat_cmnd.c
--- echat-0.04beta1/chat_cmnd.c	2003-08-20 07:52:51.000000000 +0000
+++ echat-0.04beta1-haiku/chat_cmnd.c	2009-06-13 16:40:56.000000000 +0000
@@ -208,7 +208,7 @@
         CMD_QUIT, CMD_QUIT1, CMD_QUIT2, CMD_HELP, CMD_HELP1, CMD_HELP2, CMD_VERSION, CMD_VERSION1,
         CMD_STATUS, CMD_STATUS1, CMD_KEYS, CMD_KEYS1, CMD_VERBOSE, CMD_VERBOSE1, CMD_CLEAR, CMD_CLEAR1, CMD_SHELL);
     strncat(message, buf, STR_SIZE);
-    snprintf(buf, STR_SIZE, "%s %s %s(%s) %s(%s) %s %s(%s) %s(%s) %s ",
+    snprintf(buf, STR_SIZE, "%s %s %s(%s) %s(%s) %s %s(%s) %s(%s) %s (%s)",
         CMD_LOAD, CMD_SAVE, CMD_ANSWER, CMD_ANSWER1, CMD_GENDER, CMD_GENDER1, CMD_MY,
         CMD_FAVORITE, CMD_FAVORITE1, CMD_IGNORE, CMD_IGNORE1, CMD_BAN, CMD_BCAST);
     strncat(message, buf, STR_SIZE);
