From 6f4a723e6d44b44d1508c06d278d988522c04416 Mon Sep 17 00:00:00 2001
From: Robert Kausch <robert.kausch@freac.org>
Date: Fri, 10 Mar 2023 00:59:04 +0100
Subject: [PATCH] Query CPU info at least once even if set_cpu_affinity() fails

---
 libcpuid/cpuid_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcpuid/cpuid_main.c b/libcpuid/cpuid_main.c
index 358dedd..568ce92 100644
--- a/libcpuid/cpuid_main.c
+++ b/libcpuid/cpuid_main.c
@@ -1076,7 +1076,7 @@ int cpuid_get_all_raw_data(struct cpu_raw_data_array_t* data)
 	bool affinity_saved = save_cpu_affinity();
 
 	cpu_raw_data_array_t_constructor(data, true);
-	while (set_cpu_affinity(logical_cpu)) {
+	while (set_cpu_affinity(logical_cpu) || logical_cpu == 0) {
 		debugf(2, "Getting raw dump for logical CPU %i\n", logical_cpu);
 		cpuid_grow_raw_data_array(data, logical_cpu + 1);
 		raw_ptr = &data->raw[logical_cpu];
