From 8fc175d03be69e84940012866ffdfa0b7b6c865b Mon Sep 17 00:00:00 2001
From: Luke <noryb009@gmail.com>
Date: Fri, 20 Dec 2013 17:11:40 +0000
Subject: [PATCH] Remove reference to libm

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 5437b74..f7087fe 100755
--- a/configure
+++ b/configure
@@ -305,7 +305,7 @@ sed_config="
     s#@ARFLAGS@#${ARFLAGS-cru}#g
     s#@RANLIB@#${RANLIB-ranlib}#g
     s#@LIBS@#${LIBS-}#g
-    s#@LIBM@#${LIBM--lm}#g
+    s#@LIBM@#${LIBM-}#g
     s#@LIBZ@#${LIBZ--lz}#g
     s#@LIBPNG@#${LIBPNG--lpng}#g
     s#@DIFF@#${DIFF-diff -b -u}#g
-- 
1.8.3.4

