diff -urN SDL_net-1.2.7/acinclude/libtool.m4 SDL_net-1.2.7-haiku/acinclude/libtool.m4
--- SDL_net-1.2.7/acinclude/libtool.m4	2007-07-15 01:34:07.000000000 +0000
+++ SDL_net-1.2.7-haiku/acinclude/libtool.m4	2008-07-10 23:46:27.000000000 +0000
@@ -1288,7 +1288,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([[^/]]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -2287,7 +2287,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -2505,7 +2505,7 @@
 [AC_REQUIRE([AC_CANONICAL_HOST])dnl
 LIBM=
 case $host in
-*-*-beos* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
+*-*-beos* | *-*-haiku* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
   # These system don't have libm, or don't need it
   ;;
 *-ncr-sysv4.3*)
@@ -4850,7 +4850,7 @@
       # like `-m68040'.
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
     mingw* | os2* | pw32*)
@@ -5123,7 +5123,7 @@
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
diff -urN SDL_net-1.2.7/configure.in SDL_net-1.2.7-haiku/configure.in
--- SDL_net-1.2.7/configure.in	2007-07-20 12:40:02.000000000 +0000
+++ SDL_net-1.2.7-haiku/configure.in	2008-07-10 23:48:29.000000000 +0000
@@ -92,6 +92,9 @@
     *-*-beos*)
         INETLIB=""
         ;;
+    *-*-haiku*)
+        INETLIB="-lnetwork"
+        ;;
     *)
         INETLIB=""
         ;;
