From 818d2a84a9ec2d73a082585578b8b46ad4d1dd47 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Mon, 16 Mar 2015 17:50:12 -0400
Subject: [PATCH] Allow running as root on Haiku.

---
 src/core/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/main.cpp b/src/core/main.cpp
index 5313811..73c031b 100644
--- a/src/core/main.cpp
+++ b/src/core/main.cpp
@@ -142,7 +142,7 @@ int main( int argc, char * * argv )
 		}
 	}
 
-#ifndef LMMS_BUILD_WIN32
+#if !defined(LMMS_BUILD_WIN32) && !defined(__HAIKU__)
 	if ( ( getuid() == 0 || geteuid() == 0 ) && !allow_root )
 	{
 		printf("LMMS cannot be run as root.\nUse \"--allowroot\" to override.\n\n");
-- 
2.2.2

