diff --git a/util/gifinto.c b/util/gifinto.c
index 911c793..e89388d 100644
--- a/util/gifinto.c
+++ b/util/gifinto.c
@@ -49,6 +49,7 @@ static int
 int main(int argc, char **argv)
 {
     int	NumFiles;
+    int FD;
     bool Error, MinSizeFlag = false, HelpFlag = false;
     char **FileName = NULL, FoutTmpName[STRLEN], FullPath[STRLEN], *p;
     FILE *Fin, *Fout;
@@ -104,7 +105,6 @@ int main(int argc, char **argv)
     /* then add a name for the tempfile */
     if ( (strlen(FoutTmpName) + strlen(DEFAULT_TMP_NAME))  > STRLEN-1 ) GIF_EXIT("Filename too long.");
     strcat(FoutTmpName, DEFAULT_TMP_NAME);
-    int FD;
     FD = mkstemp(FoutTmpName); /* returns filedescriptor */
     if (FD == -1 )
     {
