--- a/src/supertux/main.cpp
+++ b/src/supertux/main.cpp
@@ -242,6 +242,7 @@
   #pragma GCC diagnostic pop
 #endif
 
+#ifndef __HAIKU__
 #ifdef _WIN32
 	std::string olduserdir = FileSystem::join(physfs_userdir, PACKAGE_NAME);
 #else
@@ -282,6 +283,7 @@
 	    log_info << "Moved old config dir " << olduserdir << " to " << userdir << std::endl;
 	  }
 	}
+#endif
 
     if (!FileSystem::is_directory(userdir))
     {
