diff -urN speex-1.2rc1/include/speex/speex_types.h speex-1.2rc1-haiku/include/speex/speex_types.h
--- speex-1.2rc1/include/speex/speex_types.h	2009-02-19 00:38:33.000000000 +0000
+++ speex-1.2rc1-haiku/include/speex/speex_types.h	2009-02-19 00:35:50.000000000 +0000
@@ -73,6 +73,15 @@
    typedef int32_t spx_int32_t;
    typedef u_int32_t spx_uint32_t;
 
+#elif defined(__HAIKU__)
+
+   /* Haiku */
+#  include <sys/types.h>
+   typedef short spx_int16_t;
+   typedef unsigned short spx_uint16_t;
+   typedef int spx_int32_t;
+   typedef unsigned int spx_uint32_t;
+
 #elif defined (__EMX__)
 
    /* OS/2 GCC */
