diff -urN libmad-0.15.1b/configure.ac libmad-0.15.1b-haiku/configure.ac
--- libmad-0.15.1b/configure.ac	2004-01-23 09:41:32.061341696 +0000
+++ libmad-0.15.1b-haiku/configure.ac	2010-09-16 12:16:45.656146432 +0000
@@ -140,8 +140,6 @@
     case "$optimize" in
 	-O|"-O "*)
 	    optimize="-O"
-	    optimize="$optimize -fforce-mem"
-	    optimize="$optimize -fforce-addr"
 	    : #x optimize="$optimize -finline-functions"
 	    : #- optimize="$optimize -fstrength-reduce"
 	    optimize="$optimize -fthread-jumps"
