diff -urN flac-1.2.1/examples/cpp/encode/file/main.cpp flac-1.2.1-haiku/examples/cpp/encode/file/main.cpp
--- flac-1.2.1/examples/cpp/encode/file/main.cpp	2007-09-13 15:58:03.004718592 +0000
+++ flac-1.2.1-haiku/examples/cpp/encode/file/main.cpp	2010-09-20 12:47:33.520617984 +0000
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include "FLAC++/metadata.h"
 #include "FLAC++/encoder.h"
 
