diff --git a/configure.in b/configure.in
index f71c002..cd41fc0 100644
--- a/configure.in
+++ b/configure.in
@@ -493,7 +493,7 @@ if test x$enable_music_ogg = xyes; then
         fi
     else
         AC_CHECK_HEADER([vorbis/vorbisfile.h], [have_ogg_hdr=yes])
-        AC_CHECK_LIB([vorbisfile -lvorbis -logg -lm], [ov_open_callbacks], [have_ogg_lib=yes])
+        AC_CHECK_LIB([vorbisfile -lvorbis -logg], [ov_open_callbacks], [have_ogg_lib=yes])
         if test x$have_ogg_hdr = xyes -a x$have_ogg_lib = xyes; then
             case "$host" in
                 *-*-darwin*)
