Index: configure
===================================================================
--- configure	(revision 18820)
+++ configure	(working copy)
@@ -1554,7 +1554,7 @@
 
 # OS specific
 case $target_os in
-    beos|haiku|zeta)
+    beos|zeta)
         prefix_default="$HOME/config"
         # helps building libavcodec
         add_cflags -DPIC -fomit-frame-pointer
@@ -1580,6 +1580,13 @@
             enable beos_netserver
             network_extralibs="-lnet"
         fi ;;
+    haiku)
+        prefix_default="/boot/common"
+        SHFLAGS=' -shared'
+        disable audio_beos
+        enable beosthreads
+        network_extralibs="-lnetwork"
+        ;;
     sunos)
         FFSERVERLDFLAGS=""
         SHFLAGS='-shared -Wl,-h,$$(@F)'
