diff -urN findutils-4.4.0/find/parser.c findutils-4.4.0-haiku/find/parser.c
--- findutils-4.4.0/find/parser.c	2008-11-30 02:34:06.000000000 +0000
+++ findutils-4.4.0-haiku/find/parser.c	2008-11-30 02:30:02.000000000 +0000
@@ -1385,8 +1385,8 @@
   if (collect_arg(argv, arg_ptr, &minutes))
     {
       struct time_val tval;
-      tval.xval = xv;
       struct timespec origin = options.cur_day_start;
+      tval.xval = xv;
       origin.tv_sec += DAYSECS;
       if (get_relative_timestamp(minutes, &tval, origin, 60,
 				 "arithmetic overflow while converting %s "
diff -urN findutils-4.4.0/gnulib/lib/fpurge.c findutils-4.4.0-haiku/gnulib/lib/fpurge.c
--- findutils-4.4.0/gnulib/lib/fpurge.c	2008-11-30 02:33:58.000000000 +0000
+++ findutils-4.4.0-haiku/gnulib/lib/fpurge.c	2008-11-30 02:45:47.000000000 +0000
@@ -59,7 +59,7 @@
   /* Most systems provide FILE as a struct and the necessary bitmask in
      <stdio.h>, because they need it for implementing getc() and putc() as
      fast macros.  */
-# if defined _IO_ferror_unlocked    /* GNU libc, BeOS */
+# if defined _IO_ftrylockfile    /* GNU libc, BeOS */
   fp->_IO_read_end = fp->_IO_read_ptr;
   fp->_IO_write_ptr = fp->_IO_write_base;
   /* Avoid memory leak when there is an active ungetc buffer.  */
diff -urN findutils-4.4.0/gnulib/lib/freadahead.c findutils-4.4.0-haiku/gnulib/lib/freadahead.c
--- findutils-4.4.0/gnulib/lib/freadahead.c	2008-11-30 02:33:58.000000000 +0000
+++ findutils-4.4.0-haiku/gnulib/lib/freadahead.c	2008-11-30 03:08:04.000000000 +0000
@@ -22,7 +22,7 @@
 size_t
 freadahead (FILE *fp)
 {
-#if defined _IO_ferror_unlocked     /* GNU libc, BeOS */
+#if defined _IO_ftrylockfile     /* GNU libc, BeOS */
   if (fp->_IO_write_ptr > fp->_IO_write_base)
     return 0;
   return fp->_IO_read_end - fp->_IO_read_ptr;
diff -urN findutils-4.4.0/tests/binary-io.h findutils-4.4.0-haiku/tests/binary-io.h
--- findutils-4.4.0/tests/binary-io.h	2008-11-30 02:34:03.000000000 +0000
+++ findutils-4.4.0-haiku/tests/binary-io.h	2008-11-30 02:32:51.000000000 +0000
@@ -30,7 +30,7 @@
 # define O_BINARY _O_BINARY
 # define O_TEXT _O_TEXT
 #endif
-#ifdef __BEOS__
+#if defined __BEOS__ || defined __HAIKU__
   /* BeOS 5 has O_BINARY and O_TEXT, but they have no effect.  */
 # undef O_BINARY
 # undef O_TEXT
