diff -up opensound-232d0de71550/kernel/OS/BeOS/os_beos.c.orig opensound-232d0de71550/kernel/OS/BeOS/os_beos.c
--- opensound-232d0de71550/kernel/OS/BeOS/os_beos.c.orig	2010-05-08 16:22:20.327417856 -0600
+++ opensound-232d0de71550/kernel/OS/BeOS/os_beos.c	2010-05-08 16:23:05.004194304 -0600
@@ -1076,6 +1076,8 @@ osdev_create (dev_info_t * dip, int dev_
 				   osdev->dip->pciinfo.function,
 				   "oss", osdev) != B_OK) {
       	cmn_err (CE_WARN, "Could not reserve PCI device\n");
+      	/* make sure the card isn't uninitialized on driver unload */
+      	oss_num_cards--;
       	/* XXX: CLEANUP! */
         return NULL;
       }
diff -up opensound-232d0de71550/setup/BeOS/build.sh.orig opensound-232d0de71550/setup/BeOS/build.sh
--- opensound-232d0de71550/setup/BeOS/build.sh.orig	2010-05-08 16:22:21.177471488 -0600
+++ opensound-232d0de71550/setup/BeOS/build.sh	2010-05-08 16:23:17.108003328 -0600
@@ -36,12 +36,12 @@ mkdir -p prototype/$BEOS_SYSTEM/add-ons/
 #hack for now
 #mkdir -p prototype/$BEOS_SYSTEM/add-ons/kernel/media/oss
 mkdir -p prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/bin
-mkdir -p prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/audio/multi
+#mkdir -p prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/audio/multi
 mkdir -p prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/audio/oss
 #hack for now
 mkdir -p prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/oss
 ln -s ../../bin/${DRVPREFIX}loader prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/oss/
-ln -s ../bin/${DRVPREFIX}loader prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/
+#ln -s ../bin/${DRVPREFIX}loader prototype/$BEOS_SYSTEM/add-ons/kernel/drivers/dev/
 #hack: install bins for now
 mkdir -p prototype/home/config/bin
 mkdir -p prototype/home/config/settings/kernel/drivers
