diff -urN libedit-20110802-3.0/configure.ac libedit-20110802-3.0-haiku/configure.ac
--- libedit-20110802-3.0/configure.ac	2011-08-02 01:09:05.036962304 -0700
+++ libedit-20110802-3.0-haiku/configure.ac	2012-06-16 10:41:31.579338240 -0700
@@ -64,7 +64,7 @@
 AC_HEADER_DIRENT
 AC_HEADER_STDC
 AC_HEADER_SYS_WAIT
-AC_CHECK_HEADERS([fcntl.h limits.h malloc.h stdlib.h string.h sys/ioctl.h sys/param.h unistd.h curses.h ncurses.h sys/cdefs.h termcap.h])
+AC_CHECK_HEADERS([fcntl.h limits.h malloc.h stdint.h stdlib.h string.h sys/errno.h errno.h sys/ioctl.h sys/param.h unistd.h curses.h ncurses.h sys/cdefs.h langinfo.h])
 
 AC_CHECK_HEADER([termios.h], [], [AC_MSG_ERROR([termios.h is required!])],[])
 
diff -urN libedit-20110802-3.0/examples/fileman.c libedit-20110802-3.0-haiku/examples/fileman.c
--- libedit-20110802-3.0/examples/fileman.c	2010-04-22 12:13:17.046137344 -0700
+++ libedit-20110802-3.0-haiku/examples/fileman.c	2012-06-16 10:41:31.721944576 -0700
@@ -11,7 +11,11 @@
 #include <sys/types.h>
 #include <sys/file.h>
 #include <sys/stat.h>
+#ifdef HAVE_SYS_ERRNO_H
 #include <sys/errno.h>
+#elif HAVE_ERRNO_H
+#include <errno.h>
+#endif
 #include <ctype.h>
 #include <string.h>
 #include <stdlib.h>
diff -urN libedit-20110802-3.0/src/editline/readline.h libedit-20110802-3.0-haiku/src/editline/readline.h
--- libedit-20110802-3.0/src/editline/readline.h	2011-02-26 14:42:59.045088768 -0800
+++ libedit-20110802-3.0-haiku/src/editline/readline.h	2012-06-16 10:41:32.068943872 -0700
@@ -75,7 +75,7 @@
 
 #ifndef CTRL
 #include <sys/ioctl.h>
-#if !defined(__sun) && !defined(__hpux) && !defined(_AIX)
+#if !defined(__sun) && !defined(__hpux) && !defined(_AIX) && !defined(__HAIKU__)
 #include <sys/ttydefaults.h>
 #endif
 #ifndef CTRL
diff -urN libedit-20110802-3.0/src/el.c libedit-20110802-3.0-haiku/src/el.c
--- libedit-20110802-3.0/src/el.c	2011-08-02 00:02:46.039583744 -0700
+++ libedit-20110802-3.0-haiku/src/el.c	2012-06-16 10:41:31.823394304 -0700
@@ -55,7 +55,9 @@
 #include <stdarg.h>
 #include <ctype.h>
 #include <locale.h>
+#ifdef HAVE_LANGINFO_H
 #include <langinfo.h>
+#endif
 #include "el.h"
 
 /* el_init():
diff -urN libedit-20110802-3.0/src/vis.c libedit-20110802-3.0-haiku/src/vis.c
--- libedit-20110802-3.0/src/vis.c	2011-07-08 23:39:42.044564480 -0700
+++ libedit-20110802-3.0-haiku/src/vis.c	2012-06-16 10:41:31.985137152 -0700
@@ -68,6 +68,10 @@
 #include <errno.h>
 #include <stdlib.h>
 
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
+
 #ifdef __weak_alias
 __weak_alias(strvisx,_strvisx)
 #endif
