diff --git a/config.mk b/config.mk
index d8fc0bc2..b8010453 100644
--- a/config.mk
+++ b/config.mk
@@ -156,7 +156,7 @@ CLIENT_LDADD:=
 
 PASSWD_LDADD:=
 
-ifneq ($(or $(findstring $(UNAME),FreeBSD), $(findstring $(UNAME),OpenBSD), $(findstring $(UNAME),NetBSD)),)
+ifneq ($(or $(findstring $(UNAME),FreeBSD), $(findstring $(UNAME),OpenBSD), $(findstring $(UNAME),NetBSD), $(findstring $(UNAME),Haiku)),)
 	BROKER_LDADD:=$(BROKER_LDADD) -lm
 else
 	BROKER_LDADD:=$(BROKER_LDADD) -ldl -lm
@@ -168,6 +168,10 @@ ifeq ($(UNAME),Linux)
 	LIB_LIBADD:=$(LIB_LIBADD) -lrt
 endif
 
+ifeq ($(UNAME),Haiku)
+	BROKER_LDADD:=$(BROKER_LDADD) -lnetwork
+endif
+
 ifeq ($(WITH_SHARED_LIBRARIES),yes)
 	CLIENT_LDADD:=${CLIENT_LDADD} ../lib/libmosquitto.so.${SOVERSION}
 endif
diff --git a/src/conf_includedir.c b/src/conf_includedir.c
index b602f963..14723fc6 100644
--- a/src/conf_includedir.c
+++ b/src/conf_includedir.c
@@ -38,7 +38,7 @@ Contributors:
 #endif
 
 #if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__QNX__)
-#  include <sys/syslog.h>
+#  include <syslog.h>
 #endif
 
 #include "mosquitto_broker_internal.h"
