diff --git a/numpy/core/src/umath/cpuid.c b/numpy/core/src/umath/cpuid.c
index 6744ceb..b1aaa09 100644
--- a/numpy/core/src/umath/cpuid.c
+++ b/numpy/core/src/umath/cpuid.c
@@ -19,7 +19,9 @@
 static NPY_INLINE
 int os_avx_support(void)
 {
-#if HAVE_XGETBV
+// See, GCC bug < 7.3.1
+// https://github.com/numpy/numpy/pull/10814
+#if 0
     /*
      * use bytes for xgetbv to avoid issues with compiler not knowing the
      * instruction
