From: Robert Haist <rhaist@mailbox.org>
Date: Mon, 14 May 2018 23:45:19 +0200
Subject: Fix tests makefile

---
 tests/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index f7cc17b..69e4e43 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -75,12 +75,12 @@ oprtest: oprgen.py
 
 .PHONY: difftest
 difftest: oprtest $(builddir)/difftest.sh
-	@sh $(builddir)/difftest.sh
+	@bash $(builddir)/difftest.sh
 
 
 .PHONY: difftest-refup
 difftest-refup: $(builddir)/difftest.sh
-	@sh $(builddir)/difftest.sh refup
+	@bash $(builddir)/difftest.sh refup
 
 
 .PHONY: warn_no_yasm
