--- git-1.5.5/Makefile	2008-04-08 07:10:01.000000000 +0000
+++ git-1.5.5.patched/Makefile	2008-04-19 00:20:29.000000000 +0000
@@ -989,7 +989,7 @@
 		'-DGIT_INFO_PATH="$(infodir_SQ)"' $<
 
 $(BUILT_INS): git$X
-	$(QUIET_BUILT_IN)$(RM) $@ && ln git$X $@
+	$(QUIET_BUILT_IN)$(RM) $@ && ln -s git$X $@
 
 common-cmds.h: ./generate-cmdlist.sh command-list.txt
 
@@ -1230,7 +1230,7 @@
 		cp '$(DESTDIR_SQ)$(bindir_SQ)/git$X' \
 			'$(DESTDIR_SQ)$(gitexecdir_SQ)/git$X'; \
 	fi
-	$(foreach p,$(BUILT_INS), $(RM) '$(DESTDIR_SQ)$(gitexecdir_SQ)/$p' && ln '$(DESTDIR_SQ)$(gitexecdir_SQ)/git$X' '$(DESTDIR_SQ)$(gitexecdir_SQ)/$p' ;)
+	$(foreach p,$(BUILT_INS), $(RM) '$(DESTDIR_SQ)$(gitexecdir_SQ)/$p' && ln -s '$(DESTDIR_SQ)$(gitexecdir_SQ)/git$X' '$(DESTDIR_SQ)$(gitexecdir_SQ)/$p' ;)
 ifneq (,$X)
 	$(foreach p,$(patsubst %$X,%,$(filter %$X,$(ALL_PROGRAMS) $(BUILT_INS) git$X)), $(RM) '$(DESTDIR_SQ)$(gitexecdir_SQ)/$p';)
 endif
--- git-1.5.5/git-compat-util.h	2008-04-08 07:10:01.000000000 +0000
+++ git-1.5.5.patched/git-compat-util.h	2008-04-18 22:15:48.000000000 +0000
@@ -453,4 +453,8 @@
 # define FORCE_DIR_SET_GID 0
 #endif
 
+#ifndef INET_ADDRSTRLEN
+#define INET_ADDRSTRLEN 16
+#endif
+
 #endif
--- git-1.5.5/builtin-count-objects.c	2008-04-08 07:10:01.000000000 +0000
+++ git-1.5.5.patched/builtin-count-objects.c	2008-04-19 00:32:05.000000000 +0000
@@ -43,7 +43,8 @@
 			if (lstat(path, &st) || !S_ISREG(st.st_mode))
 				bad = 1;
 			else
-				(*loose_size) += xsize_t(st.st_blocks);
+//				(*loose_size) += xsize_t(st.st_blocks);
+				(*loose_size) += ceil(st.st_size / 512.0);
 		}
 		if (bad) {
 			if (verbose) {
--- git-1.5.5/templates/Makefile	2008-04-08 07:10:01.000000000 +0000
+++ git-1.5.5.patched/templates/Makefile	2008-04-19 00:44:46.000000000 +0000
@@ -23,7 +23,7 @@
 
 bpsrc = $(filter-out %~,$(wildcard *--*))
 boilerplates.made : $(bpsrc)
-	$(QUIET)ls *--* 2>/dev/null | \
+	$(QUIET)ls -1 *--* 2>/dev/null | \
 	while read boilerplate; \
 	do \
 		case "$$boilerplate" in *~) continue ;; esac && \
