diff -Naur libftdi1-1.0/CMakeLists.txt libftdi1-1.0-haiku/CMakeLists.txt
--- libftdi1-1.0/CMakeLists.txt	2013-01-29 13:21:38.038273024 +0000
+++ libftdi1-1.0-haiku/CMakeLists.txt	2014-01-12 22:10:44.125566976 +0000
@@ -166,7 +166,7 @@
 if ( UNIX )
   configure_file ( libftdi1-config.in ${CMAKE_CURRENT_BINARY_DIR}/libftdi1-config @ONLY )
   install ( PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/libftdi1-config
-            DESTINATION bin )
+	  DESTINATION bin${BIN_SUFFIX} )
 
   # config script install path
   if ( NOT DEFINED LIBFTDI_CMAKE_CONFIG_DIR )
diff -Naur libftdi1-1.0/ftdi_eeprom/CMakeLists.txt libftdi1-1.0-haiku/ftdi_eeprom/CMakeLists.txt
--- libftdi1-1.0/ftdi_eeprom/CMakeLists.txt	2013-01-29 13:21:38.042991616 +0000
+++ libftdi1-1.0-haiku/ftdi_eeprom/CMakeLists.txt	2014-01-12 22:11:26.599523328 +0000
@@ -28,9 +28,9 @@
 
         add_executable(ftdi_eeprom main.c)
         target_link_libraries(ftdi_eeprom ftdi1)
-        target_link_libraries(ftdi_eeprom ${Confuse_LIBRARIES})
+	target_link_libraries(ftdi_eeprom ${Confuse_LIBRARIES} ${LIBUSB})
 
-        install(TARGETS ftdi_eeprom DESTINATION bin)
+	install(TARGETS ftdi_eeprom DESTINATION bin${BIN_SUFFIX})
 
     else(Confuse_FOUND)
         message(STATUS "libConfuse not found, won't build ftdi_eeprom")
diff -Naur libftdi1-1.0/src/CMakeLists.txt libftdi1-1.0-haiku/src/CMakeLists.txt
--- libftdi1-1.0/src/CMakeLists.txt	2013-01-29 13:21:38.044826624 +0000
+++ libftdi1-1.0-haiku/src/CMakeLists.txt	2014-01-12 21:54:24.677117952 +0000
@@ -51,7 +51,7 @@
             )
 
    install( FILES ${c_headers}
-            DESTINATION include/${PROJECT_NAME}
+	   DESTINATION develop/headers${INC_SUFFIX}/${PROJECT_NAME}
             COMPONENT headers
             )
 
