diff -urN json-c-0.9/Makefile.am json-c-0.9-haiku/Makefile.am
--- json-c-0.9/Makefile.am	2009-07-31 02:57:29.052690944 +0000
+++ json-c-0.9-haiku/Makefile.am	2010-09-09 04:53:58.000000000 +0000
@@ -1,4 +1,4 @@
-AM_CFLAGS = -Wall -Wwrite-strings -Werror -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
+AM_CFLAGS = -Wall -Wwrite-strings -Werror -D_GNU_SOURCE -D_REENTRANT
 
 EXTRA_DIST = README.html README-WIN32.html config.h.win32 doc
 
diff -urN json-c-0.9/Makefile.in json-c-0.9-haiku/Makefile.in
--- json-c-0.9/Makefile.in	2009-07-31 02:58:09.042467328 +0000
+++ json-c-0.9-haiku/Makefile.in	2010-09-09 04:58:41.989855744 +0000
@@ -219,7 +219,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = -Wall -Wwrite-strings -Werror -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
+AM_CFLAGS = -Wall -Wwrite-strings -Werror -D_GNU_SOURCE -D_REENTRANT
 EXTRA_DIST = README.html README-WIN32.html config.h.win32 doc
 lib_LTLIBRARIES = libjson.la
 pkgconfigdir = $(libdir)/pkgconfig
