diff --git a/makefile b/makefile
new file mode 100644
index 0000000..d134de0
--- /dev/null
+++ b/makefile
@@ -0,0 +1,30 @@
+NAME= Cherry_Blossom
+
+TYPE= APP
+
+SRCS= interface/DrawingTidbits.cpp \
+interface/PositionToolTip.cpp \
+interface/TransportButton.cpp \
+interface/PeakView.cpp \
+interface/SeekSlider.cpp \
+interface/VolumeSlider.cpp \
+views/AlbumView.cpp \
+views/ControlView.cpp \
+views/MusicView.cpp \
+views/SearchView.cpp \
+views/ArtistView.cpp \
+views/GenreView.cpp \
+views/PlayListView.cpp \
+views/SongView.cpp \
+CherryBlossom.cpp \
+CherryBlossomWindow.cpp \
+
+SYSTEM_INCLUDE_PATHS = /boot/system/develop/headers/private/interface/ \
+	/boot/system/develop/headers/private/shared/
+
+
+LIBS= be locale tag
+
+APP_VERSION = 1.0
+
+include $(BUILDHOME)/etc/makefile-engine
