diff -ur SDL-1.2.13/acinclude.m4 SDL-1.2.13-haiku/acinclude.m4
--- SDL-1.2.13/acinclude.m4	2007-12-30 20:48:39.000000000 -0800
+++ SDL-1.2.13-haiku/acinclude.m4	2008-06-16 20:14:12.000000000 -0700
@@ -1049,6 +1049,12 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  haiku*)
+    # On Haiku, this test takes a really really long time.
+    # Haiku has a limit of 4096 for now.
+    lt_cv_sys_max_cmd_len=4096;
+    ;;
+
   amigaos*)
     # On AmigaOS with pdksh, this test takes hours, literally.
     # So we just punt and use a minimum line length of 8192.
@@ -1239,7 +1245,7 @@
   lt_cv_dlopen_libs=
 
   case $host_os in
-  beos*)
+  beos* | haiku*)
     lt_cv_dlopen="load_add_on"
     lt_cv_dlopen_libs=
     lt_cv_dlopen_self=yes
@@ -1603,7 +1609,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([[^/]]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -2602,7 +2608,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -2820,7 +2826,7 @@
 [AC_REQUIRE([AC_CANONICAL_HOST])dnl
 LIBM=
 case $host in
-*-*-beos* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
+*-*-beos* | *-*-haiku* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
   # These system don't have libm, or don't need it
   ;;
 *-ncr-sysv4.3*)
@@ -3311,7 +3317,7 @@
     fi
     ;;
 
-  beos*)
+  beos* | haiku*)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
       _LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
       # Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -5165,7 +5171,7 @@
       # like `-m68040'.
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
     mingw* | os2* | pw32*)
@@ -5438,7 +5444,7 @@
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
@@ -5805,7 +5811,7 @@
       _LT_AC_TAGVAR(ld_shlibs, $1)=no
       ;;
 
-    beos*)
+    beos* | haiku*)
       if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	# Joseph Beckenbach <jrb3@best.com> says some releases of gcc
diff -ur SDL-1.2.13/build-scripts/config.guess SDL-1.2.13-haiku/build-scripts/config.guess
--- SDL-1.2.13/build-scripts/config.guess	2007-12-30 20:47:51.000000000 -0800
+++ SDL-1.2.13-haiku/build-scripts/config.guess	2008-05-25 09:59:19.000000000 -0700
@@ -1,9 +1,10 @@
 #! /bin/sh
 # Attempt to guess a canonical system name.
 #   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
-#   2000, 2001, 2002, 2003, 2004, 2005 Free Software Foundation, Inc.
+#   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008
+#   Free Software Foundation, Inc.
 
-timestamp='2006-01-02'
+timestamp='2008-01-08'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
@@ -55,8 +56,8 @@
 GNU config.guess ($timestamp)
 
 Originally written by Per Bothner.
-Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
-Free Software Foundation, Inc.
+Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001,
+2002, 2003, 2004, 2005, 2006, 2007, 2008 Free Software Foundation, Inc.
 
 This is free software; see the source for copying conditions.  There is NO
 warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE."
@@ -106,7 +107,7 @@
 trap "exitcode=\$?; (rm -f \$tmpfiles 2>/dev/null; rmdir \$tmp 2>/dev/null) && exit \$exitcode" 0 ;
 trap "rm -f \$tmpfiles 2>/dev/null; rmdir \$tmp 2>/dev/null; exit 1" 1 2 13 15 ;
 : ${TMPDIR=/tmp} ;
- { tmp=`(umask 077 && mktemp -d -q "$TMPDIR/cgXXXXXX") 2>/dev/null` && test -n "$tmp" && test -d "$tmp" ; } ||
+ { tmp=`(umask 077 && mktemp -d "$TMPDIR/cgXXXXXX") 2>/dev/null` && test -n "$tmp" && test -d "$tmp" ; } ||
  { test -n "$RANDOM" && tmp=$TMPDIR/cg$$-$RANDOM && (umask 077 && mkdir $tmp) ; } ||
  { tmp=$TMPDIR/cg-$$ && (umask 077 && mkdir $tmp) && echo "Warning: creating insecure temp directory" >&2 ; } ||
  { echo "$me: cannot create a temporary directory in $TMPDIR" >&2 ; exit 1 ; } ;
@@ -160,6 +161,7 @@
 	    arm*) machine=arm-unknown ;;
 	    sh3el) machine=shl-unknown ;;
 	    sh3eb) machine=sh-unknown ;;
+	    sh5el) machine=sh5le-unknown ;;
 	    *) machine=${UNAME_MACHINE_ARCH}-unknown ;;
 	esac
 	# The Operating System including object format, if it has switched
@@ -210,7 +212,7 @@
 	echo ${UNAME_MACHINE}-unknown-solidbsd${UNAME_RELEASE}
 	exit ;;
     macppc:MirBSD:*:*)
-	echo powerppc-unknown-mirbsd${UNAME_RELEASE}
+	echo powerpc-unknown-mirbsd${UNAME_RELEASE}
 	exit ;;
     *:MirBSD:*:*)
 	echo ${UNAME_MACHINE}-unknown-mirbsd${UNAME_RELEASE}
@@ -328,7 +330,7 @@
     sun4*:SunOS:5.*:* | tadpole*:SunOS:5.*:*)
 	echo sparc-sun-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
-    i86pc:SunOS:5.*:*)
+    i86pc:SunOS:5.*:* | i86xen:SunOS:5.*:*)
 	echo i386-pc-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
     sun4*:SunOS:6*:*)
@@ -530,7 +532,7 @@
 		echo rs6000-ibm-aix3.2
 	fi
 	exit ;;
-    *:AIX:*:[45])
+    *:AIX:*:[456])
 	IBM_CPU_ID=`/usr/sbin/lsdev -C -c processor -S available | sed 1q | awk '{ print $1 }'`
 	if /usr/sbin/lsattr -El ${IBM_CPU_ID} | grep ' POWER' >/dev/null 2>&1; then
 		IBM_ARCH=rs6000
@@ -770,6 +772,8 @@
 	case ${UNAME_MACHINE} in
 	    pc98)
 		echo i386-unknown-freebsd`echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'` ;;
+	    amd64)
+		echo x86_64-unknown-freebsd`echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'` ;;
 	    *)
 		echo ${UNAME_MACHINE}-unknown-freebsd`echo ${UNAME_RELEASE}|sed -e 's/[-(].*//'` ;;
 	esac
@@ -777,7 +781,7 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
-    i*:MINGW*:*)
+    *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
     i*:windows32*:*)
@@ -787,9 +791,18 @@
     i*:PW*:*)
 	echo ${UNAME_MACHINE}-pc-pw32
 	exit ;;
-    x86:Interix*:[345]*)
-	echo i586-pc-interix${UNAME_RELEASE}|sed -e 's/\..*//'
-	exit ;;
+    *:Interix*:[3456]*)
+    	case ${UNAME_MACHINE} in
+	    x86)
+		echo i586-pc-interix${UNAME_RELEASE}
+		exit ;;
+	    EM64T | authenticamd)
+		echo x86_64-unknown-interix${UNAME_RELEASE}
+		exit ;;
+	    IA64)
+		echo ia64-unknown-interix${UNAME_RELEASE}
+		exit ;;
+	esac ;;
     [345]86:Windows_95:* | [345]86:Windows_98:* | [345]86:Windows_NT:*)
 	echo i${UNAME_MACHINE}-pc-mks
 	exit ;;
@@ -823,6 +836,16 @@
 	echo ${UNAME_MACHINE}-pc-minix
 	exit ;;
     arm*:Linux:*:*)
+	eval $set_cc_for_build
+	if echo __ARM_EABI__ | $CC_FOR_BUILD -E - 2>/dev/null \
+	    | grep -q __ARM_EABI__
+	then
+	    echo ${UNAME_MACHINE}-unknown-linux-gnu
+	else
+	    echo ${UNAME_MACHINE}-unknown-linux-gnueabi
+	fi
+	exit ;;
+    avr32*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
     cris:Linux:*:*)
@@ -859,7 +882,11 @@
 	#endif
 	#endif
 EOF
-	eval "`$CC_FOR_BUILD -E $dummy.c 2>/dev/null | sed -n '/^CPU/{s: ::g;p;}'`"
+	eval "`$CC_FOR_BUILD -E $dummy.c 2>/dev/null | sed -n '
+	    /^CPU/{
+		s: ::g
+		p
+	    }'`"
 	test x"${CPU}" != x && { echo "${CPU}-unknown-linux-gnu"; exit; }
 	;;
     mips64:Linux:*:*)
@@ -878,7 +905,11 @@
 	#endif
 	#endif
 EOF
-	eval "`$CC_FOR_BUILD -E $dummy.c 2>/dev/null | sed -n '/^CPU/{s: ::g;p;}'`"
+	eval "`$CC_FOR_BUILD -E $dummy.c 2>/dev/null | sed -n '
+	    /^CPU/{
+		s: ::g
+		p
+	    }'`"
 	test x"${CPU}" != x && { echo "${CPU}-unknown-linux-gnu"; exit; }
 	;;
     or32:Linux:*:*)
@@ -933,6 +964,9 @@
     x86_64:Linux:*:*)
 	echo x86_64-unknown-linux-gnu
 	exit ;;
+    xtensa*:Linux:*:*)
+    	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit ;;
     i*86:Linux:*:*)
 	# The BFD linker knows what the default object file format is, so
 	# first see if it will tell us. cd to the root directory to prevent
@@ -975,7 +1009,7 @@
 	LIBC=gnulibc1
 	# endif
 	#else
-	#if defined(__INTEL_COMPILER) || defined(__PGI)
+	#if defined(__INTEL_COMPILER) || defined(__PGI) || defined(__SUNPRO_C) || defined(__SUNPRO_CC)
 	LIBC=gnu
 	#else
 	LIBC=gnuaout
@@ -985,7 +1019,11 @@
 	LIBC=dietlibc
 	#endif
 EOF
-	eval "`$CC_FOR_BUILD -E $dummy.c 2>/dev/null | sed -n '/^LIBC/{s: ::g;p;}'`"
+	eval "`$CC_FOR_BUILD -E $dummy.c 2>/dev/null | sed -n '
+	    /^LIBC/{
+		s: ::g
+		p
+	    }'`"
 	test x"${LIBC}" != x && {
 		echo "${UNAME_MACHINE}-pc-linux-${LIBC}"
 		exit
@@ -1178,6 +1216,12 @@
     BePC:BeOS:*:*)	# BeOS running on Intel PC compatible.
 	echo i586-pc-beos
 	exit ;;
+    BeMac:Haiku:*:*)
+	echo powerpc-apple-haiku
+	exit ;;
+    BePC:Haiku:*:*)
+	echo i586-pc-haiku
+	exit ;;
     SX-4:SUPER-UX:*:*)
 	echo sx4-nec-superux${UNAME_RELEASE}
 	exit ;;
@@ -1187,6 +1231,15 @@
     SX-6:SUPER-UX:*:*)
 	echo sx6-nec-superux${UNAME_RELEASE}
 	exit ;;
+    SX-7:SUPER-UX:*:*)
+	echo sx7-nec-superux${UNAME_RELEASE}
+	exit ;;
+    SX-8:SUPER-UX:*:*)
+	echo sx8-nec-superux${UNAME_RELEASE}
+	exit ;;
+    SX-8R:SUPER-UX:*:*)
+	echo sx8r-nec-superux${UNAME_RELEASE}
+	exit ;;
     Power*:Rhapsody:*:*)
 	echo powerpc-apple-rhapsody${UNAME_RELEASE}
 	exit ;;
diff -ur SDL-1.2.13/build-scripts/config.sub SDL-1.2.13-haiku/build-scripts/config.sub
--- SDL-1.2.13/build-scripts/config.sub	2007-12-30 20:47:51.000000000 -0800
+++ SDL-1.2.13-haiku/build-scripts/config.sub	2008-05-26 00:09:31.000000000 -0700
@@ -1208,7 +1208,7 @@
 	      | -chorusos* | -chorusrdb* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
-	      | -uxpv* | -beos* | -mpeix* | -udk* \
+	      | -uxpv* | -beos* | haiku* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
diff -ur SDL-1.2.13/build-scripts/ltmain.sh SDL-1.2.13-haiku/build-scripts/ltmain.sh
--- SDL-1.2.13/build-scripts/ltmain.sh	2007-12-30 20:47:51.000000000 -0800
+++ SDL-1.2.13-haiku/build-scripts/ltmain.sh	2008-06-16 19:27:13.000000000 -0700
@@ -1553,7 +1553,7 @@
       -l*)
 	if test "X$arg" = "X-lc" || test "X$arg" = "X-lm"; then
 	  case $host in
-	  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-beos*)
+	  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-beos* | *-*-haiku*)
 	    # These systems don't actually have a C or math library (as such)
 	    continue
 	    ;;
@@ -3455,7 +3455,7 @@
       if test "$build_libtool_libs" = yes; then
 	if test -n "$rpath"; then
 	  case $host in
-	  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2* | *-*-beos*)
+	  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2* | *-*-beos* | *-*-haiku*)
 	    # these systems don't actually have a c library (as such)!
 	    ;;
 	  *-*-rhapsody* | *-*-darwin1.[012])
diff -ur SDL-1.2.13/configure SDL-1.2.13-haiku/configure
--- SDL-1.2.13/configure	2007-12-30 21:09:39.000000000 -0800
+++ SDL-1.2.13-haiku/configure	2008-05-26 00:07:55.000000000 -0700
@@ -4386,7 +4386,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -6096,10 +6096,10 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
-  beos*)
+  beos* | haiku*)
     # On BeOS, this test takes a really really long time.
     # So we just punt and use a minimum line length of 8192.
-    lt_cv_sys_max_cmd_len=8192;
+    lt_cv_sys_max_cmd_len=4096;
     ;;
 
   amigaos*)
@@ -7088,7 +7088,7 @@
       lt_prog_compiler_pic='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
@@ -7583,7 +7583,7 @@
       ld_shlibs=no
       ;;
 
-    beos*)
+    beos* | haiku*)
       if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
 	allow_undefined_flag=unsupported
 	# Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -8574,7 +8574,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([^/]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -9152,7 +9152,7 @@
   lt_cv_dlopen_libs=
 
   case $host_os in
-  beos*)
+  beos* | haiku*)
     lt_cv_dlopen="load_add_on"
     lt_cv_dlopen_libs=
     lt_cv_dlopen_self=yes
@@ -10998,7 +10998,7 @@
     fi
     ;;
 
-  beos*)
+  beos* | haiku*)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
       allow_undefined_flag_CXX=unsupported
       # Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -11877,7 +11877,7 @@
       # like `-m68040'.
       lt_prog_compiler_pic_CXX='-m68020 -resident32 -malways-restore-a4'
       ;;
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
     mingw* | os2* | pw32*)
@@ -12499,7 +12499,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([^/]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -13618,7 +13618,7 @@
       lt_prog_compiler_pic_F77='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
@@ -14113,7 +14113,7 @@
       ld_shlibs_F77=no
       ;;
 
-    beos*)
+    beos* | haiku*)
       if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
 	allow_undefined_flag_F77=unsupported
 	# Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -15084,7 +15084,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([^/]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -16194,7 +16194,7 @@
       lt_prog_compiler_pic_GCJ='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
@@ -16689,7 +16689,7 @@
       ld_shlibs_GCJ=no
       ;;
 
-    beos*)
+    beos* | haiku*)
       if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
 	allow_undefined_flag_GCJ=unsupported
 	# Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -17680,7 +17680,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([^/]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -33849,7 +33849,7 @@
         SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
         SDL_LIBS="-lmingw32 -lSDLmain $SDL_LIBS -mwindows"
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         ARCH=beos
         ac_default_prefix=/boot/develop/tools/gnupro
         CheckDummyVideo
diff -ur SDL-1.2.13/configure.in SDL-1.2.13-haiku/configure.in
--- SDL-1.2.13/configure.in	2007-12-30 20:48:39.000000000 -0800
+++ SDL-1.2.13-haiku/configure.in	2008-05-26 00:12:38.000000000 -0700
@@ -2456,7 +2456,7 @@
         SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
         SDL_LIBS="-lmingw32 -lSDLmain $SDL_LIBS -mwindows"
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         ARCH=beos
         ac_default_prefix=/boot/develop/tools/gnupro
         CheckDummyVideo
diff -ur SDL-1.2.13/include/SDL_platform.h SDL-1.2.13-haiku/include/SDL_platform.h
--- SDL-1.2.13/include/SDL_platform.h	2007-12-30 20:48:36.000000000 -0800
+++ SDL-1.2.13-haiku/include/SDL_platform.h	2008-05-26 21:58:21.000000000 -0700
@@ -29,10 +29,14 @@
 #undef __AIX__
 #define __AIX__		1
 #endif
-#if defined(__BEOS__)
+#if defined(__BEOS__) && !defined(__HAIKU__)
 #undef __BEOS__
 #define __BEOS__	1
 #endif
+#if defined(__HAIKU__)
+#undef __HAIKU__
+#define __HAIKU__ 1
+#endif
 #if defined(bsdi) || defined(__bsdi) || defined(__bsdi__)
 #undef __BSDI__
 #define __BSDI__	1
diff -ur SDL-1.2.13/include/begin_code.h SDL-1.2.13-haiku/include/begin_code.h
--- SDL-1.2.13/include/begin_code.h	2007-12-30 20:48:36.000000000 -0800
+++ SDL-1.2.13-haiku/include/begin_code.h	2008-05-26 22:01:54.000000000 -0700
@@ -33,7 +33,7 @@
 
 /* Some compilers use a special export keyword */
 #ifndef DECLSPEC
-# if defined(__BEOS__)
+# if defined(__BEOS__) || defined(__HAIKU__)
 #  if defined(__GNUC__)
 #   define DECLSPEC	__declspec(dllexport)
 #  else
Binary files SDL-1.2.13/src/events/SDL_events.c and SDL-1.2.13-haiku/src/events/SDL_events.c differ
Binary files SDL-1.2.13/src/events/SDL_events_c.h and SDL-1.2.13-haiku/src/events/SDL_events_c.h differ
Binary files SDL-1.2.13/src/hermes/COPYING.LIB and SDL-1.2.13-haiku/src/hermes/COPYING.LIB differ
Binary files SDL-1.2.13/src/joystick/SDL_sysjoystick.h and SDL-1.2.13-haiku/src/joystick/SDL_sysjoystick.h differ
Binary files SDL-1.2.13/src/joystick/beos/SDL_bejoystick.cc and SDL-1.2.13-haiku/src/joystick/beos/SDL_bejoystick.cc differ
Binary files SDL-1.2.13/src/joystick/bsd/SDL_sysjoystick.c and SDL-1.2.13-haiku/src/joystick/bsd/SDL_sysjoystick.c differ
Binary files SDL-1.2.13/src/joystick/linux/SDL_sysjoystick.c and SDL-1.2.13-haiku/src/joystick/linux/SDL_sysjoystick.c differ
Binary files SDL-1.2.13/src/loadso/macosx/SDL_dlcompat.c and SDL-1.2.13-haiku/src/loadso/macosx/SDL_dlcompat.c differ
Binary files SDL-1.2.13/src/loadso/mint/SDL_sysloadso.c and SDL-1.2.13-haiku/src/loadso/mint/SDL_sysloadso.c differ
Binary files SDL-1.2.13/src/loadso/os2/SDL_sysloadso.c and SDL-1.2.13-haiku/src/loadso/os2/SDL_sysloadso.c differ
Binary files SDL-1.2.13/src/loadso/win32/SDL_sysloadso.c and SDL-1.2.13-haiku/src/loadso/win32/SDL_sysloadso.c differ
diff -ur SDL-1.2.13/src/main/macos/exports/SDL.x SDL-1.2.13-haiku/src/main/macos/exports/SDL.x
--- SDL-1.2.13/src/main/macos/exports/SDL.x	2007-12-30 20:47:53.000000000 -0800
+++ SDL-1.2.13-haiku/src/main/macos/exports/SDL.x	2007-12-30 20:47:53.000000000 -0800
@@ -1,209 +1,108 @@
-	SDL_Init
-	SDL_InitSubSystem
-	SDL_QuitSubSystem
-	SDL_WasInit
-	SDL_Quit
-	SDL_GetAppState
-	SDL_AudioInit
-	SDL_AudioQuit
-	SDL_AudioDriverName
-	SDL_OpenAudio
-	SDL_GetAudioStatus
-	SDL_PauseAudio
-	SDL_LoadWAV_RW
-	SDL_FreeWAV
-	SDL_BuildAudioCVT
-	SDL_ConvertAudio
-	SDL_MixAudio
-	SDL_LockAudio
-	SDL_UnlockAudio
-	SDL_CloseAudio
-	SDL_CDNumDrives
-	SDL_CDName
-	SDL_CDOpen
-	SDL_CDStatus
-	SDL_CDPlayTracks
-	SDL_CDPlay
-	SDL_CDPause
-	SDL_CDResume
-	SDL_CDStop
-	SDL_CDEject
-	SDL_CDClose
-	SDL_HasRDTSC
-	SDL_HasMMX
-	SDL_HasMMXExt
-	SDL_Has3DNow
-	SDL_Has3DNowExt
-	SDL_HasSSE
-	SDL_HasSSE2
-	SDL_HasAltiVec
-	SDL_SetError
-	SDL_GetError
-	SDL_ClearError
-	SDL_Error
-	SDL_PumpEvents
-	SDL_PeepEvents
-	SDL_PollEvent
-	SDL_WaitEvent
-	SDL_PushEvent
-	SDL_SetEventFilter
-	SDL_GetEventFilter
-	SDL_EventState
-	SDL_NumJoysticks
-	SDL_JoystickName
-	SDL_JoystickOpen
-	SDL_JoystickOpened
-	SDL_JoystickIndex
-	SDL_JoystickNumAxes
-	SDL_JoystickNumBalls
-	SDL_JoystickNumHats
-	SDL_JoystickNumButtons
-	SDL_JoystickUpdate
-	SDL_JoystickEventState
-	SDL_JoystickGetAxis
-	SDL_JoystickGetHat
-	SDL_JoystickGetBall
-	SDL_JoystickGetButton
-	SDL_JoystickClose
-	SDL_EnableUNICODE
-	SDL_EnableKeyRepeat
-	SDL_GetKeyRepeat
-	SDL_GetKeyState
-	SDL_GetModState
-	SDL_SetModState
-	SDL_GetKeyName
-	SDL_LoadObject
-	SDL_LoadFunction
-	SDL_UnloadObject
-	SDL_GetMouseState
-	SDL_GetRelativeMouseState
-	SDL_WarpMouse
-	SDL_CreateCursor
-	SDL_SetCursor
-	SDL_GetCursor
-	SDL_FreeCursor
-	SDL_ShowCursor
-	SDL_CreateMutex
-	SDL_mutexP
-	SDL_mutexV
-	SDL_DestroyMutex
-	SDL_CreateSemaphore
-	SDL_DestroySemaphore
-	SDL_SemWait
-	SDL_SemTryWait
-	SDL_SemWaitTimeout
-	SDL_SemPost
-	SDL_SemValue
-	SDL_CreateCond
-	SDL_DestroyCond
-	SDL_CondSignal
-	SDL_CondBroadcast
-	SDL_CondWait
-	SDL_CondWaitTimeout
-	SDL_RWFromFile
-	SDL_RWFromFP
-	SDL_RWFromMem
-	SDL_RWFromConstMem
-	SDL_AllocRW
-	SDL_FreeRW
-	SDL_ReadLE16
-	SDL_ReadBE16
-	SDL_ReadLE32
-	SDL_ReadBE32
-	SDL_ReadLE64
-	SDL_ReadBE64
-	SDL_WriteLE16
-	SDL_WriteBE16
-	SDL_WriteLE32
-	SDL_WriteBE32
-	SDL_WriteLE64
-	SDL_WriteBE64
-	SDL_GetWMInfo
-	SDL_CreateThread
-	SDL_CreateThread
-	SDL_ThreadID
-	SDL_GetThreadID
-	SDL_WaitThread
-	SDL_KillThread
-	SDL_GetTicks
-	SDL_Delay
-	SDL_SetTimer
-	SDL_AddTimer
-	SDL_RemoveTimer
-	SDL_Linked_Version
-	SDL_VideoInit
-	SDL_VideoQuit
-	SDL_VideoDriverName
-	SDL_GetVideoSurface
-	SDL_GetVideoInfo
-	SDL_VideoModeOK
-	SDL_ListModes
-	SDL_SetVideoMode
-	SDL_UpdateRects
-	SDL_UpdateRect
-	SDL_Flip
-	SDL_SetGamma
-	SDL_SetGammaRamp
-	SDL_GetGammaRamp
-	SDL_SetColors
-	SDL_SetPalette
-	SDL_MapRGB
-	SDL_MapRGBA
-	SDL_GetRGB
-	SDL_GetRGBA
-	SDL_CreateRGBSurface
-	SDL_CreateRGBSurfaceFrom
-	SDL_FreeSurface
-	SDL_LockSurface
-	SDL_UnlockSurface
-	SDL_LoadBMP_RW
-	SDL_SaveBMP_RW
-	SDL_SetColorKey
-	SDL_SetAlpha
-	SDL_SetClipRect
-	SDL_GetClipRect
-	SDL_ConvertSurface
-	SDL_UpperBlit
-	SDL_LowerBlit
-	SDL_FillRect
-	SDL_DisplayFormat
-	SDL_DisplayFormatAlpha
-	SDL_CreateYUVOverlay
-	SDL_LockYUVOverlay
-	SDL_UnlockYUVOverlay
-	SDL_DisplayYUVOverlay
-	SDL_FreeYUVOverlay
-	SDL_GL_LoadLibrary
-	SDL_GL_GetProcAddress
-	SDL_GL_SetAttribute
-	SDL_GL_GetAttribute
-	SDL_GL_SwapBuffers
-	SDL_GL_UpdateRects
-	SDL_GL_Lock
-	SDL_GL_Unlock
-	SDL_WM_SetCaption
-	SDL_WM_GetCaption
-	SDL_WM_SetIcon
-	SDL_WM_IconifyWindow
-	SDL_WM_ToggleFullScreen
-	SDL_WM_GrabInput
-	SDL_SoftStretch
-	SDL_putenv
-	SDL_getenv
-	SDL_qsort
-	SDL_revcpy
-	SDL_strlcpy
-	SDL_strlcat
-	SDL_strdup
-	SDL_strrev
-	SDL_strupr
-	SDL_strlwr
-	SDL_ltoa
-	SDL_ultoa
-	SDL_strcasecmp
-	SDL_strncasecmp
-	SDL_snprintf
-	SDL_vsnprintf
-	SDL_iconv
-	SDL_iconv_string
-	SDL_InitQuickDraw
+_corrupted; then
+  { $as_echo "$as_me:$LINENO: error: changes in the environment can compromise the build" >&5
+$as_echo "$as_me: error: changes in the environment can compromise the build" >&2;}
+  { { $as_echo "$as_me:$LINENO: error: run \`make distclean' and/or \`rm $cache_file' and start over" >&5
+$as_echo "$as_me: error: run \`make distclean' and/or \`rm $cache_file' and start over" >&2;}
+   { (exit 1); exit 1; }; }
+fi
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+ac_ext=c
+ac_cpp='$CPP $CPPFLAGS'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compiler_gnu=$ac_cv_c_compiler_gnu
+
+
+am__api_version='1.10'
+
+ac_aux_dir=
+for ac_dir in "$srcdir" "$srcdir/.." "$srcdir/../.."; do
+  if test -f "$ac_dir/install-sh"; then
+    ac_aux_dir=$ac_dir
+    ac_install_sh="$ac_aux_dir/install-sh -c"
+    break
+  elif test -f "$ac_dir/install.sh"; then
+    ac_aux_dir=$ac_dir
+    ac_install_sh="$ac_aux_dir/install.sh -c"
+    break
+  elif test -f "$ac_dir/shtool"; then
+    ac_aux_dir=$ac_dir
+    ac_install_sh="$ac_aux_dir/shtool install -c"
+    break
+  fi
+done
+if test -z "$ac_aux_dir"; then
+  { { $as_echo "$as_me:$LINENO: error: cannot find install-sh or install.sh in \"$srcdir\" \"$srcdir/..\" \"$srcdir/../..\"" >&5
+$as_echo "$as_me: error: cannot find install-sh or install.sh in \"$srcdir\" \"$srcdir/..\" \"$srcdir/../..\"" >&2;}
+   { (exit 1); exit 1; }; }
+fi
+
+# These three variables are undocumented and unsupported,
+# and are intended to be withdrawn in a future Autoconf release.
+# They can cause serious problems if a builder's source tree is in a directory
+# whose full name contains unusual characters.
+ac_config_guess="$SHELL $ac_aux_dir/config.guess"  # Please don't use this var.
+ac_config_sub="$SHELL $ac_aux_dir/config.sub"  # Please don't use this var.
+ac_configure="$SHELL $ac_aux_dir/configure"  # Please don't use this var.
+
+
+# Find a good install program.  We prefer a C program (faster),
+# so one script is as good as another.  But avoid the broken or
+# incompatible versions:
+# SysV /etc/install, /usr/sbin/install
+# SunOS /usr/etc/install
+# IRIX /sbin/install
+# AIX /bin/install
+# AmigaOS /C/install, which installs bootblocks on floppy discs
+# AIX 4 /usr/bin/installbsd, which doesn't work without a -g flag
+# AFS /usr/afsws/bin/install, which mishandles nonexistent args
+# SVR4 /usr/ucb/install, which tries to use the nonexistent group "staff"
+# OS/2's system install, which has a completely different semantic
+# ./install, which can be erroneously created by make from ./install.sh.
+# Reject install programs that cannot install multiple files.
+{ $as_echo "$as_me:$LINENO: checking for a BSD-compatible install" >&5
+$as_echo_n "checking for a BSD-compatible install... " >&6; }
+if test -z "$INSTALL"; then
+if test "${ac_cv_path_install+set}" = set; then
+  $as_echo_n "(cached) " >&6
+else
+  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
+for as_dir in $PATH
+do
+  IFS=$as_save_IFS
+  test -z "$as_dir" && as_dir=.
+  # Account for people who put trailing slashes in PATH elements.
+case $as_dir/ in
+  ./ | .// | /cC/* | \
+  /etc/* | /usr/sbin/* | /usr/etc/* | /sbin/* | /usr/afsws/bin/* | \
+  ?:\\/os2\\/install\\/* | ?:\\/OS2\\/INSTALL\\/* | \
+  /usr/ucb/* ) ;;
+  *)
+    # OSF1 and SCO ODT 3.0 have their own names for install.
+    # Don't use installbsd from OSF since it installs stuff as root
+    # by default.
+    for ac_prog in gin
\ No newline at end of file
diff -ur SDL-1.2.13/src/main/macos/exports/gendef.pl SDL-1.2.13-haiku/src/main/macos/exports/gendef.pl
--- SDL-1.2.13/src/main/macos/exports/gendef.pl	2007-12-30 20:47:53.000000000 -0800
+++ SDL-1.2.13-haiku/src/main/macos/exports/gendef.pl	2007-12-30 20:47:53.000000000 -0800
@@ -1,43 +1,36 @@
-#!/usr/bin/perl
-#
-# Program to take a set of header files and generate DLL export definitions
+r conftest.dir
+	    if "$as_dir/$ac_prog$ac_exec_ext" -c conftest.one conftest.two "`pwd`/conftest.dir" &&
+	      test -s conftest.one && test -s conftest.two &&
+	      test -s conftest.dir/conftest.one &&
+	      test -s conftest.dir/conftest.two
+	    then
+	      ac_cv_path_install="$as_dir/$ac_prog$ac_exec_ext -c"
+	      break 3
+	    fi
+	  fi
+	fi
+      done
+    done
+    ;;
+esac
 
-# Special exports to ignore for this platform
+done
+IFS=$as_save_IFS
 
-while ( ($file = shift(@ARGV)) ) {
-	if ( ! defined(open(FILE, $file)) ) {
-		warn "Couldn't open $file: $!\n";
-		next;
-	}
-	$printed_header = 0;
-	$file =~ s,.*/,,;
-	while (<FILE>) {
-		if ( / DECLSPEC.* SDLCALL ([^\s\(]+)/ ) {
-			if ( not $exclude{$1} ) {
-				print "\t$1\n";
-			}
-		}
-	}
-	close(FILE);
-}
+rm -rf conftest.one conftest.two conftest.dir
 
-# Special exports to include for this platform
-print "\tSDL_putenv\n";
-print "\tSDL_getenv\n";
-print "\tSDL_qsort\n";
-print "\tSDL_revcpy\n";
-print "\tSDL_strlcpy\n";
-print "\tSDL_strlcat\n";
-print "\tSDL_strdup\n";
-print "\tSDL_strrev\n";
-print "\tSDL_strupr\n";
-print "\tSDL_strlwr\n";
-print "\tSDL_ltoa\n";
-print "\tSDL_ultoa\n";
-print "\tSDL_strcasecmp\n";
-print "\tSDL_strncasecmp\n";
-print "\tSDL_snprintf\n";
-print "\tSDL_vsnprintf\n";
-print "\tSDL_iconv\n";
-print "\tSDL_iconv_string\n";
-print "\tSDL_InitQuickDraw\n";
+fi
+  if test "${ac_cv_path_install+set}" = set; then
+    INSTALL=$ac_cv_path_install
+  else
+    # As a last resort, use the slow shell script.  Don't cache a
+    # value for INSTALL within a source directory, because that will
+    # break other packages using the cache if that directory is
+    # removed, or if the value is a relative name.
+    INSTALL=$ac_install_sh
+  fi
+fi
+{ $as_echo "$as_me:$LINENO: result: $INSTALL" >&5
+$as_echo "$INSTALL" >&6; }
+
+# Use test -z because SunOS4 sh mishandles braces in ${var-val}
\ No newline at end of file
Binary files SDL-1.2.13/src/main/macosx/Info.plist.in and SDL-1.2.13-haiku/src/main/macosx/Info.plist.in differ
Binary files SDL-1.2.13/src/main/macosx/SDLMain.h and SDL-1.2.13-haiku/src/main/macosx/SDLMain.h differ
Binary files SDL-1.2.13/src/main/macosx/SDLMain.m and SDL-1.2.13-haiku/src/main/macosx/SDLMain.m differ
diff -ur SDL-1.2.13/src/main/symbian/EKA2/SDL_main.cpp SDL-1.2.13-haiku/src/main/symbian/EKA2/SDL_main.cpp
--- SDL-1.2.13/src/main/symbian/EKA2/SDL_main.cpp	2007-12-30 20:47:54.000000000 -0800
+++ SDL-1.2.13-haiku/src/main/symbian/EKA2/SDL_main.cpp	2007-12-30 20:47:54.000000000 -0800
@@ -862,174 +862,186 @@
     gEpocEnv->iArgv[gEpocEnv->iArgc] = NULL;
          
     RThread thread;
-    User::LeaveIfError(thread.Create(KSDLMain, DoMain, aStackSize, NULL, NULL));
-    
-    if(aStatus != NULL)
-    	{
-    	thread.Logon(*aStatus);
-    	}
-    	
-    gEpocEnv->iId = thread.Id();
-    thread.SetPriority(EPriorityLess);
-    if((aFlags & CSDL::ERequestResume) == 0)
-        {
-        thread.Resume();
-        }
-    thread.Close();
-    return gEpocEnv->iId;
-    }
-    
-EXPORT_C TInt CSDL::AppendWsEvent(const TWsEvent& aEvent)
-    {
-    return EpocSdlEnv::EventQueue().Append(aEvent);
-    }
-    
-EXPORT_C void CSDL::SDLPanic(const TDesC& aInfo, TInt aErr)
-    {
-    EpocSdlEnv::PanicMain(aInfo, aErr);
-    }
-    
-EXPORT_C TInt CSDL::GetSDLCode(TInt aScanCode)
-    {
-    if(aScanCode < 0)
-        return MAX_SCANCODE;
-    if(aScanCode >= MAX_SCANCODE)
-        return -1;
-    return KeyMap()[aScanCode];
-    }
-    
-EXPORT_C TInt CSDL::SDLCodesCount() const
-	{
-	return MAX_SCANCODE;
-	}
-	
-EXPORT_C void CSDL::ResetSDLCodes()
-	{
-	ResetKeyMap();
-	}
-    
-EXPORT_C void CSDL::SetOrientation(TOrientationMode aMode)
-	{
-	gEpocEnv->iDsa->SetOrientation(aMode);
-	}
-    
-EXPORT_C TInt CSDL::SetSDLCode(TInt aScanCode, TInt aSDLCode)
-    {
-    const TInt current = GetSDLCode(aScanCode);
-    if(aScanCode >= 0 && aScanCode < MAX_SCANCODE)
-        KeyMap()[aScanCode] = static_cast<SDLKey>(aSDLCode);
-    return current;
-    }
-
-
-EXPORT_C MSDLObserver* CSDL::Observer()
-	{
-	return gEpocEnv->iAppSrv->Observer();
-	}    
-    
-EXPORT_C void CSDL::SetObserver(MSDLObserver* aObserver)
-	{
-	gEpocEnv->iAppSrv->SetObserver(aObserver);
-	}
-	
-EXPORT_C void CSDL::Resume()
-	{
-	EpocSdlEnv::Resume();
-	}
-	
-EXPORT_C void CSDL::Suspend()
-	{
-	gEpocEnv->iDsa->DoStop();
-	}
-	
-EXPORT_C CSDL::CSDL()
-    {
-    }
-
-EXPORT_C void CSDL::DisableKeyBlocking(CAknAppUi& aAppUi) const
-	{
-	gEpocEnv->iAppUi = &aAppUi;
-	EnvUtils::DisableKeyBlocking();
-	}
-
-EXPORT_C TInt CSDL::SetBlitter(MBlitter* aBlitter)
-	{
-	if(gEpocEnv && gEpocEnv->iDsa)
-		{
-		gEpocEnv->iDsa->SetBlitter(aBlitter);
-		return KErrNone;
-		}
-	return KErrNotReady;
-	}
-		
-	
-EXPORT_C TInt CSDL::AppendOverlay(MOverlay& aOverlay, TInt aPriority)
-	{
-	if(gEpocEnv && gEpocEnv->iDsa)
-		{
-		return gEpocEnv->iDsa->AppendOverlay(aOverlay, aPriority);
-		}
-	return KErrNotReady;
-	}
-
-EXPORT_C TInt CSDL::RemoveOverlay(MOverlay& aOverlay)	
-	{
-	if(gEpocEnv && gEpocEnv->iDsa)
-		{
-		return gEpocEnv->iDsa->RemoveOverlay(aOverlay);
-		}
-	return KErrNotReady;
-	}
-
-EXPORT_C TInt CSDL::RedrawRequest()
-	{
-	if(gEpocEnv && gEpocEnv->iDsa)
-		{
-		return gEpocEnv->iDsa->RedrawRequest();
-		}
-	return KErrNotReady;
-	}
-	
-/*
-EXPORT_C CSDL* CSDL::Current()
-    {
-    return gEpocEnv != NULL ? gEpocEnv->iSdl : NULL;
-    }
-
-    
-EXPORT_C TInt CSDL::SetVolume(TInt aVolume)
-    {
-    return EpocSdlEnv::SetVolume(aVolume);
-    } 
-    
-EXPORT_C TInt CSDL::Volume() const
-    {
-    return EpocSdlEnv::Volume();
-    }     
-	
-EXPORT_C TInt CSDL::MaxVolume() const
-    {
-    return EpocSdlEnv::MaxVolume();
-    } 	
-*/
-			
-void EnvUtils::DisableKeyBlocking()
-	{
-	if(gEpocEnv->iAppUi != NULL)
-		return CCurrentAppUi::Cast(gEpocEnv->iAppUi)->DisableKeyBlocking();
-	}
-	
-TBool EnvUtils::Rendezvous(RThread& aThread, TRequestStatus& aStatus)
-	{
-	if(gEpocEnv->iId != TThreadId(0) &&
-    		 	aThread.Open(gEpocEnv->iId) &&
-    		  	aThread.ExitType() == EExitPending)
-    			{
-    			aThread.Rendezvous(aStatus);
-    			return ETrue;
-    			}
-    return EFalse;
-	}
-	
-	
-
+    User::LeaveIfError(thread.Create(KSDLMain, DoMain, aStackSize, NUL8332BA31	0xC3C6
+0x8332BA32	0xC3C7
+0x8332BA33	0xC3C8
+0x8332BA34	0xC3C9
+0x8332BA35	0xC3CA
+0x8332BA36	0xC3CB
+0x8332BA37	0xC3CC
+0x8332BA38	0xC3CD
+0x8332BA39	0xC3CE
+0x8332BB30	0xC3CF
+0x8332BB31	0xC3D0
+0x8332BB32	0xC3D1
+0x8332BB33	0xC3D2
+0x8332BB34	0xC3D3
+0x8332BB35	0xC3D4
+0x8332BB36	0xC3D5
+0x8332BB37	0xC3D6
+0x8332BB38	0xC3D7
+0x8332BB39	0xC3D8
+0x8332BC30	0xC3D9
+0x8332BC31	0xC3DA
+0x8332BC32	0xC3DB
+0x8332BC33	0xC3DC
+0x8332BC34	0xC3DD
+0x8332BC35	0xC3DE
+0x8332BC36	0xC3DF
+0x8332BC37	0xC3E0
+0x8332BC38	0xC3E1
+0x8332BC39	0xC3E2
+0x8332BD30	0xC3E3
+0x8332BD31	0xC3E4
+0x8332BD32	0xC3E5
+0x8332BD33	0xC3E6
+0x8332BD34	0xC3E7
+0x8332BD35	0xC3E8
+0x8332BD36	0xC3E9
+0x8332BD37	0xC3EA
+0x8332BD38	0xC3EB
+0x8332BD39	0xC3EC
+0x8332BE30	0xC3ED
+0x8332BE31	0xC3EE
+0x8332BE32	0xC3EF
+0x8332BE33	0xC3F0
+0x8332BE34	0xC3F1
+0x8332BE35	0xC3F2
+0x8332BE36	0xC3F3
+0x8332BE37	0xC3F4
+0x8332BE38	0xC3F5
+0x8332BE39	0xC3F6
+0x8332BF30	0xC3F7
+0x8332BF31	0xC3F8
+0x8332BF32	0xC3F9
+0x8332BF33	0xC3FA
+0x8332BF34	0xC3FB
+0x8332BF35	0xC3FC
+0x8332BF36	0xC3FD
+0x8332BF37	0xC3FE
+0x8332BF38	0xC3FF
+0x8332BF39	0xC400
+0x8332C030	0xC401
+0x8332C031	0xC402
+0x8332C032	0xC403
+0x8332C033	0xC404
+0x8332C034	0xC405
+0x8332C035	0xC406
+0x8332C036	0xC407
+0x8332C037	0xC408
+0x8332C038	0xC409
+0x8332C039	0xC40A
+0x8332C130	0xC40B
+0x8332C131	0xC40C
+0x8332C132	0xC40D
+0x8332C133	0xC40E
+0x8332C134	0xC40F
+0x8332C135	0xC410
+0x8332C136	0xC411
+0x8332C137	0xC412
+0x8332C138	0xC413
+0x8332C139	0xC414
+0x8332C230	0xC415
+0x8332C231	0xC416
+0x8332C232	0xC417
+0x8332C233	0xC418
+0x8332C234	0xC419
+0x8332C235	0xC41A
+0x8332C236	0xC41B
+0x8332C237	0xC41C
+0x8332C238	0xC41D
+0x8332C239	0xC41E
+0x8332C330	0xC41F
+0x8332C331	0xC420
+0x8332C332	0xC421
+0x8332C333	0xC422
+0x8332C334	0xC423
+0x8332C335	0xC424
+0x8332C336	0xC425
+0x8332C337	0xC426
+0x8332C338	0xC427
+0x8332C339	0xC428
+0x8332C430	0xC429
+0x8332C431	0xC42A
+0x8332C432	0xC42B
+0x8332C433	0xC42C
+0x8332C434	0xC42D
+0x8332C435	0xC42E
+0x8332C436	0xC42F
+0x8332C437	0xC430
+0x8332C438	0xC431
+0x8332C439	0xC432
+0x8332C530	0xC433
+0x8332C531	0xC434
+0x8332C532	0xC435
+0x8332C533	0xC436
+0x8332C534	0xC437
+0x8332C535	0xC438
+0x8332C536	0xC439
+0x8332C537	0xC43A
+0x8332C538	0xC43B
+0x8332C539	0xC43C
+0x8332C630	0xC43D
+0x8332C631	0xC43E
+0x8332C632	0xC43F
+0x8332C633	0xC440
+0x8332C634	0xC441
+0x8332C635	0xC442
+0x8332C636	0xC443
+0x8332C637	0xC444
+0x8332C638	0xC445
+0x8332C639	0xC446
+0x8332C730	0xC447
+0x8332C731	0xC448
+0x8332C732	0xC449
+0x8332C733	0xC44A
+0x8332C734	0xC44B
+0x8332C735	0xC44C
+0x8332C736	0xC44D
+0x8332C737	0xC44E
+0x8332C738	0xC44F
+0x8332C739	0xC450
+0x8332C830	0xC451
+0x8332C831	0xC452
+0x8332C832	0xC453
+0x8332C833	0xC454
+0x8332C834	0xC455
+0x8332C835	0xC456
+0x8332C836	0xC457
+0x8332C837	0xC458
+0x8332C838	0xC459
+0x8332C839	0xC45A
+0x8332C930	0xC45B
+0x8332C931	0xC45C
+0x8332C932	0xC45D
+0x8332C933	0xC45E
+0x8332C934	0xC45F
+0x8332C935	0xC460
+0x8332C936	0xC461
+0x8332C937	0xC462
+0x8332C938	0xC463
+0x8332C939	0xC464
+0x8332CA30	0xC465
+0x8332CA31	0xC466
+0x8332CA32	0xC467
+0x8332CA33	0xC468
+0x8332CA34	0xC469
+0x8332CA35	0xC46A
+0x8332CA36	0xC46B
+0x8332CA37	0xC46C
+0x8332CA38	0xC46D
+0x8332CA39	0xC46E
+0x8332CB30	0xC46F
+0x8332CB31	0xC470
+0x8332CB32	0xC471
+0x8332CB33	0xC472
+0x8332CB34	0xC473
+0x8332CB35	0xC474
+0x8332CB36	0xC475
+0x8332CB37	0xC476
+0x8332CB38	0xC477
+0x8332CB39	0xC478
+0x8332CC30	0xC479
+0x8332CC31	0xC47A
+0x8332CC32	0xC47B
+0x8332CC33	0
\ No newline at end of file
Binary files SDL-1.2.13/src/stdlib/SDL_iconv.c and SDL-1.2.13-haiku/src/stdlib/SDL_iconv.c differ
Binary files SDL-1.2.13/src/thread/SDL_systhread.h and SDL-1.2.13-haiku/src/thread/SDL_systhread.h differ
Binary files SDL-1.2.13/src/thread/SDL_thread.c and SDL-1.2.13-haiku/src/thread/SDL_thread.c differ
Binary files SDL-1.2.13/src/thread/SDL_thread_c.h and SDL-1.2.13-haiku/src/thread/SDL_thread_c.h differ
Binary files SDL-1.2.13/src/thread/beos/SDL_syssem.c and SDL-1.2.13-haiku/src/thread/beos/SDL_syssem.c differ
Binary files SDL-1.2.13/src/thread/beos/SDL_systhread.c and SDL-1.2.13-haiku/src/thread/beos/SDL_systhread.c differ
Binary files SDL-1.2.13/src/thread/dc/SDL_syscond.c and SDL-1.2.13-haiku/src/thread/dc/SDL_syscond.c differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macevents.c and SDL-1.2.13-haiku/src/video/maccommon/SDL_macevents.c differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macevents_c.h and SDL-1.2.13-haiku/src/video/maccommon/SDL_macevents_c.h differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macgl.c and SDL-1.2.13-haiku/src/video/maccommon/SDL_macgl.c differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macgl_c.h and SDL-1.2.13-haiku/src/video/maccommon/SDL_macgl_c.h differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_mackeys.h and SDL-1.2.13-haiku/src/video/maccommon/SDL_mackeys.h differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macmouse.c and SDL-1.2.13-haiku/src/video/maccommon/SDL_macmouse.c differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macmouse_c.h and SDL-1.2.13-haiku/src/video/maccommon/SDL_macmouse_c.h differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macwm.c and SDL-1.2.13-haiku/src/video/maccommon/SDL_macwm.c differ
Binary files SDL-1.2.13/src/video/maccommon/SDL_macwm_c.h and SDL-1.2.13-haiku/src/video/maccommon/SDL_macwm_c.h differ
Binary files SDL-1.2.13/src/video/macdsp/SDL_dspvideo.c and SDL-1.2.13-haiku/src/video/macdsp/SDL_dspvideo.c differ
Binary files SDL-1.2.13/src/video/macdsp/SDL_dspvideo.h and SDL-1.2.13-haiku/src/video/macdsp/SDL_dspvideo.h differ
Binary files SDL-1.2.13/src/video/macrom/SDL_romvideo.c and SDL-1.2.13-haiku/src/video/macrom/SDL_romvideo.c differ
Binary files SDL-1.2.13/src/video/macrom/SDL_romvideo.h and SDL-1.2.13-haiku/src/video/macrom/SDL_romvideo.h differ
Binary files SDL-1.2.13/src/video/math_private.h and SDL-1.2.13-haiku/src/video/math_private.h differ
Binary files SDL-1.2.13/src/video/mmx.h and SDL-1.2.13-haiku/src/video/mmx.h differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxevents.c and SDL-1.2.13-haiku/src/video/nanox/SDL_nxevents.c differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxevents_c.h and SDL-1.2.13-haiku/src/video/nanox/SDL_nxevents_c.h differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nximage.c and SDL-1.2.13-haiku/src/video/nanox/SDL_nximage.c differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nximage_c.h and SDL-1.2.13-haiku/src/video/nanox/SDL_nximage_c.h differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxmodes.c and SDL-1.2.13-haiku/src/video/nanox/SDL_nxmodes.c differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxmodes_c.h and SDL-1.2.13-haiku/src/video/nanox/SDL_nxmodes_c.h differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxmouse.c and SDL-1.2.13-haiku/src/video/nanox/SDL_nxmouse.c differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxmouse_c.h and SDL-1.2.13-haiku/src/video/nanox/SDL_nxmouse_c.h differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxvideo.c and SDL-1.2.13-haiku/src/video/nanox/SDL_nxvideo.c differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxvideo.h and SDL-1.2.13-haiku/src/video/nanox/SDL_nxvideo.h differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxwm.c and SDL-1.2.13-haiku/src/video/nanox/SDL_nxwm.c differ
Binary files SDL-1.2.13/src/video/nanox/SDL_nxwm_c.h and SDL-1.2.13-haiku/src/video/nanox/SDL_nxwm_c.h differ
Binary files SDL-1.2.13/src/video/nds/SDL_ndsevents.c and SDL-1.2.13-haiku/src/video/nds/SDL_ndsevents.c differ
Binary files SDL-1.2.13/src/video/nds/SDL_ndsevents_c.h and SDL-1.2.13-haiku/src/video/nds/SDL_ndsevents_c.h differ
Binary files SDL-1.2.13/src/video/nds/SDL_ndsmouse.c and SDL-1.2.13-haiku/src/video/nds/SDL_ndsmouse.c differ
Binary files SDL-1.2.13/src/video/nds/SDL_ndsmouse_c.h and SDL-1.2.13-haiku/src/video/nds/SDL_ndsmouse_c.h differ
Binary files SDL-1.2.13/src/video/nds/SDL_ndsvideo.c and SDL-1.2.13-haiku/src/video/nds/SDL_ndsvideo.c differ
Binary files SDL-1.2.13/src/video/nds/SDL_ndsvideo.h and SDL-1.2.13-haiku/src/video/nds/SDL_ndsvideo.h differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_gl.c and SDL-1.2.13-haiku/src/video/photon/SDL_ph_gl.c differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_gl.h and SDL-1.2.13-haiku/src/video/photon/SDL_ph_gl.h differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_image.c and SDL-1.2.13-haiku/src/video/photon/SDL_ph_image.c differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_image_c.h and SDL-1.2.13-haiku/src/video/photon/SDL_ph_image_c.h differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_modes.c and SDL-1.2.13-haiku/src/video/photon/SDL_ph_modes.c differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_modes_c.h and SDL-1.2.13-haiku/src/video/photon/SDL_ph_modes_c.h differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_mouse.c and SDL-1.2.13-haiku/src/video/photon/SDL_ph_mouse.c differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_mouse_c.h and SDL-1.2.13-haiku/src/video/photon/SDL_ph_mouse_c.h differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_video.h and SDL-1.2.13-haiku/src/video/photon/SDL_ph_video.h differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_wm.c and SDL-1.2.13-haiku/src/video/photon/SDL_ph_wm.c differ
Binary files SDL-1.2.13/src/video/photon/SDL_ph_wm_c.h and SDL-1.2.13-haiku/src/video/photon/SDL_ph_wm_c.h differ
diff -ur SDL-1.2.13/test/configure SDL-1.2.13-haiku/test/configure
--- SDL-1.2.13/test/configure	2007-12-30 21:09:40.000000000 -0800
+++ SDL-1.2.13-haiku/test/configure	2008-05-26 00:19:27.000000000 -0700
@@ -2821,7 +2821,7 @@
         MATHLIB=""
         SYS_GL_LIBS="-lopengl32"
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         EXE=""
         MATHLIB=""
         SYS_GL_LIBS="-lGL"
diff -ur SDL-1.2.13/test/configure.in SDL-1.2.13-haiku/test/configure.in
--- SDL-1.2.13/test/configure.in	2007-12-30 20:48:35.000000000 -0800
+++ SDL-1.2.13-haiku/test/configure.in	2008-05-26 00:17:24.000000000 -0700
@@ -20,7 +20,7 @@
         MATHLIB=""
         SYS_GL_LIBS="-lopengl32"
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         EXE=""
         MATHLIB=""
         SYS_GL_LIBS="-lGL"
diff -ur SDL-1.2.13/test/testplatform.c SDL-1.2.13-haiku/test/testplatform.c
--- SDL-1.2.13/test/testplatform.c	2007-12-30 20:48:35.000000000 -0800
+++ SDL-1.2.13-haiku/test/testplatform.c	2008-05-26 22:21:42.000000000 -0700
@@ -154,7 +154,11 @@
 		printf("This system is running %s\n",
 #if __AIX__
 			"AIX"
-#elif __BEOS__
+#elif __HAIKU__
+			"Haiku"  
+  /* Haiku must appear here before BeOS,
+      since it also defines __BEOS__ */ 
+#elif __BEOS__ 
 			"BeOS"
 #elif __BSDI__
 			"BSDI"
