diff -urN SDL-1.2.13/README.Haiku SDL-1.2.13-haiku/README.Haiku
--- SDL-1.2.13/README.Haiku	1969-12-31 16:00:00.000000000 -0800
+++ SDL-1.2.13-haiku/README.Haiku	2008-08-21 07:13:52.000000000 -0700
@@ -0,0 +1,13 @@
+
+SDL on Haiku
+==============
+
+You can build SDL on Haiku like any other GNU style package.
+e.g. ./configure && make && make install
+By default it is installed in /boot/develop/tools/gnupro/{bin,lib,etc.}
+
+Once you install SDL, you need to copy libSDL.so to /boot/home/config/lib,
+so it can be found by the dynamic linker.
+
+Enjoy!
+	Sam Lantinga				(slouken@libsdl.org)
diff -urN SDL-1.2.13/configure.in SDL-1.2.13-haiku/configure.in
--- SDL-1.2.13/configure.in	2007-12-30 20:48:39.000000000 -0800
+++ SDL-1.2.13-haiku/configure.in	2008-08-21 07:34:03.000000000 -0700
@@ -40,7 +40,7 @@
 AC_SUBST(LT_AGE)
 
 dnl Detect the canonical build and host environments
-AC_CONFIG_AUX_DIRS($srcdir/build-scripts)
+AC_CONFIG_AUX_DIR([build-scripts])
 AC_CANONICAL_HOST
 AC_C_BIGENDIAN
 if test x$ac_cv_c_bigendian = xyes; then
@@ -2456,7 +2456,7 @@
         SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
         SDL_LIBS="-lmingw32 -lSDLmain $SDL_LIBS -mwindows"
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         ARCH=beos
         ac_default_prefix=/boot/develop/tools/gnupro
         CheckDummyVideo
diff -urN SDL-1.2.13/include/SDL_platform.h SDL-1.2.13-haiku/include/SDL_platform.h
--- SDL-1.2.13/include/SDL_platform.h	2007-12-30 20:48:36.000000000 -0800
+++ SDL-1.2.13-haiku/include/SDL_platform.h	2008-08-21 07:15:35.000000000 -0700
@@ -29,10 +29,14 @@
 #undef __AIX__
 #define __AIX__		1
 #endif
-#if defined(__BEOS__)
+#if defined(__BEOS__) && !defined(__HAIKU__)
 #undef __BEOS__
 #define __BEOS__	1
 #endif
+#if defined(__HAIKU__)
+#undef __HAIKU__
+#define __HAIKU__ 1
+#endif
 #if defined(bsdi) || defined(__bsdi) || defined(__bsdi__)
 #undef __BSDI__
 #define __BSDI__	1
diff -urN SDL-1.2.13/include/begin_code.h SDL-1.2.13-haiku/include/begin_code.h
--- SDL-1.2.13/include/begin_code.h	2007-12-30 20:48:36.000000000 -0800
+++ SDL-1.2.13-haiku/include/begin_code.h	2008-08-21 07:15:35.000000000 -0700
@@ -33,7 +33,7 @@
 
 /* Some compilers use a special export keyword */
 #ifndef DECLSPEC
-# if defined(__BEOS__)
+# if defined(__BEOS__) || defined(__HAIKU__)
 #  if defined(__GNUC__)
 #   define DECLSPEC	__declspec(dllexport)
 #  else
diff -urN SDL-1.2.13/test/configure.in SDL-1.2.13-haiku/test/configure.in
--- SDL-1.2.13/test/configure.in	2007-12-30 20:48:35.000000000 -0800
+++ SDL-1.2.13-haiku/test/configure.in	2008-08-21 07:15:35.000000000 -0700
@@ -20,7 +20,7 @@
         MATHLIB=""
         SYS_GL_LIBS="-lopengl32"
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         EXE=""
         MATHLIB=""
         SYS_GL_LIBS="-lGL"
diff -urN SDL-1.2.13/test/testplatform.c SDL-1.2.13-haiku/test/testplatform.c
--- SDL-1.2.13/test/testplatform.c	2007-12-30 20:48:35.000000000 -0800
+++ SDL-1.2.13-haiku/test/testplatform.c	2008-08-21 07:15:35.000000000 -0700
@@ -154,7 +154,11 @@
 		printf("This system is running %s\n",
 #if __AIX__
 			"AIX"
-#elif __BEOS__
+#elif __HAIKU__
+			"Haiku"  
+  /* Haiku must appear here before BeOS,
+      since it also defines __BEOS__ */ 
+#elif __BEOS__ 
 			"BeOS"
 #elif __BSDI__
 			"BSDI"
