Index: configure
===================================================================
--- configure	(revision 17861)
+++ configure	(working copy)
@@ -1512,7 +1512,7 @@
 
 # OS specific
 case $target_os in
-    beos|haiku|zeta)
+    beos|zeta)
         prefix_default="$HOME/config"
         # helps building libavcodec
         add_cflags -DPIC -fomit-frame-pointer
@@ -1538,6 +1538,13 @@
             enable beos_netserver
             network_extralibs="-lnet"
         fi ;;
+    haiku)
+        prefix_default="/boot/common"
+        disable audio_beos
+        enable beosthreads
+        network_extralibs="-lnetwork"
+        ;;
     sunos)
         FFSERVERLDFLAGS=""
         SHFLAGS='-shared -Wl,-h,$$(@F)'