diff -urN ffmpeg-0.5/configure ffmpeg-0.5-haiku/configure
--- ffmpeg-0.5/configure	2009-03-01 17:57:14.000000000 +0000
+++ ffmpeg-0.5-haiku/configure	2009-10-18 22:24:38.000000000 +0000
@@ -1483,7 +1483,7 @@
 
 # OS specific
 case $target_os in
-    beos|haiku|zeta)
+    beos|zeta)
         prefix_default="$HOME/config"
         # helps building libavcodec
         add_cflags -DPIC -fomit-frame-pointer
@@ -1509,6 +1509,13 @@
             enable beos_netserver
             network_extralibs="-lnet"
         fi ;;
+    haiku)
+        prefix_default="/boot/common"
+        SHFLAGS=' -shared'
+        disable audio_beos
+        enable beosthreads
+        network_extralibs="-lnetwork"
+        ;;
     sunos)
         FFSERVERLDFLAGS=""
         SHFLAGS='-shared -Wl,-h,$$(@F)'
