--- ffmpeg-0.6/configure	2010-06-15 19:44:30.020447232 +0000
+++ ffmpeg-0.6-haiku/configure	2010-07-13 20:40:48.244056064 +0000
@@ -2127,7 +2127,7 @@
 
 # OS specific
 case $target_os in
-    beos|haiku|zeta)
+    beos|zeta)
         prefix_default="$HOME/config"
         # 3 gcc releases known for BeOS, each with ugly bugs
         gcc_version="$($cc -v 2>&1 | grep version | cut -d ' ' -f3-)"
@@ -2151,6 +2148,10 @@
             enable beos_netserver
             network_extralibs="-lnet"
         fi ;;
+    haiku)
+        prefix_default="/boot/common"
+        network_extralibs="-lnetwork"
+        ;;
     sunos)
         FFSERVERLDFLAGS=""
         SHFLAGS='-shared -Wl,-h,$$(@F)'
