diff -urN ffmpeg-23081/configure ffmpeg-23081-haiku/configure
--- ffmpeg-23081/configure	2010-05-10 22:20:12.867958784 +0000
+++ ffmpeg-23081-haiku/configure	2010-05-10 22:26:05.354680832 +0000
@@ -2121,7 +2121,7 @@
 
 # OS specific
 case $target_os in
-    beos|haiku|zeta)
+    beos|zeta)
         prefix_default="$HOME/config"
         # 3 gcc releases known for BeOS, each with ugly bugs
         gcc_version="$($cc -v 2>&1 | grep version | cut -d ' ' -f3-)"
@@ -2145,6 +2145,16 @@
             enable beos_netserver
             network_extralibs="-lnet"
         fi ;;
+    haiku)
+        prefix_default="/boot/common"
+        add_cflags "-DPIC -fomit-frame-pointer"
+        beosthreads="yes"
+        extralibs=""
+        audio_oss="no"
+        video4linux="no"
+        video4linux2="no"
+        network_extralibs="-lnetwork"
+        ;;
     sunos)
         FFSERVERLDFLAGS=""
         SHFLAGS='-shared -Wl,-h,$$(@F)'
