From b44fcd01d625972a8a4ee18cb72be1d0bd4cde38 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Mon, 28 Jul 2014 15:54:14 -0400
Subject: [PATCH] configure: remove invalid option. From a StackOverflow post.

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 4ed43a0..6a81e3b 100755
--- a/configure
+++ b/configure
@@ -4297,7 +4297,7 @@ void foo(void) { __asm__ volatile ("" ::); }
 EOF
 
 _restrict=
-for restrict_keyword in restrict __restrict__ __restrict; do
+for restrict_keyword in __restrict__ __restrict; do
     check_cc <<EOF && _restrict=$restrict_keyword && break
 void foo(char * $restrict_keyword p);
 EOF
-- 
1.8.3.4

