diff --git a/src/Makefile b/src/Makefile
index 430c5ab..979b3ba 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -8,7 +8,7 @@ SDL_FLAGS := $(shell sdl-config --cflags)
 
 CFLAGS := $(CFLAGS) -Wall -fsigned-char
 LDFLAGS := $(LDFLAGS)
-LIBS :=
+LIBS := -lintl -liconv
 
 ifdef DEBUG
 CFLAGS := $(CFLAGS) -O0 -g -pedantic -DWITH_DEBUG
@@ -70,6 +70,9 @@ endif
 ifeq ($(OS),Linux)
 PLATFORM := all
 endif
+ifeq ($(OS),Haiku)
+PLATFORM := all
+endif
 endif
 
 include Makefile.$(PLATFORM)
