diff -up tar-1.22/lib/string.in.h.orig tar-1.22/lib/string.in.h
--- tar-1.22/lib/string.in.h.orig	2009-12-29 20:13:44.000000000 -0700
+++ tar-1.22/lib/string.in.h	2009-12-29 20:14:22.000000000 -0700
@@ -39,6 +39,7 @@
 /* The attribute __pure__ was added in gcc 2.96.  */
 # if __GNUC__ < 2 || (__GNUC__ == 2 && __GNUC_MINOR__ < 96)
 #  define __pure__ /* empty */
+#  define restrict __restrict
 # endif
 #endif
 
