From 12b60ca7f246e0d55d6c2bcac983c74737366e5a Mon Sep 17 00:00:00 2001
From: hanya <hanya.runo@gmail.com>
Date: Fri, 2 Mar 2018 09:37:04 +0900
Subject: [PATCH] Support OpenType fonts

---
 platforms/qhaikuplatform/haiku/qhaikuplatformfontdatabase.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platforms/qhaikuplatform/haiku/qhaikuplatformfontdatabase.cpp b/platforms/qhaikuplatform/haiku/qhaikuplatformfontdatabase.cpp
index c95abb7..d72a2d3 100644
--- a/platforms/qhaikuplatform/haiku/qhaikuplatformfontdatabase.cpp
+++ b/platforms/qhaikuplatform/haiku/qhaikuplatformfontdatabase.cpp
@@ -70,7 +70,7 @@ void QHaikuPlatformFontDatabase::populateFontDatabase()
 	for (int32 i = 0; i < fontPaths.CountStrings(); i++) {
 	    QDir dir(QLatin1String(fontPaths.StringAt(i).String()));
 		QDirIterator qdi(dir.absolutePath(),
-            QStringList() << "*.ttf",
+            QStringList() << "*.ttf" << "*.otf",
             QDir::Files, QDirIterator::Subdirectories);
     	while (qdi.hasNext()) {
 			const QByteArray file = QFile::encodeName(qdi.next());
