diff -up which-2.20/which.c.orig which-2.20/which.c
--- which-2.20/which.c.orig	2013-10-20 12:30:35.994574336 -0600
+++ which-2.20/which.c	2013-10-20 12:30:21.694681600 -0600
@@ -428,6 +428,7 @@ static uid_t const superuser = 0;
 
 int main(int argc, char *argv[])
 {
+  int function_start_type = 0;
   const char *path_list = getenv("PATH");
   int short_option, fail_count = 0;
   static int long_option;
@@ -537,7 +538,6 @@ int main(int argc, char *argv[])
     return -1;
   }
 
-  int function_start_type = 0;
   if (read_alias || read_functions)
   {
     char buf[1024];
