diff --git a/configure.ac b/configure.ac
index c4acf24..f70ea37 100644
--- a/configure.ac
+++ b/configure.ac
@@ -99,6 +99,28 @@ if test "x$fds_bits_found" = xfalse ; then
 fi
 
 if test "x$fds_bits_found" = xfalse ; then
+   AC_CHECK_MEMBER(fd_set.bits,
+			[
+			    fds_bits_found=plain
+			    USE_FDS_BITS="bits"
+			],,
+			[
+                               #ifdef HAVE_SYS_PARAM_H
+                               #include <sys/param.h>
+                               #endif
+                               #ifdef HAVE_SYS_TYPES_H
+                               #include <sys/types.h>
+                               #endif
+                               #ifdef HAVE_SYS_TIME_H
+                               #include <sys/time.h>
+                               #endif
+                               #ifdef HAVE_SYS_SELECT_H
+			       #include <sys/select.h>
+                               #endif
+			])
+fi
+
+if test "x$fds_bits_found" = xfalse ; then
     AC_MSG_ERROR([Could not determine how to access the fds_bits or equivalent
                   structure in fd_set on your platform.])
 fi
