diff --git a/boost/config/platform/haiku.hpp b/boost/config/platform/haiku.hpp
index 1b15c7e..c1b8e74 100644
--- a/boost/config/platform/haiku.hpp
+++ b/boost/config/platform/haiku.hpp
@@ -11,12 +11,18 @@
 
 #define BOOST_NO_INTRINSIC_WCHAR_T
 #define BOOST_HAS_UNISTD_H
+#define BOOST_NO_AUTO_PTR
+#define GLIBC_HAVE_LONG_LONG
+#define BOOST_HAS_STDINT_H
 	
 #define BOOST_HAS_BETHREADS
 	
 #ifndef BOOST_DISABLE_THREADS
 #   define BOOST_HAS_THREADS
 #endif
+#define BOOST_NO_CXX11_HDR_TYPE_TRAITS
+#define BOOST_NO_CXX11_ATOMIC_SMART_PTR
+
 
 //
 // thread API's not auto detected:
