diff -urN ImageMagick-6.6.1-10/configure.ac ImageMagick-6.6.1-10-haiku/configure.ac
--- ImageMagick-6.6.1-10/configure.ac	2010-04-26 00:45:55.061865984 +0000
+++ ImageMagick-6.6.1-10-haiku/configure.ac	2010-05-21 04:07:59.830472192 +0000
@@ -2556,7 +2556,7 @@
 AC_SUBST(EXECUTABLE_PATH)
 
 # Path to ImageMagick lib
-LIBRARY_PATH="${LIB_DIR}/${LIBRARY_RELATIVE_PATH}"
+LIBRARY_PATH="${LIB_DIR}/${LIBRARY_RELATIVE_PATH}:${LIBRARY_PATH}"
 DEFINE_LIBRARY_PATH="${LIB_DIR}/${LIBRARY_RELATIVE_PATH}/"
 case "${build_os}" in
   mingw* )
