diff -ur Organizer/Source/App/calenderView.cpp Organizer-fix/Source/App/calenderView.cpp
--- Organizer/Source/App/calenderView.cpp	2013-11-22 11:46:04.618659840 +0000
+++ Organizer-fix/Source/App/calenderView.cpp	2013-11-22 11:46:00.566755328 +0000
@@ -55,7 +55,7 @@
 //Load Settings
 MSave load("load");
 
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 String.SetTo(Path.Path());
 String.Append(SAVE_SETTINGS_PATH);
 Path.SetTo(String.String());
diff -ur Organizer/Source/App/editScriptView.cpp Organizer-fix/Source/App/editScriptView.cpp
--- Organizer/Source/App/editScriptView.cpp	2013-11-22 11:46:04.629669888 +0000
+++ Organizer-fix/Source/App/editScriptView.cpp	2013-11-22 11:46:00.575143936 +0000
@@ -145,7 +145,7 @@
 BString tempString3;
 BString FilePath;
 
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &fPath);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &fPath);
 tempString.SetTo(fPath.Path());
 tempString.Append(SAVE_HISTORY_PATH);
 fPath = tempString.String();
@@ -183,7 +183,7 @@
 int ItemNumber = 1;
 MSave Save("Save");
 
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 tempString.SetTo(Path.Path());
 tempString.Append(SAVE_HISTORY_PATH);
 	for (int a = 1; a <= INT_MAXFILES; a++)
diff -ur Organizer/Source/App/editWindow.cpp Organizer-fix/Source/App/editWindow.cpp
--- Organizer/Source/App/editWindow.cpp	2013-11-22 11:46:04.633864192 +0000
+++ Organizer-fix/Source/App/editWindow.cpp	2013-11-22 11:46:00.578027520 +0000
@@ -302,7 +302,7 @@
 	MSave Save("save");
 		if(!strcmp(fPath.String(), ""))
 		{
-		find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+		find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 		fPath.SetTo(Path.Path());
 		fPath.Append(SAVE_NOTE_PATH);
 		Path = fPath.String();
@@ -359,7 +359,7 @@
 	MSave Save("save");
 		if(!strcmp(fPath.String(), ""))
 		{
-		find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+		find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 		fPath.SetTo(Path.Path());
 		fPath.Append(SAVE_NOTE_PATH);
 		Path = fPath.String();
@@ -430,7 +430,7 @@
 	MSave Save("save");
 		if(!strcmp(fPath.String(), ""))
 		{
-		find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+		find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 		fPath.SetTo(Path.Path());
 		fPath.Append(SAVE_HISTORY_PATH);
 		Path = fPath.String();
@@ -509,7 +509,7 @@
 	MSave Save("save");
 		if(!strcmp(fPath.String(), ""))
 		{
-		find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+		find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 		fPath.SetTo(Path.Path());
 		fPath.Append(SAVE_NOTE_PATH);
 		Path = fPath.String();
diff -ur Organizer/Source/App/exportWindow.cpp Organizer-fix/Source/App/exportWindow.cpp
--- Organizer/Source/App/exportWindow.cpp	2013-11-22 11:46:04.641466368 +0000
+++ Organizer-fix/Source/App/exportWindow.cpp	2013-11-22 11:46:00.583794688 +0000
@@ -114,7 +114,7 @@
 BEntry Entry("/boot/home");
 MSave Load("Load");
 
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 fFolderPath.SetTo(Path.Path());
 fFolderPath.Append(SAVE_NOTE_PATH);
 
diff -ur Organizer/Source/App/main.cpp Organizer-fix/Source/App/main.cpp
--- Organizer/Source/App/main.cpp	2013-11-22 11:46:04.650641408 +0000
+++ Organizer-fix/Source/App/main.cpp	2013-11-22 11:46:00.590086144 +0000
@@ -11,7 +11,7 @@
 BPath path;
 BString tmp;
 MSave savefile("savefile");
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &path);
 tmp.SetTo(path.Path());
 tmp.Append(SAVE_SETTINGS_PATH);
 path.SetTo(tmp.String());
@@ -62,7 +62,7 @@
 								\Notes
 *********************************************/
 //Make sure all the folders are there
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &path);
 //Make fSub folder
 tmp.SetTo("mkdir ");
 tmp.Append(path.Path());
@@ -364,7 +364,7 @@
 					BString tempString;
 					BString tempString2;
 					BPath Path;
-					find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+					find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 					tempString.SetTo(path.Path());
 					tempString.Append(SAVE_SETTINGS_PATH);
 					save.FindString(tempString.String(), SAVE_FILE_NAME, NAME_EXEC_DIR, &tempString2, "M_ERROR");
@@ -476,7 +476,7 @@
 mApp::CheckForUpdateThread_static(void *data)
 {
 	mApp *check4updates = (mApp *)data;
-	check4updates->CheckForUpdateThread();
+	//check4updates->CheckForUpdateThread();
 	return 0;
 }
 
@@ -577,7 +577,7 @@
 	tmpString << _mWindow->_CalenderView->mCalender->ThisYear();
 	_mWindow->Unlock();
 	
-	find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+	find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 	FolderPath.SetTo(Path.Path());
 	FolderPath << SAVE_NOTE_PATH;
 		for (int a = 1; a <= INT_MAXFILES; a++)
@@ -631,7 +631,7 @@
 					if (NowMinute == TodayMinute && NowHour == TodayHour)
 					{
 					Save.FindString(FolderPath.String(), FileName.String(), SAVE_FILE_NOTE, &tmpString, "M_ERROR");
-					find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+					find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 					FolderPath.SetTo(Path.Path());
 					FolderPath << SAVE_PATH_PART_1 << SAVE_PATH_PART_2;
 					FileName.SetTo("TempRunScript");
diff -ur Organizer/Source/App/makefile Organizer-fix/Source/App/makefile
--- Organizer/Source/App/makefile	2013-11-22 11:46:04.654835712 +0000
+++ Organizer-fix/Source/App/makefile	2013-11-22 11:46:00.593231872 +0000
@@ -29,7 +29,7 @@
 	translation \
 	fSub
 DEFINES= 
-LIBPATHS= ../../lib
+LIBPATHS= ../lib
 
 SYSTEM_INCLUDE_PATHS = 
 LOCAL_INCLUDE_PATHS = 
diff -ur Organizer/Source/App/mWindow.cpp Organizer-fix/Source/App/mWindow.cpp
--- Organizer/Source/App/mWindow.cpp	2013-11-22 11:46:04.646184960 +0000
+++ Organizer-fix/Source/App/mWindow.cpp	2013-11-22 11:46:00.586940416 +0000
@@ -339,7 +339,7 @@
 {
 BPath Path;
 BString string;
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 string.SetTo(Path.Path());
 string.Append(SAVE_SETTINGS_PATH);
 path->SetTo(string.String());
@@ -358,7 +358,7 @@
 BString FilePath;
 BEntry Entry("/boot/home");
 
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &fPath);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &fPath);
 tempString.SetTo(fPath.Path());
 tempString.Append(SAVE_NOTE_PATH);
 fPath = tempString.String();
diff -ur Organizer/Source/App/obsolete/Language.cpp Organizer-fix/Source/App/obsolete/Language.cpp
--- Organizer/Source/App/obsolete/Language.cpp	2013-11-22 11:46:04.667942912 +0000
+++ Organizer-fix/Source/App/obsolete/Language.cpp	2013-11-22 11:46:00.659292160 +0000
@@ -201,7 +201,7 @@
 {
 BPath Path;
 BString string;
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 string.SetTo(Path.Path());
 string.Append(SAVE_SETTINGS_PATH);
 path->SetTo(string.String());
diff -ur Organizer/Source/App/preferencesView.cpp Organizer-fix/Source/App/preferencesView.cpp
--- Organizer/Source/App/preferencesView.cpp	2013-11-22 11:46:04.678428672 +0000
+++ Organizer-fix/Source/App/preferencesView.cpp	2013-11-22 11:46:00.665583616 +0000
@@ -49,7 +49,7 @@
 
 MSave load("load");
 
-find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 String.SetTo(Path.Path());
 String.Append(SAVE_SETTINGS_PATH);
 Path.SetTo(String.String());
diff -ur Organizer/Source/App/preferencesWindow.cpp Organizer-fix/Source/App/preferencesWindow.cpp
--- Organizer/Source/App/preferencesWindow.cpp	2013-11-22 11:46:04.681836544 +0000
+++ Organizer-fix/Source/App/preferencesWindow.cpp	2013-11-22 11:46:00.668729344 +0000
@@ -39,7 +39,7 @@
 		BPath Path;
 		BString String;
 		
-		find_directory(B_COMMON_SETTINGS_DIRECTORY, &Path);
+		find_directory(B_SYSTEM_SETTINGS_DIRECTORY, &Path);
 		String.SetTo(Path.Path());
 		String.Append(SAVE_SETTINGS_PATH);
 		String.Append(SAVE_FILE_NAME);
diff -ur Organizer/Source/App/Secret/glView.cpp Organizer-fix/Source/App/Secret/glView.cpp
--- Organizer/Source/App/Secret/glView.cpp	2013-11-22 11:46:04.602669056 +0000
+++ Organizer-fix/Source/App/Secret/glView.cpp	2013-11-22 11:46:00.556531712 +0000
@@ -22,7 +22,7 @@
 // Constructor/destructor
 TutGLView::TutGLView( Wind *owner_wind, BRect frame, const char *name, 
 					  int32 mode, int32 flags )
-	: BGLView( frame, const_cast<char *>( name ), mode, flags, 
+	: BGLView( frame, name, mode, flags,
 			   BGL_RGB | BGL_DOUBLE | BGL_DEPTH ),
 	  pad_x( 1.0 ),
 	  pad_y(-2.0),
@@ -131,7 +131,7 @@
 	// out.  You'll have to run the application from a Terminal to see
 	// these.
 	fprintf( stderr, "Unexpected error occured (%ld):\n", err );
-	fprintf( stderr, "\t%s\n", gluErrorString( err ) );
+	//fprintf( stderr, "\t%s\n", gluErrorString( err ) );
 }
 
 void TutGLView::Render( void )
@@ -164,7 +164,7 @@
 	glMatrixMode( GL_PROJECTION );
 	glLoadIdentity();
 	BRect r = Bounds();
-	gluPerspective( 45.0, r.Width() / r.Height(), 0.1, 100.0 );
+	//gluPerspective( 45.0, r.Width() / r.Height(), 0.1, 100.0 );
 	glMatrixMode( GL_MODELVIEW );
 }
 
@@ -178,7 +178,7 @@
 	glViewport( 0, 0, width, height );
 	glMatrixMode( GL_PROJECTION );
 	glLoadIdentity();
-	gluPerspective( 45.0, (float)width/(float)height, 0.1, 100.0 );
+	//gluPerspective( 45.0, (float)width/(float)height, 0.1, 100.0 );
 	glMatrixMode( GL_MODELVIEW );
 }
 
diff -ur Organizer/Source/lib/makefile Organizer-fix/Source/lib/makefile
--- Organizer/Source/lib/makefile	2013-11-22 11:46:04.719585280 +0000
+++ Organizer-fix/Source/lib/makefile	2013-11-22 11:46:00.701235200 +0000
@@ -1,4 +1,4 @@
-NAME= ../../../lib/libfSub.so
+NAME= ../libfSub.so
 TYPE= SHARED
 SRCS= Bitmap.cpp \
 	Bubble.cpp \
