Index: configure.in
===================================================================
--- configure.in	(revision 30382)
+++ configure.in	(working copy)
@@ -184,7 +184,7 @@
 dnl So we only turn it on for platforms where we know we really need it.
 AC_MSG_CHECKING([whether libtool needs -no-undefined])
 case $host in
-  *-*-cygwin*)
+  *-*-cygwin* | *-*-beos* | *-*-haiku*)
     AC_MSG_RESULT([yes])
     LT_NO_UNDEFINED="-no-undefined"
     ;;
