From b81f77d8d57d72bddf8aab1139203a5b86575c55 Mon Sep 17 00:00:00 2001
From: Jerome Duval <jerome.duval@gmail.com>
Date: Wed, 6 Apr 2022 16:16:47 +0200
Subject: [PATCH] reset xmouse_seq when setting xmouse_extended_seq

---
 lib/tty/tty.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/tty/tty.c b/lib/tty/tty.c
index 05b3a49..b72c935 100644
--- a/lib/tty/tty.c
+++ b/lib/tty/tty.c
@@ -405,6 +405,7 @@ tty_init_xterm_support (gboolean is_xterm)
             xmouse_seq = ESC_STR "[M";
 
         xmouse_extended_seq = ESC_STR "[<";
+        xmouse_seq = NULL;
     }
 }
 
-- 
2.30.2

