Only in bam-0.4.0: bam
diff -ur bam-0.4.0-orig/make_beos.sh bam-0.4.0/make_beos.sh
--- bam-0.4.0-orig/make_beos.sh	2010-08-09 21:08:24.043778048 +0300
+++ bam-0.4.0/make_beos.sh	2014-12-15 21:08:44.414187520 +0200
@@ -1,4 +1,4 @@
 #!/bin/sh
 gcc -Wall -ansi -pedantic src/tools/txt2c.c -o src/tools/txt2c
 src/tools/txt2c src/base.lua src/tools.lua src/driver_gcc.lua src/driver_cl.lua > src/internal_base.h
-gcc -Wall -ansi -pedantic src/lua/*.c src/*.c -o bam -I src/lua -lpthread -O2 $*
+gcc -Wall -ansi -pedantic src/lua/*.c src/*.c -o bam -I src/lua -O2 $*
diff -ur bam-0.4.0-orig/src/support.c bam-0.4.0/src/support.c
--- bam-0.4.0-orig/src/support.c	2010-08-09 21:08:24.045875200 +0300
+++ bam-0.4.0/src/support.c	2014-12-15 23:47:55.438042624 +0200
@@ -140,7 +140,11 @@
 	#include <dlfcn.h>
 	#include <unistd.h>
 	#include <sys/types.h>
+#ifdef BAM_FAMILY_BEOS
+	#include <signal.h>
+#else
 	#include <sys/signal.h>
+#endif
 	#include <sys/stat.h>
 	#include <utime.h>
 	#include <pthread.h>
Binary files bam-0.4.0-orig/src/tools/txt2c and bam-0.4.0/src/tools/txt2c differ
