--- fossil-src-20120317175325/src/main.c.org	2012-04-21 05:23:25.719847424 +0200
+++ fossil-src-20120317175325/src/main.c	2012-04-21 05:24:46.773062656 +0200
@@ -1625,7 +1625,7 @@
 }
 
 #if !defined(_WIN32)
-#if !defined(__DARWIN__) && !defined(__APPLE__)
+#if !defined(__DARWIN__) && !defined(__APPLE__) && !defined(__HAIKU__)
 /*
 ** Search for an executable on the PATH environment variable.
 ** Return true (1) if found and false (0) if not found.
@@ -1721,7 +1721,7 @@
 #if !defined(_WIN32)
   /* Unix implementation */
   if( isUiCmd ){
-#if !defined(__DARWIN__) && !defined(__APPLE__)
+#if !defined(__DARWIN__) && !defined(__APPLE__) && !defined(__HAIKU__)
     zBrowser = db_get("web-browser", 0);
     if( zBrowser==0 ){
       static char *azBrowserProg[] = { "xdg-open", "gnome-open", "firefox" };
