diff -urN libvorbis-1.3.1-orig/configure.ac libvorbis-1.3.1/configure.ac
--- libvorbis-1.3.1-orig/configure.ac	2010-03-26 06:56:58.061079552 +0000
+++ libvorbis-1.3.1/configure.ac	2010-05-12 05:56:55.536870912 +0000
@@ -206,6 +206,10 @@
 		DEBUG="-g -Wall -W -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
 		CFLAGS="-O20 -Wall -W -ffast-math -D_REENTRANT -fsigned-char"
 		PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char";;
+ 	*-*-haiku*)
+		DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
+		CFLAGS="-O20 -Wall -D__NO_MATH_INLINES -fsigned-char"
+		PROFILE="-O20 -g -pg -D__NO_MATH_INLINES -fsigned-char" ;;
  	*)
 		DEBUG="-g -Wall -Wextra -D__NO_MATH_INLINES -fsigned-char"
 		CFLAGS="-O20 -Wall -Wextra -D__NO_MATH_INLINES -fsigned-char"
