diff -urN tnftpd-20091122/configure.ac tnftpd-20091122-haiku/configure.ac
--- tnftpd-20091122/configure.ac	2009-12-30 01:42:42.036962304 +0000
+++ tnftpd-20091122-haiku/configure.ac	2010-08-30 03:35:13.000000000 +0000
@@ -123,7 +123,7 @@
                              [LIBS="-lsocket -lnsl $LIBS"],
                              [],
                              [-lnsl])])
-
+AC_CHECK_LIB(network, socket, LIBS="-lnetwork $LIBS")                              
 #
 # Checks for header files.
 #
