diff --git a/Master/tlpkg/TeXLive/TLUtils.pm b/Master/tlpkg/TeXLive/TLUtils.pm
index afcd157..b8bcc4a 100644
--- a/Master/tlpkg/TeXLive/TLUtils.pm
+++ b/Master/tlpkg/TeXLive/TLUtils.pm
@@ -293,7 +293,7 @@ sub platform_name {
     $CPU = $guessed_platform =~ /hf$/ ? "armhf" : "armel";
   }
 
-  my @OSs = qw(aix cygwin darwin freebsd hpux irix
+  my @OSs = qw(aix cygwin darwin freebsd haiku hpux irix
                kfreebsd linux netbsd openbsd solaris);
   for my $os (@OSs) {
     # Match word boundary at the beginning of the os name so that
@@ -381,6 +381,7 @@ sub platform_desc {
     'hppa-hpux'        => 'HP-UX',
     'i386-cygwin'      => 'Cygwin on Intel x86',
     'i386-darwin'      => 'MacOSX legacy (10.5-10.6) on Intel x86',
+    'i386-haiku'       => 'Intel x86 with Haiku',
     'i386-freebsd'     => 'FreeBSD on Intel x86',
     'i386-kfreebsd'    => 'GNU/kFreeBSD on Intel x86',
     'i386-linux'       => 'GNU/Linux on Intel x86',
diff --git a/Master/tlpkg/texlive.tlpdb b/Master/tlpkg/texlive.tlpdb
index f917900..42f1100 100644
--- a/Master/tlpkg/texlive.tlpdb
+++ b/Master/tlpkg/texlive.tlpdb
@@ -3947,7 +3947,7 @@ longdesc installer to work.  Concerning the 00* names: All packages
 longdesc starting with 00texlive are ``virtual packages'', in the sense
 longdesc that no containers are generated and these packages are never
 longdesc split into .src and .doc sub-packages in the tlpdb.
-depend setting_available_architectures:aarch64-linux amd64-freebsd amd64-netbsd armhf-linux i386-cygwin i386-freebsd i386-linux i386-netbsd i386-solaris win32 x86_64-cygwin x86_64-darwin x86_64-darwinlegacy x86_64-linux x86_64-linuxmusl x86_64-solaris
+depend setting_available_architectures:aarch64-linux amd64-freebsd amd64-netbsd armhf-linux i386-cygwin i386-freebsd i386-haiku i386-linux i386-netbsd i386-solaris win32 x86_64-cygwin x86_64-darwin x86_64-darwinlegacy x86_64-linux x86_64-linuxmusl x86_64-solaris
 depend opt_autobackup:1
 depend opt_backupdir:tlpkg/backups
 depend opt_create_formats:1
-- 
2.24.0
