--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,23 +48,10 @@
 i128_support = []
 nightly = ["i128_support"]
 serde1 = ["serde", "serde_derive", "rand_core/serde1"]
-std = ["rand_core/std", "alloc", "libc", "winapi", "cloudabi", "fuchsia-zircon"]
-[target."cfg(target_os = \"cloudabi\")".dependencies.cloudabi]
-version = "0.0.3"
-optional = true
-[target."cfg(target_os = \"fuchsia\")".dependencies.fuchsia-zircon]
-version = "0.3.2"
-optional = true
+std = ["rand_core/std", "alloc", "libc"]
 [target."cfg(unix)".dependencies.libc]
 version = "0.2"
 optional = true
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3"
-features = ["minwindef", "ntsecapi", "profileapi", "winnt"]
-optional = true
-[target.wasm32-unknown-unknown.dependencies.stdweb]
-version = "0.4"
-optional = true
 [badges.appveyor]
 repository = "alexcrichton/rand"
 
