--- a/unsquash-2.c
+++ b/unsquash-2.c
@@ -118,11 +118,11 @@ int read_fragment_table_2(long long *directory_table_end)
 void read_fragment_2(unsigned int fragment, long long *start_block, int *size)
 {
 	TRACE("read_fragment: reading fragment %d\n", fragment);
-
+{
 	squashfs_fragment_entry_2 *fragment_entry = &fragment_table[fragment];
 	*start_block = fragment_entry->start_block;
 	*size = fragment_entry->size;
-}
+}}
 
 
 struct inode *read_inode_2(unsigned int start_block, unsigned int offset)
--- a/unsquash-3.c
+++ b/unsquash-3.c
@@ -105,11 +105,11 @@ int read_fragment_table_3(long long *directory_table_end)
 void read_fragment_3(unsigned int fragment, long long *start_block, int *size)
 {
 	TRACE("read_fragment: reading fragment %d\n", fragment);
-
+{
 	squashfs_fragment_entry_3 *fragment_entry = &fragment_table[fragment];
 	*start_block = fragment_entry->start_block;
 	*size = fragment_entry->size;
-}
+}}
 
 
 struct inode *read_inode_3(unsigned int start_block, unsigned int offset)
--- a/unsquash-4.c
+++ b/unsquash-4.c
@@ -85,13 +85,13 @@ int read_fragment_table_4(long long *directory_table_end)
 void read_fragment_4(unsigned int fragment, long long *start_block, int *size)
 {
 	TRACE("read_fragment: reading fragment %d\n", fragment);
-
+{
 	struct squashfs_fragment_entry *fragment_entry;
 
 	fragment_entry = &fragment_table[fragment];
 	*start_block = fragment_entry->start_block;
 	*size = fragment_entry->size;
-}
+}}
 
 
 struct inode *read_inode_4(unsigned int start_block, unsigned int offset)
