diff --git a/APlayer/Players/GameMusic/libgme/Music_Emu.h b/APlayer/Players/GameMusic/libgme/Music_Emu.h
index 0a8bc0a..fd5dbc1 100644
--- a/APlayer/Players/GameMusic/libgme/Music_Emu.h
+++ b/APlayer/Players/GameMusic/libgme/Music_Emu.h
@@ -107,14 +107,6 @@ public:
 	
 	// Set frequency equalizer parameters
 	void set_equalizer( equalizer_t const& );
-
-	// Construct equalizer of given treble/bass settings
-	static const equalizer_t make_equalizer( double treble, double bass )
-	{
-	    const Music_Emu::equalizer_t e = { treble, bass,
-		0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 };
-	    return e;
-	}
 	
 	// Equalizer settings for TV speaker
 	static equalizer_t const tv_eq;
diff --git a/APlayer/Players/GameMusic/libgme/Nsfe_Emu.h b/APlayer/Players/GameMusic/libgme/Nsfe_Emu.h
index 32b05d5..880b951 100644
--- a/APlayer/Players/GameMusic/libgme/Nsfe_Emu.h
+++ b/APlayer/Players/GameMusic/libgme/Nsfe_Emu.h
@@ -46,7 +46,9 @@ public:
 public:
 	// deprecated
 	struct header_t { char tag [4]; };
-	using Music_Emu::load;
+	#ifdef (!defined(__GNUC__) || (__GNUC__ > 2))
+	Music_Emu::load;
+	#endif
 	blargg_err_t load( header_t const& h, Data_Reader& in ) // use Remaining_Reader
 			{ return load_remaining_( &h, sizeof h, in ); }
 	void disable_playlist( bool = true ); // use clear_playlist()
