diff --git a/src/AppSettings.cpp b/src/AppSettings.cpp
index 1bc0035..5e1ae36 100644
--- a/src/AppSettings.cpp
+++ b/src/AppSettings.cpp
@@ -483,7 +483,11 @@ AppSettings::AppSettings()
         userDir = QFileInfo(settings.fileName()).dir().absolutePath();
 
         QStringList translationPaths;
+#ifdef Q_OS_HAIKU
+        translationPaths.append((appDir + "/../data/ghostwriter/translations"));
+#else
         translationPaths.append(appDir + "/translations");
+#endif
         translationPaths.append(appDir + "/../share/" +
             QCoreApplication::applicationName().toLower() +
             "/translations");
