diff -Naur OpenJPEG_v1_3/Makefile OpenJPEG_v1_3-haiku/Makefile
--- OpenJPEG_v1_3/Makefile	2007-12-21 10:39:41.000000000 +0000
+++ OpenJPEG_v1_3-haiku/Makefile	2009-11-03 21:15:52.000000000 +0000
@@ -11,7 +11,7 @@
 CC = gcc
 AR = ar
 
-PREFIX = /usr
+PREFIX = /boot/common
 INSTALL_LIBDIR = $(PREFIX)/lib
 INSTALL_INCLUDE = $(PREFIX)/include
 
@@ -58,11 +58,11 @@
 
 install: OpenJPEG
 	install -d '$(DESTDIR)$(INSTALL_LIBDIR)' '$(DESTDIR)$(INSTALL_INCLUDE)'
-	install -m 644 -o root -g root $(STATICLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
+	install -m 644 -o baron -g users $(STATICLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
 	ranlib '$(DESTDIR)$(INSTALL_LIBDIR)/$(STATICLIB)'
-	install -m 755 -o root -g root $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
+	install -m 755 -o baron -g users $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)'
 	ln -sf $(SHAREDLIB) '$(DESTDIR)$(INSTALL_LIBDIR)/$(LIBNAME)'
-	install -m 644 -o root -g root libopenjpeg/openjpeg.h '$(DESTDIR)$(INSTALL_INCLUDE)'
+	install -m 644 -o baron -g users libopenjpeg/openjpeg.h '$(DESTDIR)$(INSTALL_INCLUDE)'
 	-ldconfig
 
 clean:
