From 6ed4f1f0b316a6819e4f38e36ee7f8863133c907 Mon Sep 17 00:00:00 2001
From: Scott McCreary <scottmc2@gmail.com>
Date: Mon, 30 Jan 2017 18:41:03 -0800
Subject: [PATCH] Fix gcc2 build

---
 joe/bw.c | 2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/joe/bw.c b/joe/bw.c
index ecbfb32..f620585 100644
--- a/joe/bw.c
+++ b/joe/bw.c
@@ -1208,8 +1208,8 @@ int ucrawll(W *w, int k)
 {
 	BW *bw;
 	off_t amnt;
-	WIND_BW(bw, w);
 	int rtn = -1;
+	WIND_BW(bw, w);

 	if (opt_left < 0)
 		amnt = bw->w / (-opt_left);
--
2.7.0
