diff --git a/JamMin.rdef b/JamMin.rdef
index 1707392..a46587d 100644
--- a/JamMin.rdef
+++ b/JamMin.rdef
@@ -13,10 +13,7 @@ resource app_version {
 	variety = B_APPV_ALPHA,
 	internal = 0,
 
-	short_info = "The Jamfile Administrator",
-	long_info = #'CSTR' array {
-
-	}
+	short_info = "The Jamfile Administrator"
 };
 
 resource app_flags B_SINGLE_LAUNCH;
diff --git a/JamMinWindow.cpp b/JamMinWindow.cpp
index 2b0077f..2b3c9af 100644
--- a/JamMinWindow.cpp
+++ b/JamMinWindow.cpp
@@ -1001,10 +1001,6 @@ JamMinWindow::IsSystemLibraryPath(BPath* libPath)
 	//if (ret == B_OK) {
 		if ( strcmp(libPath->Path(), path.Path()) == 0 ) return true;
 	//}
-	ret = find_directory(B_COMMON_LIB_DIRECTORY, &path);
-	if (ret == B_OK) {
-		if ( strcmp(libPath->Path(), path.Path()) == 0 ) return true;
-	}
 	
 	ret = find_directory(B_USER_LIB_DIRECTORY, &path); //the same as B_COMMON_LIB_DIRECTORY
 	if (ret == B_OK) {
diff --git a/makefile b/makefile
new file mode 100644
index 0000000..8514e9d
--- /dev/null
+++ b/makefile
@@ -0,0 +1,14 @@
+## JamMin makefile for Haiku OS ##
+
+NAME= JamApp
+
+TYPE= APP
+
+SRCS= TargetSettingsWin.cpp TargetConfig.cpp main.cpp LinkerSettingsView.cpp JMOpenFilePanel.cpp JamMinWindow.cpp JamMinApp.cpp JamFile.cpp IncludeSettingsView.cpp GeneralSettingsView.cpp CompilerSettingsView.cpp
+RDEFS= JamMin.rdef
+LIBS= be stdc++.r4 root tracker
+OPTIMIZE= FULL
+WARNINGS = ALL
+
+## include makefile engine
+include $(BUILDHOME)/etc/makefile-engine
