--- browser/app/firefox-branding.js copy	2006-09-12 11:50:19.000000000 -0400
+++ browser/app/firefox-branding.js	2009-03-17 16:26:04.000000000 -0400
@@ -1,14 +1,5 @@
-pref("startup.homepage_override_url","http://www.mozilla.org/projects/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/%APP%/%VERSION%/firstrun/");
-// URL user can browse to manually if for some reason all update installation
-// attempts fail.
-pref("app.update.url.manual", "http://www.mozilla.org/products/%APP%/");
-// A default value for the "More information about this update" link
-// supplied in the "An update is available" page of the update wizard. 
-pref("app.update.url.details", "http://www.mozilla.org/projects/%APP%/");
-
 // Release notes URL
-pref("app.releaseNotesURL", "http://www.mozilla.org/projects/%APP%/%VERSION%/releasenotes/");
+pref("app.releaseNotesURL", "about:buildconfig");
 
 // Search codes belong only in builds with official branding
 pref("browser.search.param.yahoo-fr", "");
