Index: mozilla/widget/src/beos/nsLookAndFeel.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/beos/nsLookAndFeel.cpp,v
retrieving revision 1.27
diff -u -8 -p -r1.27 nsLookAndFeel.cpp
--- mozilla/widget/src/beos/nsLookAndFeel.cpp	26 Oct 2004 01:45:26 -0000	1.27
+++ mozilla/widget/src/beos/nsLookAndFeel.cpp	19 Feb 2009 04:37:06 -0000
@@ -157,19 +157,17 @@ nsresult nsLookAndFeel::NativeGetColor(c
       aColor = NS_RGB(0x00, 0x00, 0x00);
       break;
     case eColor_graytext:
       aColor = NS_RGB(0x77, 0x77, 0x77);
       break;
     case eColor_highlight:
     case eColor__moz_menuhover:
       {
-        // B_MENU_SELECTION_BACKGROUND_COLOR  is used for text selection
-        // this blue colors seems more suitable
-        color = ui_color(B_KEYBOARD_NAVIGATION_COLOR);
+        color = ui_color(B_MENU_SELECTION_BACKGROUND_COLOR );
         aColor = NS_RGB(color.red, color.green, color.blue);
       }
       break;
     case eColor_highlighttext:
     case eColor__moz_menuhovertext:
       {
         color = ui_color(B_MENU_SELECTED_ITEM_TEXT_COLOR);
         aColor = NS_RGB(color.red, color.green, color.blue);
