Index: mozilla/toolkit/xre/nsAppRunner.cpp
===================================================================
RCS file: /cvsroot/mozilla/toolkit/xre/nsAppRunner.cpp,v
retrieving revision 1.113.2.24
diff -u -4 -r1.113.2.24 nsAppRunner.cpp
--- mozilla/toolkit/xre/nsAppRunner.cpp	8 Nov 2007 11:56:25 -0000	1.113.2.24
+++ mozilla/toolkit/xre/nsAppRunner.cpp	16 Feb 2009 19:19:26 -0000
@@ -1360,12 +1360,13 @@
 
 #elif defined(XP_BEOS)
   int32 cookie = 0;
   image_info info;
-
-  if(get_next_image_info(0, &cookie, &info) != B_OK)
-    return NS_ERROR_FAILURE;
-
+  do {
+    if(get_next_image_info(0, &cookie, &info) != B_OK)
+      return NS_ERROR_FAILURE;
+  } while(info.type != B_APP_IMAGE);
+  
   rv = NS_NewNativeLocalFile(nsDependentCString(info.name), PR_TRUE,
                              getter_AddRefs(lf));
   if (NS_FAILED(rv))
     return rv;
