Index: mozilla/nsprpub/pr/src/io/prpolevt.c
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/src/io/prpolevt.c,v
retrieving revision 3.15
diff -u -8 -p -r3.15 prpolevt.c
--- mozilla/nsprpub/pr/src/io/prpolevt.c	25 Apr 2004 15:00:58 -0000	3.15
+++ mozilla/nsprpub/pr/src/io/prpolevt.c	18 Feb 2009 02:06:41 -0000
@@ -443,20 +443,20 @@ PR_IMPLEMENT(PRFileDesc *) PR_NewPollabl
 #else
     if (PR_NewTCPSocketPair(fd) == PR_FAILURE) {
         fd[0] = fd[1] = NULL;
         goto errorExit;
     }
 	/*
 	 * set the TCP_NODELAY option to reduce notification latency
 	 */
-    socket_opt.option = PR_SockOpt_NoDelay;
-    socket_opt.value.no_delay = PR_TRUE;
-    rv = PR_SetSocketOption(fd[1], &socket_opt);
-    PR_ASSERT(PR_SUCCESS == rv);
+//    socket_opt.option = PR_SockOpt_NoDelay;
+//    socket_opt.value.no_delay = PR_TRUE;
+//    rv = PR_SetSocketOption(fd[1], &socket_opt);
+//    PR_ASSERT(PR_SUCCESS == rv);
 #endif
 
     event->secret->writeEnd = fd[1];
     if (PR_PushIOLayer(fd[0], PR_TOP_IO_LAYER, event) == PR_FAILURE) {
         goto errorExit;
     }
 
     return fd[0];
