--- ../Makefile	2004-05-07 05:39:47.034340864 +0000
+++ Makefile-new	2013-12-13 22:51:00.000000000 +0000
@@ -2,7 +2,7 @@
 CFLAGS+=-O2 -Wall -W -g
 #CFLAGS+=-O2 -Wall -W -g -fprofile-arcs -ftest-coverage -pg
 CFLAGS+=`sdl-config --cflags`
-LFLAGS=`sdl-config --libs` -lSDL_image -L/usr/X11R6/lib -lX11
+LFLAGS=`sdl-config --libs` -lSDL_image
 SRCS= \
 #	SFont.c \
 	board.c \
--- ../main.c	2004-05-07 05:36:26.035389440 +0000
+++ main-new.c	2013-12-13 23:41:06.150994944 +0000
@@ -2,6 +2,8 @@
 #include <stdlib.h>
 #include <sys/stat.h>
 #include <SDL.h>
+#include <unistd.h>
+#include <libgen.h>
 
 #include "board.h"
 #include "graphics.h"
@@ -27,11 +29,12 @@
 /*  3. Start the game loop */
 /*  4. Clean up before the program exits */
 /* ************************************************************************ */
-int main()
+int main(int argc, char *argv[])
 {
     game_t game;
     screen_t screen;
-    
+    // For tracker
+    chdir(dirname(argv[0]));
     /* initialize various systems (sounds, graphics, input, etc) */
     if (init(&screen)) 
     {       
