From 5791b4969c8acb81a6879035269319d3140d2bfa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
Date: Wed, 18 Dec 2013 14:10:05 +0100
Subject: [PATCH] C89
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: François Revol <revol@free.fr>
---
 ls-caps.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ls-caps.c b/ls-caps.c
index f192510..865a999 100644
--- a/ls-caps.c
+++ b/ls-caps.c
@@ -1260,8 +1260,8 @@ show_caps(struct device *d, int where)
 
   if (get_conf_word(d, PCI_STATUS) & PCI_STATUS_CAP_LIST)
     {
-      where = get_conf_byte(d, where) & ~3;
       byte been_there[256];
+      where = get_conf_byte(d, where) & ~3;
       memset(been_there, 0, 256);
       while (where)
 	{
-- 
1.8.3.4

