diff -Naur libdvdcss-1.2.12/configure.ac libdvdcss-1.2.12-haiku/configure.ac
--- libdvdcss-1.2.12/configure.ac	2012-03-11 23:07:47.053739520 +0000
+++ libdvdcss-1.2.12-haiku/configure.ac	2012-12-15 02:07:39.623378432 +0000
@@ -35,7 +35,7 @@
                  AC_DEFINE(SYS_CYGWIN, 1, Have a Cygwin system.))
     AC_DEFINE(WIN32, 1, Using Win32.)
     ;;
-  xbeos*)
+  xbeos*|xhaiku*)
     AC_DEFINE(SYS_BEOS, 1, Have a BeOS system.)
     ;;
   xos2*)
