diff -Naur libdvdcss-1.2.12/configure.ac libdvdcss-1.2.12-haiku/configure.ac
--- libdvdcss-1.2.12/configure.ac	2012-03-11 18:07:47.032505856 -0500
+++ libdvdcss-1.2.12-haiku/configure.ac	2013-11-26 20:16:56.198180864 -0600
@@ -5,7 +5,7 @@
 AC_CANONICAL_SYSTEM
 
 AM_INIT_AUTOMAKE(libdvdcss, 1.2.12)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_PROG_CC
 AC_STDC_HEADERS
@@ -35,7 +35,7 @@
                  AC_DEFINE(SYS_CYGWIN, 1, Have a Cygwin system.))
     AC_DEFINE(WIN32, 1, Using Win32.)
     ;;
-  xbeos*)
+  xbeos*|xhaiku*)
     AC_DEFINE(SYS_BEOS, 1, Have a BeOS system.)
     ;;
   xos2*)
