diff -urN libdvdcss-1.2.10/configure.ac libdvdcss-1.2.10-haiku/configure.ac
--- libdvdcss-1.2.10/configure.ac	2009-02-28 19:51:59.000000000 +0000
+++ libdvdcss-1.2.10-haiku/configure.ac	2009-02-28 19:39:44.000000000 +0000
@@ -35,7 +35,7 @@
                  AC_DEFINE(SYS_CYGWIN, 1, Have a Cygwin system.))
     AC_DEFINE(WIN32, 1, Using Win32.)
     ;;
-  xbeos*)
+  xbeos*|xhaiku*)
     AC_DEFINE(SYS_BEOS, 1, Have a BeOS system.)
     ;;
   x*msvc*)
