diff -ur fontconfig-2.10.0/src/fcatomic.c fontconfig-2.10.0-haiku/src/fcatomic.c
--- fontconfig-2.10.0/src/fcatomic.c	2012-06-17 21:49:30.023855104 -0400
+++ fontconfig-2.10.0-haiku/src/fcatomic.c	2012-07-18 19:39:56.629145600 -0400
@@ -132,7 +132,7 @@
 	return FcFalse;
     }
     ret = link ((char *) atomic->tmp, (char *) atomic->lck);
-    if (ret < 0 && errno == EPERM)
+    if (ret < 0 && (errno == EPERM || errno == B_UNSUPPORTED))
     {
 	/* the filesystem where atomic->lck points to may not supports
 	 * the hard link. so better try to fallback
