diff -Naur libtorrent-0.12.5/rak/socket_address.h libtorrent-0.12.5-haiku/rak/socket_address.h
--- libtorrent-0.12.5/rak/socket_address.h	2008-05-07 12:19:14.000000000 +0000
+++ libtorrent-0.12.5-haiku/rak/socket_address.h	2009-11-25 03:19:59.000000000 +0000
@@ -54,6 +54,8 @@
 #include <netinet/in.h>
 #include <sys/types.h>
 #include <sys/socket.h>
+ 
+typedef unsigned short  sa_family_t;
 
 namespace rak {
 
