--- tuxracer-0.61/src/game_config.c.org	2016-04-26 17:44:17.903086080 +0200
+++ tuxracer-0.61/src/game_config.c	2016-04-26 17:44:05.612892672 +0200
@@ -505,7 +505,7 @@
         "# decreasing this number, at the cost of lower image quality." );
 
     INIT_PARAM_BOOL( 
-	fullscreen, True,
+	fullscreen, False,
 	"# If true then the game will run in full-screen mode." );
 
     INIT_PARAM_INT( 
--- tuxracer-0.61/src/main.c.org	2016-04-26 22:09:04.763887616 +0200
+++ tuxracer-0.61/src/main.c	2016-04-26 22:08:49.627572736 +0200
@@ -56,6 +56,11 @@
 #include "loading.h"
 #include "tcl_util.h"
 
+#ifdef __HAIKU__
+#include <unistd.h>
+#include <libgen.h>
+#endif
+
 #define WINDOW_TITLE "Tux Racer " VERSION
 
 #define GAME_INIT_SCRIPT "tuxracer_init.tcl"
@@ -122,6 +127,7 @@
 
 int main( int argc, char **argv ) 
 {
+	chdir(dirname(argv[0]));
     /* Print copyright notice */
     fprintf( stderr, "Tux Racer " VERSION " -- a Sunspire Studios Production "
 	     "(http://www.sunspirestudios.com)\n"
