diff -Naur rasqal-0.9.29/src/rasqal_expr_numerics.c rasqal-0.9.29-haiku/src/rasqal_expr_numerics.c
--- rasqal-0.9.29/src/rasqal_expr_numerics.c	2012-05-14 02:37:45.000000000 +0000
+++ rasqal-0.9.29-haiku/src/rasqal_expr_numerics.c	2012-12-11 15:24:37.000000000 +0000
@@ -36,6 +36,7 @@
 #include <stdlib.h>
 #endif
 #include <stdarg.h>
+#include <stdint.h>
 
 #include "rasqal.h"
 #include "rasqal_internal.h"
