diff -urp cdrtools-3.01/DEFAULTS/Defaults.haiku cdrtools-3.01-haiku/DEFAULTS/Defaults.haiku
--- cdrtools-3.01/DEFAULTS/Defaults.haiku	2013-11-04 20:58:20.000000000 +0000
+++ cdrtools-3.01-haiku/DEFAULTS/Defaults.haiku
@@ -30,16 +30,16 @@ CWARNOPTS=
 
 DEFINCDIRS=	$(SRCROOT)/include
 DEFOSINCDIRS=
-LDPATH=		-L/opt/schily/lib
+LDPATH=		-L$(shell finddir B_SYSTEM_LIB_DIRECTORY)
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	-R$(INS_BASE)/lib
 
 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/boot/opt/schily
+#INS_BASE=	/boot/opt/schily
 INS_KBASE=	/
 INS_RBASE=	/
 #
diff -urp cdrtools-3.01/libscg/scsi-beos.c cdrtools-3.01-haiku/libscg/scsi-beos.c
--- cdrtools-3.01/libscg/scsi-beos.c	2009-06-30 18:34:03.000000000 +0000
+++ cdrtools-3.01-haiku/libscg/scsi-beos.c
@@ -292,7 +292,11 @@ scgo_havebus(scgp, busno)
 	char		buf[128];
 
 	if (busno < 8)
+#ifdef	__HAIKU__
+		js_snprintf(buf, sizeof (buf), "/dev/disk/scsi/%d", busno);
+#else
 		js_snprintf(buf, sizeof (buf), "/dev/bus/scsi/%d", busno);
+#endif
 	else
 #ifdef	__HAIKU__
 		js_snprintf(buf, sizeof (buf), "/dev/disk/atapi/%d", busno-8);
@@ -320,9 +324,15 @@ scgo_fileno(scgp, busno, tgt, tlun)
 			return (f->fd);
 	}
 	if (busno < 8) {
+#ifdef __HAIKU__
+		js_snprintf(buf, sizeof (buf),
+					"/dev/disk/scsi/%d/%d/%d/raw",
+					busno, tgt, tlun);
+#else
 		js_snprintf(buf, sizeof (buf),
 					"/dev/bus/scsi/%d/%d/%d/raw",
 					busno, tgt, tlun);
+#endif
 	} else {
 		char *tgtstr = (tgt == 0) ? "master" : (tgt == 1) ? "slave" : "dummy";
 		js_snprintf(buf, sizeof (buf),
