diff -urN cdrtools-3.01/DEFAULTS/Defaults.haiku cdrtools-3.01-haiku/DEFAULTS/Defaults.haiku
--- cdrtools-3.01/DEFAULTS/Defaults.haiku	
+++ cdrtools-3.01-haiku/DEFAULTS/Defaults.haiku	
@@ -29,16 +29,16 @@
 CWARNOPTS=
 
 DEFINCDIRS=	$(SRCROOT)/include
-LDPATH=		-L/opt/schily/lib
+LDPATH=		-L$(shell finddir B_COMMON_LIB_DIRECTORY)
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	-R$(INS_BASE)/lib

 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/boot/opt/schily
+INS_BASE=	$(shell finddir B_COMMON_DIRECTORY)
 INS_KBASE=	/
 INS_RBASE=	/
 #
