diff -urN cdrtools-3.00/DEFAULTS/Defaults.haiku cdrtools-3.00-haiku/DEFAULTS/Defaults.haiku
--- cdrtools-3.00/DEFAULTS/Defaults.haiku	2009-07-25 15:36:53.000000000 +0000
+++ cdrtools-3.00-haiku/DEFAULTS/Defaults.haiku	2009-10-12 04:53:52.000000000 +0000
@@ -1,7 +1,7 @@
-#ident "@(#)Defaults.haiku	1.3 09/07/25 "
+#ident "@(#)Defaults.beos	1.5 09/06/04 "
 ###########################################################################
 #
-# global definitions for Haiku Systems
+# global definitions for BeOS Systems
 #
 ###########################################################################
 #
@@ -29,16 +29,16 @@
 CWARNOPTS=
 
 DEFINCDIRS=	$(SRCROOT)/include
-LDPATH=		-L/opt/schily/lib
+LDPATH=		-L$(shell finddir B_COMMON_LIB_DIRECTORY)
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	-R$(INS_BASE)/lib
 
 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/boot/opt/schily
+INS_BASE=	$(shell finddir B_COMMON_DIRECTORY)
 INS_KBASE=	/
 INS_RBASE=	/
 #
