diff -urN cdrtools-3.01/DEFAULTS/Defaults.haiku cdrtools-3.01-haiku/DEFAULTS/Defaults.haiku
--- cdrtools-3.01/DEFAULTS/Defaults.haiku	
+++ cdrtools-3.01-haiku/DEFAULTS/Defaults.haiku	
@@ -29,16 +29,16 @@
 CWARNOPTS=
 
 DEFINCDIRS=	$(SRCROOT)/include
-LDPATH=		-L/opt/schily/lib
+LDPATH=		-L$(shell finddir B_SYSTEM_LIB_DIRECTORY)
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	-R$(INS_BASE)/lib

 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/boot/opt/schily
+INS_BASE=	$(shell finddir B_SYSTEM_DIRECTORY)
 INS_KBASE=	/
 INS_RBASE=	/
 #
diff -urN cdrtools-3.01/RULES/x86_64-haiku-cc.rul cdrtools-3.01-haiku/RULES/x86_64-haiku-cc.rul
--- cdrtools-3.01/RULES/x86_64-haiku-cc.rul	1970-01-01 00:00:00.000000000 +0000
+++ cdrtools-3.01-haiku/RULES/x86_64-haiku-cc.rul	2012-12-08 19:24:57.418643968 +0000
@@ -0,0 +1,54 @@
+#ident "@(#)bepc-haiku-cc.rul	1.3 09/02/05 "
+###########################################################################
+# Written 1996 by J. Schilling
+###########################################################################
+#
+# Platform dependent MACROS for Haiku, a BeOS like OS
+#
+###########################################################################
+# Copyright (c) J. Schilling
+###########################################################################
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License, Version 1.0 only
+# (the "License").  You may not use this file except in compliance
+# with the License.
+#
+# See the file CDDL.Schily.txt in this distribution for details.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file CDDL.Schily.txt from this distribution.
+###########################################################################
+include $(SRCROOT)/$(RULESDIR)/rules.prg
+###########################################################################
+include $(SRCROOT)/$(RULESDIR)/cc-gcc.rul
+
+#
+# XXX Old BeOS needs libroot.so to get [efg]cvt()
+# XXX New BeOS seems to have this in libc and in addition we include
+# XXX strtod.c for floating point conversion on OS without such code.
+# XXX For this reason, -DNO_FLOATINGPOINT -DUSE_FLOATINGARGS has been
+# XXX removed and printing floating numbers will work with old/new BeOS
+#
+#OSDEFS +=	-DNO_FLOATINGPOINT -DUSE_FLOATINGARGS
+OSDEFS +=
+
+KDEFINES=	-DKERNEL -D_KERNEL
+
+#CONFFLAGS=	i586
+
+LIB_PREFIX=	lib
+LIB_SUFFIX=	.a
+SHL_SUFFIX=	.so
+
+#LIB_SOCKET=	-lsocket -lbind
+LIB_SOCKET=	-lnetwork
+LIB_MATH=
+LIB_KVM=	
+
+#LDOPTS=		$(LIBS_PATH) $(LDPATH) $(RUNPATH:-R%=-Wl,-R%)
+LDOPTS=		$(LIBS_PATH) $(LDPATH)
+
+LORDER=		echo
+TSORT=		cat
+
+LN=             /bin/ln -s
diff -urN cdrtools-3.01/RULES/x86_64-haiku-gcc.rul cdrtools-3.01-haiku/RULES/x86_64-haiku-gcc.rul
--- cdrtools-3.01/RULES/x86_64-haiku-gcc.rul	1970-01-01 00:00:00.000000000 +0000
+++ cdrtools-3.01-haiku/RULES/x86_64-haiku-gcc.rul	2012-12-08 19:24:57.421265408 +0000
@@ -0,0 +1,54 @@
+#ident "@(#)bepc-haiku-gcc.rul	1.3 09/02/05 "
+###########################################################################
+# Written 1996 by J. Schilling
+###########################################################################
+#
+# Platform dependent MACROS for Haiku, a BeOS like OS
+#
+###########################################################################
+# Copyright (c) J. Schilling
+###########################################################################
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License, Version 1.0 only
+# (the "License").  You may not use this file except in compliance
+# with the License.
+#
+# See the file CDDL.Schily.txt in this distribution for details.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file CDDL.Schily.txt from this distribution.
+###########################################################################
+include $(SRCROOT)/$(RULESDIR)/rules.prg
+###########################################################################
+include $(SRCROOT)/$(RULESDIR)/cc-gcc.rul
+
+#
+# XXX Old BeOS needs libroot.so to get [efg]cvt()
+# XXX New BeOS seems to have this in libc and in addition we include
+# XXX strtod.c for floating point conversion on OS without such code.
+# XXX For this reason, -DNO_FLOATINGPOINT -DUSE_FLOATINGARGS has been
+# XXX removed and printing floating numbers will work with old/new BeOS
+#
+#OSDEFS +=	-DNO_FLOATINGPOINT -DUSE_FLOATINGARGS
+OSDEFS +=
+
+KDEFINES=	-DKERNEL -D_KERNEL
+
+#CONFFLAGS=	i586
+
+LIB_PREFIX=	lib
+LIB_SUFFIX=	.a
+SHL_SUFFIX=	.so
+
+#LIB_SOCKET=	-lsocket -lbind
+LIB_SOCKET=	-lnetwork
+LIB_MATH=
+LIB_KVM=	
+
+#LDOPTS=		$(LIBS_PATH) $(LDPATH) $(RUNPATH:-R%=-Wl,-R%)
+LDOPTS=		$(LIBS_PATH) $(LDPATH)
+
+LORDER=		echo
+TSORT=		cat
+
+LN=             /bin/ln -s
diff -up cdrtools-3.01/libscg/scsi-beos.c.orig cdrtools-3.01/libscg/scsi-beos.c
--- cdrtools-3.01/libscg/scsi-beos.c.orig	2009-06-30 12:34:03.022282240 -0600
+++ cdrtools-3.01/libscg/scsi-beos.c	2012-07-19 18:25:01.146800640 -0600
@@ -292,7 +292,11 @@ scgo_havebus(scgp, busno)
 	char		buf[128];
 
 	if (busno < 8)
+#ifdef	__HAIKU__
+		js_snprintf(buf, sizeof (buf), "/dev/disk/scsi/%d", busno);
+#else
 		js_snprintf(buf, sizeof (buf), "/dev/bus/scsi/%d", busno);
+#endif
 	else
 #ifdef	__HAIKU__
 		js_snprintf(buf, sizeof (buf), "/dev/disk/atapi/%d", busno-8);
@@ -320,9 +324,15 @@ scgo_fileno(scgp, busno, tgt, tlun)
 			return (f->fd);
 	}
 	if (busno < 8) {
+#ifdef __HAIKU__
+		js_snprintf(buf, sizeof (buf),
+					"/dev/disk/scsi/%d/%d/%d/raw",
+					busno, tgt, tlun);
+#else
 		js_snprintf(buf, sizeof (buf),
 					"/dev/bus/scsi/%d/%d/%d/raw",
 					busno, tgt, tlun);
+#endif
 	} else {
 		char *tgtstr = (tgt == 0) ? "master" : (tgt == 1) ? "slave" : "dummy";
 		js_snprintf(buf, sizeof (buf),
