diff -up cdrtools-3.02/DEFAULTS/Defaults.haiku cdrtools-3.02-haiku/DEFAULTS/Defaults.haiku
--- cdrtools-3.02/DEFAULTS/Defaults.haiku	2013-11-04 20:58:20.000000000 +0000
+++ cdrtools-3.02-haiku/DEFAULTS/Defaults.haiku
@@ -30,22 +30,22 @@ CWARNOPTS=
 
 DEFINCDIRS=	$(SRCROOT)/include
 DEFOSINCDIRS=
-LDPATH=		-L/opt/schily/lib
+LDPATH=		-L$(shell finddir B_SYSTEM_LIB_DIRECTORY)
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+RUNPATH=	-R$(INS_BASE)/lib
 
 ###########################################################################
 #
 # Installation config stuff
 #
 ###########################################################################
-INS_BASE=	/boot/opt/schily
+#INS_BASE=	/boot/opt/schily
 INS_KBASE=	/
 INS_RBASE=	/
 #
 DEFUMASK=	002
 #
 DEFINSMODEF=	444
 DEFINSMODEX=	755
 DEFINSUSR=	root
-DEFINSGRP=	bin
+DEFINSGRP=	root
 
