diff -up cdrtools-3.02/libscg/scsi-beos.c cdrtools-3.02-haiku/libscg/scsi-beos.c
--- cdrtools-3.02/libscg/scsi-beos.c	2009-06-30 18:34:03.000000000 +0000
+++ cdrtools-3.02-haiku/libscg/scsi-beos.c
@@ -292,7 +292,11 @@ scgo_havebus(scgp, busno)
 	char		buf[128];
 
 	if (busno < 8)
+#ifdef	__HAIKU__
+		js_snprintf(buf, sizeof (buf), "/dev/disk/scsi/%d", busno);
+#else
 		js_snprintf(buf, sizeof (buf), "/dev/bus/scsi/%d", busno);
+#endif
 	else
 #ifdef	__HAIKU__
 		js_snprintf(buf, sizeof (buf), "/dev/disk/atapi/%d", busno-8);
@@ -320,9 +324,15 @@ scgo_fileno(scgp, busno, tgt, tlun)
 			return (f->fd);
 	}
 	if (busno < 8) {
+#ifdef __HAIKU__
+		js_snprintf(buf, sizeof (buf),
+					"/dev/disk/scsi/%d/%d/%d/raw",
+					busno, tgt, tlun);
+#else
 		js_snprintf(buf, sizeof (buf),
 					"/dev/bus/scsi/%d/%d/%d/raw",
 					busno, tgt, tlun);
+#endif
 	} else {
 		char *tgtstr = (tgt == 0) ? "master" : (tgt == 1) ? "slave" : "dummy";
 		js_snprintf(buf, sizeof (buf),
