diff -urN abiword-2.8.6/configure.in abiword-2.8.6-haiku/configure.in
--- abiword-2.8.6/configure.in	2010-06-13 13:52:21.002883584 +0000
+++ abiword-2.8.6-haiku/configure.in	2011-04-14 13:24:55.000000000 +0000
@@ -84,6 +84,9 @@
   ${gsf_req}
   ${wv_req}
 "
+haiku_pkgs="
+  ${fribidi_req}
+"
 
 # optional deps
 enchant_req='enchant >= 1.2.0'
@@ -119,6 +122,9 @@
 # win32 deps
 win_pkgs="$enchant_req"
 
+#haiku deps
+haiku_pkgs=""
+
 #
 # System tests
 #
@@ -144,6 +150,10 @@
     PLATFORM="win"
     TOOLKIT="win"
     ;;
+  *haiku*)
+    PLATFORM="haiku"
+    TOOLKIT="haiku"
+    ;;
   *)
     PLATFORM="unix"
     TOOLKIT="gtk"
@@ -408,6 +418,16 @@
         fi
 ])
 
+AC_ARG_WITH([haiku],
+	[AS_HELP_STRING([--with-haiku], [(Haiku only) add `finddir B_COMMON_DIRECTORY' prefix to CPP/LDFLAGS])], 
+[
+        if test "x$withval" != "xno"; then
+                CPPFLAGS="$CPPFLAGS -I`finddir B_COMMON_DIRECTORY`/include"
+                LDFLAGS="$LDFLAGS -L`finddir B_COMMON_DIRECTORY`/lib"
+		CC=$CXX
+        fi
+])
+
 AC_ARG_WITH([fink],
 	[AS_HELP_STRING([--with-fink], [(Mac OSX only) add `/sw' prefix to CPP/LDFLAGS])], 
 [
@@ -514,11 +534,14 @@
 	deps_pkgs="$deps_pkgs $cocoa_pkgs"
 	SYSTEM_LIBS="-framework Cocoa"
 	AC_CHECK_PROG([CONVERT], [convert], convert, AC_MSG_ERROR(Cannot find ImageMagick convert))
+elif test "$TOOLKIT" = "haiku"; then
+	deps_pkgs="$haiku_pkgs"
+	SYSTEM_LIBS="-lbe -lroot -lnetwork"
 else
 	deps_pkgs="$deps_pkgs $gtk_pkgs"
 fi
 
-PKG_CHECK_MODULES(DEPS,[$deps_pkgs])
+#PKG_CHECK_MODULES(DEPS,[$deps_pkgs])
 dnl specific gtk214 test.
 if test "$TOOLKIT" = "gtk"; then
 	PKG_CHECK_MODULES(GTK214, gtk+-2.0 > 2.14, [abi_cv_gtk214=yes], 
@@ -647,6 +670,7 @@
 AM_CONDITIONAL([TOOLKIT_COCOA], test "$TOOLKIT" == "cocoa")
 AM_CONDITIONAL([TOOLKIT_GTK], test "$TOOLKIT" == "gtk")
 AM_CONDITIONAL([TOOLKIT_WIN], test "$TOOLKIT" == "win")
+AM_CONDITIONAL([TOOLKIT_HAIKU], test "$TOOLKIT" == "haiku")
 
 if test "$enable_dynamic" == ""; then
 	enable_dynamic="no"
@@ -783,7 +807,7 @@
 		eval $(echo "enable_$plugin")="auto"
 	done
 else
-	default_plugins="opendocument"
+	default_plugins="" 
 
 	if test "$abi_cv_disable_default_plugins" != "yes"; then
 		PLUGINS="$default_plugins"
