--- libyahoo2-1.0.1-haiku/configure.ac	2010-07-27 01:13:13.015466496 +0000
+++ libyahoo2-1.0.1/configure.ac	2012-12-11 17:26:38.095682560 +0000
@@ -43,9 +43,9 @@
 
 # Checks for libraries.
 enable_sample_client="yes"
-AC_SEARCH_LIBS(socket, socket, ,
-	[AC_SEARCH_LIBS(socket, socket, LIBS="$LIBS -lnsl", 
-		enable_sample_client="no", -lnsl)])
+AC_SEARCH_LIBS(socket, socket network)
+AC_SEARCH_LIBS(gethostbyname2, bind nsl resolv)
+AC_CHECK_LIB(rt, clock_gettime)
 
 # Checks for header files.
 AC_HEADER_STDC
