diff --git a/Makefile "b/Makefile"
index ed15791..2043f32 100644
--- a/Makefile
+++ "b/Makefile"	
@@ -21,7 +21,7 @@ CFLAGS += -D_FILE_OFFSET_BITS=64
 FFMPEG_CFLAGS += $(shell pkg-config --cflags libswresample)
 FFMPEG_LIBS += $(shell pkg-config --libs libswresample)
 
-CMUS_LIBS = $(PTHREAD_LIBS) $(NCURSES_LIBS) $(ICONV_LIBS) $(DL_LIBS) $(DISCID_LIBS) $(CUE_LIBS) -lm $(COMPAT_LIBS)
+CMUS_LIBS = $(PTHREAD_LIBS) $(NCURSES_LIBS) $(ICONV_LIBS) $(DL_LIBS) $(DISCID_LIBS) $(CUE_LIBS)  $(COMPAT_LIBS)
 
 input.o main.o ui_curses.o pulse.lo: .version
 input.o main.o ui_curses.o pulse.lo: CFLAGS += -DVERSION=\"$(VERSION)\"
diff --git a/configure "b/configure"
index 224feaa..31b7e25 100755
--- a/configure
+++ "b/configure"	
@@ -10,7 +10,6 @@ check_cflags()
 
 	for i in -Wold-style-definition \
 		-Wno-pointer-sign \
-		-Werror-implicit-function-declaration \
 		-Wno-unused-parameter
 	do
 		check_cc_flag $i
@@ -156,7 +155,7 @@ check_mpc()
 	else
 		check_header mpcdec/mpcdec.h || return $?
 	fi
-	check_library MPC "" "-lmpcdec -lm"
+	check_library MPC "" "-lmpcdec "
 	return $?
 }
 
@@ -168,13 +167,13 @@ check_cddb()
 
 check_cdio()
 {
-	pkg_config CDIO "libcdio_cdda" "" "-lcdio_cdio -lcdio -lm"
+	pkg_config CDIO "libcdio_cdda" "" "-lcdio_cdio -lcdio "
 	return $?
 }
 
 check_flac()
 {
-	pkg_config FLAC "flac" "" "-lFLAC -lm" || return $?
+	pkg_config FLAC "flac" "" "-lFLAC " || return $?
 
 	# Make sure the FLAC_CFLAGS value is sane, strip trailing '/FLAC'.
 	FLAC_CFLAGS=`echo $FLAC_CFLAGS | sed "s/FLAC$//"`
@@ -183,7 +182,7 @@ check_flac()
 
 check_mad()
 {
-	pkg_config MAD "mad" "" "-lmad -lm"
+	pkg_config MAD "mad" "" "-lmad "
 	return $?
 }
 
@@ -213,7 +212,7 @@ check_mikmod()
 
 check_modplug()
 {
-	pkg_config MODPLUG "libmodplug" "-I/usr/include/libmodplug" "-lmodplug -lstdc++ -lm" || return $?
+	pkg_config MODPLUG "libmodplug" "-I/usr/include/libmodplug" "-lmodplug -lstdc++ " || return $?
 	MODPLUG_API_8=0
 	if check_function "ModPlug_GetModuleType" $MODPLUG_CFLAGS $MODPLUG_LIBS
 	then
@@ -233,10 +232,10 @@ check_vorbis()
 {
 	if test "$CONFIG_TREMOR" = y
 	then
-		pkg_config VORBIS "vorbisidec" "" "-lvorbisidec -lm"
+		pkg_config VORBIS "vorbisidec" "" "-lvorbisidec "
 		return $?
 	else
-		pkg_config VORBIS "vorbisfile" "" "-lvorbisfile -lvorbis -lm -logg"
+		pkg_config VORBIS "vorbisfile" "" "-lvorbisfile -lvorbis  -logg"
 		return $?
 	fi
 }
@@ -364,14 +363,14 @@ check_mp4()
 		check_header mp4.h || return $?
 	fi
 	check_header neaacdec.h &&
-	check_library MP4 "" "-lmp4v2 -lfaad -lm"
+	check_library MP4 "" "-lmp4v2 -lfaad "
 	return $?
 }
 
 check_aac()
 {
 	check_header neaacdec.h &&
-	check_library AAC "" "-lfaad -lm"
+	check_library AAC "" "-lfaad "
 	return $?
 }
 
diff --git a/scripts/checks.sh "b/scripts/checks.sh"
index 64cbbf3..22ebf58 100644
--- a/scripts/checks.sh
+++ "b/scripts/checks.sh"	
@@ -643,9 +643,12 @@ check_pthread()
 	OpenBSD)
 		PTHREAD_LIBS="$PTHREAD_LIBS -pthread"
 		;;
+	Haiku)
+		PTHREAD_LIBS="$PTHREAD_LIBS"
+		;;
 	esac
 
-	for __libs in "$PTHREAD_LIBS" -lpthread -lc_r -lkse
+	for __libs in "$PTHREAD_LIBS" -lpthread -lc_r -lkse -lroot
 	do
 		test -z "$__libs" && continue
 		check_library PTHREAD "-D_REENTRANT" "$__libs" && return 0
