diff -Naur sharutils-4.6.3-original/configure.ac sharutils-4.6.3/configure.ac
--- sharutils-4.6.3-original/configure.ac	2009-04-07 00:40:07.000000000 -0400
+++ sharutils-4.6.3/configure.ac	2009-04-07 00:41:01.000000000 -0400
@@ -9,6 +9,7 @@
 AC_INIT([sharutils], sharutils_version, sharutils_eaddr)
 AC_CONFIG_SRCDIR([src/shar.c])
 AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIR([m4])
 
 AM_INIT_AUTOMAKE([1.9 gnits dist-bzip2])
 AC_SUBST(DIST_ALPHA)
@@ -20,7 +21,7 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+LT_INIT
 
 dnl FIXME: Check that it is GNU diff indeed.
 # Carefully avoid gdiff for X as found on SGI systems.
