diff -up tcl8.5.9/unix/tcl.m4.orig tcl8.5.9/unix/tcl.m4
--- tcl8.5.9/unix/tcl.m4.orig	2011-04-14 15:28:58.862191616 -0600
+++ tcl8.5.9/unix/tcl.m4	2011-04-14 15:18:45.910163968 -0600
@@ -669,9 +669,16 @@ AC_DEFUN([SC_ENABLE_THREADS], [
 			# The space is needed
 			THREADS_LIBS=" -pthread"
 		    else
+		    AC_CHECK_LIB(root, pthread_mutex_init,
+		    tcl_ok=yes, tcl_ok=no)
+		    if test "$tcl_ok" = "yes"; then
+		    # The space is needed
+		    THREAD_LIBS=" "
+		    else
 			TCL_THREADS=0
 			AC_MSG_WARN([Don't know how to find pthread lib on your system - you must disable thread support or edit the LIBS in the Makefile...])
 		    fi
+		    fi
 		fi
 	    fi
 	fi
@@ -1276,6 +1283,7 @@ dnl AC_CHECK_TOOL(AR, ar)
 	    SHLIB_LD='${CC} -shared ${CFLAGS} ${LDFLAGS}'
 	    DL_OBJS="tclLoadDl.o"
 	    DL_LIBS="-lroot"
+	    LD_LIBRARY_PATH_VAR="LIBRARY_PATH"
 	    AC_CHECK_LIB(network, inet_ntoa, [LIBS="$LIBS -lnetwork"])
 	    ;;
 	HP-UX-*.11.*)
