diff --git a/configure.ac b/configure.ac
index 49ff391..7404001 100644
--- a/configure.ac
+++ b/configure.ac
@@ -100,6 +100,19 @@ echo "CXXFLAGS: $CXXFLAGS"
 #
 AC_CONFIG_HEADERS(config.h)
 
+dnl ----------------------------------------------
+dnl CHECK FOR MACHINE/VENDOR/OPERATING SYSTEM TYPE
+dnl ----------------------------------------------
+dnl Find out machine type, vendor, and operating system
+dnl What type of host is this?
+AC_CANONICAL_HOST
+AC_DEFINE_UNQUOTED(MACHTYPE, "$host_cpu",
+[Define to be the machine type (microprocessor class or machine model).])
+AC_DEFINE_UNQUOTED(VENDOR,   "$host_vendor",
+[Define to be a string corresponding the vendor of the machine.])
+AC_DEFINE_UNQUOTED(OSTYPE,   "$host_os",
+[Define to be the name of the operating system.])
+
 
 #
 # This adds markup to the code that results in a few extra compile
@@ -210,9 +223,15 @@ CXXFLAGS="$CXXFLAGS -Wall -Wno-sign-compare"
 # This is needed in order to get the really cool backtraces on Linux
 #
 
-if test `uname` != "Darwin"; then
-   LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
-fi
+case "$host_os" in
+	darwin* )
+		;;
+	haiku*)
+		;;
+	*)
+		LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
+		;;
+esac
 
 #
 # On Cygwin, we need to add some flags for ncurses.
@@ -349,7 +368,7 @@ AC_DEFINE([NOMACROS], [1], [Define to 1 to disable curses macros that conflict w
 #
 
 # Check for os dependant libraries for all binaries.
-AC_SEARCH_LIBS( connect, socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
+AC_SEARCH_LIBS( connect, network socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
 AC_SEARCH_LIBS( nanosleep, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( shm_open, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( pthread_create, pthread, , [AC_MSG_ERROR([Cannot find the pthread library, needed to build this package.] )] )
diff --git a/env_universal_common.cpp b/env_universal_common.cpp
index b493de2..0b9c8c9 100644
--- a/env_universal_common.cpp
+++ b/env_universal_common.cpp
@@ -36,6 +36,10 @@
 #include <notify.h>
 #endif
 
+#ifdef __HAIKU__
+#include <ifaddrs.h>
+#endif
+
 // NAME_MAX is not defined on Solaris and suggests the use of pathconf()
 // There is no obvious sensible pathconf() for shared memory and _XPG_NAME_MAX
 // seems a reasonable choice.
@@ -982,7 +986,7 @@ void env_universal_t::parse_message_internal(const wcstring &msgstr, var_table_t
 /* Thanks to Jan Brittenson
  http://lists.apple.com/archives/xcode-users/2009/May/msg00062.html
  */
-#ifdef SIOCGIFHWADDR
+#if defined (SIOCGIFHWADDR) && ! defined (__HAIKU__)
 
 /* Linux */
 #include <net/if.h>
@@ -1005,7 +1009,7 @@ static bool get_mac_address(unsigned char macaddr[MAC_ADDRESS_MAX_LEN], const ch
     return result;
 }
 
-#elif defined(HAVE_GETIFADDRS)
+#elif defined(HAVE_GETIFADDRS) && ! defined (__HAIKU__)
 
 /* OS X and BSD */
 #include <ifaddrs.h>
@@ -1102,6 +1106,10 @@ class universal_notifier_shmem_poller_t : public universal_notifier_t
 #define SHMEM_MAGIC_NUMBER 0xF154
 #define SHMEM_VERSION_CURRENT 1000
 
+#ifndef MAP_FILE
+#define MAP_FILE 0
+#endif
+
     private:
     long long last_change_time;
     uint32_t last_seed;
diff --git a/intern.cpp b/intern.cpp
index 56dec21..cc8a299 100644
--- a/intern.cpp
+++ b/intern.cpp
@@ -20,6 +20,10 @@
 #include "common.h"
 #include "intern.h"
 
+#ifdef __HAIKU__
+#include <posix/pthread.h>
+#endif
+
 /** Comparison function for intern'd strings */
 class string_table_compare_t
 {
diff --git a/iothread.cpp b/iothread.cpp
index 1fcd137..f621f8c 100644
--- a/iothread.cpp
+++ b/iothread.cpp
@@ -12,6 +12,12 @@
 #include <fcntl.h>
 #include <queue>
 
+#ifdef __HAIKU__
+#include <posix/pthread.h>
+#include <sys/types.h>
+#include <sys/select.h>
+#endif
+
 #ifdef _POSIX_THREAD_THREADS_MAX
 #if _POSIX_THREAD_THREADS_MAX < 64
 #define IO_MAX_THREADS _POSIX_THREAD_THREADS_MAX
