diff --git a/configure.ac b/configure.ac
index 49ff391..7d22bf9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -100,6 +100,18 @@ echo "CXXFLAGS: $CXXFLAGS"
 #
 AC_CONFIG_HEADERS(config.h)
 
+dnl ----------------------------------------------
+dnl CHECK FOR MACHINE/VENDOR/OPERATING SYSTEM TYPE
+dnl ----------------------------------------------
+dnl Find out machine type, vendor, and operating system
+dnl What type of host is this?
+AC_CANONICAL_HOST
+AC_DEFINE_UNQUOTED(MACHTYPE, "$host_cpu",
+[Define to be the machine type (microprocessor class or machine model).])
+AC_DEFINE_UNQUOTED(VENDOR,   "$host_vendor",
+[Define to be a string corresponding the vendor of the machine.])
+AC_DEFINE_UNQUOTED(OSTYPE,   "$host_os",
+[Define to be the name of the operating system.])
 
 #
 # This adds markup to the code that results in a few extra compile
@@ -210,9 +222,16 @@ CXXFLAGS="$CXXFLAGS -Wall -Wno-sign-compare"
 # This is needed in order to get the really cool backtraces on Linux
 #
 
-if test `uname` != "Darwin"; then
-   LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
-fi
+case "$host_os" in
+	darwin* )
+		;;
+	haiku*)
+		CFLAGS="$CFLAGS -D_BSD_SOURCE"  CPPFLAGS="$CPPFLAGS -D_BSD_SOURCE" CXXFLAGS="$CXXFLAGS -D_BSD_SOURCE"
+		;;
+	*)
+		LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
+		;;
+esac
 
 #
 # On Cygwin, we need to add some flags for ncurses.
@@ -349,7 +368,7 @@ AC_DEFINE([NOMACROS], [1], [Define to 1 to disable curses macros that conflict w
 #
 
 # Check for os dependant libraries for all binaries.
-AC_SEARCH_LIBS( connect, socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
+AC_SEARCH_LIBS( connect, network socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
 AC_SEARCH_LIBS( nanosleep, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( shm_open, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( pthread_create, pthread, , [AC_MSG_ERROR([Cannot find the pthread library, needed to build this package.] )] )
diff --git a/env_universal_common.cpp b/env_universal_common.cpp
index b493de2..8e8046b 100644
--- a/env_universal_common.cpp
+++ b/env_universal_common.cpp
@@ -1102,6 +1102,10 @@ class universal_notifier_shmem_poller_t : public universal_notifier_t
 #define SHMEM_MAGIC_NUMBER 0xF154
 #define SHMEM_VERSION_CURRENT 1000
 
+#ifndef MAP_FILE
+#define MAP_FILE 0
+#endif
+
     private:
     long long last_change_time;
     uint32_t last_seed;
