diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2015-07-03 19:46:59.002359296 +0000
+++ b/configure.ac	2015-08-29 17:09:39.068157440 +0000
@@ -211,7 +211,13 @@
 #
 
 if test `uname` != "Darwin"; then
-   LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
+	case $target_os in
+		haiku*)
+			;;
+		*)
+			LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
+			;;
+	esac
 fi
 
 #
@@ -349,7 +355,7 @@
 #
 
 # Check for os dependant libraries for all binaries.
-AC_SEARCH_LIBS( connect, socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
+AC_SEARCH_LIBS( connect, network socket, , [AC_MSG_ERROR([Cannot find the socket library, needed to build this package.] )] )
 AC_SEARCH_LIBS( nanosleep, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( shm_open, rt, , [AC_MSG_ERROR([Cannot find the rt library, needed to build this package.] )] )
 AC_SEARCH_LIBS( pthread_create, pthread, , [AC_MSG_ERROR([Cannot find the pthread library, needed to build this package.] )] )

diff -ur a/env_universal_common.cpp b/env_universal_common.cpp
--- a/env_universal_common.cpp	2015-07-03 19:46:59.011534336 +0000
+++ b/env_universal_common.cpp	2015-08-29 16:48:54.951844864 +0000
@@ -36,6 +36,10 @@
 #include <notify.h>
 #endif
 
+#ifdef __HAIKU__
+#include <ifaddrs.h>
+#endif
+
 // NAME_MAX is not defined on Solaris and suggests the use of pathconf()
 // There is no obvious sensible pathconf() for shared memory and _XPG_NAME_MAX
 // seems a reasonable choice.
@@ -982,7 +986,7 @@
 /* Thanks to Jan Brittenson
  http://lists.apple.com/archives/xcode-users/2009/May/msg00062.html
  */
-#ifdef SIOCGIFHWADDR
+#if defined (SIOCGIFHWADDR) && ! defined (__HAIKU__)
 
 /* Linux */
 #include <net/if.h>
@@ -1005,7 +1009,7 @@
     return result;
 }
 
-#elif defined(HAVE_GETIFADDRS)
+#elif defined(HAVE_GETIFADDRS) && ! defined (__HAIKU__)
 
 /* OS X and BSD */
 #include <ifaddrs.h>
@@ -1102,6 +1106,10 @@
 #define SHMEM_MAGIC_NUMBER 0xF154
 #define SHMEM_VERSION_CURRENT 1000
 
+#ifndef MAP_FILE
+#define MAP_FILE 0
+#endif
+
     private:
     long long last_change_time;
     uint32_t last_seed;

diff -ur a/intern.cpp b/intern.cpp
--- a/intern.cpp	2015-07-03 19:46:59.016777216 +0000
+++ b/intern.cpp	2015-08-29 16:54:24.978321408 +0000
@@ -20,6 +20,10 @@
 #include "common.h"
 #include "intern.h"
 
+#ifdef __HAIKU__
+#include <posix/pthread.h>
+#endif
+
 /** Comparison function for intern'd strings */
 class string_table_compare_t
 {

diff -ur a/iothread.cpp b/iothread.cpp
--- a/iothread.cpp	2015-07-03 19:46:59.017039360 +0000
+++ b/iothread.cpp	2015-08-29 16:58:04.486801408 +0000
@@ -12,6 +12,12 @@
 #include <fcntl.h>
 #include <queue>
 
+#ifdef __HAIKU__
+#include <posix/pthread.h>
+#include <sys/types.h>
+#include <sys/select.h>
+#endif
+
 #ifdef _POSIX_THREAD_THREADS_MAX
 #if _POSIX_THREAD_THREADS_MAX < 64
 #define IO_MAX_THREADS _POSIX_THREAD_THREADS_MAX