diff --git a/miner.h b/miner.h
index 3f9b2f4..4a0b5be 100644
--- a/miner.h
+++ b/miner.h
@@ -83,6 +83,7 @@ static inline uint32_t swab32(uint32_t v)
 #endif
 
 #if !HAVE_DECL_BE32DEC
+#define be32dec be32dec_haiku
 static inline uint32_t be32dec(const void *pp)
 {
 	const uint8_t *p = (uint8_t const *)pp;
@@ -101,6 +102,7 @@ static inline uint32_t le32dec(const void *pp)
 #endif
 
 #if !HAVE_DECL_BE32ENC
+#define be32enc be32enc_haiku
 static inline void be32enc(void *pp, uint32_t x)
 {
 	uint8_t *p = (uint8_t *)pp;
