diff -up mercurial-1.5.2/mercurial/ui.py.orig mercurial-1.5.2/mercurial/ui.py
--- mercurial-1.5.2/mercurial/ui.py.orig	2010-05-01 15:02:48.041156608 -0600
+++ mercurial-1.5.2/mercurial/ui.py	2010-05-01 17:04:56.068943872 -0600
@@ -384,7 +384,7 @@ class ui(object):
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
-                os.environ.get("EDITOR", "vi"))
+                os.environ.get("EDITOR", "nano"))
 
     def progress(self, topic, pos, item="", unit="", total=None):
         '''show a progress message
