diff -ur convmv-1.15.org/Makefile convmv-1.15/Makefile
--- convmv-1.15.org/Makefile	2008-11-29 14:49:38.012845056 +0100
+++ convmv-1.15/Makefile	2012-04-21 02:00:10.777519104 +0200
@@ -6,6 +6,7 @@
 SED=sed
 
 all: manpage
+	sed -i "s#/usr/bin/perl#`which perl`#" convmv
 
 install: all
 	mkdir -p $(DESTDIR)$(MANDIR)/man1/
@@ -22,6 +23,7 @@
 
 test:
 	test -d suite || $(TAR) xf testsuite.tar
+	sed -i "s#/usr/bin/perl#`which perl`#" suite/parsable_tester.pl
 	cd suite ; ./dotests.sh
 
 dist: clean
