--- orc-0.4.16/configure.ac	2011-10-03 03:34:22.054788096 +0000
+++ orc-0.4.16-haiku/configure.ac	2012-04-08 21:58:47.729808896 +0000
@@ -97,7 +97,7 @@
   mingw*|pw32*|cygwin*)
     AC_DEFINE(HAVE_CODEMEM_VIRTUALALLOC, 1, [Use VirtualAlloc to allocate code for execution])
     ;;
-  linux*|darwin*|solaris*|netbsd*|freebsd*|openbsd*|kfreebsd*|dragonfly*|gnu*)
+  linux*|darwin*|solaris*|netbsd*|freebsd*|openbsd*|kfreebsd*|dragonfly*|gnu*|haiku*)
     AC_DEFINE(HAVE_CODEMEM_MMAP, 1, [Use mmap to allocate code for execution])
     ;;
   *)
