diff --git a/src/third_party/abseil-cpp/absl/base/config.h b/src/third_party/abseil-cpp/absl/base/config.h
index 9544996..f878679 100644
--- a/src/third_party/abseil-cpp/absl/base/config.h	2021-08-05 16:07:34.771751936 +0900
+++ b/src/third_party/abseil-cpp/absl/base/config.h	2021-08-05 14:02:43.693370880 +0900
@@ -375 +375 @@ static_assert(ABSL_INTERNAL_INLINE_NAMESPACE_STR[0] != 'h' ||
-    defined(__ASYLO__) || defined(__myriad2__)
+    defined(__ASYLO__) || defined(__myriad2__) || defined(__HAIKU__)
