diff -urN SDL_image-1.2.6/acinclude/libtool.m4 SDL_image-1.2.6-haiku/acinclude/libtool.m4
--- SDL_image-1.2.6/acinclude/libtool.m4	2007-07-13 03:38:28.000000000 +0000
+++ SDL_image-1.2.6-haiku/acinclude/libtool.m4	2008-06-30 09:50:10.000000000 +0000
@@ -734,6 +734,12 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  haiku*)
+    # On Haiku, this test takes a really really long time.
+    # Haiku has a maximum line length of 4096 for now.
+    lt_cv_sys_max_cmd_len=4096;
+    ;;
+
   amigaos*)
     # On AmigaOS with pdksh, this test takes hours, literally.
     # So we just punt and use a minimum line length of 8192.
@@ -1288,7 +1294,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([[^/]]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -2287,7 +2293,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -2505,7 +2511,7 @@
 [AC_REQUIRE([AC_CANONICAL_HOST])dnl
 LIBM=
 case $host in
-*-*-beos* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
+*-*-beos* | haiku* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
   # These system don't have libm, or don't need it
   ;;
 *-ncr-sysv4.3*)
@@ -2996,7 +3002,7 @@
     fi
     ;;
 
-  beos*)
+  beos* | haiku*)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
       _LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
       # Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -4850,7 +4856,7 @@
       # like `-m68040'.
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
     mingw* | os2* | pw32*)
@@ -5123,7 +5129,7 @@
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
@@ -5490,7 +5496,7 @@
       _LT_AC_TAGVAR(ld_shlibs, $1)=no
       ;;
 
-    beos*)
+    beos* | haiku*)
       if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	# Joseph Beckenbach <jrb3@best.com> says some releases of gcc
