--- libsdl-1.2-hg/src/audio/baudio/SDL_beaudio.cc   dim. nov. 04 19:07:49 2012 +0100
+++ b/src/audio/baudio/SDL_beaudio.cc   mar. janv. 01 22:41:22 2013 +0100
@@ -199,11 +199,11 @@
        return (-1);
    }

-    format.buffer_size = spec->samples;
-
	/* Calculate the final parameters for this audio specification */
	SDL_CalculateAudioSpec(spec);

+    format.buffer_size = spec->size;
+
	/* Subscribe to the audio stream (creates a new thread) */
	{ sigset_t omask;
		SDL_MaskSignals(&omask);
         return (-1);
     }
