diff -r 31b0f2e06e3c src/main/beos/SDL_BeApp.cc
--- a/src/main/beos/SDL_BeApp.cc	Sun Apr 25 20:16:38 2010 -0400
+++ b/src/main/beos/SDL_BeApp.cc	Sun Jun 06 07:56:39 2010 +0000
@@ -39,13 +39,15 @@
 
 static int StartBeApp(void *unused)
 {
+  if(!be_app) {
 	BApplication *App;
 
 	App = new BApplication("application/x-SDL-executable");
 
 	App->Run();
 	delete App;
-	return(0);
+  }
+ return(0);
 }
 
 /* Initialize the Be Application, if it's not already started */
@@ -86,7 +88,7 @@
 	++SDL_BeAppActive;
 
 	/* The app is running, and we're ready to go */
-	return(0);
+ return(0);
 }
 
 /* Quit the Be Application, if there's nothing left to do */
