diff -r 930614179450 libsdl-1.2-hg/src/video/bwindow/SDL_BWin.h
--- libsdl-1.2-hg/src/video/bwindow/SDL_BWin.h	Fri Dec 03 16:55:25 2010 -0800
+++ libsdl-1.2-hg-haiku/src/video/bwindow/SDL_BWin.h	Thu Dec 23 09:38:48 2010 +0000
@@ -113,7 +113,9 @@
 				SDL_GLView = new BGLView(Bounds(), "SDL GLView",
 					 	B_FOLLOW_ALL_SIDES, (B_WILL_DRAW|B_FRAME_EVENTS),
 					 	gl_flags);
-				SDL_GLView->EnableDirectMode(true);
+				//TODO: Figure out why directmode isn't working
+				//SDL_GLView->EnableDirectMode(true);
+				SDL_GLView->EnableDirectMode(false);
 			}
 			if ( the_view != SDL_GLView ) {
 				if ( the_view ) {
@@ -133,17 +135,17 @@
 			}
 			if ( the_view != SDL_View ) {
 				if ( the_view ) {
-#if SDL_VIDEO_OPENGL
-					if ( the_view == SDL_GLView ) {
-						SDL_GLView->UnlockGL();
-					}
-#endif
 					RemoveChild(the_view);
 				}
 				AddChild(SDL_View);
 				the_view = SDL_View;
 			}
 		}
+#if SDL_VIDEO_OPENGL
+		if ( the_view == SDL_GLView ) {
+			SDL_GLView->UnlockGL();
+		}
+#endif
 		Unlock();
 		return(retval);
 	}
 
