diff -r 0d4756e57224 src/video/bwindow/SDL_sysevents.cc
--- a/src/video/bwindow/SDL_sysevents.cc	Sun Sep 19 08:56:53 2010 -0700
+++ b/src/video/bwindow/SDL_sysevents.cc	Tue Nov 30 13:38:17 2010 +0000
@@ -386,9 +386,9 @@
 		case B_DIRECT_MODIFY:
 			{
 				int32 width = info->window_bounds.right -
-					info->window_bounds.left + 1;
+					info->window_bounds.left;
 				int32 height = info->window_bounds.bottom -
-					info->window_bounds.top + 1;
+					info->window_bounds.top;
 				SDL_PrivateResize(width, height);
 				break;
 			}
