diff -urN SDL_gfx-2.0.20/Test/configure.in SDL_gfx-2.0.20-haiku/Test/configure.in
--- SDL_gfx-2.0.20/Test/configure.in	2009-04-21 04:47:51.000000000 +0000
+++ SDL_gfx-2.0.20-haiku/Test/configure.in	2009-10-09 16:17:59.000000000 +0000
@@ -44,7 +44,7 @@
     *-*-cygwin* | *-*-mingw32*)
         MATHLIB=""
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         MATHLIB=""
         ;;
     *-*-darwin*)
diff -urN SDL_gfx-2.0.20/configure.in SDL_gfx-2.0.20-haiku/configure.in
--- SDL_gfx-2.0.20/configure.in	2009-09-24 02:07:16.000000000 +0000
+++ SDL_gfx-2.0.20-haiku/configure.in	2009-10-09 15:36:08.000000000 +0000
@@ -81,7 +81,7 @@
     *-*-cygwin* | *-*-mingw32*)
         MATHLIB=""
         ;;
-    *-*-beos*)
+    *-*-beos* | *-*-haiku*)
         MATHLIB=""
         ;;
     *-*-darwin*)
