diff -urN libdvbpsi-0.1.7/examples/decode_mpeg.c libdvbpsi-0.1.7-haiku/examples/decode_mpeg.c
--- libdvbpsi-0.1.7/examples/decode_mpeg.c	2010-02-12 10:17:47.066584576 +0000
+++ libdvbpsi-0.1.7-haiku/examples/decode_mpeg.c	2010-12-16 09:59:32.000000000 +0000
@@ -234,21 +234,21 @@
 #ifdef HAVE_GETTIMEOFDAY
 static mtime_t report_PCRPacketTiming( int i_cc, ts_pid_t *ts_pid,
                     mtime_t i_prev_pcr, mtime_t time_prev, int32_t i_bytes )
+{
+    mtime_t time_current;
+    mtime_t tv_delta;
+    struct timeval tv;
 #else
 static void report_PCRPacketTiming( int i_cc, ts_pid_t *ts_pid,
                                     mtime_t i_prev_pcr, int32_t i_bytes )
-#endif
 {
+#endif
     mtime_t i_delta = 0;
 
     /* sequence number and program_id */
     printf( "%.2d %d ", i_cc, ts_pid->i_pid );
 
 #ifdef HAVE_GETTIMEOFDAY
-    mtime_t time_current;
-    mtime_t tv_delta;
-    struct timeval tv;
-
     /* arrival time of packet */
     gettimeofday( &tv, NULL );
     time_current = ((mtime_t)tv.tv_sec*1000) + ((mtime_t)tv.tv_usec/1000);
