diff --git a/Makefile b/Makefile
index 1c01e5c..b3fe4b5 100644
--- a/Makefile
+++ b/Makefile
@@ -9,13 +9,13 @@ PKG_CFLAGS:=$(shell $(PKG_CONFIG) --cflags $(LIBS) || echo "FAILED")
 ifeq ($(PKG_CFLAGS),FAILED)
        $(error "$(PKG_CONFIG) failed")
 endif
-CFLAGS+=$(PKG_CFLAGS) -fPIC -DPIC
+CFLAGS+=$(PKG_CFLAGS) -fPIC -DPIC -Ivendor/http-parser
 
 PKG_LDLIBS:=$(shell $(PKG_CONFIG) --libs $(LIBS) || echo "FAILED")
 ifeq ($(PKG_LDLIBS),FAILED)
        $(error "$(PKG_CONFIG) failed")
 endif
-LDLIBS+=$(PKG_LDLIBS) -lhttp_parser
+LDLIBS+=$(PKG_LDLIBS) vendor/http-parser/http_parser.o
 
 ifndef MATRIX_NO_E2E
 LDLIBS+=-lolm -lgcrypt
