diff -urN cairo-1.10.0/configure.ac cairo-1.10.0-haiku/configure.ac
--- cairo-1.10.0/configure.ac	2010-02-19 18:38:03.064487424 +0000
+++ cairo-1.10.0-haiku/configure.ac	2010-06-19 08:32:47.591134720 +0000
@@ -148,6 +148,11 @@
       AC_CHECK_LIB(be,main,beos_LIBS="$beos_LIBS -lbe")
       AC_CHECK_LIB(zeta,main,beos_LIBS="$beos_LIBS -lzeta")
       ;;
+    *-*-haiku)
+      beos_LIBS=""
+      AC_CHECK_LIB(be,main,beos_LIBS="$beos_LIBS -lbe")
+      AC_CHECK_LIB(network,main,beos_LIBS="$beos_LIBS -lnetwork")
+      ;;
     *)
       use_beos="no (requires a BeOS platform)"
       ;;
