--- curl-7.22.0/m4/curl-openssl.m4	2011-06-13 21:09:52.046137344 +0000
+++ curl-7.22.0-haiku/m4/curl-openssl.m4	2011-09-14 22:16:54.307232768 +0000
@@ -61,6 +61,12 @@
         tst_verfix=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 7`
         tst_api=0x$tst_vermaj$tst_vermin$tst_verfix
         ;;
+      x12)
+        tst_vermaj=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 3`
+        tst_vermin=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 5`
+        tst_verfix=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 8`
+        tst_api=0x$tst_vermaj$tst_vermin$tst_verfix
+        ;;        
       *)
         tst_api="unknown"
         ;;
