diff -urN curl-7.25.0/m4/curl-openssl.m4 curl-7.25.0-haiku/m4/curl-openssl.m4
--- curl-7.25.0/m4/curl-openssl.m4	2012-03-08 11:35:25.043515904 -0800
+++ curl-7.25.0-haiku/m4/curl-openssl.m4	2012-04-01 21:44:11.908066816 -0700
@@ -55,7 +55,7 @@
         tst_verfix=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 5`
         tst_api=0x$tst_vermaj$tst_vermin$tst_verfix
         ;;
-      x11|x10)
+      x12|x11|x10)
         tst_vermaj=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 3`
         tst_vermin=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 5`
         tst_verfix=`echo $curl_cv_def_OPENSSL_VERSION_NUMBER | cut -c 7`
