diff -urN pjdfstest-20100807/Makefile pjdfstest-20100807-haiku/Makefile
--- pjdfstest-20100807/Makefile	2010-08-06 17:38:26.008912896 +0000
+++ pjdfstest-20100807-haiku/Makefile	2010-11-05 16:31:34.000524288 +0000
@@ -17,6 +17,10 @@
 CFLAGS+= -D__OS_Darwin__
 CFLAGS+= -DHAS_LCHMOD -DHAS_CHFLAGS -DHAS_LCHFLAGS
 
+.elif ${OSTYPE} == "Haiku"
+CFLAGS+= -D__OS_Haiku__
+CFLAGS+= -DHAS_LCHMOD -DHAS_CHFLAGS -DHAS_LCHFLAGS
+
 .elif ${OSTYPE} == "Linux"
 CFLAGS+= -D__OS_Linux__
 CFLAGS+= -D_GNU_SOURCE
diff -urN pjdfstest-20100807/tests/conf pjdfstest-20100807-haiku/tests/conf
--- pjdfstest-20100807/tests/conf	2010-08-06 21:53:43.009961472 +0000
+++ pjdfstest-20100807-haiku/tests/conf	2010-11-05 16:44:11.000000000 +0000
@@ -1,7 +1,7 @@
 # $FreeBSD: head/tools/regression/fstest/tests/conf 210970 2010-08-06 21:53:50Z pjd $
 # fstest configuration file
 
-# Known operating systems: FreeBSD, SunOS, Linux
+# Known operating systems: FreeBSD, SunOS, Linux, Haiku
 os=`uname`
 
 case "${os}" in
@@ -17,6 +17,10 @@
 	fs=`mount -v | egrep "^${pattern}" | awk '{print $5}' | \
 	    tr -s '[:lower:]' '[:upper:]'`
 	;;
+Haiku)
+	GREP=grep
+	fs=`df -b | tail -1 | awk '{print toupper($2)}'`
+	;;	
 Linux)
 	GREP=grep
 	fs=`df -PT . | tail -1 | awk '{print toupper($2)}'`
