diff -up colordiff-1.0.13/colordiff.pl.orig colordiff-1.0.13/colordiff.pl
--- colordiff-1.0.13/colordiff.pl.orig	2013-10-15 17:51:27.583270400 -0600
+++ colordiff-1.0.13/colordiff.pl	2013-10-15 17:58:48.368836608 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!perl -w
 
 ########################################################################
 #                                                                      #
@@ -67,7 +67,7 @@ my $HOME   = $ENV{HOME};
 my $etcdir = '/etc';
 my ($setting, $value);
 my @config_files = ("$etcdir/colordiffrc");
-push (@config_files, "$ENV{HOME}/.colordiffrc") if (defined $ENV{HOME});
+push (@config_files, "$ENV{HOME}/config/settings/colordiffrc") if (defined $ENV{HOME});
 my $config_file;
 my $diff_type = 'unknown';
 
