diff -up colordiff-1.0.10/colordiff.pl.orig colordiff-1.0.10/colordiff.pl
--- colordiff-1.0.10/colordiff.pl.orig	2012-08-05 14:43:30.500432896 -0600
+++ colordiff-1.0.10/colordiff.pl	2012-08-05 14:43:41.898367488 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!/boot/common/bin/perl -w
 
 ########################################################################
 #                                                                      #
