diff -Naur tree-1.6.0/Makefile tree-1.6.0-haiku/Makefile
--- tree-1.6.0/Makefile	2011-06-24 16:25:27.060030976 +0200
+++ tree-1.6.0-haiku/Makefile	2011-12-18 01:40:48.786169856 +0100
@@ -31,8 +31,8 @@
 
 # Linux defaults:
 #CFLAGS=-ggdb -Wall -DLINUX -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS=-O4 -Wall  -DLINUX -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-LDFLAGS=-s
+#CFLAGS=-O4 -Wall  -DLINUX -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+#LDFLAGS=-s
 
 # Uncomment for FreeBSD:
 #CFLAGS=-O2 -Wall -fomit-frame-pointer
@@ -58,6 +58,13 @@
 #LDFLAGS=
 #OBJS+=strverscmp.o
 
+# Uncomment for Haiku:
+prefix=$(shell finddir B_COMMON_DIRECTORY)
+MANDIR=$(shell finddir B_COMMON_DOCUMENTATION_DIRECTORY)/man/man1
+CFLAGS=-O2 -Wall -fomit-frame-pointer
+LDFLAGS=
+OBJS+=strverscmp.o
+
 # Uncomment for HP/UX:
 #CC=cc
 #CFLAGS=-O2 -DAportable -Wall
