--- alex4_BeOS_src/Makefile.orig	2003-08-07 22:25:19.000000000 +0000
+++ alex4_BeOS_src/Makefile
@@ -1,11 +1,11 @@
 CC=g++
 CFLAGS=-c -O1 -c `allegro-config --cflags`
-LDFLAGS=`allegro-config --libs` -ldumb
+LDFLAGS=`allegro-config --libs` -ldumb -laldmb
 SOURCES=actor.c edit.c map.c player.c shooter.c bullet.c hisc.c options.c script.c timer.c control.c main.c particle.c scroller.c token.c
 OBJECTS :=$(SOURCES:%.c=%.o)
 all:  clean compile
 	strip --strip-all alex4
-	xres -o alex4 alex4.rsrc
+	xres -o alex4 ../alex4.rsrc
 	mimeset -f alex4
 	mv alex4 ../alex4
 
