diff -up git-1.7.1/configure.ac.orig git-1.7.1/configure.ac
--- git-1.7.1/configure.ac.orig	2010-04-23 20:38:35.035127296 -0600
+++ git-1.7.1/configure.ac	2010-05-13 18:03:30.180355072 -0600
@@ -547,17 +547,13 @@ test -n "$NEEDS_SOCKET" && LIBS="$LIBS -
 # Define NEEDS_RESOLV if linking with -lnsl and/or -lsocket is not enough.
 # Notably on Solaris hstrerror resides in libresolv and on Solaris 7
 # inet_ntop and inet_pton additionally reside there.
-AC_CHECK_LIB([c], [hstrerror],
-[NEEDS_RESOLV=],
-[NEEDS_RESOLV=YesPlease])
+AC_SEARCH_LIBS([hstrerror], [resolv],
+[AC_DEFINE([NEEDS_RESOLV], [YesPlease])])
 AC_SUBST(NEEDS_RESOLV)
-test -n "$NEEDS_RESOLV" && LIBS="$LIBS -lresolv"
 
-AC_CHECK_LIB([c], [basename],
-[NEEDS_LIBGEN=],
-[NEEDS_LIBGEN=YesPlease])
+AC_SEARCH_LIBS([basename], [gen],
+[AC_DEFINE([NEEDS_LIBGEN], [YesPlease])])
 AC_SUBST(NEEDS_LIBGEN)
-test -n "$NEEDS_LIBGEN" && LIBS="$LIBS -lgen"
 
 ## Checks for header files.
 AC_MSG_NOTICE([CHECKS for header files])
