diff -Naur libvpx-0.9.2-git/build/make/configure.sh libvpx-0.9.2-git-haiku/build/make/configure.sh
--- libvpx-0.9.2-git/build/make/configure.sh	2010-10-01 11:22:21.319815680 +0000
+++ libvpx-0.9.2-git-haiku/build/make/configure.sh	2010-10-01 11:30:22.888406016 +0000
@@ -905,8 +908,14 @@
     # Almost every platform uses pthreads.
     if enabled multithread; then
         case ${toolchain} in
-            *-win*);;
-            *) check_header pthread.h && add_extralibs -lpthread
+            *-win*)
+            ;;
+            *linux*|*bsd*)
+            check_header pthread.h && add_extralibs -lpthread
+            ;;
+            *-haiku*)
+        	check_header pthread.h
+        	;;
         esac
     fi
 
diff -Naur libvpx-0.9.2-git/examples.mk libvpx-0.9.2-git-haiku/examples.mk
--- libvpx-0.9.2-git/examples.mk	2010-10-01 11:22:21.332398592 +0000
+++ libvpx-0.9.2-git-haiku/examples.mk	2010-10-01 11:19:40.000000000 +0000
@@ -72,7 +72,9 @@
 
 
 # Handle extra library flags depending on codec configuration
+ifeq ($(TOOLCHAIN),Unix)
 CODEC_EXTRA_LIBS-$(CONFIG_VP8)         += m
+endif
 
 #
 # End of specified files. The rest of the build rules should happen
diff -Naur libvpx-0.9.2-git/libs.mk libvpx-0.9.2-git-haiku/libs.mk
--- libvpx-0.9.2-git/libs.mk	2010-10-01 11:22:21.525074432 +0000
+++ libvpx-0.9.2-git-haiku/libs.mk	2010-10-01 11:13:12.000000000 +0000
@@ -182,6 +182,9 @@
 LIBS-$(BUILD_LIBVPX_SO) += $(BUILD_PFX)$(LIBVPX_SO)
 $(BUILD_PFX)$(LIBVPX_SO): $(LIBVPX_OBJS) libvpx.ver
 $(BUILD_PFX)$(LIBVPX_SO): extralibs += -lm -pthread
+ifeq ($(TOOLCHAIN),Haiku)
+	extralibs +=
+endif
 $(BUILD_PFX)$(LIBVPX_SO): SONAME = libvpx.so.$(VERSION_MAJOR)
 $(BUILD_PFX)$(LIBVPX_SO): SO_VERSION_SCRIPT = libvpx.ver
 LIBVPX_SO_SYMLINKS      := $(addprefix $(LIBSUBDIR)/, \
