diff -urN libtheora-1.0/configure.ac libtheora-1.0-haiku/configure.ac
--- libtheora-1.0/configure.ac	2008-10-24 04:07:32.000000000 +0000
+++ libtheora-1.0-haiku/configure.ac	2009-09-14 02:21:57.000000000 +0000
@@ -404,7 +404,7 @@
 if test x$HAVE_SDL = xyes -a x$HAVE_OSS = xyes -a x$HAVE_VORBIS = xyes; then
   BUILDABLE_EXAMPLES="$BUILDABLE_EXAMPLES player_example"
 fi
-if test x$HAVE_PNG = xyes; then
+if test x$HAVE_PNG = xyes -a $target_os != 'haiku'; then
   BUILDABLE_EXAMPLES="$BUILDABLE_EXAMPLES png2theora"
 fi
 AC_SUBST(BUILDABLE_EXAMPLES)
