--- libgcrypt-1.5.3/configure.ac	2011-06-29 08:53:22.065011712 +0000
+++ libgcrypt-1.5.3-haiku/configure.ac	2012-04-09 17:24:30.365690880 +0000
@@ -619,6 +619,7 @@
 AC_SEARCH_LIBS(setsockopt, [socket], ,
 	[AC_SEARCH_LIBS(setsockopt, [socket], , , [-lnsl])])
 AC_SEARCH_LIBS(setsockopt, [nsl])
+AC_SEARCH_LIBS(socket, [network])
 
 ##################################
 #### Checks for header files. ####
