diff -Naur libgcrypt-1.4.4-orig/configure.ac libgcrypt-1.4.4/configure.ac
--- libgcrypt-1.4.4-orig/configure.ac	2009-03-08 17:29:52.000000000 +0000
+++ libgcrypt-1.4.4/configure.ac	2009-03-08 17:59:06.000000000 +0000
@@ -564,6 +564,8 @@
 AC_SEARCH_LIBS(setsockopt, [socket], ,
 	[AC_SEARCH_LIBS(setsockopt, [socket], , , [-lnsl])])
 AC_SEARCH_LIBS(setsockopt, [nsl])
+# Haiku has everything in -lnetwork
+AC_SEARCH_LIBS(socket, [network])
 
 ##################################
 #### Checks for header files. ####
diff -Naur libgcrypt-1.4.4-orig/src/ath.c libgcrypt-1.4.4/src/ath.c
--- libgcrypt-1.4.4-orig/src/ath.c	2009-03-08 17:29:52.000000000 +0000
+++ libgcrypt-1.4.4/src/ath.c	2009-03-08 17:59:06.000000000 +0000
@@ -24,11 +24,6 @@
 
 #include <assert.h>  /* Right: We need to use assert and not gcry_assert.  */
 #include <unistd.h>
-#ifdef HAVE_SYS_SELECT_H
-# include <sys/select.h>
-#else
-# include <sys/time.h>
-#endif
 #include <sys/types.h>
 #ifndef _WIN32
 #include <sys/wait.h>
diff -Naur libgcrypt-1.4.4-orig/src/ath.h libgcrypt-1.4.4/src/ath.h
--- libgcrypt-1.4.4-orig/src/ath.h	2009-03-08 17:29:52.000000000 +0000
+++ libgcrypt-1.4.4/src/ath.h	2009-03-08 17:59:06.000000000 +0000
@@ -25,6 +25,11 @@
 #include <windows.h>
 #else
 #include <sys/types.h>
+#ifdef HAVE_SYS_SELECT_H
+# include <sys/select.h>
+#else
+# include <sys/time.h>
+#endif
 #include <sys/socket.h>
 #endif
 #include <gpg-error.h>
