diff -Naur libgcrypt-1.4.5/configure.ac libgcrypt-1.4.5-haiku/configure.ac
--- libgcrypt-1.4.5/configure.ac	2009-12-11 15:43:15.025165824 +0000
+++ libgcrypt-1.4.5-haiku/configure.ac	2010-04-27 00:48:26.807141376 +0000
@@ -574,12 +574,16 @@
 	[AC_SEARCH_LIBS(setsockopt, [socket], , , [-lnsl])])
 AC_SEARCH_LIBS(setsockopt, [nsl])
 
+# Haiku has everything in -lnetwork
+AC_SEARCH_LIBS(socket, [network]) 
+
 ##################################
 #### Checks for header files. ####
 ##################################
 
 AC_HEADER_STDC
 AC_CHECK_HEADERS(unistd.h sys/select.h)
+AC_CHECK_HEADERS(sys/select.h sys/time.h)
 
 ##########################################
 #### Checks for typedefs, structures, ####
