diff --git a/configure.in b/configure.in
index 9cd3a85..0138f84 100644
--- a/configure.in
+++ b/configure.in
@@ -20,21 +20,6 @@ esac
 AC_SUBST(CPU)
 
 dnl -------------------------------------------------------------------
-dnl Check for BeOS R4
-dnl -------------------------------------------------------------------
-AC_MSG_CHECKING( [BeOS Release] )
-release=[$(uname -r | sed 's/\([^\.]*\)\..*/\1/')]
-case "$release" in
-	5|4)
-		AC_MSG_RESULT( [post-R4] )
-		;;
-	*)
-		AC_MSG_RESULT( [not R4] )
-		AC_MSG_ERROR( [BeOS Release 4 is required] )
-		;;
-esac
-
-dnl -------------------------------------------------------------------
 dnl Check for libprefs source distribution
 dnl -------------------------------------------------------------------
 AC_ARG_WITH( libprefs, [ --with-libprefs=PATH	path to the libprefs distribution ] )
