diff --git a/src/Makefile.in b/src/Makefile.in
index 15486d7..c8c4802 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -25,7 +25,7 @@ RSRCS = LaunchPad_ppc.rsrc
 
 OBJS = $(addsuffix .o, $(foreach file, $(SRCS), $(basename $(notdir $(file)))))
 
-LIBS = $(LIBPREFSDIR)/bigprefs.$(CPU).a -lbe
+LIBS =  $(LIBPREFSDIR)/libprefs.so -lbe # Was first $(LIBPREFSDIR)/bigprefs.$(CPU).a
 
 all:	LaunchPad
 
