--- pkg-config-0.25/parse.c	2010-05-27 20:21:36.003670016 +0000
+++ pkg-config-0.25-haiku/parse.c 2010-08-05 02:44:44.243269632 +0000
@@ -880,8 +880,9 @@
             pkg->other_cflags = g_slist_prepend (pkg->other_cflags,
                                                  g_strdup (arg));
 	  if (strcmp("-idirafter", arg) == 0) {
-              tmp = trim_string(argv[++i]);
-	      char *n = strdup_escape_shell(tmp);
+          char *n;
+          tmp = trim_string(argv[++i]);
+	      n = strdup_escape_shell(tmp);
 	      pkg->other_cflags = g_slist_prepend(pkg->other_cflags, n);
 	      g_free(tmp);
 	  }
diff -urN pkg-config-0.25/pkg.m4 pkg-config-0.25-dnlfixed/pkg.m4
--- pkg-config-0.25/pkg.m4	2009-12-06 20:39:02.000524288 +0000
+++ pkg-config-0.25-dnlfixed/pkg.m4	2010-09-07 20:24:30.345505792 +0000
@@ -135,7 +135,7 @@
 Consider adjusting the PKG_CONFIG_PATH environment variable if you
 installed software in a non-standard prefix.
 
-_PKG_TEXT])dnl
+_PKG_TEXT])[]dnl
         ])
 elif test $pkg_failed = untried; then
      	AC_MSG_RESULT([no])
