--- pkg-config-0.25/parse.c	2010-05-27 20:21:36.003670016 +0000
+++ pkg-config-0.25-haiku/parse.c 2010-08-05 02:44:44.243269632 +0000
@@ -880,8 +880,9 @@
             pkg->other_cflags = g_slist_prepend (pkg->other_cflags,
                                                  g_strdup (arg));
 	  if (strcmp("-idirafter", arg) == 0) {
-              tmp = trim_string(argv[++i]);
-	      char *n = strdup_escape_shell(tmp);
+          char *n;
+          tmp = trim_string(argv[++i]);
+	      n = strdup_escape_shell(tmp);
 	      pkg->other_cflags = g_slist_prepend(pkg->other_cflags, n);
 	      g_free(tmp);
 	  }
