diff -urN libcdio-0.83/configure.ac libcdio-0.83-haiku/configure.ac
--- libcdio-0.83/configure.ac	2009-10-27 11:26:32.055574528 +0000
+++ libcdio-0.83-haiku/configure.ac	2010-10-05 11:54:28.000000000 +0000
@@ -246,7 +246,7 @@
 dnl empty_array_size
 AC_MSG_CHECKING([how to create empty arrays])
 
-empty_array_size="xxxx"
+empty_array_size="xxx"
 AC_TRY_COMPILE([],[struct { int foo; int bar[]; } doo;], empty_array_size="")
 
 if test "x$empty_array_size" = "xxxx";then
@@ -572,6 +572,7 @@
     [AC_MSG_WARN([new enough libcddb not found. CDDB access disabled. Get libcddb from http://libcddb.sourceforge.net])
     HAVE_CDDB=no])
     AC_CHECK_LIB(socket, connect)
+    AC_CHECK_LIB(network, connect)
     AC_CHECK_FUNC(gethostbyname, , AC_CHECK_LIB(nsl, gethostbyname))
 fi
 
