diff -urN cdrdao-1.2.3/configure.ac cdrdao-1.2.3-haiku/configure.ac
--- cdrdao-1.2.3/configure.ac	2009-09-02 12:29:35.004194304 +0000
+++ cdrdao-1.2.3-haiku/configure.ac	2010-09-15 14:44:31.406061056 +0000
@@ -8,7 +8,7 @@
 AM_CONFIG_HEADER(config.h)
 AM_MAINTAINER_MODE
 AC_CANONICAL_HOST
-AM_GCONF_SOURCE_2
+dnlAM_GCONF_SOURCE_2
 
 AC_ARG_WITH(pcctsbin,[  --with-pcctsbin=dir     set directory of PCCTS parser generator executables],[pcctsbin=$withval],[pcctsbin=default])
 
@@ -86,12 +86,17 @@
   AC_CHECK_LIB(posix4,sched_setscheduler,[use_libposix4=yes;AC_DEFINE(HAVE_SCHED_SETSCHEDULER,1,"")]))
 
 AC_CHECK_LIB(socket,socket,[use_libsocket=yes])
+AC_CHECK_LIB(network,socket,[use_libnetwork=yes])
 AC_CHECK_LIB(socket,connect,[use_libsocket=yes])
 
 dnl Add 'libsocket' if required
 if test "$use_libsocket" = yes; then
   LIBS="$LIBS -lsocket"
 fi
+if test "$use_libnetwork" = yes; then
+  LIBS="$LIBS -lnetwork"
+fi
+
 
 dnl Check if libposix4 must be linked
 if test "$use_libposix4" = yes; then
@@ -282,6 +287,11 @@
                 scsilib_objs="ScsiIf-win.o"
                 ;;
 
+	*-haiku*)
+				scsilib_libs="$scsilib_libs -lscg -lschily"
+				scsilib_objs="ScsiIf-lib.o"
+				;;
+
     *)
        AC_MSG_ERROR([No native interface for operating system $host_os, use --with-libscg]) 
        ;;
