diff -Naur MediaInfo_CLI_GNU_FromSource/MediaInfoLib/Project/GNU/Library/libmediainfo.pc.in MediaInfo_CLI_GNU_FromSource-haiku/MediaInfoLib/Project/GNU/Library/libmediainfo.pc.in
--- MediaInfo_CLI_GNU_FromSource/MediaInfoLib/Project/GNU/Library/libmediainfo.pc.in	2012-10-21 21:32:39.055050240 +0000
+++ MediaInfo_CLI_GNU_FromSource-haiku/MediaInfoLib/Project/GNU/Library/libmediainfo.pc.in	2012-12-03 20:00:08.240123904 +0000
@@ -3,11 +3,11 @@
 libdir=@libdir@
 includedir=@includedir@
 Unicode=@MediaInfoLib_Unicode@
-Libs_Static=@libdir@/lib@MediaInfoLib_LibName@.a ${libdir}/libzen.a -lpthread -lz
+Libs_Static=@libdir@/lib@MediaInfoLib_LibName@.a ${libdir}/libzen.a  -lz
 la_name=lib@MediaInfoLib_LibName@.la
 
 Name: libmediainfo
 Version: @PACKAGE_VERSION@
 Description: MediaInfoLib
-Libs: -L@libdir@ -l@MediaInfoLib_LibName@ -lzen -lz -lpthread
+Libs: -L@libdir@ -l@MediaInfoLib_LibName@ -lzen -lz 
 Cflags: -I@includedir@ @MediaInfoLib_CXXFLAGS@
diff -Naur MediaInfo_CLI_GNU_FromSource/ZenLib/Project/GNU/Library/configure MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Project/GNU/Library/configure
--- MediaInfo_CLI_GNU_FromSource/ZenLib/Project/GNU/Library/configure	2012-10-21 21:32:46.003670016 +0000
+++ MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Project/GNU/Library/configure	2012-12-03 20:00:08.639893504 +0000
@@ -19110,8 +19110,8 @@
 fi
 
 
-ZenLib_LIBS="-L$(pwd) -lzen -lpthread -lstdc++ $ZenLib_LIBS"
-ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a -lpthread -lstdc++ $ZenLib_LIBS_Static"
+ZenLib_LIBS="-L$(pwd) -lzen  -lstdc++ $ZenLib_LIBS"
+ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a  -lstdc++ $ZenLib_LIBS_Static"
 
 
 if test "$enable_wstring_missing" = "yes"; then
@@ -19288,7 +19288,7 @@
 	fi
 fi
 
-LDFLAGS="$LDFLAGS -lpthread -lstdc++"
+LDFLAGS="$LDFLAGS  -lstdc++"
 
 
 
diff -Naur MediaInfo_CLI_GNU_FromSource/ZenLib/Project/GNU/Library/configure.ac MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Project/GNU/Library/configure.ac
--- MediaInfo_CLI_GNU_FromSource/ZenLib/Project/GNU/Library/configure.ac	2012-10-21 21:32:46.002359296 +0000
+++ MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Project/GNU/Library/configure.ac	2012-12-03 20:00:08.646184960 +0000
@@ -122,8 +122,8 @@
 dnl -------------------------------------------------------------------------
 dnl Common
 dnl
-ZenLib_LIBS="-L$(pwd) -lzen -lpthread -lstdc++ $ZenLib_LIBS"
-ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a -lpthread -lstdc++ $ZenLib_LIBS_Static"
+ZenLib_LIBS="-L$(pwd) -lzen  -lstdc++ $ZenLib_LIBS"
+ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a  -lstdc++ $ZenLib_LIBS_Static"
 
 dnl #########################################################################
 dnl ### C/C++ compiler options used to compile
@@ -221,7 +221,7 @@
 dnl -------------------------------------------------------------------------
 dnl External libs
 dnl
-LDFLAGS="$LDFLAGS -lpthread -lstdc++"
+LDFLAGS="$LDFLAGS  -lstdc++"
 
 dnl #########################################################################
 dnl ### Output
diff -Naur MediaInfo_CLI_GNU_FromSource/ZenLib/Project/GNU/Library/libzen.pc.in MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Project/GNU/Library/libzen.pc.in
--- MediaInfo_CLI_GNU_FromSource/ZenLib/Project/GNU/Library/libzen.pc.in	2012-10-21 21:32:46.003932160 +0000
+++ MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Project/GNU/Library/libzen.pc.in	2012-12-03 20:00:08.653262848 +0000
@@ -4,10 +4,10 @@
 includedir=@includedir@
 Unicode=@ZenLib_Unicode@
 WstringMissing=@ZenLib_wstring_missing@
-Libs_Static=@libdir@/libzen.a -lpthread
+Libs_Static=@libdir@/libzen.a 
 
 Name: libzen
 Version: @PACKAGE_VERSION@
 Description: ZenLib
-Libs: -L@libdir@ -lzen -lpthread -lstdc++
+Libs: -L@libdir@ -lzen  -lstdc++
 Cflags: -I${includedir} @ZenLib_CXXFLAGS@
diff -Naur MediaInfo_CLI_GNU_FromSource/ZenLib/Source/ZenLib/Conf.h MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Source/ZenLib/Conf.h
--- MediaInfo_CLI_GNU_FromSource/ZenLib/Source/ZenLib/Conf.h	2012-10-21 21:32:46.000262144 +0000
+++ MediaInfo_CLI_GNU_FromSource-haiku/ZenLib/Source/ZenLib/Conf.h	2012-12-03 20:29:36.678428672 +0000
@@ -80,7 +80,7 @@
     || defined(__unix) || defined(__unix__) \
     || defined(____SVR4____) || defined(__LINUX__) || defined(__sgi) \
     || defined(__hpux) || defined(sun) || defined(__SUN__) || defined(_AIX) \
-    || defined(__EMX__) || defined(__VMS) || defined(__BEOS__)
+    || defined(__EMX__) || defined(__VMS) || defined(__BEOS__) || defined(__HAIKU__)
     #ifndef UNIX
         #define UNIX
     #endif
