diff -Naur ne-2.1/makefile ne-2.1-haiku/makefile
--- ne-2.1/makefile	2010-05-10 09:42:12.052953088 -0700
+++ ne-2.1-haiku/makefile	2011-07-29 16:03:27.170131456 -0700
@@ -42,6 +42,19 @@
 	make install PREFIX=/usr CMDSUFFIX=.exe
 	tar zcvf ne-cygwin-$(VERSION).tar.gz /usr/share/ne /usr/bin/ne.exe /usr/share/doc/ne /usr/share/info/ne.info.gz /usr/share/man/man1/ne.1
 
+haiku:
+	(cd src; make NE_GLOBAL_DIR=$(shell finddir B_COMMON_DATA_DIRECTORY)/ne)
+	mkdir -p $(DESTDIR)$(shell finddir B_COMMON_BIN_DIRECTORY)
+	mkdir -p $(DESTDIR)$(shell finddir B_COMMON_DATA_DIRECTORY)/ne/syntax
+	mkdir -p $(DESTDIR)$(shell finddir B_COMMON_DOCUMENTATION_DIRECTORY)/man/man1
+	mkdir -p $(DESTDIR)$(shell finddir B_COMMON_DATA_DIRECTORY)/doc/ne
+	mkdir -p $(DESTDIR)$(shell finddir B_COMMON_DATA_DIRECTORY)/info
+	cp -pf src/ne$(CMDSUFFIX) $(DESTDIR)$(shell finddir B_COMMON_BIN_DIRECTORY)
+	cp -p syntax/*.jsf $(DESTDIR)$(shell finddir B_COMMON_DATA_DIRECTORY)/ne/syntax
+	cp -p doc/ne.1 $(DESTDIR)$(shell finddir B_COMMON_DOCUMENTATION_DIRECTORY)/man/man1
+	cp -pr doc/ne.pdf doc/html doc/ne.txt doc/default.* README COPYING NEWS CHANGES $(DESTDIR)$(shell finddir B_COMMON_DATA_DIRECTORY)/doc/ne
+	cp -p doc/ne.info.gz $(DESTDIR)$(shell finddir B_COMMON_DATA_DIRECTORY)/info
+
 install:
 	(cd src; make NE_GLOBAL_DIR=$(PREFIX)/share/ne)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
diff -Naur ne-2.1/src/makefile ne-2.1-haiku/src/makefile
--- ne-2.1/src/makefile	2010-04-28 06:47:06.005767168 +0000
+++ ne-2.1-haiku/src/makefile	2011-04-26 12:54:53.738721792 +0000
@@ -101,10 +101,10 @@
 	$(if $(NE_TERMCAP), -DTERMCAP,) \
 	$(if $(NE_ANSI),    -DTERMCAP -DANSI,)
 
-LIBS=$(if $(NE_TERMCAP)$(NE_ANSI),,-lcurses)
+LIBS=$(if $(NE_TERMCAP)$(NE_ANSI),,-lncurses)
 
 ne:	$(OBJS) $(if $(NE_TERMCAP)$(NE_ANSI),$(TERMCAPOBJS),)
-	$(CC) -lm $(OPTS) $(LDFLAGS) $^ $(LIBS) -o $(PROGRAM)
+	$(CC) $(OPTS) $(LDFLAGS) $^ $(LIBS) -o $(PROGRAM)
 
 clean:
 	rm -f *.o core
