From 829e5a683e69191a1af9a38102339b9ab972794f Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sun, 6 Mar 2016 14:57:13 -0500
Subject: [PATCH] main: Don't use the custom signal handler.

From d78f45afa5d352032213e32bda3cb3a7010c9d8e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
Date: Sat, 13 Apr 2019 18:52:51 +0200
Subject: 2


diff --git a/src/drivers/main.cpp b/src/drivers/main.cpp
index ff7d59a..1f999ab 100644
--- a/src/drivers/main.cpp
+++ b/src/drivers/main.cpp
@@ -481,7 +481,7 @@ static void CreateDirs(void)
 }
 
 static bool volatile SignalSafeExitWanted = false;
-#if defined(HAVE_SIGNAL) || defined(HAVE_SIGACTION)
+#if 0
 static const char *SiginfoString = NULL;
 typedef struct
 {
-- 
2.21.0

