diff -Naur mednafen/mednafen/tremor/misc.h mednafen-haiku/mednafen/tremor/misc.h
--- mednafen/mednafen/tremor/misc.h	2012-03-29 20:58:54.021757952 +0000
+++ mednafen-haiku/mednafen/tremor/misc.h	2013-01-05 18:21:51.628883456 +0000
@@ -15,6 +15,9 @@
 
  ********************************************************************/
 
+//Haiku
+#include <endian.h>
+
 #ifndef _V_RANDOM_H_
 #define _V_RANDOM_H_
 #include "ivorbiscodec.h"
