From 829e5a683e69191a1af9a38102339b9ab972794f Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sun, 6 Mar 2016 14:57:13 -0500
Subject: [PATCH] main: Don't use the custom signal handler.

---
 src/drivers/main.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/drivers/main.cpp b/src/drivers/main.cpp
index 36b5408..dcfb48d 100644
--- a/src/drivers/main.cpp
+++ b/src/drivers/main.cpp
@@ -286,7 +286,7 @@ static void CreateDirs(void)
  }
 }
 
-#if defined(HAVE_SIGNAL) || defined(HAVE_SIGACTION)
+#if 0
 
 static const char *SiginfoString = NULL;
 static bool volatile SignalSafeExitWanted = false;
@@ -1273,7 +1273,7 @@ void PumpWrap(void)
   krepeat = 0;
  }
 
- #if defined(HAVE_SIGNAL) || defined(HAVE_SIGACTION)
+ #if 0
  if(SignalSafeExitWanted)
   NeedExitNow = true;
  #endif
@@ -1739,7 +1739,7 @@ int main(int argc, char *argv[])
 
         SDL_EnableUNICODE(1);
 
-        #if defined(HAVE_SIGNAL) || defined(HAVE_SIGACTION)
+        #if 0
         SetSignals(CloseStuff);
         #endif
 
@@ -1912,7 +1912,7 @@ for(int zgi = 1; zgi < argc; zgi++)// start game load test loop
 	MDFND_DestroyMutex(VTMutex);
         MDFND_DestroyMutex(EVMutex);
 
-	#if defined(HAVE_SIGNAL) || defined(HAVE_SIGACTION)
+	#if 0
 	SetSignals(SIG_IGN);
 	#endif
 
-- 
2.7.0

