diff -urN cherokee-1.2.101/configure.in cherokee-1.2.101-haiku/configure.in
--- cherokee-1.2.101/configure.in	2011-10-18 18:49:42.034865152 +0000
+++ cherokee-1.2.101-haiku/configure.in	2012-07-19 15:37:24.169869312 +0000
@@ -109,6 +109,9 @@
 		setenv_is_threadsafe="yes"
           libdl="-ldl"
           ;;
+	*-*-haiku*)
+          libdl=""
+          ;;          
 	*-*-hpux*)
           libdl="-ldl"
           ;;
@@ -324,6 +327,7 @@
 
 AC_CHECK_FUNC(socket, :, [
 	AC_CHECK_LIB(socket, main)
+	AC_CHECK_LIB(network, socket)
 	AC_CHECK_LIB(net, main)
 	AC_CHECK_LIB(nsl_s, main)
 	AC_CHECK_LIB(nsl, main)
