diff -urN libtool-2.4/libltdl/m4/libtool.m4 libtool-2.4-haiku/libltdl/m4/libtool.m4
--- libtool-2.4/libltdl/m4/libtool.m4	2010-09-22 01:41:19.046923776 -0700
+++ libtool-2.4-haiku/libltdl/m4/libtool.m4	2011-07-26 09:06:47.362807296 -0700
@@ -2472,7 +2472,7 @@
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}${major} ${libname}${shared_ext}'
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LIBRARY_PATH
-  shlibpath_overrides_runpath=yes
+  shlibpath_overrides_runpath=no
   sys_lib_dlsearch_path_spec='/boot/home/config/lib /boot/common/lib /boot/system/lib'
   hardcode_into_libs=yes
   ;;
