diff -ur libtool-2.4/libltdl/m4/libtool.m4 libtool-2.4/libltdl/m4/libtool.m4
--- libtool-2.4/libltdl/m4/libtool.m4	2010-09-22 08:41:19.064749568 +0000
+++ libtool-2.4/libltdl/m4/libtool.m4	2011-07-24 16:58:02.885260288 +0000
@@ -2472,8 +2472,10 @@
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}${major} ${libname}${shared_ext}'
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LIBRARY_PATH
-  shlibpath_overrides_runpath=yes
-  sys_lib_dlsearch_path_spec='/boot/home/config/lib /boot/common/lib /boot/system/lib'
+  shlibpath_overrides_runpath=no
+  # TODO: The /boot/common search path should also develop/lib!
+  sys_lib_search_path_spec='/boot/common/non-packaged/lib /boot/common/lib /boot/system/develop/lib'
+  sys_lib_dlsearch_path_spec='/boot/home/config/non-packaged/lib /boot/home/config/lib /boot/common/non-packaged/lib /boot/common/lib /boot/system/lib'
   hardcode_into_libs=yes
   ;;
 
