diff --git a/src/fonts.c b/src/fonts.c
index a81a2e6..b1e34aa 100644
--- a/src/fonts.c
+++ b/src/fonts.c
@@ -966,10 +966,12 @@ static void loadfonts(SDL_Surface * screen, const char *const dir)
        status_t result;
     result = find_directory(B_SYSTEM_FONTS_DIRECTORY, volume, false, buffer, sizeof(buffer));
        loadfonts(screen, buffer);
-    result = find_directory(B_COMMON_FONTS_DIRECTORY, volume, false, buffer, sizeof(buffer));
+    result = find_directory(B_SYSTEM_NONPACKAGED_FONTS_DIRECTORY, volume, false, buffer, sizeof(buffer));
        loadfonts(screen, buffer);
     result = find_directory(B_USER_FONTS_DIRECTORY, volume, false, buffer, sizeof(buffer));
        loadfonts(screen, buffer);
+    result = find_directory(B_USER_NONPACKAGED_FONTS_DIRECTORY, volume, false, buffer, sizeof(buffer));
+       loadfonts(screen, buffer);
 #elif defined(__APPLE__)
     loadfonts(screen, "/System/Library/Fonts");
     loadfonts(screen, "/Library/Fonts");
diff --git a/src/i18n.c b/src/i18n.c
index 4df46be..2e2298e 100644
--- a/src/i18n.c
+++ b/src/i18n.c
@@ -880,10 +880,11 @@ static int set_current_language(const char *restrict locale_choice) MUST_CHECK;
 static int set_current_language(const char *restrict loc)
 {
   int i;
+  int j = 0;
   int y_nudge = 0;
   char * oldloc;
   char *env_language;
-
+  char *env_language_lang;
 
   if (strlen(loc) > 0)
   {
@@ -974,8 +975,6 @@ printf ("Locale AFTER is: %s\n", setlocale(LC_ALL,NULL));//EP
       mysetenv("LANGUAGE", "C");
   }
     env_language = strdup(getenv("LANGUAGE"));
-    int j = 0;
-    char *env_language_lang;
     if (*env_language)
       {
       env_language_lang = strtok(env_language, ":");
