From 1d2cfa905f7200749055f8b28a2fe5d442d822e8 Mon Sep 17 00:00:00 2001
From: David Murphy <murphman@gmail.com>
Date: Mon, 2 Sep 2019 14:03:10 -0400
Subject: Websocketpp: CMake configuration updates for port to Haiku


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2786aba..c58ebf7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,7 @@
 # Minimum cmake requirement. We should require a quite recent
 # cmake for the dependency find macros etc. to be up to date.
 cmake_minimum_required (VERSION 2.8.8)
+set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
 
 ############ Paths
 
@@ -123,7 +124,7 @@ if (BUILD_TESTS OR BUILD_EXAMPLES)
 
     # g++
     if (CMAKE_COMPILER_IS_GNUCXX)
-        set (WEBSOCKETPP_PLATFORM_LIBS pthread rt)
+        set (WEBSOCKETPP_PLATFORM_LIBS pthread network)
         set (WEBSOCKETPP_PLATFORM_TLS_LIBS ssl crypto)
         set (WEBSOCKETPP_BOOST_LIBS system thread)
         set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
-- 
2.23.0

