diff -urN pixman-0.16.2-orig/test/blitters-test.c pixman-0.16.2/test/blitters-test.c
--- pixman-0.16.2-orig/test/blitters-test.c	2009-09-28 05:58:31.000000000 -0700
+++ pixman-0.16.2/test/blitters-test.c	2009-11-16 20:39:32.000000000 -0800
@@ -174,7 +174,7 @@
 {
     int stride = pixman_image_get_stride (img);
     uint32_t *data = pixman_image_get_data (img);
-    int height = pixman_image_get_height (img);;
+    int height = pixman_image_get_height (img);
     int i, j;
 
     /* swap bytes only on big endian systems */
