diff -urN pixman-0.19.4/configure.ac pixman-0.19.4-haiku/configure.ac
--- pixman-0.19.4/configure.ac	2010-09-21 14:11:23.028049408 +0000
+++ pixman-0.19.4-haiku/configure.ac	2010-10-18 14:12:25.000000000 +0000
@@ -726,6 +726,7 @@
 
     PIXMAN_CHECK_PTHREAD([CFLAGS="-D_REENTRANT"; LIBS="-lpthread"])
     PIXMAN_CHECK_PTHREAD([CFLAGS="-pthread"; LDFLAGS="-pthread"])
+    PIXMAN_CHECK_PTHREAD([CFLAGS="-D_REENTRANT"; LDFLAGS="-lroot"])
 
     if test $support_for_pthread_setspecific = yes; then
 	CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
