diff -up pngcrush-1.7.67-nolib/pngcrush.c.orig pngcrush-1.7.67-nolib/pngcrush.c
--- pngcrush-1.7.67-nolib/pngcrush.c.orig	2013-10-15 00:27:02.442236928 -0600
+++ pngcrush-1.7.67-nolib/pngcrush.c	2013-10-15 00:27:50.849084416 -0600
@@ -6146,6 +6146,8 @@ int main(int argc, char *argv[])
                         png_uint_32 zbuf_size;
                         png_uint_32 required_window;
                         int channels = 0;
+                        png_uint_32 rowbytes;
+                        png_uint_32 h = height;
                         png_set_compression_strategy(write_ptr,
                                                      z_strategy);
                         png_set_compression_mem_level(write_ptr,
@@ -6167,8 +6169,6 @@ int main(int argc, char *argv[])
                          * a png_uint_32, by limiting the width and height used
                          * to 15 bits.
                          */
-                        png_uint_32 rowbytes;
-                        png_uint_32 h = height;
 
                         rowbytes = png_get_rowbytes(read_ptr, read_info_ptr);
 #ifndef PNGCRUSH_H
