diff -up pngcrush-1.7.11-nolib/Makefile.orig pngcrush-1.7.11-nolib/Makefile
--- pngcrush-1.7.11-nolib/Makefile.orig	2010-02-25 21:21:23.004980736 -0700
+++ pngcrush-1.7.11-nolib/Makefile	2010-05-18 14:10:18.201588736 -0600
@@ -31,7 +31,11 @@ E =
 PNGCRUSH  = pngcrush
 
 # LIBS = -lm
+ifeq ($(shell uname), Haiku)
+LIBS = -L/boot/system/lib -lpng -lz
+else
 LIBS = -L$(PNGLIB) -lpng -lz -lm
+endif
 # LIBS = -lpng -lz -lm
 #LIBS = ${ZLIB}/libz.a -lm
 
