diff -ur pngcrush-1.7.6-nolib/Makefile pngcrush-1.7.6-nolib-haiku/Makefile
--- pngcrush-1.7.6-nolib/Makefile	2006-03-30 18:39:16.000000000 -0700
+++ pngcrush-1.7.6-nolib-haiku/Makefile	2009-12-29 01:02:29.000000000 -0700
@@ -36,7 +36,11 @@
 PNGCRUSH  = pngcrush
 
 # LIBS = -lm
+ifeq ($(shell uname), Haiku)
+LIBS = -L/boot/system/lib -lpng -lz
+else
 LIBS = -L$(PNGLIB) -lpng -lz -lm
+endif
 # LIBS = -lpng -lz -lm
 #LIBS = ${ZLIB}/libz.a -lm
 
