diff --git a/tuxpuck.h b/tuxpuck.h
index 4a68aae..3e48d81 100644
--- a/tuxpuck.h
+++ b/tuxpuck.h
@@ -146,6 +146,9 @@ void sprite_erase(Sprite *);
 Uint8 sprite_update(Sprite *, Uint32);
 void sprite_set_position(Sprite *, Uint32, Uint32);
 void sprite_set_animation(Sprite *, Uint8);
+#ifdef __HAIKU__
+#define timer_create timer_create_tp
+#endif
 Timer *timer_create(void);
 void timer_free(Timer *);
 void timer_reset(Timer *);
