Index: mozilla/directory/c-sdk/config/BeOS.mk
===================================================================
RCS file: /cvsroot/mozilla/directory/c-sdk/config/BeOS.mk,v
retrieving revision 5.0.138.1
diff -u -8 -p -r5.0.138.1 BeOS.mk
--- mozilla/directory/c-sdk/config/BeOS.mk	3 Feb 2006 14:41:11 -0000	5.0.138.1
+++ mozilla/directory/c-sdk/config/BeOS.mk	19 Feb 2009 02:42:41 -0000
@@ -64,17 +64,18 @@ ifeq (PC,$(findstring PC,$(OS_TEST)))
 CPU_ARCH		= x86
 CC                      = gcc
 CCC                     = g++
 LD                      = gcc
 RANLIB                  = ranlib
 DSO_LDOPTS              = -nostart
 PORT_FLAGS		= -DHAVE_STRERROR
 ifdef BUILD_OPT
-OPTIMIZER		= -O2
+#OPTIMIZER		= -O2
+OPTIMIZER		= -O1 -march=pentium -mcpu=pentiumpro
 LDFLAGS			+= -s
 else
 OPTIMIZER		= -gdwarf-2 -O0
 endif
 else
 CPU_ARCH		= ppc
 CC                      = mwcc
 CCC                     = mwcc
@@ -85,17 +86,18 @@ DSO_LDOPTS              = -xms -export p
 					-term _term_routine_ \
 					-lroot -lnet \
 					/boot/develop/lib/ppc/glue-noinit.a \
 					/boot/develop/lib/ppc/init_term_dyn.o \
 					/boot/develop/lib/ppc/start_dyn.o 
 
 PORT_FLAGS		= -DHAVE_STRERROR -D_POSIX_SOURCE
 ifdef BUILD_OPT
-OPTIMIZER		= -O2
+#OPTIMIZER		= -O2
+OPTIMIZER		= -O1 -march=pentium -mcpu=pentiumpro
 else
 OPTIMIZER		= -g -O0
 endif
 endif
 CPU_ARCH_TAG		= _$(CPU_ARCH)
 
 OS_INCLUDES		=  -I- -I. 
 #G++INCLUDES		= -I/usr/include/g++
Index: mozilla/directory/c-sdk/configure.in
===================================================================
RCS file: /cvsroot/mozilla/directory/c-sdk/configure.in,v
retrieving revision 5.0.2.34.4.3
diff -u -8 -p -r5.0.2.34.4.3 configure.in
--- mozilla/directory/c-sdk/configure.in	11 Jul 2007 20:43:19 -0000	5.0.2.34.4.3
+++ mozilla/directory/c-sdk/configure.in	19 Feb 2009 02:43:12 -0000
@@ -819,17 +819,17 @@ case "$target" in
     AC_DEFINE(_POSIX_SOURCE)
     DSO_LDOPTS=-nostart
     MDCPUCFG_H=_beos.cfg
     USE_BTHREADS=1
     PR_MD_ARCH_DIR=beos
     RESOLVE_LINK_SYMBOLS=1
     case "${target_cpu}" in
     i*86)
-        _OPTIMIZE_FLAGS=-O2
+        _OPTIMIZE_FLAGS=-O1 -march=pentium -mcpu=pentiumpro
         _DEBUG_FLAGS='-gdwarf-2 -O0'
         MKSHLIB='$(CCC) $(DSO_LDOPTS) -o $@'
         AC_CHECK_LIB(bind, gethostbyaddr, [OS_LIBS="$OS_LIBS -lbind -lsocket"])
         ;;
     powerpc)
         CC=mwcc
         CCC=mwcc
         LD=mwld
Index: mozilla/nsprpub/configure.in
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/configure.in,v
retrieving revision 1.199.2.17
diff -u -8 -p -r1.199.2.17 configure.in
--- mozilla/nsprpub/configure.in	31 Oct 2007 18:07:38 -0000	1.199.2.17
+++ mozilla/nsprpub/configure.in	19 Feb 2009 02:59:37 -0000
@@ -874,17 +874,17 @@ case "$target" in
     AC_DEFINE(_POSIX_SOURCE)
     DSO_LDOPTS=-nostart
     MDCPUCFG_H=_beos.cfg
     USE_BTHREADS=1
     PR_MD_ARCH_DIR=beos
     RESOLVE_LINK_SYMBOLS=1
     case "${target_cpu}" in
     i*86)
-        _OPTIMIZE_FLAGS=-O2
+        _OPTIMIZE_FLAGS=-O1 -march=pentium -mcpu=pentiumpro
         _DEBUG_FLAGS='-gdwarf-2 -O0'
         MKSHLIB='$(CCC) $(DSO_LDOPTS) -o $@'
         AC_CHECK_LIB(bind, gethostbyaddr, [OS_LIBS="$OS_LIBS -lbind -lsocket"])
         ;;
     powerpc)
         CC=mwcc
         CCC=mwcc
         LD=mwld
Index: mozilla/security/coreconf/BeOS.mk
===================================================================
RCS file: /cvsroot/mozilla/security/coreconf/BeOS.mk,v
retrieving revision 1.2
diff -u -8 -p -r1.2 BeOS.mk
--- mozilla/security/coreconf/BeOS.mk	25 Apr 2004 15:02:17 -0000	1.2
+++ mozilla/security/coreconf/BeOS.mk	19 Feb 2009 03:02:58 -0000
@@ -56,17 +56,17 @@ ifeq ($(OS_TEST),ppc)
 	CPU_ARCH	= ppc
 else
 	OS_REL_CFLAGS	= -Di386
 	CPU_ARCH	= x86
 endif
 
 MKSHLIB		= $(CC) -nostart -Wl,-soname -Wl,$(@:$(OBJDIR)/%.so=%.so)
 ifdef BUILD_OPT
-	OPTIMIZER	= -O2
+	OPTIMIZER	= -O1 -march=pentium -mcpu=pentiumpro
 endif
 
 OS_CFLAGS		= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wall -pipe
 OS_LIBS			= -lbe
 
 DEFINES			+= -DBEOS
 
 ifdef USE_PTHREADS
Index: mozilla/xpcom/reflect/xptcall/src/md/unix/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/xpcom/reflect/xptcall/src/md/unix/Makefile.in,v
retrieving revision 1.80.4.6
diff -u -8 -p -r1.80.4.6 Makefile.in
--- mozilla/xpcom/reflect/xptcall/src/md/unix/Makefile.in	19 Feb 2008 21:11:34 -0000	1.80.4.6
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/Makefile.in	19 Feb 2009 04:24:38 -0000
@@ -94,16 +94,17 @@ CPPSRCS		:= xptcinvoke_ipf64.cpp xptcstu
 ASFILES		:= xptcstubs_asm_ipf64.s xptcinvoke_asm_ipf64.s
 endif
 endif
 #
 # BeOS/Intel (uses the same unixish_x86 code)
 #
 ifeq ($(OS_ARCH)$(OS_TEST),BeOSBePC)
 CPPSRCS		:= xptcinvoke_gcc_x86_unix.cpp xptcstubs_gcc_x86_unix.cpp
+CXXFLAGS += -O1 -march=pentium -mcpu=pentiumpro
 endif
 #
 # Neutrino/Intel (uses the same unixish_x86 code)
 #
 ifeq ($(OS_TARGET),NTO)
 ifeq ($(OS_TEST),x86)
 CPPSRCS		:= xptcinvoke_unixish_x86.cpp xptcstubs_unixish_x86.cpp
 endif
Index: mozilla/security/nss-fips/lib/freebl/mpi/target.mk
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/freebl/mpi/target.mk,v
retrieving revision 1.4.2.1
diff -u -8 -p -r1.4.2.1 target.mk
--- mozilla/security/nss-fips/lib/freebl/mpi/target.mk	16 Mar 2006 16:37:28 -0000	1.4.2.1
+++ mozilla/security/nss-fips/lib/freebl/mpi/target.mk	19 Feb 2009 04:29:59 -0000
@@ -40,17 +40,18 @@
 ##
 ## Define CFLAGS to contain any local options your compiler
 ## setup requires.
 ##
 ## Conditional compilation options are no longer here; see
 ## the file 'mpi-config.h' instead.
 ##
 MPICMN = -I. -DMP_API_COMPATIBLE -DMP_IOFUNC
-CFLAGS= -O $(MPICMN)
+CFLAGS= -O1 -march=pentium -mcpu=pentiumpro
+#CFLAGS= -O $(MPICMN)
 #CFLAGS=-ansi -fullwarn -woff 1521 -O3 $(MPICMN)
 #CFLAGS=-ansi -pedantic -Wall -O3 $(MPICMN)
 #CFLAGS=-ansi -pedantic -Wall -g -O2 -DMP_DEBUG=1 $(MPICMN)
 
 ifeq ($(TARGET),mipsIRIX)
 #IRIX
 #MPICMN += -DMP_MONT_USE_MP_MUL 
 MPICMN += -DMP_ASSEMBLY_MULTIPLY -DMP_ASSEMBLY_SQUARE
Index: mozilla/security/nss/lib/freebl/mpi/target.mk
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/freebl/mpi/target.mk,v
retrieving revision 1.4.2.1
diff -u -8 -p -r1.4.2.1 target.mk
--- mozilla/security/nss/lib/freebl/mpi/target.mk	16 Mar 2006 16:37:28 -0000	1.4.2.1
+++ mozilla/security/nss/lib/freebl/mpi/target.mk	19 Feb 2009 04:30:56 -0000
@@ -40,17 +40,18 @@
 ##
 ## Define CFLAGS to contain any local options your compiler
 ## setup requires.
 ##
 ## Conditional compilation options are no longer here; see
 ## the file 'mpi-config.h' instead.
 ##
 MPICMN = -I. -DMP_API_COMPATIBLE -DMP_IOFUNC
-CFLAGS= -O $(MPICMN)
+CFLAGS= -O1 -march=pentium -mcpu=pentiumpro
+#CFLAGS= -O $(MPICMN)
 #CFLAGS=-ansi -fullwarn -woff 1521 -O3 $(MPICMN)
 #CFLAGS=-ansi -pedantic -Wall -O3 $(MPICMN)
 #CFLAGS=-ansi -pedantic -Wall -g -O2 -DMP_DEBUG=1 $(MPICMN)
 
 ifeq ($(TARGET),mipsIRIX)
 #IRIX
 #MPICMN += -DMP_MONT_USE_MP_MUL 
 MPICMN += -DMP_ASSEMBLY_MULTIPLY -DMP_ASSEMBLY_SQUARE
