diff -Naur libgpg-error-1.7-orig/src/mkerrcodes2.awk libgpg-error-1.7/src/mkerrcodes2.awk
--- libgpg-error-1.7-orig/src/mkerrcodes2.awk	2009-03-08 16:04:38.000000000 +0000
+++ libgpg-error-1.7/src/mkerrcodes2.awk	2009-03-08 15:50:42.000000000 +0000
@@ -81,7 +81,7 @@
 /^#/ { next; }
 
 header {
-  if ($1 ~ /^[0123456789]+$/)
+  if ($1 ~ /^-?[0123456789]+$/)
     {
       print "static const int err_code_from_index[] = {";
       header = 0;
