diff -urN joe-3.5/Makefile.am joe-3.5-haiku/Makefile.am
--- joe-3.5/Makefile.am	2006-07-18 17:01:52.000000000 -0700
+++ joe-3.5-haiku/Makefile.am	2008-08-18 22:46:57.000000000 -0700
@@ -24,7 +24,7 @@
 bin_PROGRAMS = joe termidx
 INCLUDES = -DJOERC="\"$(sysconf_joedir)/\""
 
-joe_LDADD = -lm
+joe_LDADD =  @M_LIBS@
 
 joe_SOURCES = b.c blocks.c bw.c cmd.c hash.c help.c kbd.c macro.c main.c menu.c \
 	path.c poshist.c pw.c queue.c qw.c rc.c regex.c scrn.c tab.c termcap.c \
diff -urN joe-3.5/configure.ac joe-3.5-haiku/configure.ac
--- joe-3.5/configure.ac	2006-07-19 07:11:13.000000000 -0700
+++ joe-3.5-haiku/configure.ac	2008-08-18 23:01:03.000000000 -0700
@@ -31,9 +31,8 @@
 	*)	;;
 esac
 
-search_libs="$search_libs m"
-
 # Checks for libraries.
+
 AC_ARG_ENABLE(curses,
 [  --disable-curses        disable use of curses library],
 	curses=$enableval, curses=yes)
@@ -42,6 +41,15 @@
 	search_libs="ncurses curses"
 fi
 
+# Check for libm
+M_PRESENT=""
+AC_CHECK_LIB( m, sin, M_PRESENT="yes",, $M_LIBS -lm )
+if test "x$M_PRESENT" = "xyes"; then
+AC_DEFINE(HAVE_LIBM, 1, [Define if libm is present])
+M_LIBS="-lm"
+AC_SUBST(M_LIBS)
+fi
+
 AC_ARG_ENABLE(termcap,
 [  --disable-termcap       disable use of termcap library],
 	termcap=$enableval, termcap=yes)
