Index: bepdf-1.1.1b2-532/bepdf/Jamfile
===================================================================
--- bepdf-1.1.1b2-532/bepdf/Jamfile	(revision 532)
+++ bepdf-1.1.1b2-532/bepdf/Jamfile	(working copy)
@@ -96,10 +96,14 @@
 #		naming scheme you need to specify the path to the library
 #		and it's name
 #		library: my_lib.a entry: my_lib.a or path/my_lib.a
-LIBS = 
+LIBS = stdc++.r4 ;
+if $(BEPDF_BUILD_GCC) = gcc4 {
+	# Use the standard libstdc++ name when building with gcc 4.
+	LIBS = stdc++ ;
+}
+LIBS +=
 # BeOS
 be
-stdc++.r4
 textencoding
 tracker
 translation
