From 60be00ed00d8b6c33e8601d3fd2b1bce48c4f197 Mon Sep 17 00:00:00 2001
From: Akshay Jaggi <akshay1994.leo@gmail.com>
Date: Wed, 25 Jun 2014 20:37:44 +0000
Subject: [PATCH] Patch for Build

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 58a54f2..a5c8f8a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,9 +5,9 @@ AC_PREREQ(2.59)
 #AC_INIT(dfu-programmer, VERSION, BUG-REPORT-ADDRESS)
 AC_INIT([dfu-programmer],[0.6.2])
 AC_CONFIG_AUX_DIR(m4)
-AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
+AM_INIT_AUTOMAKE
 AC_CONFIG_SRCDIR([src/atmel.c])
-AM_CONFIG_HEADER([src/config.h])
+AC_CONFIG_HEADERS([src/config.h])
 
 AM_MAINTAINER_MODE
 
-- 
1.8.3.4

