diff -up b43-fwcutter-012/Makefile.orig b43-fwcutter-012/Makefile
--- b43-fwcutter-012/Makefile.orig	2009-06-02 09:52:55.058982400 -0600
+++ b43-fwcutter-012/Makefile	2012-06-16 16:13:04.182452224 -0600
@@ -24,8 +24,8 @@ $(BIN): $(OBJECTS)
 install: all
 	-install -d -o 0 -g 0 -m 755 $(PREFIX)/bin/
 	-install -o 0 -g 0 -m 755 $(BIN) $(PREFIX)/bin/
-	-install -d -o 0 -g 0 -m 755 $(PREFIX)/man/man1/
-	-install -o 0 -g 0 -m 644 $(BIN).1 $(PREFIX)/man/man1/
+	-install -d -o 0 -g 0 -m 755 $(PREFIX)/documentation/man/man1/
+	-install -o 0 -g 0 -m 644 $(BIN).1 $(PREFIX)/documentation/man/man1/
 
 clean:
 	-rm -f *.o $(BIN)
