--- fluidsynth-1.1.6/src/utils/fluid_sys.h
+++ fluidsynth-1.1.6/src/utils/fluid_sys.h
@@ -335,7 +335,7 @@ extern fluid_profile_data_t fluid_profile_data[];
     sample data.
  */
 
-#if defined(HAVE_SYS_MMAN_H) && !defined(__OS2__)
+#if defined(HAVE_SYS_MMAN_H) && !defined(__OS2__) && !defined(__HAIKU__)
 #define fluid_mlock(_p,_n)      mlock(_p, _n)
 #define fluid_munlock(_p,_n)    munlock(_p,_n)
 #else
-- 
1.8.3.4

--- fluidsynth-1.1.6/configure.ac.org	2012-08-16 06:01:13.000000000 +0200
+++ fluidsynth-1.1.6/configure.ac	2014-07-21 20:22:51.415760384 +0200
@@ -35,7 +35,7 @@
 LT_VERSION_INFO="lt_current:lt_revision:lt_age"
 AC_SUBST(LT_VERSION_INFO)
 
-AM_CONFIG_HEADER(src/config.h)
+AC_CONFIG_HEADER(src/config.h)
 AC_CONFIG_MACRO_DIR([m4])
 
 CFLAGS="$CFLAGS"

--- fluidsynth-1.1.6/configure.ac.org	2014-07-26 14:02:31.369360896 +0200
+++ fluidsynth-1.1.6/configure.ac	2014-07-26 13:59:47.159383552 +0200
@@ -147,10 +147,10 @@
 # Those are only valid for GCC
 if test "$GCC" = yes; then
   if test "$ENABLE_DEBUG" = "yes"; then
-    CFLAGS="${CFLAGS} ${FCCFLAGS} -g -Wall -W -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wstrict-prototypes -Wno-unused-parameter -Wno-cast-qual"
+    CFLAGS="${CFLAGS} ${FCCFLAGS} -g -Wall -W -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wstrict-prototypes -Wno-cast-qual"
     AC_DEFINE(DEBUG, 1, [Define to activate debugging message]) 
   else
-    CFLAGS="${CFLAGS} ${FCCFLAGS} -O2 -fomit-frame-pointer -funroll-all-loops -finline-functions -Wall -W -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wstrict-prototypes -Winline -Wno-unused-parameter -Wno-cast-qual"
+    CFLAGS="${CFLAGS} ${FCCFLAGS} -O2 -fomit-frame-pointer -funroll-all-loops -finline-functions -Wall -W -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wstrict-prototypes -Winline -Wno-cast-qual"
     AC_DEFINE(DEBUG, 0, [Define to activate debugging message])
   fi
 fi
