diff -Naur fluidsynth-1.1.3/CMakeLists.txt fluidsynth-1.1.3-haiku/CMakeLists.txt
--- fluidsynth-1.1.3/CMakeLists.txt	2010-10-10 22:16:30.031457280 -0300
+++ fluidsynth-1.1.3-haiku/CMakeLists.txt	2011-04-24 14:21:01.000000000 -0300
@@ -152,9 +152,18 @@
   endif  ( MINGW )
 else ( WIN32 )
 # Check PThreads, but not in Windows 
+
+if (UNIX AND NOT BEOS)
+# Check PThreads, but not in Windows 
   find_package ( Pthreads REQUIRED )
   set ( HAVE_LIBPTHREAD ${PTHREADS_FOUND} )
   set ( LIBFLUID_LIBS "m" )
+endif (UNIX AND NOT BEOS)
+
+if (BEOS)
+set ( LIBFLUID_LIBS "network")
+endif (BEOS)
+
 endif ( WIN32 )
 
 # IBM OS/2
diff -Naur fluidsynth-1.1.3/src/utils/fluid_sys.h fluidsynth-1.1.3-haiku/src/utils/fluid_sys.h
--- fluidsynth-1.1.3/src/utils/fluid_sys.h	2010-10-10 22:16:30.022806528 -0300
+++ fluidsynth-1.1.3-haiku/src/utils/fluid_sys.h	2011-04-24 14:17:52.000000000 -0300
@@ -335,7 +335,7 @@
     sample data.
  */
 
-#if defined(HAVE_SYS_MMAN_H) && !defined(__OS2__)
+#if defined(HAVE_SYS_MMAN_H) && !defined(__OS2__) && !defined(__HAIKU__)
 #define fluid_mlock(_p,_n)      mlock(_p, _n)
 #define fluid_munlock(_p,_n)    munlock(_p,_n)
 #else
