diff -urN fluidsynth-1.1.6/CMakeLists.txt fluidsynth-1.1.6-haiku/CMakeLists.txt
--- fluidsynth-1.1.6/CMakeLists.txt
+++ fluidsynth-1.1.6/haiku/CMakeLists.txt
@@ -168,9 +168,16 @@
   endif  ( MINGW )
 else ( WIN32 )
 # Check PThreads, but not in Windows
-  find_package ( Pthreads REQUIRED )
-  set ( HAVE_LIBPTHREAD ${PTHREADS_FOUND} )
-  set ( LIBFLUID_LIBS "m" )
+if (UNIX AND NOT HAIKU)
+# Check PThreads, but not in Windows 
+   find_package ( Pthreads REQUIRED )
+   set ( HAVE_LIBPTHREAD ${PTHREADS_FOUND} )
+   set ( LIBFLUID_LIBS "m" )
+endif (UNIX AND NOT HAIKU)
+
+if (HAIKU)
+set ( LIBFLUID_LIBS "network")
+endif (HAIKU)
 endif ( WIN32 )
 
 # IBM OS/2
diff -Naur fluidsynth-1.1.6/src/utils/fluid_sys.h fluidsynth-1.1.3-haiku/src/utils/fluid_sys.h
--- fluidsynth-1.1.6/src/utils/fluid_sys.h
+++ fluidsynth-1.1.6-haiku/src/utils/fluid_sys.h
@@ -335,7 +335,7 @@
     sample data.
  */
 
-#if defined(HAVE_SYS_MMAN_H) && !defined(__OS2__)
+#if defined(HAVE_SYS_MMAN_H) && !defined(__OS2__) && !defined(__HAIKU__)
 #define fluid_mlock(_p,_n)      mlock(_p, _n)
 #define fluid_munlock(_p,_n)    munlock(_p,_n)
 #else
