diff -urN vorbis-tools-1.2.0/configure.ac vorbis-tools-1.2.0-haiku/configure.ac
--- vorbis-tools-1.2.0/configure.ac	2009-02-28 18:51:08.000000000 +0000
+++ vorbis-tools-1.2.0-haiku/configure.ac	2009-02-28 18:36:00.000000000 +0000
@@ -156,6 +156,8 @@
 SOCKET_LIBS=
 AC_CHECK_LIB(socket, socket, SOCKET_LIBS="-lsocket")
 AC_CHECK_LIB(nsl, gethostbyname, SOCKET_LIBS="-lnsl $SOCKET_LIBS")
+AC_CHECK_LIB(network, socket, SOCKET_LIBS="-lnetwork")
+AC_CHECK_LIB(network, gethostbyname, SOCKET_LIBS="$SOCKET_LIBS")
 
 
 dnl --------------------------------------------------
diff -urN vorbis-tools-1.2.0/oggenc/oggenc.c vorbis-tools-1.2.0-haiku/oggenc/oggenc.c
--- vorbis-tools-1.2.0/oggenc/oggenc.c	2009-02-28 18:51:09.000000000 +0000
+++ vorbis-tools-1.2.0-haiku/oggenc/oggenc.c	2009-02-28 18:49:09.000000000 +0000
@@ -262,12 +262,13 @@
             {
                 /* Create a filename from existing filename, replacing extension with .ogg or .oga */
                 char *start, *end;
+                char *extension;
 
                 start = infiles[i];
                 end = strrchr(infiles[i], '.');
                 end = end?end:(start + strlen(infiles[i])+1);
 
-                char *extension = (opt.with_skeleton) ? ".oga" : ".ogg";
+                extension = (opt.with_skeleton) ? ".oga" : ".ogg";
                 out_fn = malloc(end - start + 5);
                 strncpy(out_fn, start, end-start);
                 out_fn[end-start] = 0;
