diff -ur netsurf-2.9/libcss-0.1.2/include/libcss/font_face.h netsurf-2.9-c89/libcss-0.1.2/include/libcss/font_face.h
--- netsurf-2.9/libcss-0.1.2/include/libcss/font_face.h	2011-12-04 21:06:24.023592960 +0000
+++ netsurf-2.9-c89/libcss-0.1.2/include/libcss/font_face.h	2012-08-30 23:10:26.000000000 +0000
@@ -33,7 +33,7 @@
 	CSS_FONT_FACE_FORMAT_SVG		= 0x08,
 		/* SVG Font; .svg, .svgz */
 	
-	CSS_FONT_FACE_FORMAT_UNKNOWN 		= 0x10,
+	CSS_FONT_FACE_FORMAT_UNKNOWN 		= 0x10
 		/* Format specified, but not recognised */
 	
 	/* We don't define CSS_FONT_FACE_SRC_FORMAT_TRUETYPE as might be 
@@ -49,7 +49,7 @@
 typedef enum css_font_face_location_type {
 	CSS_FONT_FACE_LOCATION_TYPE_UNSPECIFIED	= 0,
 	CSS_FONT_FACE_LOCATION_TYPE_LOCAL	= 1,
-	CSS_FONT_FACE_LOCATION_TYPE_URI		= 2,
+	CSS_FONT_FACE_LOCATION_TYPE_URI		= 2
 } css_font_face_location_type;
 
 	
diff -ur netsurf-2.9/libcss-0.1.2/include/libcss/properties.h netsurf-2.9-c89/libcss-0.1.2/include/libcss/properties.h
--- netsurf-2.9/libcss-0.1.2/include/libcss/properties.h	2012-01-28 20:25:01.023330816 +0000
+++ netsurf-2.9-c89/libcss-0.1.2/include/libcss/properties.h	2012-08-30 23:11:48.000000000 +0000
@@ -287,7 +287,7 @@
 enum css_column_rule_color_e {
 	CSS_COLUMN_RULE_COLOR_INHERIT		= CSS_BACKGROUND_COLOR_INHERIT,
 	CSS_COLUMN_RULE_COLOR_COLOR		= CSS_BACKGROUND_COLOR_COLOR,
-	CSS_COLUMN_RULE_COLOR_CURRENT_COLOR	= CSS_BACKGROUND_COLOR_CURRENT_COLOR,
+	CSS_COLUMN_RULE_COLOR_CURRENT_COLOR	= CSS_BACKGROUND_COLOR_CURRENT_COLOR
 };
 
 enum css_column_rule_style_e {
diff -ur netsurf-2.9/libcss-0.1.2/src/parse/properties/content.c netsurf-2.9-c89/libcss-0.1.2/src/parse/properties/content.c
--- netsurf-2.9/libcss-0.1.2/src/parse/properties/content.c	2011-01-26 12:49:58.027000832 +0000
+++ netsurf-2.9-c89/libcss-0.1.2/src/parse/properties/content.c	2012-08-30 23:23:00.000000000 +0000
@@ -33,7 +33,7 @@
 		css_style *result)
 {
 	int orig_ctx = *ctx;
-	css_error error;
+	css_error error = 0;
 	const css_token *token;
 	bool match;
 
diff -ur netsurf-2.9/libcss-0.1.2/src/parse/properties/quotes.c netsurf-2.9-c89/libcss-0.1.2/src/parse/properties/quotes.c
--- netsurf-2.9/libcss-0.1.2/src/parse/properties/quotes.c	2011-01-26 12:49:58.027787264 +0000
+++ netsurf-2.9-c89/libcss-0.1.2/src/parse/properties/quotes.c	2012-08-30 23:25:08.000000000 +0000
@@ -32,7 +32,7 @@
 		css_style *result)
 {
 	int orig_ctx = *ctx;
-	css_error error;
+	css_error error = 0;
 	const css_token *token;
 	bool match;
 
diff -ur netsurf-2.9/netsurf-2.9/Makefile.defaults netsurf-2.9-c89/netsurf-2.9/Makefile.defaults
--- netsurf-2.9/netsurf-2.9/Makefile.defaults	2012-01-01 21:42:38.027000832 +0000
+++ netsurf-2.9-c89/netsurf-2.9/Makefile.defaults	2012-08-31 16:45:50.000000000 +0000
@@ -186,13 +186,13 @@
 ifeq ($(TARGET),beos)
 
 
+  # HAIKU use finddir
   # Where to install the netsurf binary
-  NETSURF_BEOS_BIN := /boot/apps/netsurf/
+  NETSURF_BEOS_BIN := $(shell finddir B_APPS_DIRECTORY)/NetSurf/
 
-  # TODO:HAIKU -- not sure if ~/.netsurf applies in beos
   # Where to search for NetSurf's resources after looking in ~/.netsurf and
   # $NETSURFRES.  It must have a trailing /
-  NETSURF_BEOS_RESOURCES := /boot/apps/netsurf/res/
+  NETSURF_BEOS_RESOURCES := $(shell finddir B_APPS_DIRECTORY)/NetSurf/res/
 
   # Enable NetSurf's use of libsvgtiny for displaying SVGs
   # Valid options: YES, NO, AUTO
