diff -up git-1.7.11.3/Makefile.orig git-1.7.11.3/Makefile
--- git-1.7.11.3/Makefile.orig	2012-07-22 14:44:07.027525120 -0600
+++ git-1.7.11.3/Makefile	2012-07-28 22:12:40.271056896 -0600
@@ -1297,6 +1297,31 @@ ifeq ($(uname_S),Minix)
 	NO_CURL =
 	NO_EXPAT =
 endif
+ifeq ($(uname_S),Haiku)
+	NO_LIBGEN_H = YesPlease
+	NO_MEMMEM = YesPlease
+	NO_MKSTEMPS = YesPlease
+	NEEDS_LIBICONV = YesPlease
+	PERL_PATH = perl
+	PYTHON_PATH = python
+#	TCL_PATH = tclsh
+	mandir = /boot/common/documentation/man
+	infodir = /boot/common/documentation/info
+	htmldir = /boot/common/documentation/doc/git-1.7.11.3
+	gitexecdir = /boot/common/bin/git-core
+	gitwebdir = /boot/common/data/gitweb
+	template_dir = /boot/common/data/git-core/templates
+	NO_TCLTK = YesPlease
+	DEFAULT_EDITOR = nano
+	CURL_DIR = /boot/common
+	EXPAT_DIR = /boot/common
+	BASIC_CFLAGS += -I/boot/common/include
+	BASIC_LDFLAGS += -lnetwork -lbsd -L/boot/common/lib
+	PTHREAD_LIBS =
+	prefix = /boot/common
+	NO_CROSS_DIRECTORY_HARDLINKS = YesPlease
+	NO_GETTEXT = YesPlease
+endif
 ifneq (,$(findstring MINGW,$(uname_S)))
 	pathsep = ;
 	NO_PREAD = YesPlease
diff -up git-1.7.11.3/git-web--browse.sh.orig git-1.7.11.3/git-web--browse.sh
--- git-1.7.11.3/git-web--browse.sh.orig	2012-07-22 14:44:07.017039360 -0600
+++ git-1.7.11.3/git-web--browse.sh	2012-07-28 22:12:40.283115520 -0600
@@ -116,7 +116,7 @@ if test -z "$browser" ; then
 			browser_candidates="konqueror $browser_candidates"
 		fi
 	else
-		browser_candidates="w3m elinks links lynx"
+		browser_candidates="w3m elinks links lynx open"
 	fi
 	# SECURITYSESSIONID indicates an OS X GUI login session
 	if test -n "$SECURITYSESSIONID" \
