diff -up git-1.7.6/Makefile.orig git-1.7.6/Makefile
--- git-1.7.6/Makefile.orig	2011-06-26 16:26:13.012582912 -0600
+++ git-1.7.6/Makefile	2011-07-02 12:04:18.592183296 -0600
@@ -1142,6 +1142,30 @@ ifeq ($(uname_S),Interix)
 		NO_INET_PTON = YesPlease
 	endif
 endif
+ifeq ($(uname_S),Haiku)
+	NO_LIBGEN_H = YesPlease
+	NO_MEMMEM = YesPlease
+	NO_MKSTEMPS = YesPlease
+	NEEDS_LIBICONV = YesPlease
+	NO_IPV6 = YesPlease
+	PERL_PATH = perl
+	PYTHON_PATH = python
+#	TCL_PATH = tclsh
+	mandir = /boot/common/documentation/man
+	infodir = /boot/common/documentation/info
+	gitexecdir = /boot/common/bin/git-core
+	gitwebdir = /boot/common/data/gitweb
+	template_dir = /boot/common/data/git-core/templates
+	NO_TCLTK = YesPlease
+	DEFAULT_EDITOR = nano
+	CURL_DIR = /boot/common
+	EXPAT_DIR = /boot/common
+	BASIC_CFLAGS += -I/boot/common/include
+	BASIC_LDFLAGS += -lnetwork -lbsd -L/boot/common/lib
+	PTHREAD_LIBS =
+	prefix = /boot/common
+	NO_CROSS_DIRECTORY_HARDLINKS = YesPlease
+endif
 ifneq (,$(findstring MINGW,$(uname_S)))
 	pathsep = ;
 	NO_PREAD = YesPlease
diff -up git-1.7.6/git-gui/Makefile.orig git-1.7.6/git-gui/Makefile
--- git-1.7.6/git-gui/Makefile.orig	2011-06-26 16:26:13.052690944 -0600
+++ git-1.7.6/git-gui/Makefile	2011-07-02 12:04:18.594804736 -0600
@@ -57,7 +57,7 @@ INSTALL_X1 =
 INSTALL_A0 = find # space is required here
 INSTALL_A1 = | cpio -pud
 INSTALL_L0 = rm -f # space is required here
-INSTALL_L1 = && ln # space is required here
+INSTALL_L1 = && ln -s # space is required here
 INSTALL_L2 =
 INSTALL_L3 =
 
@@ -87,7 +87,7 @@ ifndef V
 	INSTALL_L0 = dst=
 	INSTALL_L1 = && src=
 	INSTALL_L2 = && dst=
-	INSTALL_L3 = && echo '   ' 'LINK       ' `basename "$$dst"` '->' `basename "$$src"` && rm -f "$$dst" && ln "$$src" "$$dst"
+	INSTALL_L3 = && echo '   ' 'LINK       ' `basename "$$dst"` '->' `basename "$$src"` && rm -f "$$dst" && ln -s "$$src" "$$dst"
 
 	CLEAN_DST = echo ' ' UNINSTALL
 	REMOVE_D0 = dir=
