diff -Naur dwarf-20130729/dwarfdump/Makefile.in dwarf-20130729-c89/dwarfdump/Makefile.in
--- dwarf-20130729/dwarfdump/Makefile.in	2013-02-02 19:13:55.027000832 +0000
+++ dwarf-20130729-c89/dwarfdump/Makefile.in	2014-01-02 23:57:54.130809856 +0000
@@ -154,9 +154,11 @@
 # will work and leave sensible permissions on the resulting files.
 # Some adjustment might be required, see README.
 install: all
-	$(INSTALL) dwarfdump $(bindir)/dwarfdump
-	$(INSTALL) $(srcdir)/dwarfdump.conf $(libdir)/dwarfdump.conf
-	$(INSTALL) $(srcdir)/dwarfdump.1 $(man1dir)/dwarfdump.1
+	install -m 0755 -d $(bindir)
+	install -m 0755 -d $(man1dir)
+	install -m 0755  dwarfdump $(bindir)
+	install -m 0755  dwarfdump.conf $(libdir)
+	install -m 0755  dwarfdump.1 $(man1dir)
 
 uninstall:
 	-rm -f $(bindir)/dwarfdump
diff -Naur dwarf-20130729/dwarfdump/print_die.c dwarf-20130729-c89/dwarfdump/print_die.c
--- dwarf-20130729/dwarfdump/print_die.c	2013-02-06 20:48:07.019922944 +0000
+++ dwarf-20130729-c89/dwarfdump/print_die.c	2014-01-02 23:11:00.841744384 +0000
@@ -3380,6 +3380,7 @@
                     we have a serious botch. this FORM
                     defines the value as a .debug_info
                     global offset. */
+                int res = 0;
                 Dwarf_Off cuoff = 0;
                 Dwarf_Off culen = 0;
                 Dwarf_Off die_overall_offset = 0;
@@ -3399,7 +3400,7 @@
                     print_error(dbg,small_buf,DW_DLV_OK,0);
                 }
 
-                int res = dwarf_die_CU_offset_range(die,&cuoff,
+                res = dwarf_die_CU_offset_range(die,&cuoff,
                     &culen,&err);
                 DWARF_CHECK_COUNT(tag_tree_result,1);
                 if (res != DW_DLV_OK) {
diff -Naur dwarf-20130729/libdwarf/gennames.c dwarf-20130729-c89/libdwarf/gennames.c
--- dwarf-20130729/libdwarf/gennames.c	2013-01-25 19:09:11.052166656 +0000
+++ dwarf-20130729-c89/libdwarf/gennames.c	2014-01-02 22:57:20.856162304 +0000
@@ -486,8 +486,8 @@
 
     /* Process each line from 'dwarf.h' */
     while (!feof(f_dwarf_in)) {
-        errno = 0;
         char *fgbad = fgets(line_in,sizeof(line_in),f_dwarf_in);
+        errno = 0;
         if(!fgbad) {
             if(feof(f_dwarf_in)) {
                 break;
diff -Naur dwarf-20130729/libdwarf/Makefile.in dwarf-20130729-c89/libdwarf/Makefile.in
--- dwarf-20130729/libdwarf/Makefile.in	2013-07-28 17:58:54.051904512 +0000
+++ dwarf-20130729-c89/libdwarf/Makefile.in	2014-01-03 00:36:16.817627136 +0000
@@ -46,8 +46,9 @@
 
 prefix =	@prefix@
 exec_prefix =	@exec_prefix@
-bindir =	$(exec_prefix)/bin
-libdir =	$(exec_prefix)/lib
+bindir =       $(DESTDIR)$(prefix)/bin
+libdir =       $(DESTDIR)$(prefix)/lib
+incdir =       $(DESTDIR)$(prefix)/develop/include/libdwarf
 
 INSTALL =	@INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -213,7 +214,11 @@
 	rm -f dwarf_names_enum.h dwarf_names_new.h dwarf_names.c dwarf_names.h
 
 install: all
-	echo "No install provided, see comments in the README"
+	install -m 0755 -d $(incdir)
+	install -m 0755 -d $(libdir)
+	install -m 0755 libdwarf.h $(incdir)
+	install -m 0755 dwarf.h $(incdir)
+	install -m 0755 libdwarf.so $(libdir)
 
 distclean:	clean
 	rm -f config.status config.log config.cache config.h
