diff --git a/Paladin/LibWindow.cpp b/Paladin/LibWindow.cpp
index c75e6ee..fb1537f 100644
--- a/Paladin/LibWindow.cpp
+++ b/Paladin/LibWindow.cpp
@@ -208,7 +208,7 @@ LibraryWindow::ScanThread(void *data)
 		
 		r.OffsetBy(0,r.Height() + 10);
 		
-		out = win->ScanFolder(r.LeftTop(),"/boot/common/lib",&maxwidth);
+		out = win->ScanFolder(r.LeftTop(),"/boot/system/lib",&maxwidth);
 		if (out != BRect(0,0,-1,-1));
 		{
 			r = out;
