--- barrage-1.0.4/src/Makefile.in.org	2011-07-28 15:28:20.000000000 +0200
+++ barrage-1.0.4/src/Makefile.in	2014-07-14 13:05:29.277086208 +0200
@@ -185,7 +185,7 @@
 	SFont.c
 
 barrage_LDFLAGS = 
-barrage_LDADD = @audio_lib@ -lm
+barrage_LDADD = @audio_lib@
 EXTRA_DIST = \
 	defs.h particle.h shots.h units.h\
 	data.h bfield.h menu.h SFont.h chart.h
--- barrage-1.0.4/src/main.c.org	2014-07-14 13:28:39.039321600 +0200
+++ barrage-1.0.4/src/main.c	2014-07-14 13:27:14.832831488 +0200
@@ -28,7 +28,7 @@
 #define BITDEPTH 16
 
 int delay = 0;
-int fullscreen = 1;
+int fullscreen = 0;
 int audio_on = 1;
 int audio_freq = 22050;
 int audio_format = AUDIO_S16LSB;
