diff -Naur W6Haiku/InfoView.cpp W6Haiku-haiku/InfoView.cpp
--- W6Haiku/InfoView.cpp	2008-08-16 02:57:03.065798144 +0000
+++ W6Haiku-haiku/InfoView.cpp	2013-01-13 11:59:46.657457152 +0000
@@ -280,7 +280,7 @@
 	sprintf(str2,_jour->month_name);  	strcat(str,str2);
 	sprintf(str2," %i, %i",_jour->month_counter, _jour->year);	strcat(str,str2);
 	strcat(str,")");
-	Window()->SetTitle(str);
+	//Window()->SetTitle(str);
 
 	
 	//A l'ancienne
@@ -567,4 +567,4 @@
    			x->Close();
 			delete upBitmap;
    			delete downBitmap;  			
-}
\ No newline at end of file
+}
diff -Naur W6Haiku/W6.cpp W6Haiku-haiku/W6.cpp
--- W6Haiku/W6.cpp	2008-08-16 04:35:06.002359296 +0000
+++ W6Haiku-haiku/W6.cpp	2013-01-13 12:00:52.285736960 +0000
@@ -89,9 +89,9 @@
  	
  	
  	//classe qui tourne en permanence et qui fait les daypassed (et par conséquent, les battle etc. 
-	run_me = new Running(_country,_group,_politic,_weapon,_jour);
+	//run_me = new Running(_country,_group,_politic,_weapon,_jour);
 	runWin = new BWindow(BRect(0,0,0,0), "W6 time run", B_TITLED_WINDOW,  B_NOT_RESIZABLE); 
-	runWin->AddChild(run_me); //sinon le pulse fonctionne pas...					  
+	//runWin->AddChild(run_me); //sinon le pulse fonctionne pas...					  
 	runWin->Lock();  
 	runWin->Show(); //faut faire show pour lancer le pulse dans les views
 	runWin->Hide(); //quitte à hider la window ensuite
