diff -Naur premake-4.3/build/gmake.unix/Premake4.make premake-4.3-haiku/build/gmake.unix/Premake4.make
--- premake-4.3/build/gmake.unix/Premake4.make	2010-11-16 11:29:14.028311552 +0000
+++ premake-4.3-haiku/build/gmake.unix/Premake4.make	2012-12-27 23:50:16.043253760 +0000
@@ -28,8 +28,8 @@
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -Wall -Os
   CXXFLAGS  += $(CFLAGS) 
-  LDFLAGS   += -s -rdynamic
-  LIBS      += -lm -ldl
+  LDFLAGS   += -s
+  LIBS      +=
   RESFLAGS  += $(DEFINES) $(INCLUDES) 
   LDDEPS    += 
   LINKCMD    = $(CC) -o $(TARGET) $(OBJECTS) $(LDFLAGS) $(RESOURCES) $(ARCH) $(LIBS)
diff -Naur premake-4.3/src/host/lua-5.1.4/src/luaconf.h premake-4.3-haiku/src/host/lua-5.1.4/src/luaconf.h
--- premake-4.3/src/host/lua-5.1.4/src/luaconf.h	2010-11-16 11:29:12.042991616 +0000
+++ premake-4.3-haiku/src/host/lua-5.1.4/src/luaconf.h	2012-12-27 23:50:16.050331648 +0000
@@ -55,7 +55,6 @@
 #define LUA_USE_MKSTEMP
 #define LUA_USE_ISATTY
 #define LUA_USE_POPEN
-#define LUA_USE_ULONGJMP
 #endif
 
 
diff -Naur premake-4.3/src/host/premake.h premake-4.3-haiku/src/host/premake.h
--- premake-4.3/src/host/premake.h	2010-11-16 11:29:12.047185920 +0000
+++ premake-4.3-haiku/src/host/premake.h	2012-12-27 23:50:16.052166656 +0000
@@ -12,7 +12,7 @@
 
 /* Identify the current platform I'm not sure how to reliably detect
  * Windows but since it is the most common I use it as the default */
-#if defined(__linux__)
+#if defined(__linux__)  || defined(__HAIKU__)
 #define PLATFORM_LINUX    (1)
 #define PLATFORM_STRING   "linux"
 #elif defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
