diff -urN vlc-1.1.5/configure.ac vlc-1.1.5-haiku/configure.ac
--- vlc-1.1.5/configure.ac	2010-11-06 18:02:38.013369344 -0700
+++ vlc-1.1.5-haiku/configure.ac	2010-11-07 11:39:30.000000000 -0800
@@ -379,6 +379,18 @@
         VLC_ADD_LIBS([filesystem access_ftp access_mms access_output_udp oldtelnet netsync sap libvlccore growl_udp],[-lnet])
     fi
     ;;
+  haiku)
+    SYS=beos
+    CFLAGS_save="${CFLAGS_save} -Wno-multichar"; CFLAGS="${CFLAGS_save}"
+    CXXFLAGS_save="${CXXFLAGS_save} -Wno-multichar"; CXXFLAGS="${CXXFLAGS_save}"
+    VLC_ADD_CXXFLAGS([beos],[])
+    VLC_ADD_LIBS([vlc libvlccore logger],[-lbe])
+    VLC_ADD_LIBS([dvdnav dvdread],[-ldl])
+    VLC_ADD_LIBS([filesystem],[-lpoll])
+    LDFLAGS_save="${LDFLAGS_save} -lintl"; LDFLAGS="${LDFLAGS_save}"
+    VLC_ADD_LIBS([filesystem access_ftp access_mms access_output_udp oldtelnet netsync sap libvlccore growl_udp],[-lnetwork])
+    ;;
+
   *)
     SYS="${host_os}"
     ;;
@@ -4713,7 +4725,7 @@
 dnl  Stuff used by the program
 dnl
 VERSION_MESSAGE="${VERSION} ${CODENAME}"
-COPYRIGHT_MESSAGE="Copyright © ${COPYRIGHT_YEARS} the VideoLAN team"
+COPYRIGHT_MESSAGE="Copyright C ${COPYRIGHT_YEARS} the VideoLAN team"
 AC_DEFINE_UNQUOTED(VERSION_MESSAGE, "${VERSION_MESSAGE}", [Simple version string])
 AC_DEFINE_UNQUOTED(COPYRIGHT_MESSAGE, "${COPYRIGHT_MESSAGE}", [Copyright string])
 AC_DEFINE_UNQUOTED(COPYRIGHT_YEARS, "${COPYRIGHT_YEARS}", [The copyright years])
