diff -urN axel-2.4/configure axel-2.4-haiku/configure
--- axel-2.4/configure	2009-06-12 09:21:27.000000000 +0000
+++ axel-2.4-haiku/configure	2009-06-12 09:16:10.000000000 +0000
@@ -200,6 +200,26 @@
 	echo 'Please keep in mind that you need GNU make to make Axel!'
 	echo ''
 ;;
+Haiku* )
+	echo 'LFLAGS+=-lnetwork' >> Makefile.settings
+	echo '#define NOGETOPTLONG' >> config.h
+#	intl not yet working on Haiku
+#	if [ "$i18n" = "1" ]; then
+#		echo 'LFLAGS+=-lintl' >> Makefile.settings; 
+#	fi
+	echo '#define _IO(x,y)        (IOC_VOID|((x)<<8)|y)' >> config.h
+	echo '#define _IOR(x,y,t)     (IOC_OUT|((sizeof(t)&IOCPARM_MASK)<<16)|((x)<<8)|y)' >> config.h
+	echo '#define _IOW(x,y,t)     (IOC_IN|((sizeof(t)&IOCPARM_MASK)<<16)|((x)<<8)|y)' >> config.h
+#	echo '/* this should be _IORW, but stdio got there first */' >> config.h
+	echo '#define _IOWR(x,y,t)    (IOC_INOUT|((sizeof(t)&IOCPARM_MASK)<<16)|((x)<<8)|y)' >> config.h
+	echo "#define SIOCGIFADDR     _IOWR('i',13, struct ifreq)     /* get ifnet address */" >> config.h
+	echo '#define IOCPARM_MASK    0x7f            /* parameters must be < 128 bytes */' >> config.h
+	echo '#define IOC_VOID        0x20000000      /* no parameters */' >> config.h
+	echo '#define IOC_OUT         0x40000000      /* copy out parameters */' >> config.h
+	echo '#define IOC_IN          0x80000000      /* copy in parameters */' >> config.h
+	echo '#define IOC_INOUT       (IOC_IN|IOC_OUT)' >> config.h
+#	echo '/* the 0x20000000 is so we can distinguish new ioctl's from old */' >> config.h	
+;;
 CYGWIN_* )
 	echo 'LFLAGS+=-lpthread' >> Makefile.settings
 	if [ "$i18n" = "1" ]; then
Binary files axel-2.4/conn.o and axel-2.4-haiku/conn.o differ
Binary files axel-2.4/ftp.o and axel-2.4-haiku/ftp.o differ
Binary files axel-2.4/http.o and axel-2.4-haiku/http.o differ
Binary files axel-2.4/search.o and axel-2.4-haiku/search.o differ
Binary files axel-2.4/tcp.o and axel-2.4-haiku/tcp.o differ
Binary files axel-2.4/text.o and axel-2.4-haiku/text.o differ
