diff -urN taglib-1.7/taglib/ape/apefooter.cpp taglib-1.7-haiku/taglib/ape/apefooter.cpp
--- taglib-1.7/taglib/ape/apefooter.cpp	2011-03-11 13:15:38.020709376 +0000
+++ taglib-1.7-haiku/taglib/ape/apefooter.cpp	2011-04-11 03:16:26.167772160 +0000
@@ -24,7 +24,7 @@
  *   http://www.mozilla.org/MPL/                                           *
  ***************************************************************************/
 
-#include <ostream>
+#include <iostream>
 #include <bitset>
 
 #include <tstring.h>
diff -urN taglib-1.7/taglib/mpeg/id3v2/id3v2header.cpp taglib-1.7-haiku/taglib/mpeg/id3v2/id3v2header.cpp
--- taglib-1.7/taglib/mpeg/id3v2/id3v2header.cpp	2011-03-11 13:15:38.031719424 +0000
+++ taglib-1.7-haiku/taglib/mpeg/id3v2/id3v2header.cpp	2011-04-11 02:24:30.000000000 +0000
@@ -23,7 +23,7 @@
  *   http://www.mozilla.org/MPL/                                           *
  ***************************************************************************/
 
-#include <ostream>
+#include <iostream>
 #include <bitset>
 
 #include <tstring.h>
diff -urN taglib-1.7/taglib/mpeg/id3v2/id3v2synchdata.cpp taglib-1.7-haiku/taglib/mpeg/id3v2/id3v2synchdata.cpp
--- taglib-1.7/taglib/mpeg/id3v2/id3v2synchdata.cpp	2011-03-11 13:15:38.031981568 +0000
+++ taglib-1.7-haiku/taglib/mpeg/id3v2/id3v2synchdata.cpp	2011-04-11 02:23:58.000000000 +0000
@@ -23,7 +23,7 @@
  *   http://www.mozilla.org/MPL/                                           *
  ***************************************************************************/
 
-#include <ostream>
+#include <iostream>
 
 #include "id3v2synchdata.h"
 
diff -urN taglib-1.7/taglib/toolkit/tbytevector.cpp taglib-1.7-haiku/taglib/toolkit/tbytevector.cpp
--- taglib-1.7/taglib/toolkit/tbytevector.cpp	2011-03-11 13:15:38.038797312 +0000
+++ taglib-1.7-haiku/taglib/toolkit/tbytevector.cpp	2011-04-11 02:26:04.000000000 +0000
@@ -23,7 +23,7 @@
  *   http://www.mozilla.org/MPL/                                           *
  ***************************************************************************/
 
-#include <ostream>
+#include <iostream>
 
 #include <tstring.h>
 #include <tdebug.h>
diff -urN taglib-1.7/taglib/toolkit/tbytevector.h taglib-1.7-haiku/taglib/toolkit/tbytevector.h
--- taglib-1.7/taglib/toolkit/tbytevector.h	2011-03-11 13:15:38.038797312 +0000
+++ taglib-1.7-haiku/taglib/toolkit/tbytevector.h	2011-04-11 02:22:38.000000000 +0000
@@ -30,7 +30,7 @@
 #include "taglib_export.h"
 
 #include <vector>
-#include <ostream>
+#include <iostream>
 
 namespace TagLib {
 
diff -urN taglib-1.7/taglib/toolkit/tstring.cpp taglib-1.7-haiku/taglib/toolkit/tstring.cpp
--- taglib-1.7/taglib/toolkit/tstring.cpp	2011-03-11 13:15:38.040108032 +0000
+++ taglib-1.7-haiku/taglib/toolkit/tstring.cpp	2011-04-11 02:25:40.000000000 +0000
@@ -27,7 +27,7 @@
 #include "unicode.h"
 #include "tdebug.h"
 
-#include <ostream>
+#include <iostream>
 
 #include <string.h>
 
diff -urN taglib-1.7/taglib/toolkit/tstring.h taglib-1.7-haiku/taglib/toolkit/tstring.h
--- taglib-1.7/taglib/toolkit/tstring.h	2011-03-11 13:15:38.040108032 +0000
+++ taglib-1.7-haiku/taglib/toolkit/tstring.h	2011-04-11 02:23:07.000000000 +0000
@@ -31,7 +31,7 @@
 #include "tbytevector.h"
 
 #include <string>
-#include <ostream>
+#include <iostream>
 
 /*!
  * \relates TagLib::String
diff -urN taglib-1.7/taglib/toolkit/tstringlist.h taglib-1.7-haiku/taglib/toolkit/tstringlist.h
--- taglib-1.7/taglib/toolkit/tstringlist.h	2011-03-11 13:15:38.040370176 +0000
+++ taglib-1.7-haiku/taglib/toolkit/tstringlist.h	2011-04-11 02:23:30.000000000 +0000
@@ -31,7 +31,7 @@
 #include "tbytevectorlist.h"
 #include "taglib_export.h"
 
-#include <ostream>
+#include <iostream>
 
 namespace TagLib {
 
