From 2ae489b85228a74d925ac4ff3bc54537172ec9c5 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Fri, 23 Feb 2018 15:09:04 +0100
Subject: [PATCH] Small fixes for compilation on Haiku


diff --git a/src/Makefile.am b/src/Makefile.am
index bc906a14..589a024e 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -15,7 +15,7 @@
 datarootdir = @datarootdir@
 mandir = @mandir@
 man1dir = $(mandir)/man1
-docdir = $(prefix)/share/doc
+docdir = @docdir@
 builddir = @builddir@
 top_srcdir = @top_srcdir@
 srcdir = $(top_srcdir)/src
@@ -267,8 +267,12 @@
 	$(BEOS_XRES) -o $(APP) $(APP).rsrc
 endif
 ifneq ($(BEOS_SETVERSION),)
+	versfile="$(srcdir)/include/version.h"; \
+	major=`sed -n -e 's/#define[ \\t]*VER_MAJOR[ \\t]*\\([0-9]*\\)/\\1/p' $$versfile`; \
+	minor=`sed -n -e 's/#define[ \\t]*VER_MINOR[ \\t]*\\([0-9]*\\)/\\1/p' $$versfile`; \
+	micro=`sed -n -e 's/#define[ \\t]*VER_MICRO[ \\t]*\\([0-9]*\\)/\\1/p' $$versfile`; \
 	$(BEOS_SETVERSION) $(APP) \
-		-app 0 9 0 d 0 \
+		-app $$major $$minor $$micro d 0 \
 		-short "$(APP) $(PACKAGE_VERSION)" \
 		-long "$(APP) $(PACKAGE_VERSION)"
 endif
diff --git a/src/Unix/beos/clipbrd_beos.cpp b/src/Unix/beos/clipbrd_beos.cpp
index 9c073cc6..b9dfa64a 100644
--- a/src/Unix/beos/clipbrd_beos.cpp
+++ b/src/Unix/beos/clipbrd_beos.cpp
@@ -18,6 +18,7 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
+#include "sysdeps.h"
 #if defined(OS_beos)
 #include <Clipboard.h>
 #include <String.h>
@@ -51,7 +55,7 @@ void write_aclip(char *data, size_t len)
 char * read_aclip( size_t *len)
 {
 	const char *text;
-	int32 textLen;
+	ssize_t textLen;
 	BMessage *clip = NULL;
 	char *data = NULL;
 	fprintf(stderr, "%s()\n", __FUNCTION__);
-- 
2.16.1

