--- Source/JavaScriptCore/runtime/Options.cpp.ori	2016-07-06 09:04:18.145489920 +0200
+++ Source/JavaScriptCore/runtime/Options.cpp	2016-07-06 08:32:30.864550912 +0200
@@ -28,8 +28,8 @@
 
 #include "HeapStatistics.h"
 #include <algorithm>
+#include <cmath>
 #include <limits>
-#include <math.h>
 #include <mutex>
 #include <stdlib.h>
 #include <string.h>
--- Source/WebCore/platform/haiku/RenderThemeHaiku.cpp.ori	2016-07-05 14:50:40.034078720 +0200
+++ Source/WebCore/platform/haiku/RenderThemeHaiku.cpp	2016-07-05 14:54:06.209453056 +0200
@@ -33,6 +33,7 @@
 #include "InputTypeNames.h"
 #include "NotImplemented.h"
 #include "PaintInfo.h"
+#include "RenderElement.h"
 #include "UserAgentScripts.h"
 #include "UserAgentStyleSheets.h"
 #include <ControlLook.h>
