--- a/Source/WebCore/platform/haiku/KURLHaiku.cpp	2017-11-01 21:26:35.336330752 +0100
+++ b/Source/WebCore/platform/haiku/KURLHaiku.cpp	2017-11-01 21:14:24.063700992 +0100
@@ -37,16 +37,7 @@
 
 URL::URL(const BUrl& url)
 {
-    setProtocol(url.Protocol());
-    setHost(url.Host());
-    if (url.Port() != 0)
-        setPort(url.Port());
-    setUser(url.UserName());
-    setPass(url.Password());
-    setPath(url.Path());
-    setQuery(url.Request());
-    if (url.Fragment().Length() != 0)
-        setFragmentIdentifier(url.Fragment());
+	parse(String::fromUTF8(url.UrlString()));
 }
 
 URL::operator BUrl() const
