diff -Naur confuse-2.7/configure.ac confuse-2.7-haiku/configure.ac
--- confuse-2.7/configure.ac	2010-02-22 00:27:44.009961472 +0000
+++ confuse-2.7-haiku/configure.ac	2014-01-12 12:40:30.637796352 +0000
@@ -6,7 +6,7 @@
 AC_CONFIG_AUX_DIR(support)
 AM_MAINTAINER_MODE
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_SRCDIR(src/confuse.c)
 
 # Checks for programs.
