diff -urN php-5.2.9/acinclude.m4 php-5.2.9-haiku/acinclude.m4
--- php-5.2.9/acinclude.m4	2009-03-01 12:13:19.000000000 +0000
+++ php-5.2.9-haiku/acinclude.m4	2009-03-01 12:56:26.000000000 +0000
@@ -2332,7 +2332,7 @@
   if test "$found_openssl" = "no"; then
   
     if test "$PHP_OPENSSL_DIR" = "yes"; then
-      PHP_OPENSSL_DIR="/usr/local/ssl /usr/local /usr /usr/local/openssl"
+      PHP_OPENSSL_DIR="/usr/local/ssl /usr/local /usr /usr/local/openssl /boot/common/ssl"
     fi
 
     for i in $PHP_OPENSSL_DIR; do
@@ -2444,7 +2444,7 @@
   dnl
   if test "$found_iconv" = "no"; then
 
-    for i in $PHP_ICONV /usr/local /usr; do
+    for i in $PHP_ICONV /usr/local /usr /boot/common; do
       if test -r $i/include/giconv.h; then
         AC_DEFINE(HAVE_GICONV_H, 1, [ ])
         ICONV_DIR=$i
@@ -2500,7 +2500,7 @@
 AC_DEFUN([PHP_SETUP_LIBXML], [
 AC_CACHE_CHECK([for xml2-config path], ac_cv_php_xml2_config_path,
 [
-  for i in $PHP_LIBXML_DIR /usr/local /usr; do
+  for i in $PHP_LIBXML_DIR /usr/local /boot/common /usr; do
     if test -x "$i/bin/xml2-config"; then
       ac_cv_php_xml2_config_path="$i/bin/xml2-config"
       break
diff -urN php-5.2.9/configure.in php-5.2.9-haiku/configure.in
--- php-5.2.9/configure.in	2009-03-01 12:13:18.000000000 +0000
+++ php-5.2.9-haiku/configure.in	2009-03-01 12:02:35.000000000 +0000
@@ -345,6 +345,11 @@
 PHP_CHECK_FUNC(gethostbyaddr, nsl)
 PHP_CHECK_FUNC(yp_get_default_domain, nsl)
 
+PHP_CHECK_FUNC(socket, network)
+PHP_CHECK_FUNC(socketpair, network)
+PHP_CHECK_FUNC(gethostname, network)
+PHP_CHECK_FUNC(gethostbyaddr, network)
+
 PHP_CHECK_FUNC(dlopen, dl)
 if test "$ac_cv_func_dlopen" = "yes"; then
   AC_DEFINE(HAVE_LIBDL, 1, [ ])
diff -urN php-5.2.9/ext/standard/microtime.c php-5.2.9-haiku/ext/standard/microtime.c
--- php-5.2.9/ext/standard/microtime.c	2009-03-01 12:12:26.000000000 +0000
+++ php-5.2.9-haiku/ext/standard/microtime.c	2009-03-01 12:06:02.000000000 +0000
@@ -132,7 +132,7 @@
 	array_init(return_value);
 #define PHP_RUSAGE_PARA(a) \
 		add_assoc_long(return_value, #a, usg.a)
-#if !defined( _OSD_POSIX) && !defined(__BEOS__) /* BS2000 has only a few fields in the rusage struct */
+#if !defined( _OSD_POSIX) && !defined(__BEOS__) && !defined(__HAIKU__)/* BS2000 has only a few fields in the rusage struct */
 	PHP_RUSAGE_PARA(ru_oublock);
 	PHP_RUSAGE_PARA(ru_inblock);
 	PHP_RUSAGE_PARA(ru_msgsnd);
