From fc9ee0911f7ea1881eed295c0cca7470b1c35a14 Mon Sep 17 00:00:00 2001
From: Luke <noryb009@gmail.com>
Date: Mon, 25 Nov 2013 21:02:41 +0000
Subject: [PATCH] COMMON -> SYSTEM

---
 src/ExeBox.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ExeBox.cpp b/src/ExeBox.cpp
index b4ee544..b3c3bb2 100644
--- a/src/ExeBox.cpp
+++ b/src/ExeBox.cpp
@@ -212,7 +212,7 @@ ExeBox::InitializeAutocompletion(void)
 			sPathData->AddItem(new entry_ref(ref));
 	}
 	
-	find_directory(B_COMMON_BIN_DIRECTORY,&path);
+	find_directory(B_SYSTEM_BIN_DIRECTORY,&path);
 	dir.SetTo(path.Path());
 	dir.Rewind();
 	while (dir.GetNextRef(&ref) == B_OK)
@@ -271,7 +271,7 @@ ExeBox::QueryThread(void *data)
 	{
 		if ( ref.directory == B_USER_ADDONS_DIRECTORY 
 			|| ref.directory == B_BEOS_ADDONS_DIRECTORY
-			|| ref.directory == B_COMMON_ADDONS_DIRECTORY)
+			|| ref.directory == B_SYSTEM_ADDONS_DIRECTORY)
 			continue;
 
 		if (ref.directory == -1)
@@ -341,7 +341,7 @@ ExeBox::UpdateThread(void *data)
 		queryList.AddItem(new entry_ref(ref));
 	}
 	
-	find_directory(B_COMMON_BIN_DIRECTORY,&path);
+	find_directory(B_SYSTEM_BIN_DIRECTORY,&path);
 	dir.SetTo(path.Path());
 	dir.Rewind();
 
-- 
1.8.3.4

