diff -ur texinfo-4.13.orig/gnulib/lib/config.charset texinfo-4.13/gnulib/lib/config.charset
--- texinfo-4.13.orig/gnulib/lib/config.charset	2007-12-25 23:57:18.000000000 +0000
+++ texinfo-4.13/gnulib/lib/config.charset	2008-11-09 15:48:51.000000000 +0000
@@ -478,8 +478,8 @@
 	# space nevertheless.
 	echo "* UTF-8"
 	;;
-    beos*)
-	# BeOS has a single locale, and it has UTF-8 encoding.
+    beos*|haiku*)
+	# BeOS has a single locale, and it has UTF-8 encoding. Same for Haiku.
 	echo "* UTF-8"
 	;;
     msdosdjgpp*)
diff -ur texinfo-4.13.orig/system.h texinfo-4.13/system.h
--- texinfo-4.13.orig/system.h	2008-07-26 22:54:54.000000000 +0000
+++ texinfo-4.13/system.h	2008-11-09 15:43:55.000000000 +0000
@@ -151,7 +151,7 @@
 # define va_end(args)
 #endif
 
-#if O_BINARY
+#if O_BINARY && !defined(__HAIKU__)
 # ifdef HAVE_IO_H
 #  include <io.h>
 # endif
