diff -Naur burnitnow-beta5-svn/const.h burnitnow-beta5-svn-patch/const.h
--- burnitnow-beta5-svn/const.h	2012-11-20 19:48:42.045350912 +0100
+++ burnitnow-beta5-svn-patch/const.h	2012-11-20 17:57:36.000000000 +0100
@@ -64,7 +64,7 @@
 const uint32 AUDIO_NOFIX = 'AuNF';
 
 const uint32 MISC_DAO = 'MiDA';
-const uint32 MISC_BURNPROOF = 'MiBP';
+const uint32 MISC_BURNFREE = 'MiBP';
 
 
 const uint32 BURN_WITH_CDRECORD = 'BWCR';
diff -Naur burnitnow-beta5-svn/jpWindow.cpp burnitnow-beta5-svn-patch/jpWindow.cpp
--- burnitnow-beta5-svn/jpWindow.cpp	2012-11-20 19:48:42.007602176 +0100
+++ burnitnow-beta5-svn-patch/jpWindow.cpp	2012-11-20 18:03:09.000000000 +0100
@@ -65,7 +65,7 @@
 char* BURN_DIR;
 BPath CDRTOOLS_DIR;
 
-char BURNPROOF[30]; // driveropts = burnproof
+char BURNFREE[30]; // driveropts=burnfree
 char PAD[10]; // -pad (audio)
 char DAO[10]; // -dao
 char NOFIX[10]; // -nofix (audio)
@@ -714,10 +714,10 @@
 	else
 		strcpy(DAO, " ");
 
-	if (fBurnItPrefs->FindString("BURNPROOF", &tr) == B_OK)
-		strcpy(BURNPROOF, tr);
+	if (fBurnItPrefs->FindString("BURNFREE", &tr) == B_OK)
+		strcpy(BURNFREE, tr);
 	else
-		strcpy(BURNPROOF, " ");
+		strcpy(BURNFREE, " ");
 
 	if (fBurnItPrefs->FindString("NOFIX", &tr) == B_OK)
 		strcpy(NOFIX, tr);
@@ -1398,11 +1398,11 @@
 				strcpy(DAO, " ");
 			break;
 
-		case MISC_BURNPROOF:
-			if (fPrefsView->fBurnProofCheckBox->Value() == 1)
-				strcpy(BURNPROOF, "driveropts = burnproof");
+		case MISC_BURNFREE:
+			if (fPrefsView->fBurnFreeCheckBox->Value() == 1)
+				strcpy(BURNFREE, "driveropts=burnfree");
 			else
-				strcpy(BURNPROOF, " ");
+				strcpy(BURNFREE, " ");
 			break;
 
 		case MENU_FILE_ABOUT:
@@ -1643,7 +1643,7 @@
 	fBurnItPrefs->SetString("DATA_STRING", DATA_STRING);
 	fBurnItPrefs->SetString("PAD", PAD);
 	fBurnItPrefs->SetString("DAO", DAO);
-	fBurnItPrefs->SetString("BURNPROOF", BURNPROOF);
+	fBurnItPrefs->SetString("BURNFREE", BURNFREE);
 	fBurnItPrefs->SetString("SWAB", SWAB);
 	fBurnItPrefs->SetString("NOFIX", NOFIX);
 	fBurnItPrefs->SetString("PREEMP", PREEMP);
@@ -1812,7 +1812,7 @@
 				commandstr << BURN_DIR << '"' << " | " << CDRTOOLS_DIR.Path();
 				commandstr << "/cdrecord dev=" << fBurnDevice->scsiid;
 				commandstr << " speed=" << BURN_SPD ;
-				commandstr << " "  << BURNPROOF;
+				commandstr << " "  << BURNFREE;
 				commandstr << "tsize" << tsize << " "; 
 				commandstr << DAO << " -data ";
 				commandstr << DUMMYMODE ;
@@ -1822,7 +1822,7 @@
 				commandstr.SetTo(CDRTOOLS_DIR.Path());
 				commandstr << "/";
 				commandstr << "cdrecord dev=" << fBurnDevice->scsiid << " speed=" << BURN_SPD ;
-				commandstr << " " << BURNPROOF << " " << DAO;
+				commandstr << " " << BURNFREE << " " << DAO;
 				commandstr << " -data " << DUMMYMODE ;
 				commandstr << " " << EJECT ;
 				commandstr << " -v" ;	// aw 110910
@@ -1847,7 +1847,7 @@
 			commandstr << "//";
 			commandstr << "cdrecord dev=" << fBurnDevice->scsiid;
 			commandstr << " speed=" << BURN_SPD; 
-			commandstr << " " << BURNPROOF ;
+			commandstr << " " << BURNFREE ;
 			commandstr << " " << DAO; 
 			commandstr << " " << PAD;
 			commandstr << " " << PREEMP ;
@@ -1880,7 +1880,7 @@
 			 	commandstr << '"' << VOL_NAME << '"' << BURN_DIR << '"' << " |";
 				commandstr << CDRTOOLS_DIR.Path() << "/cdrecord dev=" ;
 			 	commandstr <<  fBurnDevice->scsiid << "speed=" << BURN_SPD ;
-			 	commandstr << " " << BURNPROOF;
+			 	commandstr << " " << BURNFREE;
 				commandstr << " tsize=" << tsize ;
 			 	commandstr << " " << DAO << " " << DUMMYMODE ;
 			 	commandstr << " " << EJECT << " " << PAD ;
@@ -1893,7 +1893,7 @@
 				commandstr << "//";
 				commandstr << "cdrecord dev=" << fBurnDevice->scsiid;
 				commandstr << " speed=" << BURN_SPD ;
-			 	commandstr << " " << BURNPROOF << " " << DAO << " " << PAD ;
+			 	commandstr << " " << BURNFREE << " " << DAO << " " << PAD ;
 			 	commandstr << " " << PREEMP << " " << SWAB ;
 			 	commandstr << " " << NOFIX ;
 			 	commandstr << " " << DUMMYMODE ;
diff -Naur burnitnow-beta5-svn/PrefsView.cpp burnitnow-beta5-svn-patch/PrefsView.cpp
--- burnitnow-beta5-svn/PrefsView.cpp	2012-11-20 19:48:42.031981568 +0100
+++ burnitnow-beta5-svn-patch/PrefsView.cpp	2012-11-20 18:02:46.000000000 +0100
@@ -16,7 +16,7 @@
 
 
 extern char DAO[10];
-extern char BURNPROOF[30];
+extern char BURNFREE[30];
 
 
 PrefsView::PrefsView(BRect size)
@@ -59,9 +59,9 @@
 
 	r.top += 15;
 	r.bottom += 15;
-	fBurnProofCheckBox = new BCheckBox(r, "burnproof", "BurnProof (read help)", new BMessage(MISC_BURNPROOF));
-	if (!strncmp(BURNPROOF, "driveropts = burnproof", 22))
-		fBurnProofCheckBox->SetValue(B_CONTROL_ON);
+	fBurnFreeCheckBox = new BCheckBox(r, "burnfree", "BurnFree (read help)", new BMessage(MISC_BURNFREE));
+	if (!strncmp(BURNFREE, "driveropts=burnfree", 22))
+		fBurnFreeCheckBox->SetValue(B_CONTROL_ON);
 
-	miscOptBox->AddChild(fBurnProofCheckBox);
+	miscOptBox->AddChild(fBurnFreeCheckBox);
 }
diff -Naur burnitnow-beta5-svn/PrefsView.h burnitnow-beta5-svn-patch/PrefsView.h
--- burnitnow-beta5-svn/PrefsView.h	2012-11-20 19:48:42.015990784 +0100
+++ burnitnow-beta5-svn-patch/PrefsView.h	2012-11-20 17:57:08.000000000 +0100
@@ -15,7 +15,7 @@
 	PrefsView(BRect size);
 
 	BMenu* fRecordersMenu;
-	BCheckBox* fDAOCheckBox, *fBurnProofCheckBox;
+	BCheckBox* fDAOCheckBox, *fBurnFreeCheckBox;
 };
 
 
