diff -urN vorbis-tools-1.4.0/vcut/vcut.c vorbis-tools-1.4.0-gcc2/vcut/vcut.c
--- vorbis-tools-1.4.0/vcut/vcut.c	2010-03-26 07:07:07.022020096 +0000
+++ vorbis-tools-1.4.0-gcc2/vcut/vcut.c	2010-06-05 23:51:34.494927872 +0000
@@ -233,6 +233,7 @@
  * Returns 0 for success, or -1 on failure. */
 static int open_output_stream(vcut_state *s)
 {
+	int rv;
 	if(!s->out && !s->drop_output)
 	{
 		if(open_output_file(s, s->output_filename)!=0)
@@ -240,7 +241,7 @@
 	}
 
 	/* ogg_stream_init should only fail if stream_out is null */
-	int rv = ogg_stream_init(&s->stream_out, ++s->serial_out);
+	rv = ogg_stream_init(&s->stream_out, ++s->serial_out);
 	assert(rv == 0);
 
 	s->output_stream_open = 1;
