From 493df5691a40ee6d9a8b29653090281424b67187 Mon Sep 17 00:00:00 2001
From: Ken Mays <kmays2000@gmail.com>
Date: Tue, 26 Jan 2021 14:58:40 +0000
Subject: Fix pointcloud Partio::ParticleIndex for Haiku x86


diff --git a/src/liboslexec/pointcloud.cpp b/src/liboslexec/pointcloud.cpp
index c0a0818..81c65e3 100644
--- a/src/liboslexec/pointcloud.cpp
+++ b/src/liboslexec/pointcloud.cpp
@@ -383,8 +383,8 @@ RendererServices::pointcloud_get (ShaderGlobals *sg,
         count = maxn;
     }
 
-    static_assert (sizeof(size_t) == sizeof(Partio::ParticleIndex),
-            "Only will work if Partio ParticleIndex is the size of a size_t");
+//    static_assert (sizeof(size_t) == sizeof(Partio::ParticleIndex),
+//            "Only will work if Partio ParticleIndex is the size of a size_t");
     // FIXME -- if anybody cares about an architecture in which that is not
     // the case, we can easily allocate local space to retrieve the indices,
     // then copy them back to the caller's indices.
-- 
2.30.0
