diff -ur orig/neon-0.29.6/Makefile.in neon-0.29.6/Makefile.in
--- orig/neon-0.29.6/Makefile.in	2009-09-04 09:36:26.062914560 +0200
+++ neon-0.29.6/Makefile.in	2013-04-30 19:33:32.347340800 +0200
@@ -22,7 +22,7 @@
 man3dir = $(mandir)/man3
 datarootdir = @datarootdir@
 datadir = @datadir@
-docdir = $(datadir)/doc/neon-@NEON_VERSION@
+docdir = @docdir@
 includedir = @includedir@
 neonincludes = $(includedir)/neon
 pkgconfigdir = $(libdir)/pkgconfig
