diff --git a/src/SDL/i_system.c b/src/SDL/i_system.c
index bf8cbce..b7ea0fb 100644
--- a/src/SDL/i_system.c
+++ b/src/SDL/i_system.c
@@ -396,7 +396,7 @@ const char* I_GetTempDir(void)
 // cph - V.Aguilar (5/30/99) suggested return ~/.lxdoom/, creating
 //  if non-existant
 // cph 2006/07/23 - give prboom+ its own dir
-static const char prboom_dir[] = {"/.prboom-plus"}; // Mead rem extra slash 8/21/03
+static const char prboom_dir[] = {"/config/settings/prboom-plus"}; // Mead rem extra slash 8/21/03
 
 const char *I_DoomExeDir(void)
 {
