From 2952d33eb44c0415576161fb8ff141e842658f84 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Wed, 22 Jul 2026 18:09:40 +0200
Subject: [PATCH] Fix build for r2-6.1.9 (r2abi >= 125)

---
 c/r2dec-plugin.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/c/r2dec-plugin.c b/c/r2dec-plugin.c
index 20d231c7..6c11c673 100644
--- a/c/r2dec-plugin.c
+++ b/c/r2dec-plugin.c
@@ -142,8 +142,11 @@ static void usage(const RCore* const core) {
 		"pddi", "",                   "generate issue data",
 		NULL
 	};
-
+#if R2_ABIVERSION >= 125
+	r_cons_cmd_help(core->cons, help);
+#else
 	r_cons_cmd_help(core->cons, help, core->print->flags & R_PRINT_FLAGS_COLOR);
+#endif
 }
 
 static void _cmd_pdd(RCore *core, const char *input) {
