diff --git a/data/core/start.lua b/data/core/start.lua
index 46f3cbe..32956f5 100644
--- a/data/core/start.lua
+++ b/data/core/start.lua
@@ -10,7 +10,7 @@ if MACOS_RESOURCES then
   DATADIR = MACOS_RESOURCES
 else
   local prefix = EXEDIR:match("^(.+)[/\\]bin$")
-  DATADIR = prefix and (prefix .. PATHSEP .. 'share' .. PATHSEP .. 'lite-xl') or (EXEDIR .. PATHSEP .. 'data')
+  DATADIR = os.getenv('XDG_DATA_DIRS'):match('^.-:(.-)$') .. '/lite-xl'
 end
 USERDIR = (system.get_file_info(EXEDIR .. PATHSEP .. 'user') and (EXEDIR .. PATHSEP .. 'user'))
        or os.getenv("LITE_USERDIR")
diff --git a/src/main.c b/src/main.c
index 4642a85..6433370 100644
--- a/src/main.c
+++ b/src/main.c
@@ -214,7 +214,7 @@ init_lua:
     "  HOME = os.getenv('" LITE_OS_HOME "')\n"
     "  local exedir = match(EXEFILE, '^(.*)" LITE_PATHSEP_PATTERN LITE_NONPATHSEP_PATTERN "$')\n"
     "  local prefix = match(exedir, '^(.*)" LITE_PATHSEP_PATTERN "bin$')\n"
-    "  dofile((MACOS_RESOURCES or (prefix and prefix .. '/share/lite-xl' or exedir .. '/data')) .. '/core/start.lua')\n"
+    "  dofile(os.getenv('XDG_DATA_DIRS'):match('^.-:(.-)$') .. '/lite-xl/core/start.lua')\n"
     "  core = require(os.getenv('LITE_XL_RUNTIME') or 'core')\n"
     "  core.init()\n"
     "  core.run()\n"
