diff -urN otp_src_R14B/lib/test_server/src/configure.in otp_src_R14B-haiku/lib/test_server/src/configure.in
--- otp_src_R14B/lib/test_server/src/configure.in	2010-09-13 17:00:22.037748736 +0000
+++ otp_src_R14B-haiku/lib/test_server/src/configure.in	2010-10-04 13:15:31.969932800 +0000
@@ -279,6 +279,10 @@
     LIBS="$LIBS -lsocket -lnsl"
     AC_CHECK_FUNC(accept, erl_checkNsl=0, [LIBS=$tk_oldLibs])
 fi
+
+# Haiku uses -lnetwork instead of -lsocket
+AC_CHECK_LIB(network, socket, LIBS="$LIBS -lnetwork", LIBS="$LIBS")
+
 AC_CHECK_FUNC(gethostbyname, , AC_CHECK_LIB(nsl, main, [LIBS="$LIBS -lnsl"]))
 
 dnl Checks for library functions.
