diff -urN otp_src_R14B/lib/test_server/src/configure.in otp_src_R14B-haiku/lib/test_server/src/configure.in
--- otp_src_R14B/lib/test_server/src/configure.in	2010-09-13 17:00:22.037748736 +0000
+++ otp_src_R14B-haiku/lib/test_server/src/configure.in	2010-10-04 13:15:31.969932800 +0000
@@ -279,6 +279,10 @@
     LIBS="$LIBS -lsocket -lnsl"
     AC_CHECK_FUNC(accept, erl_checkNsl=0, [LIBS=$tk_oldLibs])
 fi
+
+# Haiku uses -lnetwork instead of -lsocket
+AC_CHECK_LIB(network, socket, LIBS="$LIBS -lnetwork", LIBS="$LIBS")
+
 AC_CHECK_FUNC(gethostbyname, , AC_CHECK_LIB(nsl, main, [LIBS="$LIBS -lnsl"]))
 
 dnl Checks for library functions.
diff -urN otp_src_R14B/erts/emulator/drivers/common/inet_drv.c otp_src_R14B-haiku/erts/emulator/drivers/common/inet_drv.c
--- otp_src_R14B/erts/emulator/drivers/common/inet_drv.c	2010-02-19 13:03:42.000000000 -0500
+++ otp_src_R14B-haiku/erts/emulator/drivers/common/inet_drv.c	2010-10-08 15:13:14.000000000 -0400
@@ -621,6 +621,7 @@
 #define INET_IFF_NPOINTTOPOINT   0x0800
 /* #define INET_IFF_NRUNNING     0x1000 */
 /* #define INET_IFF_NMULTICAST   0x2000 */
+#define IFF_RUNNING 		0x0001
 
 /* Flags for "sctp_sndrcvinfo". Used in a bitmask -- must be powers of 2:
 ** INET_REQ_SETOPTS:SCTP_OPT_DEFAULT_SEND_PARAM
diff -urN otp_src_R14B/erts/emulator/sys/unix/sys.c otp_src_R14B-haiku/erts/emulator/sys/unix/sys.c
--- otp_src_R14B/erts/emulator/sys/unix/sys.c	2010-02-19 13:03:16.000000000 -0500
+++ otp_src_R14B-haiku/erts/emulator/sys/unix/sys.c	2010-10-08 15:13:14.000000000 -0400
@@ -709,7 +709,7 @@
 	if (nice_val > 39) {
 	    nice_val = 39;
 	}
-	erts_silence_warn_unused_result(nice(nice_val));
+	erts_silence_warn_unused_result(nice_val);
     }
     
     envsz = sizeof(env);
diff -urN otp_src_R14B/lib/appmon/SKIP otp_src_R14B-haiku/lib/appmon/SKIP
--- otp_src_R14B/lib/appmon/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/appmon/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/asn1/SKIP otp_src_R14B-haiku/lib/asn1/SKIP
--- otp_src_R14B/lib/asn1/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/asn1/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/common_test/SKIP otp_src_R14B-haiku/lib/common_test/SKIP
--- otp_src_R14B/lib/common_test/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/common_test/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosEvent/SKIP otp_src_R14B-haiku/lib/cosEvent/SKIP
--- otp_src_R14B/lib/cosEvent/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosEvent/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosEventDomain/SKIP otp_src_R14B-haiku/lib/cosEventDomain/SKIP
--- otp_src_R14B/lib/cosEventDomain/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosEventDomain/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosFileTransfer/SKIP otp_src_R14B-haiku/lib/cosFileTransfer/SKIP
--- otp_src_R14B/lib/cosFileTransfer/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosFileTransfer/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosNotification/SKIP otp_src_R14B-haiku/lib/cosNotification/SKIP
--- otp_src_R14B/lib/cosNotification/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosNotification/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosProperty/SKIP otp_src_R14B-haiku/lib/cosProperty/SKIP
--- otp_src_R14B/lib/cosProperty/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosProperty/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosTime/SKIP otp_src_R14B-haiku/lib/cosTime/SKIP
--- otp_src_R14B/lib/cosTime/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosTime/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/cosTransactions/SKIP otp_src_R14B-haiku/lib/cosTransactions/SKIP
--- otp_src_R14B/lib/cosTransactions/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/cosTransactions/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/crypto/SKIP otp_src_R14B-haiku/lib/crypto/SKIP
--- otp_src_R14B/lib/crypto/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/crypto/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/debugger/SKIP otp_src_R14B-haiku/lib/debugger/SKIP
--- otp_src_R14B/lib/debugger/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/debugger/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/dialyzer/SKIP otp_src_R14B-haiku/lib/dialyzer/SKIP
--- otp_src_R14B/lib/dialyzer/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/dialyzer/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/docbuilder/SKIP otp_src_R14B-haiku/lib/docbuilder/SKIP
--- otp_src_R14B/lib/docbuilder/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/docbuilder/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/edoc/SKIP otp_src_R14B-haiku/lib/edoc/SKIP
--- otp_src_R14B/lib/edoc/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/edoc/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/erl_docgen/SKIP otp_src_R14B-haiku/lib/erl_docgen/SKIP
--- otp_src_R14B/lib/erl_docgen/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/erl_docgen/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/erl_interface/SKIP otp_src_R14B-haiku/lib/erl_interface/SKIP
--- otp_src_R14B/lib/erl_interface/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/erl_interface/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/et/SKIP otp_src_R14B-haiku/lib/et/SKIP
--- otp_src_R14B/lib/et/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/et/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/eunit/SKIP otp_src_R14B-haiku/lib/eunit/SKIP
--- otp_src_R14B/lib/eunit/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/eunit/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/gs/SKIP otp_src_R14B-haiku/lib/gs/SKIP
--- otp_src_R14B/lib/gs/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/gs/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/hipe/SKIP otp_src_R14B-haiku/lib/hipe/SKIP
--- otp_src_R14B/lib/hipe/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/hipe/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/ic/SKIP otp_src_R14B-haiku/lib/ic/SKIP
--- otp_src_R14B/lib/ic/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/ic/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/inets/SKIP otp_src_R14B-haiku/lib/inets/SKIP
--- otp_src_R14B/lib/inets/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/inets/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/inviso/SKIP otp_src_R14B-haiku/lib/inviso/SKIP
--- otp_src_R14B/lib/inviso/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/inviso/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/jinterface/SKIP otp_src_R14B-haiku/lib/jinterface/SKIP
--- otp_src_R14B/lib/jinterface/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/jinterface/SKIP	2010-10-08 15:13:14.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/megaco/SKIP otp_src_R14B-haiku/lib/megaco/SKIP
--- otp_src_R14B/lib/megaco/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/megaco/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/mnesia/SKIP otp_src_R14B-haiku/lib/mnesia/SKIP
--- otp_src_R14B/lib/mnesia/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/mnesia/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/observer/SKIP otp_src_R14B-haiku/lib/observer/SKIP
--- otp_src_R14B/lib/observer/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/observer/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/odbc/SKIP otp_src_R14B-haiku/lib/odbc/SKIP
--- otp_src_R14B/lib/odbc/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/odbc/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/orber/SKIP otp_src_R14B-haiku/lib/orber/SKIP
--- otp_src_R14B/lib/orber/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/orber/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/os_mon/SKIP otp_src_R14B-haiku/lib/os_mon/SKIP
--- otp_src_R14B/lib/os_mon/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/os_mon/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/otp_mibs/SKIP otp_src_R14B-haiku/lib/otp_mibs/SKIP
--- otp_src_R14B/lib/otp_mibs/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/otp_mibs/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/parsetools/SKIP otp_src_R14B-haiku/lib/parsetools/SKIP
--- otp_src_R14B/lib/parsetools/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/parsetools/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/percept/SKIP otp_src_R14B-haiku/lib/percept/SKIP
--- otp_src_R14B/lib/percept/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/percept/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/pman/SKIP otp_src_R14B-haiku/lib/pman/SKIP
--- otp_src_R14B/lib/pman/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/pman/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/public_key/SKIP otp_src_R14B-haiku/lib/public_key/SKIP
--- otp_src_R14B/lib/public_key/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/public_key/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/reltool/SKIP otp_src_R14B-haiku/lib/reltool/SKIP
--- otp_src_R14B/lib/reltool/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/reltool/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/runtime_tools/SKIP otp_src_R14B-haiku/lib/runtime_tools/SKIP
--- otp_src_R14B/lib/runtime_tools/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/runtime_tools/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/sasl/SKIP otp_src_R14B-haiku/lib/sasl/SKIP
--- otp_src_R14B/lib/sasl/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/sasl/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/snmp/SKIP otp_src_R14B-haiku/lib/snmp/SKIP
--- otp_src_R14B/lib/snmp/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/snmp/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/ssh/SKIP otp_src_R14B-haiku/lib/ssh/SKIP
--- otp_src_R14B/lib/ssh/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/ssh/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/ssl/SKIP otp_src_R14B-haiku/lib/ssl/SKIP
--- otp_src_R14B/lib/ssl/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/ssl/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/syntax_tools/SKIP otp_src_R14B-haiku/lib/syntax_tools/SKIP
--- otp_src_R14B/lib/syntax_tools/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/syntax_tools/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/test_server/SKIP otp_src_R14B-haiku/lib/test_server/SKIP
--- otp_src_R14B/lib/test_server/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/test_server/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/toolbar/SKIP otp_src_R14B-haiku/lib/toolbar/SKIP
--- otp_src_R14B/lib/toolbar/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/toolbar/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/tools/SKIP otp_src_R14B-haiku/lib/tools/SKIP
--- otp_src_R14B/lib/tools/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/tools/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/tv/SKIP otp_src_R14B-haiku/lib/tv/SKIP
--- otp_src_R14B/lib/tv/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/tv/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/typer/SKIP otp_src_R14B-haiku/lib/typer/SKIP
--- otp_src_R14B/lib/typer/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/typer/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/webtool/SKIP otp_src_R14B-haiku/lib/webtool/SKIP
--- otp_src_R14B/lib/webtool/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/webtool/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
diff -urN otp_src_R14B/lib/wx/SKIP otp_src_R14B-haiku/lib/wx/SKIP
--- otp_src_R14B/lib/wx/SKIP	1969-12-31 19:00:00.000000000 -0500
+++ otp_src_R14B-haiku/lib/wx/SKIP	2010-10-08 15:13:16.000000000 -0400
@@ -0,0 +1 @@
+Minimal installation
