diff --git a/Makefile.am b/Makefile.am
index 268b399..a8236c1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -328,16 +328,6 @@ $(srcdir)/doc/notes.txt: $(srcdir)/doc/notes.texi
 	  $(MAKEINFO) --no-headers $(MAKEINFOFLAGS) -o notes.txt notes.texi
 
 dist_man1_MANS		= $(srcdir)/doc/libtool.1 $(srcdir)/doc/libtoolize.1
-MAINTAINERCLEANFILES	+= $(dist_man1_MANS)
-update_mans = \
-  PATH=".$(PATH_SEPARATOR)$$PATH"; export PATH; \
-  $(HELP2MAN) --output=$@
-$(srcdir)/doc/libtool.1: $(srcdir)/$(auxdir)/ltmain.sh
-	$(update_mans) --help-option=--help-all libtool
-$(srcdir)/doc/libtoolize.1: $(srcdir)/libtoolize.in
-	$(update_mans) libtoolize
-
-
 ## ------------- ##
 ## Installation. ##
 ## ------------- ##
