--- a/src/common/Game.cpp	2020-02-26 21:09:52.337117184 +0800
+++ b/src/common/Game.cpp	2020-02-26 21:09:41.111411200 +0800
@@ -8,7 +8,7 @@
 #include <windows.h>
 #endif
 
-#if defined(unix) || defined(__unix__) || defined(__unix) || defined(__APPLE__)
+#if defined(unix) || defined(__unix__) || defined(__unix) || defined(__APPLE__) || defined(__HAIKU__)
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
--- a/src/common/path.cpp	2020-02-26 21:08:47.775684096 +0800
+++ b/src/common/path.cpp	2020-02-26 21:08:50.727187456 +0800
@@ -11,6 +11,8 @@
         #include <shlobj.h>
         #include "SDL_platform.h"
     #endif
+#elif __HAIKU__
+#include <FindDirectory.h>
 #else
 #include <stdlib.h>
 #endif
@@ -47,6 +49,12 @@
     result += "/supermariowar/";
     return result;
 
+#elif __HAIKU__
+    char path[B_PATH_NAME_LENGTH];
+    find_directory(B_USER_SETTINGS_DIRECTORY, 0, false, path, B_PATH_NAME_LENGTH);
+    std::string result(std::string(path) + "/supermariowar/");
+    return result;
+
 #else // catch-all for Linux-based systems
     std::string result(".smw/");
     char* folder = getenv("HOME");
