diff -urN nasm-2.03.01/float.h nasm-2.03.01-haiku/float.h
--- nasm-2.03.01/float.h	2008-06-17 04:31:16.000000000 +0000
+++ nasm-2.03.01-haiku/float.h	2008-08-14 11:37:33.000000000 +0000
@@ -10,11 +10,16 @@
 #ifndef NASM_FLOAT_H
 #define NASM_FLOAT_H
 
+#ifdef __HAIKU__
+  #include <stdint.h>
+  #include "nasm.h"
+#endif
+
 enum float_round {
     FLOAT_RC_NEAR,
     FLOAT_RC_ZERO,
     FLOAT_RC_DOWN,
-    FLOAT_RC_UP,
+    FLOAT_RC_UP
 };
 
 int float_const(const char *string, int sign, uint8_t *result, int bytes,
diff -urN nasm-2.03.01/nasm.c nasm-2.03.01-haiku/nasm.c
--- nasm-2.03.01/nasm.c	2008-06-17 04:31:16.000000000 +0000
+++ nasm-2.03.01-haiku/nasm.c	2008-08-14 09:51:50.000000000 +0000
@@ -87,7 +87,7 @@
 enum op_type {
     op_normal,                  /* Preprocess and assemble */
     op_preprocess,              /* Preprocess only */
-    op_depend,                  /* Generate dependencies */
+    op_depend                  /* Generate dependencies */
 };
 static enum op_type operating_mode;
 /* Dependency flags */
diff -urN nasm-2.03.01/nasm.h nasm-2.03.01-haiku/nasm.h
--- nasm-2.03.01/nasm.h	2008-06-17 04:31:16.000000000 +0000
+++ nasm-2.03.01-haiku/nasm.h	2008-08-14 09:49:45.000000000 +0000
@@ -74,7 +74,7 @@
     OUT_RESERVE,		/* Reserved bytes (RESB et al) */
     OUT_REL2ADR,		/* 2-byte relative address */
     OUT_REL4ADR,		/* 4-byte relative address */
-    OUT_REL8ADR,		/* 8-byte relative address */
+    OUT_REL8ADR		/* 8-byte relative address */
 };
 
 /*
@@ -176,7 +176,7 @@
     TOKEN_DBL_AND, TOKEN_DBL_OR, TOKEN_DBL_XOR, /* &&, || and ^^ */
     TOKEN_SEG, TOKEN_WRT,       /* SEG and WRT */
     TOKEN_FLOAT,                /* floating-point constant */
-    TOKEN_FLOATIZE,		/* __floatX__ */
+    TOKEN_FLOATIZE		/* __floatX__ */
 };
 
 enum floatize {
@@ -187,7 +187,7 @@
     FLOAT_80M,
     FLOAT_80E,
     FLOAT_128L,
-    FLOAT_128H,
+    FLOAT_128H
 };
 
 /*
diff -urN nasm-2.03.01/pptok.c nasm-2.03.01-haiku/pptok.c
--- nasm-2.03.01/pptok.c	2008-06-17 04:31:23.000000000 +0000
+++ nasm-2.03.01-haiku/pptok.c	2008-08-14 09:50:38.000000000 +0000
@@ -105,7 +105,7 @@
     "%strlen",
     "%substr",
     "%undef",
-    "%xdefine",
+    "%xdefine"
 };
 enum preproc_token pp_token_hash(const char *token)
 {
diff -urN nasm-2.03.01/pptok.h nasm-2.03.01-haiku/pptok.h
--- nasm-2.03.01/pptok.h	2008-06-17 04:31:23.000000000 +0000
+++ nasm-2.03.01-haiku/pptok.h	2008-08-14 09:51:17.000000000 +0000
@@ -93,7 +93,7 @@
     PPC_IFMACRO      =  14,
     PPC_IFNUM        =  16,
     PPC_IFSTR        =  18,
-    PPC_IFTOKEN      =  20,
+    PPC_IFTOKEN      =  20
 };
 
 #define PP_COND(x)     ((enum pp_conditional)((x) & 0x1e))
diff -urN nasm-2.03.01/quote.c nasm-2.03.01-haiku/quote.c
--- nasm-2.03.01/quote.c	2008-06-17 04:31:16.000000000 +0000
+++ nasm-2.03.01-haiku/quote.c	2008-08-14 11:57:46.000000000 +0000
@@ -197,7 +197,7 @@
 	st_backslash,
 	st_hex,
 	st_oct,
-	st_ucs,
+	st_ucs
     } state;
     int ndig = 0;
     int32_t nval = 0;
@@ -392,7 +392,7 @@
     char c;
     enum unq_state {
 	st_start,
-	st_backslash,
+	st_backslash
     } state;
 
     bq = str[0];
