diff -ur GUIlib-1.2.1/acinclude.m4 GUIlib-1.2.1-haiku/acinclude.m4
--- GUIlib-1.2.1/acinclude.m4	2006-05-01 07:39:58.000000000 +0000
+++ GUIlib-1.2.1-haiku/acinclude.m4	2008-06-24 09:01:16.000000000 +0000
@@ -910,10 +910,10 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
-  beos*)
+  beos* | haiku*)
     # On BeOS, this test takes a really really long time.
     # So we just punt and use a minimum line length of 8192.
-    lt_cv_sys_max_cmd_len=8192;
+    lt_cv_sys_max_cmd_len=4096;
     ;;
 
   amigaos*)
@@ -1470,7 +1470,7 @@
   finish_eval='for lib in `ls $libdir/*.ixlibrary 2>/dev/null`; do libname=`$echo "X$lib" | $Xsed -e '\''s%^.*/\([[^/]]*\)\.ixlibrary$%\1%'\''`; test $rm /sys/libs/${libname}_ixlibrary.a; $show "cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a"; cd /sys/libs && $LN_S $lib ${libname}_ixlibrary.a || exit 1; done'
   ;;
 
-beos*)
+beos* | haiku*)
   library_names_spec='${libname}${shared_ext}'
   dynamic_linker="$host_os ld.so"
   shlibpath_var=LIBRARY_PATH
@@ -2469,7 +2469,7 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-beos*)
+beos* | haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -2687,7 +2687,7 @@
 [AC_REQUIRE([AC_CANONICAL_HOST])dnl
 LIBM=
 case $host in
-*-*-beos* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
+*-*-beos* | *-*-haiku* | *-*-cygwin* | *-*-pw32* | *-*-darwin*)
   # These system don't have libm, or don't need it
   ;;
 *-ncr-sysv4.3*)
@@ -3178,7 +3178,7 @@
     fi
     ;;
 
-  beos*)
+  beos* | haiku*)
     if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
       _LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
       # Joseph Beckenbach <jrb3@best.com> says some releases of gcc
@@ -5032,7 +5032,7 @@
       # like `-m68040'.
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
     mingw* | os2* | pw32*)
@@ -5305,7 +5305,7 @@
       _LT_AC_TAGVAR(lt_prog_compiler_pic, $1)='-m68020 -resident32 -malways-restore-a4'
       ;;
 
-    beos* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
+    beos* | haiku* | cygwin* | irix5* | irix6* | nonstopux* | osf3* | osf4* | osf5*)
       # PIC is the default for these OSes.
       ;;
 
@@ -5672,7 +5672,7 @@
       _LT_AC_TAGVAR(ld_shlibs, $1)=no
       ;;
 
-    beos*)
+    beos* | haiku*)
       if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	# Joseph Beckenbach <jrb3@best.com> says some releases of gcc
