diff -up c-ares-1.10.0/ares.h.orig c-ares-1.10.0/ares.h
--- c-ares-1.10.0/ares.h.orig	2013-10-30 14:59:47.195297280 -0600
+++ c-ares-1.10.0/ares.h	2013-10-30 15:00:19.284950528 -0600
@@ -38,7 +38,7 @@
    require it! */
 #if defined(_AIX) || defined(__NOVELL_LIBC__) || defined(__NetBSD__) || \
     defined(__minix) || defined(__SYMBIAN32__) || defined(__INTEGRITY) || \
-    defined(ANDROID) || defined(__ANDROID__)
+    defined(ANDROID) || defined(__ANDROID__) || defined(__HAIKU__)
 #include <sys/select.h>
 #endif
 #if (defined(NETWARE) && !defined(__NOVELL_LIBC__))
diff -up c-ares-1.10.0/ares_private.h.orig c-ares-1.10.0/ares_private.h
--- c-ares-1.10.0/ares_private.h.orig	2013-02-13 06:58:04.034078720 -0700
+++ c-ares-1.10.0/ares_private.h	2013-10-30 14:59:30.440664064 -0600
@@ -65,6 +65,11 @@
 
 #define PATH_HOSTS             "InetDBase:Hosts"
 
+#elif defined(__HAIKU__)
+
+#define PATH_RESOLV_CONF        "/boot/system/settings/network/resolv.conf"
+#define PATH_HOSTS              "/boot/system/settings/network/hosts"
+
 #else
 
 #define PATH_RESOLV_CONF        "/etc/resolv.conf"
