diff -Naur gnutls-2.8.6/doc/scripts/sort2.pl gnutls-2.8.6-haiku/doc/scripts/sort2.pl
--- gnutls-2.8.6/doc/scripts/sort2.pl	2009-06-02 15:59:32.012058624 -0300
+++ gnutls-2.8.6-haiku/doc/scripts/sort2.pl	2010-05-14 23:31:26.698875904 -0300
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/bin/env perl
 
 sub key_of_record {
   local($record) = @_;
diff -Naur gnutls-2.8.6/gl/string.in.h gnutls-2.8.6-haiku/gl/string.in.h
--- gnutls-2.8.6/gl/string.in.h	2009-06-02 15:59:32.064749568 -0300
+++ gnutls-2.8.6-haiku/gl/string.in.h	2010-05-14 23:33:36.399245312 -0300
@@ -36,7 +36,7 @@
 # endif
 /* The attribute __pure__ was added in gcc 2.96.  */
 # if __GNUC__ < 2 || (__GNUC__ == 2 && __GNUC_MINOR__ < 96)
-#  define __pure__ /* empty */
+#  define __pure__ none /* empty */
 # endif
 #endif
 
