diff -urN grep-2.7/src/kwsearch.c grep-2.7-haiku/src/kwsearch.c
--- grep-2.7/src/kwsearch.c	2010-08-13 23:36:17.039059456 +0000
+++ grep-2.7-haiku/src/kwsearch.c	2010-09-22 12:39:11.677642240 +0000
@@ -111,8 +111,9 @@
           /* The match was a part of multibyte character, advance at least
              one byte to ensure no infinite loop happens.  */
           mbstate_t s;
+          size_t mb_len;
           memset (&s, 0, sizeof s);
-          size_t mb_len = mbrlen (mb_start, (buf + size) - (beg + offset), &s);
+          mb_len = mbrlen (mb_start, (buf + size) - (beg + offset), &s);
           if (mb_len == (size_t) -2)
             goto failure;
           beg = mb_start;
