diff -ru jpeg-8d/jpeglib.h jpeg-8d/jpeglib.h
--- jpeg-8d/jpeglib.h	2012-08-23 22:37:22.000000000 +0200
+++ jpeg-8d/jpeglib.h	2012-01-04 23:58:28.723311670 +0100
@@ -26,6 +26,9 @@
 #endif
 #include "jmorecfg.h"		/* seldom changed options */
 
+#ifdef __HAIKU__
+#include <setjmp.h>
+#endif
 
 #ifdef __cplusplus
 #ifndef DONT_USE_EXTERN_C
@@ -732,6 +733,10 @@
   const char * const * addon_message_table; /* Non-library errors */
   int first_addon_message;	/* code for first string in addon table */
   int last_addon_message;	/* code for last string in addon table */
+  
+#ifdef __HAIKU__
+  jmp_buf long_jump_buffer;
+#endif
 };
 
 
