--- a/fff	2019-01-28 21:36:41.029622272 +0000
+++ b/fff	2019-01-30 08:43:58.977797120 +0000
@@ -14,6 +14,11 @@
 
         haiku)
             opener="open"
+
+            [[ -z $FFF_TRASH_CMD ]] && {
+                FFF_TRASH_CMD="trash"
+                FFF_TRASH="$(finddir -v "$PWD" B_TRASH_DIRECTORY)"
+            }
         ;;
     esac
 }
@@ -376,8 +381,9 @@
 
     if [[ $FFF_TRASH_CMD ]]; then
         # Pass all but the last argument to the user's
-        # custom script.
-        "$FFF_TRASH_CMD" "${@:1:$#-1}"
+        # custom script. command is used to prevent this function
+        # from conflicting with commands named "trash".
+        command "$FFF_TRASH_CMD" "${@:1:$#-1}"
 
     else
         cd "$FFF_TRASH" && mv -f "$@"
@@ -849,6 +855,8 @@
 
         # Go to trash.
         "${FFF_KEY_GO_TRASH:=t}")
+            [[ $OSTYPE == haiku && $FFF_TRASH_CMD == trash ]] &&
+                FFF_TRASH="$(finddir -v "$PWD" B_TRASH_DIRECTORY)"
             open "$FFF_TRASH"
         ;;
 
