diff -urN a/fff b/fff
--- a/fff	2019-01-05 17:49:17.027787264 +0000
+++ b/fff	2019-01-09 06:22:50.749731840 +0000
@@ -41,6 +41,7 @@
     [[ -d $1/ ]] && { l=0; refresh; PWD="${1:-/}"; get_dir; f_print; }
     [[ -f $1  ]] && {
         [[ "$OSTYPE" == darwin* ]] && { oc=open; fa="bIL"; }
+        [[ "$OSTYPE" == haiku ]] && oc=open
         case "$(file "-${fa:-biL}" "$1")" in
             text/*|*x-empty*|*json*) "${EDITOR:-vi}" "$1"; printf '\e[?25l' ;;
             *) nohup "${FFF_OPENER:-${oc:-xdg-open}}" "$1" &>/dev/null & disown
@@ -100,6 +101,8 @@
 
 main() {
     shopt -s nocaseglob nullglob checkwinsize
+    [[ "$OSTYPE" == haiku ]] &&
+        : "${FFF_TRASH:=$(finddir B_USER_CACHE_DIRECTORY)/fff/trash/}"
     mkdir -p "${FFF_TRASH:=$HOME/.cache/fff/trash/}"
     pushd "$1" &>/dev/null ||:; refresh; get_dir; f_print
 
diff -urN a/README.md b/README.md
--- a/README.md	2019-01-09 06:16:34.507248640 +0000
+++ b/README.md	2019-01-09 06:16:39.964952064 +0000
@@ -14,7 +14,7 @@
 ## Dependencies
 
 - `bash 3.2+`
-- program handling (non-text): `xdg-utils` (*not needed on macos*)
+- program handling (non-text): `xdg-utils` (*not needed on macos or Haiku*)
 - file operations: `coreutils`
 - mime types: `file`
 
@@ -64,7 +64,7 @@
 p: paste/move
 c: clear file selections
 r: rename
-d: trash (~/.cache/fff/trash/)
+d: trash (~/config/cache/fff/trash/)
 
 q: exit
 ```
