diff -urN bzflag-2.0.14/configure.ac bzflag-2.0.14-haiku/configure.ac
--- bzflag-2.0.14/configure.ac	2010-02-14 19:35:21.006553600 +0000
+++ bzflag-2.0.14-haiku/configure.ac	2010-06-20 06:37:51.000000000 +0000
@@ -15,7 +15,7 @@
 AC_CONFIG_SRCDIR(src/bzflag/bzflag.cxx)
 AC_REVISION($Revision $)
 
-AC_CONFIG_AUX_DIR(misc)
+AC_CONFIG_AUX_DIR(m4)
 
 # sed magic to get version numbers from src/date/buildDate.cxx
 MAJOR_VERSION=`sed -e 's/#.*define.*BZ_MAJOR_VERSION[^0-9]*\(.*\)/\1/' -e t -e d < \$srcdir/src/date/buildDate.cxx`
@@ -293,6 +293,10 @@
     GLIBS=" -lmedia -lgame $GLIBS"
     LIBS="-lbe"
     ;;
+  haiku*)
+    GLIBS=" -lmedia -lgame $GLIBS"
+    LIBS="-lbe"
+    ;;
   macos|darwin*)
     ;;
   *)
@@ -303,9 +307,12 @@
 AC_CHECK_LIB(m, sqrtf)
 
 # for BeOS - old network stack don't have those libs ( move it in the case switch ?)
+AC_CHECK_LIB([network], [socket])
 AC_CHECK_LIB([socket], [socket])
 AC_CHECK_LIB([socket], [gethostent], [], AC_CHECK_LIB([bind], [gethostent]))
 
+
+
 # check for various IEEE and c99 standard math functions in libm
 AC_CHECK_LIB([m], [acosf], [AC_DEFINE([HAVE_ACOSF],[1],[libm includes acosf])])
 AC_CHECK_LIB([m], [asinf], [AC_DEFINE([HAVE_ASINF],[1],[libm includes asinf])])
@@ -328,7 +335,7 @@
 case $host_os in
   macos|darwin*)
     ;;
-  beos*)
+  beos*|haiku*)
     ;;
   *)
     if test x$no_x = x ; then
@@ -670,7 +677,7 @@
 	   FLAGS="FLAGS -march=mips2"
 	fi
 	;;
-     beos*)
+     beos*|haiku*)
 	AC_DEFINE(HAVE_DEFINED_TOLOWER, 1, [tolower and toupper are not functions])
 	;;
 esac
@@ -692,7 +699,7 @@
 	  FLAGS="$FLAGS -pg"
 	fi
 	;;
-     beos)
+     beos|haiku)
 	FLAGS="$FLAGS -Werror"
 	;;
     esac
@@ -720,7 +727,7 @@
 		FLAGS="$FLAGS -O2"
 	    fi
 	    ;;
-	beos)
+	beos|haiku)
 	    FLAGS="$FLAGS -O2 -ffast-math -fomit-frame-pointer"
 	    FLAGS="$FLAGS -fno-exceptions"
 	    ;;
diff -urN bzflag-2.0.14/include/bzares.h bzflag-2.0.14-haiku/include/bzares.h
--- bzflag-2.0.14/include/bzares.h	2008-04-12 07:40:02.045350912 +0000
+++ bzflag-2.0.14-haiku/include/bzares.h	2010-06-20 06:36:21.000000000 +0000
@@ -20,7 +20,7 @@
 
 #include <sys/types.h>
 
-#if defined(_AIX) || defined(NETWARE)
+#if defined(_AIX) || defined(NETWARE) || defined (__HAIKU__)
 /* HP-UX systems version 9, 10 and 11 lack sys/select.h and so does oldish
    libc5-based Linux systems. Only include it on system that are known to
    require it! */
diff -urN bzflag-2.0.14/src/common/TimeKeeper.cxx bzflag-2.0.14-haiku/src/common/TimeKeeper.cxx
--- bzflag-2.0.14/src/common/TimeKeeper.cxx	2010-02-13 01:09:52.038797312 +0000
+++ bzflag-2.0.14-haiku/src/common/TimeKeeper.cxx	2010-06-20 06:44:57.000000000 +0000
@@ -20,7 +20,7 @@
 #ifdef HAVE_UNISTD_H
 #  include <unistd.h>
 #endif
-#ifdef __BEOS__
+#if defined __BEOS__ || defined __HAIKU__ 
 #  include <OS.h>
 #endif
 #if !defined(_WIN32)
