From 365ad3f760d1a00c9bd02215de122b32099c16ef Mon Sep 17 00:00:00 2001
From: Gerasim Troeglazov <3dEyes@gmail.com>
Date: Tue, 8 May 2018 23:50:45 +1000
Subject: Disable python scripting plugin


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ef8100..a361806 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -710,7 +710,7 @@ ENDIF(TIFF_FOUND)
 
 #<< PYTHON
 #SET(PythonLibs_FIND_VERSION 2)
-FIND_PACKAGE(PythonLibs 2 REQUIRED)
+FIND_PACKAGE(PythonLibs 2)
 IF (PYTHON_LIBRARY)
 	MESSAGE("Python Library Found OK")
 	SET(HAVE_PYTHON 1)
diff --git a/scribus/CMakeLists.txt b/scribus/CMakeLists.txt
index 718b6ee..64e10b3 100644
--- a/scribus/CMakeLists.txt
+++ b/scribus/CMakeLists.txt
@@ -1216,7 +1216,7 @@ TARGET_LINK_LIBRARIES(${EXE_NAME}
 	${FREETYPE_LIBRARIES}
 	${FONTCONFIG_LIBRARIES}
 	${LIBXML2_LIBRARIES}
-	${PYTHON_LIBRARIES}
+#	${PYTHON_LIBRARIES}
 	${TIFF_LIBRARIES}
 	${JPEG_LIBRARIES}
 	${ZLIB_LIBRARIES}
diff --git a/scribus/plugins/CMakeLists.txt b/scribus/plugins/CMakeLists.txt
index 733acc1..3e3d444 100644
--- a/scribus/plugins/CMakeLists.txt
+++ b/scribus/plugins/CMakeLists.txt
@@ -11,12 +11,12 @@ IF(NOT WANT_SCRIPTER2)
   #scripter1
   IF(NOT WIN32)
     MESSAGE(STATUS "Building with Scripter 1")
-    ADD_SUBDIRECTORY(scriptplugin)
+#    ADD_SUBDIRECTORY(scriptplugin)
   ENDIF(NOT WIN32)
 ELSE(NOT WANT_SCRIPTER2)
   #scripter2
   MESSAGE(STATUS "Building with Scripter 2")
-  ADD_SUBDIRECTORY(scripter)
+#  ADD_SUBDIRECTORY(scripter)
 ENDIF(NOT WANT_SCRIPTER2)
 ADD_SUBDIRECTORY(short-words)
 ADD_SUBDIRECTORY(tools)
-- 
2.16.2

