diff -up mercurial-4.8.1/mercurial/ui.py.orig mercurial-4.8.1/mercurial/ui.py
--- mercurial-4.8.1/mercurial/ui.py.orig       2018-12-23 05:12:19.544473088 -0300
+++ mercurial-4.8.1/mercurial/ui.py   2018-12-23 05:10:23.565968896 -0300
@@ -1592,7 +1592,7 @@
             # avoid confusion.
             editor = 'E'
         else:
-            editor = 'vi'
+            editor = 'nano'
         return (encoding.environ.get("HGEDITOR") or
                 self.config("ui", "editor", editor))

