diff --git a/contrib/python-zstandard/setup_zstd.py b/contrib/python-zstandard/setup_zstd.py
index 26bc3c9..4acccb7 100644
--- a/contrib/python-zstandard/setup_zstd.py
+++ b/contrib/python-zstandard/setup_zstd.py
@@ -178,8 +178,8 @@ def get_c_extension(
         extra_args.append("-DZDICTLIB_VISIBILITY=")
         extra_args.append("-DZSTDERRORLIB_VISIBILITY=")
 
-        if compiler_type == "unix":
-            extra_args.append("-fvisibility=hidden")
+#        if compiler_type == "unix":
+#            extra_args.append("-fvisibility=hidden")
 
     if not system_zstd and support_legacy:
         extra_args.append("-DZSTD_LEGACY_SUPPORT=1")
diff --git a/mercurial/ui.py b/mercurial/ui.py
index 2c5d9cf..406e714 100644
--- a/mercurial/ui.py
+++ b/mercurial/ui.py
@@ -1921,7 +1921,7 @@ class ui(object):
             # while s/vi/vim/ doesn't.
             editor = b'vim'
         else:
-            editor = b'vi'
+            editor = b'nano'
         return encoding.environ.get(b"HGEDITOR") or self.config(
             b"ui", b"editor", editor
         )
