diff -ur mercurial-4.8.1-orig/contrib/python-zstandard/setup_zstd.py mercurial-4.8.1/contrib/python-zstandard/setup_zstd.py
--- mercurial-4.8.1-orig/contrib/python-zstandard/setup_zstd.py	2018-12-04 19:04:15.000000000 -0300
+++ mercurial-4.8.1/contrib/python-zstandard/setup_zstd.py	2019-01-05 01:29:50.432396926 -0300
@@ -159,8 +159,8 @@
         extra_args.append('-DZDICTLIB_VISIBILITY=')
         extra_args.append('-DZSTDERRORLIB_VISIBILITY=')
 
-        if compiler_type == 'unix':
-            extra_args.append('-fvisibility=hidden')
+#        if compiler_type == 'unix':
+#            extra_args.append('-fvisibility=hidden')
 
     if not system_zstd and support_legacy:
         extra_args.append('-DZSTD_LEGACY_SUPPORT=1')
diff -ur mercurial-4.8.1-orig/mercurial/cext/osutil.c mercurial-4.8.1/mercurial/cext/osutil.c
--- mercurial-4.8.1-orig/mercurial/cext/osutil.c	2018-12-04 19:04:15.000000000 -0300
+++ mercurial-4.8.1/mercurial/cext/osutil.c	2019-01-05 01:26:32.297876574 -0300
@@ -1159,9 +1159,9 @@
 {
 	int sig = 0;
 	int r;
+	sigset_t set;
 	if (!PyArg_ParseTuple(args, "i", &sig))
 		return NULL;
-	sigset_t set;
 	r = sigemptyset(&set);
 	if (r != 0)
 		return PyErr_SetFromErrno(PyExc_OSError);
diff -ur mercurial-4.8.1-orig/mercurial/ui.py mercurial-4.8.1/mercurial/ui.py
--- mercurial-4.8.1-orig/mercurial/ui.py	2018-12-04 19:04:15.000000000 -0300
+++ mercurial-4.8.1/mercurial/ui.py	2019-01-05 01:33:09.805735796 -0300
@@ -1592,7 +1592,7 @@
             # avoid confusion.
             editor = 'E'
         else:
-            editor = 'vi'
+            editor = 'nano'
         return (encoding.environ.get("HGEDITOR") or
                 self.config("ui", "editor", editor))
 
