diff --git a/mercurial/ui.py b/mercurial/ui.py
index 2c5d9cf..406e714 100644
--- a/mercurial/ui.py
+++ b/mercurial/ui.py
@@ -1921,7 +1921,7 @@ class ui(object):
             # while s/vi/vim/ doesn't.
             editor = b'vim'
         else:
-            editor = b'vi'
+            editor = b'nano'
         return encoding.environ.get(b"HGEDITOR") or self.config(
             b"ui", b"editor", editor
         )
