diff -up mercurial-1.9/mercurial/ui.py.orig mercurial-1.9/mercurial/ui.py
--- mercurial-1.9/mercurial/ui.py.orig	2011-07-01 12:54:23.057933824 -0600
+++ mercurial-1.9/mercurial/ui.py	2011-07-02 12:23:10.208666624 -0600
@@ -656,7 +656,7 @@ class ui(object):
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
-                os.environ.get("EDITOR", "vi"))
+                os.environ.get("EDITOR", "nano"))
 
     def progress(self, topic, pos, item="", unit="", total=None):
         '''show a progress message
