diff --git a/contrib/python-zstandard/setup_zstd.py b/contrib/python-zstandard/setup_zstd.py
index 23b6fa1..9cb9713 100644
--- a/contrib/python-zstandard/setup_zstd.py
+++ b/contrib/python-zstandard/setup_zstd.py
@@ -159,8 +159,8 @@ def get_c_extension(support_legacy=False, system_zstd=False, name='zstd',
         extra_args.append('-DZDICTLIB_VISIBILITY=')
         extra_args.append('-DZSTDERRORLIB_VISIBILITY=')
 
-        if compiler_type == 'unix':
-            extra_args.append('-fvisibility=hidden')
+        #if compiler_type == 'unix':
+        #    extra_args.append('-fvisibility=hidden')
 
     if not system_zstd and support_legacy:
         extra_args.append('-DZSTD_LEGACY_SUPPORT=1')
diff --git a/mercurial/ui.py b/mercurial/ui.py
index 63ecc01..b76dd92 100644
--- a/mercurial/ui.py
+++ b/mercurial/ui.py
@@ -1722,7 +1722,7 @@ class ui(object):
             # avoid confusion.
             editor = 'E'
         else:
-            editor = 'vi'
+            editor = 'nano'
         return (encoding.environ.get("HGEDITOR") or
                 self.config("ui", "editor", editor))
 
