diff -up mercurial-2.7.1/mercurial/ui.py.orig mercurial-2.7.1/mercurial/ui.py
--- mercurial-2.7.2/mercurial/ui.py.orig	2012-06-01 22:48:21.025690112 -0600
+++ mercurial-2.7.2/mercurial/ui.py	2012-06-04 17:15:16.317980672 -0600
@@ -693,7 +693,7 @@ class ui(object):
             # avoid confusion.
             editor = 'E'
         else:
-            editor = 'vi'
+            editor = 'nano'
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
