diff -up mercurial-1.5.1/mercurial/ui.py.orig mercurial-1.5.1/mercurial/ui.py
--- mercurial-1.5.1/mercurial/ui.py.orig	2010-04-30 23:51:42.741605376 -0600
+++ mercurial-1.5.1/mercurial/ui.py	2010-04-30 23:52:08.309854208 -0600
@@ -384,7 +384,7 @@ class ui(object):
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
-                os.environ.get("EDITOR", "vi"))
+                os.environ.get("EDITOR", "nano"))
 
     def progress(self, topic, pos, item="", unit="", total=None):
         '''show a progress message
