diff -up mercurial-2.1/mercurial/ui.py.orig mercurial-2.1/mercurial/ui.py
--- mercurial-2.1/mercurial/ui.py.orig	2012-02-01 13:23:54.044040192 -0700
+++ mercurial-2.1/mercurial/ui.py	2012-02-20 23:13:31.630980608 -0700
@@ -687,7 +687,7 @@ class ui(object):
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
-                os.environ.get("EDITOR", "vi"))
+                os.environ.get("EDITOR", "nano"))
 
     def progress(self, topic, pos, item="", unit="", total=None):
         '''show a progress message
