diff -up mercurial-2.3/mercurial/ui.py.orig mercurial-2.3/mercurial/ui.py
--- mercurial-2.3/mercurial/ui.py.orig	2012-08-02 00:49:31.022544384 -0600
+++ mercurial-2.3/mercurial/ui.py	2012-08-02 01:06:19.942407680 -0600
@@ -701,7 +701,7 @@ class ui(object):
             # avoid confusion.
             editor = 'E'
         else:
-            editor = 'vi'
+            editor = 'nano'
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
