diff -up mercurial-2.2.1/mercurial/ui.py.orig mercurial-2.2.1/mercurial/ui.py
--- mercurial-2.2.1/mercurial/ui.py.orig	2012-05-03 15:06:29.013369344 -0600
+++ mercurial-2.2.1/mercurial/ui.py	2012-05-03 20:21:26.126091264 -0600
@@ -693,7 +693,7 @@ class ui(object):
             # avoid confusion.
             editor = 'E'
         else:
-            editor = 'vi'
+            editor = 'nano'
         return (os.environ.get("HGEDITOR") or
                 self.config("ui", "editor") or
                 os.environ.get("VISUAL") or
