diff --git a/Box2D/CMakeLists.txt b/Box2D/CMakeLists.txt
index a8ad550..17b3069 100644
--- a/Box2D/CMakeLists.txt
+++ b/Box2D/CMakeLists.txt
@@ -26,7 +26,9 @@ if(BOX2D_BUILD_EXAMPLES)
 
   # Testbed and dependencies.
   find_package(OpenGL REQUIRED)
+  if(NOT HAIKU) # Haiku has glut, so no need for freeglut. Disable GLUI because it's not ported.
   add_subdirectory(freeglut)
   add_subdirectory(glui)
   add_subdirectory(Testbed)
+  endif(NOT HAIKU)
 endif(BOX2D_BUILD_EXAMPLES)
diff --git a/Box2D/Box2D/CMakeLists.txt b/Box2D/Box2D/CMakeLists.txt
--- a/Box2D/Box2D/CMakeLists.txt
+++ b/Box2D/Box2D/CMakeLists.txt
@@ -207,8 +207,8 @@ 
 	# install build system hooks for third-party apps
 	install(EXPORT Box2D-targets DESTINATION ${LIB_INSTALL_DIR}/Box2D)
 
-        set (BOX2D_INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/include)
+        set (BOX2D_INCLUDE_DIR ${CMAKE_INSTALL_PREFIX}/develop/headers )
-        set (BOX2D_INCLUDE_DIRS ${BOX2D_INCLUDE_DIR} )
+        set (BOX2D_INCLUDE_DIRS ${BOX2D_INCLUDE_DIR} )
-        set (BOX2D_LIBRARY_DIRS ${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR})
+        set (BOX2D_LIBRARY_DIRS ${CMAKE_INSTALL_PREFIX}/develop/lib )
         set (BOX2D_LIBRARY Box2D)
         set (BOX2D_LIBRARIES ${BOX2D_LIBRARY})
@@ -185,12 +185,12 @@
 if(BOX2D_INSTALL)
 	# install headers
-	install(FILES ${BOX2D_General_HDRS} DESTINATION include/Box2D)
+	install(FILES ${BOX2D_General_HDRS} DESTINATION develop/headers/Box2D)
-	install(FILES ${BOX2D_Collision_HDRS} DESTINATION include/Box2D/Collision)
+	install(FILES ${BOX2D_Collision_HDRS} DESTINATION develop/headers/Box2D/Collision)
-	install(FILES ${BOX2D_Shapes_HDRS} DESTINATION include/Box2D/Collision/Shapes)
+	install(FILES ${BOX2D_Shapes_HDRS} DESTINATION develop/headers/Box2D/Collision/Shapes)
-	install(FILES ${BOX2D_Common_HDRS} DESTINATION include/Box2D/Common)
+	install(FILES ${BOX2D_Common_HDRS} DESTINATION develop/headers/Box2D/Common)
-	install(FILES ${BOX2D_Dynamics_HDRS} DESTINATION include/Box2D/Dynamics)
+	install(FILES ${BOX2D_Dynamics_HDRS} DESTINATION develop/headers/Box2D/Dynamics)
-	install(FILES ${BOX2D_Contacts_HDRS} DESTINATION include/Box2D/Dynamics/Contacts)
+	install(FILES ${BOX2D_Contacts_HDRS} DESTINATION develop/headers/Box2D/Dynamics/Contacts)
-	install(FILES ${BOX2D_Joints_HDRS} DESTINATION include/Box2D/Dynamics/Joints)
+	install(FILES ${BOX2D_Joints_HDRS} DESTINATION develop/headers/Box2D/Dynamics/Joints)
-	install(FILES ${BOX2D_Rope_HDRS} DESTINATION include/Box2D/Rope)
+	install(FILES ${BOX2D_Rope_HDRS} DESTINATION develop/headers/Box2D/Rope)
 
 	# install libraries
