diff -Naur flobopuyo-0.20/Makefile flobopuyo-0.20-haiku/Makefile
--- flobopuyo-0.20/Makefile	2004-10-08 22:41:12.000000000 +0000
+++ flobopuyo-0.20-haiku/Makefile	2009-11-01 19:26:32.000000000 +0000
@@ -12,7 +12,7 @@
 DEBUG_MODE=false
 
 # Unix/Linux settings
-PREFIX=/usr/local
+PREFIX=/boot/common
 DATADIR=$(PREFIX)/share/games/flobopuyo
 INSTALL_BINDIR=$(DESTDIR)/$(PREFIX)/games
 INSTALL_DATADIR=$(DESTDIR)/$(DATADIR)
