diff -Naur libmpeg2-0.5.1/bootstrap libmpeg2-0.5.1-haiku/bootstrap
--- libmpeg2-0.5.1/bootstrap	2008-07-16 18:47:58.024641536 -0500
+++ libmpeg2-0.5.1-haiku/bootstrap	2013-11-26 21:51:26.839647232 -0600
@@ -47,17 +47,7 @@
 done
 
 if test "${amvers}" = "no" && automake --version > /dev/null 2>&1; then
-  amvers="`automake --version | sed -e '1s/[^0-9]*//' -e q`"
-  if expr "$amvers" "<" "1.5" > /dev/null 2>&1; then
-    amvers="no"
-  else
-    amvers=""
-  fi
-fi
-
-if test "$amvers" = "no"; then
-  echo "$0: you need automake version 1.5 or later"
-  exit 1
+  amvers=""
 fi
 
 # Check for autoconf
