diff -Naur tcl8.6b1/generic/tclPosixStr.c tcl8.6b1-haiku/generic/tclPosixStr.c
--- tcl8.6b1/generic/tclPosixStr.c	2008-04-28 00:21:32.028049408 +0200
+++ tcl8.6b1-haiku/generic/tclPosixStr.c	2011-06-11 21:57:53.249036800 +0200
@@ -941,7 +941,9 @@
     case SIGALRM: return "SIGALRM";
 #endif
 #ifdef SIGBUS
-    case SIGBUS: return "SIGBUS";
+#  if !defined(SIGSEGV) || (SIGBUS != SIGSEGV)
+ 	case SIGBUS: return "SIGBUS";
+#  endif
 #endif
 #ifdef SIGCHLD
     case SIGCHLD: return "SIGCHLD";
@@ -1072,7 +1074,9 @@
     case SIGALRM: return "alarm clock";
 #endif
 #ifdef SIGBUS
-    case SIGBUS: return "bus error";
+#  if !defined(SIGSEGV) || (SIGBUS != SIGSEGV)
+ 	case SIGBUS: return "bus error";
+#  endif
 #endif
 #ifdef SIGCHLD
     case SIGCHLD: return "child status changed";
diff -Naur tcl8.6b1/pkgs/tdbc1.0b1/tclconfig/tcl.m4 tcl8.6b1-haiku/pkgs/tdbc1.0b1/tclconfig/tcl.m4
--- tcl8.6b1/pkgs/tdbc1.0b1/tclconfig/tcl.m4	2008-12-22 18:45:06.004456448 +0100
+++ tcl8.6b1-haiku/pkgs/tdbc1.0b1/tclconfig/tcl.m4	2011-06-11 21:58:19.403177472 +0200
@@ -761,8 +761,15 @@
 			    # The space is needed
 			    THREADS_LIBS=" -pthread"
 			else
-			    TCL_THREADS=0
-			    AC_MSG_WARN([Do not know how to find pthread lib on your system - thread support disabled])
+				# Haiku
+	    		AC_CHECK_LIB(root,pthread_mutex_init,tcl_ok=yes,tcl_ok=no)
+				if test "$tcl_ok" = "yes"; then
+				    # The space is needed
+				    THREADS_LIBS=" -lroot"
+				else
+				    TCL_THREADS=0
+				    AC_MSG_WARN([Do not know how to find pthread lib on your system - thread support disabled])
+				fi
 			fi
 		    fi
 		fi
@@ -1459,6 +1466,15 @@
 	    CC_SEARCH_FLAGS=""
 	    LD_SEARCH_FLAGS=""
 	    ;;
+	Haiku*)
+	    SHLIB_CFLAGS="-fPIC"
+	    SHLIB_LD='${CC} -nostart'
+	    SHLIB_LD_LIBS='${LIBS}'
+	    SHLIB_SUFFIX=".so"
+	    DL_OBJS="tclLoadDl.o"
+	    DL_LIBS="-lroot"
+	    LIBS="$LIBS -lnetwork"
+	    ;;
 	HP-UX-*.11.*)
 	    # Use updated header definitions where possible
 	    AC_DEFINE(_XOPEN_SOURCE_EXTENDED, 1, [Do we want to use the XOPEN network library?])
diff -Naur tcl8.6b1/unix/tcl.m4 tcl8.6b1-haiku/unix/tcl.m4
--- tcl8.6b1/unix/tcl.m4	2008-12-22 18:45:06.034603008 +0100
+++ tcl8.6b1-haiku/unix/tcl.m4	2011-06-11 22:04:12.228327424 +0200
@@ -667,8 +667,15 @@
 			# The space is needed
 			THREADS_LIBS=" -pthread"
 		    else
-			TCL_THREADS=0
-			AC_MSG_WARN([Don't know how to find pthread lib on your system - you must disable thread support or edit the LIBS in the Makefile...])
+				# Haiku
+	    		AC_CHECK_LIB(root,pthread_mutex_init,tcl_ok=yes,tcl_ok=no)
+				if test "$tcl_ok" = "yes"; then
+				    # The space is needed
+				    THREADS_LIBS=" -lroot"
+				else
+				    TCL_THREADS=0
+					AC_MSG_WARN([Don't know how to find pthread lib on your system - you must disable thread support or edit the LIBS in the Makefile...])
+				fi
 		    fi
 		fi
 	    fi
@@ -1252,6 +1259,15 @@
 	    CC_SEARCH_FLAGS=""
 	    LD_SEARCH_FLAGS=""
 	    ;;
+	Haiku*)
+	    SHLIB_CFLAGS="-fPIC"
+	    SHLIB_LD='${CC} -nostart'
+	    SHLIB_LD_LIBS='${LIBS}'
+	    SHLIB_SUFFIX=".so"
+	    DL_OBJS="tclLoadDl.o"
+	    DL_LIBS="-lroot"
+	    LIBS="$LIBS -lnetwork"
+	    ;;
 	HP-UX-*.11.*)
 	    # Use updated header definitions where possible
 	    AC_DEFINE(_XOPEN_SOURCE_EXTENDED, 1, [Do we want to use the XOPEN network library?])
