diff -urN libcddb-1.3.0/config.h.in libcddb-1.3.0-haiku/config.h.in
--- libcddb-1.3.0/config.h.in	2006-10-15 02:08:27.000000000 -0700
+++ libcddb-1.3.0-haiku/config.h.in	2008-08-16 13:04:55.000000000 -0700
@@ -47,6 +47,9 @@
 /* Define to 1 if you have the `socket' library (-lsocket). */
 #undef HAVE_LIBSOCKET
 
+/* Define to 1 if you have the `network' library (-lnetwork). */
+#undef HAVE_LIBNETWORK
+
 /* Define to 1 if your system has a GNU libc compatible `malloc' function, and
    to 0 otherwise. */
 #undef HAVE_MALLOC
diff -urN libcddb-1.3.0/configure.ac libcddb-1.3.0-haiku/configure.ac
--- libcddb-1.3.0/configure.ac	2006-10-15 02:04:53.000000000 -0700
+++ libcddb-1.3.0-haiku/configure.ac	2008-08-16 13:12:07.000000000 -0700
@@ -73,6 +73,13 @@
          On MacOS, <cddb_net.h> needs this but that header doesn't
          include it.])
         ;;
+    *haiku*)
+        AC_DEFINE([HAIKU],1,[Define if compiling on Haiku system.])
+        AC_DEFINE([CDDB_NEED_UNISTD_H],1,
+        [Define if you have <unistd.h> and need it included.])
+        AC_DEFINE([CDDB_NEED_SYS_SOCKET_H],1,
+        [Define if you have <sys/socket.h> and need it included.])
+        ;;
 esac
 
 dnl Checks for programs.
@@ -87,6 +94,7 @@
 
 dnl Checks for libraries.
 AC_CHECK_LIB([socket], [connect])
+AC_CHECK_LIB([network], [connect])
 
 dnl Checks for header files.
 AC_HEADER_STDC
