diff -ur libxml2-2.8.0.orig/configure.in libxml2-2.8.0/configure.in
--- libxml2-2.8.0.orig/configure.in	2012-05-23 10:33:55.027787264 +0200
+++ libxml2-2.8.0/configure.in	2013-04-26 13:50:43.429391872 +0200
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.59)
 AC_INIT(entities.c)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])
 AC_CANONICAL_HOST
 
diff -ur libxml2-2.8.0.orig/Makefile.am libxml2-2.8.0/Makefile.am
--- libxml2-2.8.0.orig/Makefile.am	2012-05-23 10:56:18.030932992 +0200
+++ libxml2-2.8.0/Makefile.am	2013-04-26 13:50:43.432537600 +0200
@@ -1208,8 +1208,8 @@
 #
 # Install the tests program sources as examples 
 #
-BASE_DIR=$(datadir)/doc
-DOC_MODULE=libxml2-$(VERSION)
+BASE_DIR=$(docdir)
+DOC_MODULE=
 EXAMPLES_DIR=$(BASE_DIR)/$(DOC_MODULE)/examples
 
 install-data-local: 
