diff -urN libxml2-2.7.3/configure.in libxml2-2.7.3-haiku/configure.in
--- libxml2-2.7.3/configure.in	2009-01-19 06:27:55.000000000 +0000
+++ libxml2-2.7.3-haiku/configure.in	2009-01-19 06:24:30.000000000 +0000
@@ -473,7 +473,7 @@
 
 dnl Checks for inet libraries:
 AC_SEARCH_LIBS(gethostent, [nsl])
-AC_SEARCH_LIBS(setsockopt, [socket net])
+AC_SEARCH_LIBS(setsockopt, [socket net network])
 AC_SEARCH_LIBS(connect, [inet])
 
 dnl Determine what socket length (socklen_t) data type is
@@ -1259,7 +1259,7 @@
 case "$host" in
 	*mingw*) M_LIBS=""
 	;;
-	*beos*) M_LIBS=""
+	*beos*|*haiku*) M_LIBS=""
 	;;
 	*) M_LIBS="-lm"
 	;;
