--- blobwars-1.19/makefile.org	2014-08-11 23:06:33.176685056 +0200
+++ blobwars-1.19/makefile	2014-08-11 23:06:19.469762048 +0200
@@ -21,8 +21,8 @@
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\" -DMEDAL_SERVER_HOST=\"$(MEDAL_SERVER_HOST)\" -DMEDAL_SERVER_PORT=$(MEDAL_SERVER_PORT)
-CXXFLAGS += $(CFLAGS) -Werror
-LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz
+CXXFLAGS += $(CFLAGS)
+LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz -lintl
 
 OBJS += CAudio.o
 OBJS += CBoss.o
--- blobwars-1.19/src/main.cpp.org	2014-08-31 18:48:07.870842368 +0200
+++ blobwars-1.19/src/main.cpp	2014-08-31 18:45:04.260833280 +0200
@@ -17,6 +17,10 @@
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 */
+#ifdef __HAIKU__
+#include <unistd.h>
+#include <libgen.h>
+#endif
 
 #include "main.h"
 
@@ -103,6 +107,11 @@
 	debug(("Not Using PAK...\n"));
 	#endif
 
+	// For tracker (Thanks PulkoMandy)
+	#ifdef __HAIKU__
+	chdir(dirname(argv[0]));
+	#endif
+	
 	#if RELEASE
 	chdir(PAKLOCATION);
 	#endif
