diff -urN mercurial-1.0.2/mercurial/bdiff.c mercurial-1.0.2-haiku/mercurial/bdiff.c
--- mercurial-1.0.2/mercurial/bdiff.c	2008-08-13 22:11:47.000000000 +0000
+++ mercurial-1.0.2-haiku/mercurial/bdiff.c	2008-09-09 15:43:50.000000000 +0000
@@ -34,7 +34,7 @@
 }
 #else
 #include <sys/types.h>
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 #include <ByteOrder.h>
 #else
 #include <arpa/inet.h>
diff -urN mercurial-1.0.2/mercurial/mpatch.c mercurial-1.0.2-haiku/mercurial/mpatch.c
--- mercurial-1.0.2/mercurial/mpatch.c	2008-08-13 22:11:47.000000000 +0000
+++ mercurial-1.0.2-haiku/mercurial/mpatch.c	2008-09-09 15:44:29.000000000 +0000
@@ -52,7 +52,7 @@
 #else
 /* not windows */
 # include <sys/types.h>
-# ifdef __BEOS__
+# if defined __BEOS__ && !defined __HAIKU__
 #  include <ByteOrder.h>
 # else
 #  include <arpa/inet.h>
