Index: configure.in
===================================================================
--- configure.in	(revision 7)
+++ configure.in	(working copy)
@@ -880,11 +880,17 @@
     OS_RELEASE=
     OS_TEST="${target_cpu}"
     case "${target_os}" in
-        linux*)       OS_ARCH=Linux ;;
-        solaris*)     OS_ARCH=SunOS OS_RELEASE=5 ;;
-        mingw*)       OS_ARCH=WINNT ;;
-        wince*)       OS_ARCH=WINCE ;;
-        darwin*)      OS_ARCH=Darwin OS_TARGET=Darwin ;;
+        linux*)         OS_ARCH=Linux ;;
+        solaris*)       OS_ARCH=SunOS OS_RELEASE=5 ;;
+        mingw*)         OS_ARCH=WINNT ;;
+        wince*)         OS_ARCH=WINCE ;;
+        darwin*)        OS_ARCH=Darwin OS_TARGET=Darwin ;;   
+        beos* | haiku*) OS_ARCH=BeOS OS_TARGET=BeOS ;;
+dnl -------------------------------------------------------
+dnl Cross-compiling-haiku - This should probably be 
+dnl        beos*)        OS_ARCH=BeOS OS_TARGET=BeOS ;;
+dnl        haiku*)       OS_ARCH=Haiku OS_TARGET=Haiku ;;
+dnl -------------------------------------------------------
     esac
 else
     OS_TARGET=`uname -s`
@@ -1002,6 +1008,14 @@
 UnixWare)
     HOST_OS_ARCH=UNIXWARE
     ;;
+beos* | haiku*)
+    HOST_OS_ARCH=BeOS
+    ;;
+dnl -------------------------------------------------------
+dnl This should probably be 
+dnl        beos*)        HOST_OS_ARCH=BeOS ;;
+dnl        haiku*)       HOST_OS_ARCH=Haiku ;;
+dnl -------------------------------------------------------
 esac
 
 case "$OS_ARCH" in
@@ -1338,7 +1352,7 @@
 dnl System overrides of the defaults for host
 dnl ========================================================
 case "$host" in
-*-beos*)
+*-beos*|*-haiku*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_BEOS -DBeOS -DBEOS -D_POSIX_SOURCE -DNO_X11"
     HOST_NSPR_MDCPUCFG='\"md/_beos.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
@@ -1482,17 +1496,35 @@
     AC_DEFINE(NSCAP_DISABLE_DEBUG_PTR_TYPES)
     ;;
 
-*-beos*)
+*-beos*|*-haiku*)
     no_x=yes
-    MKSHLIB='$(CXX) $(CXXFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
     _PLATFORM_DEFAULT_TOOLKIT="beos"
+    TARGET_NSPR_MDCPUCFG='\"md/_beos.cfg\"'
+    TARGET_MD_ARCH=beos
+
+    DSO_PIC_CFLAGS=-fPIC
     DSO_LDOPTS='-nostart'
+    MKSHLIB='$(CXX) $(CXXFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
+    MKCSHLIB='$(CC) $(CFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -Wl,-h,$@ -o $@'
     TK_LIBS='-lbe -lroot'
     LIBS="$LIBS -lbe"
+
+    _HAIKU_BUILD=0
+
     if test "$COMPILE_ENVIRONMENT"; then
+        AC_CHECK_LIB(network,main,_HAIKU_BUILD=1)
+    fi
+    if test "$_HAIKU_BUILD" = 1; then
+      LIBS="$LIBS -lnetwork"
+      AC_DEFINE(XP_HAIKU)
+      AC_DEFINE(__BEOS__)
+      AC_DEFINE(BONE_VERSION)
+    else
         AC_CHECK_LIB(bind,main,LIBS="$LIBS -lbind")
         AC_CHECK_LIB(zeta,main,LIBS="$LIBS -lzeta")
+      AC_DEFINE(BEOS_ADDON_WORKAROUND)
     fi
+
     _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Wno-multichar"
     _WARNINGS_CXXFLAGS="${_WARNINGS_CXXFLAGS} -Wno-multichar"
     _MOZ_USE_RTTI=1
@@ -2692,8 +2724,10 @@
                        ])
         if test "$ac_cv_have_visibility_builtin_bug" = "no" -a \
                 "$ac_cv_have_visibility_class_bug" = "no"; then
-          VISIBILITY_FLAGS='-I$(DIST)/include/system_wrappers -include $(topsrcdir)/config/gcc_hidden.h'
-          WRAP_SYSTEM_INCLUDES=1
+          # HACK for Haiku GCC4 -- enforce '-fvisibility=hidden'
+          #VISIBILITY_FLAGS='-I$(DIST)/include/system_wrappers -include $(topsrcdir)/config/gcc_hidden.h'
+          #WRAP_SYSTEM_INCLUDES=1
+          VISIBILITY_FLAGS='-fvisibility=hidden'
         else
           VISIBILITY_FLAGS='-fvisibility=hidden'
         fi # have visibility pragma bug
@@ -2761,6 +2795,8 @@
 case $target in
 *-darwin*)
 	;;
+*-beos*|*-haiku*)
+	;;
 *)
 	AC_CHECK_LIB(dl, dlopen,
         AC_CHECK_HEADER(dlfcn.h, 
@@ -3125,7 +3161,7 @@
 fi
 
 case $target_os in
-    os2*|msvc*|mks*|cygwin*|mingw*|darwin*|wince*)
+    os2*|msvc*|mks*|cygwin*|mingw*|darwin*|wince*|beos*|haiku*)
         ;;
     *)
     
@@ -5242,7 +5278,7 @@
 dnl Image decoders
 dnl ========================================================
 case "$target_os" in
-cygwin*|mingw*|msvc*|mks*|os2*|wince*|beos*)
+cygwin*|mingw*|msvc*|mks*|os2*|wince*|beos*|haiku*)
     MOZ_IMG_DECODERS_DEFAULT="$MOZ_IMG_DECODERS_DEFAULT icon"
     ;;
 darwin*)
@@ -5689,7 +5725,7 @@
 
 MOZ_DEBUG_ENABLE_DEFS="-DDEBUG -D_DEBUG"
  case "${target_os}" in
-    beos*)
+    beos*|haiku*)
         MOZ_DEBUG_ENABLE_DEFS="$MOZ_DEBUG_ENABLE_DEFS -DDEBUG_${USER}"
         ;;
     msvc*|mks*|cygwin*|mingw*|os2*|wince*)
@@ -6725,7 +6761,7 @@
   os2*)
       MOZ_LIBART_LIBS='-lmoz_art -lm'
       ;;
-  beos*)
+  beos*|haiku*)
       MOZ_LIBART_LIBS='-lmoz_art_lgpl -lroot -lbe'
       ;;
   *)
@@ -7377,6 +7413,9 @@
 elif test "$OS_ARCH" = "BeOS"; then
   AC_DEFINE(XP_BEOS)
   MOZ_MOVEMAIL=1
+elif test "$OS_ARCH" = "Haiku"; then
+  AC_DEFINE(XP_BEOS)
+  MOZ_MOVEMAIL=1	
 elif test "$OS_ARCH" = "Darwin"; then
   AC_DEFINE(XP_UNIX)
   AC_DEFINE(UNIX_ASYNC_DNS)
Index: directory/c-sdk/configure.in
===================================================================
--- directory/c-sdk/configure.in	(revision 7)
+++ directory/c-sdk/configure.in	(working copy)
@@ -360,7 +360,7 @@
     AC_DEFINE(DEBUG)
     DEFINES="$DEFINES -UNDEBUG"
     case "${target_os}" in
-    beos*)
+    beos*|haiku*)
         DEFINES="$DEFINES -DDEBUG_${USER}"
         ;;
     msvc*|mks*|cygwin*|mingw*|os2*)
@@ -563,10 +563,16 @@
     OS_RELEASE=
     OS_TEST="${target_cpu}"
     case "${target_os}" in
-        linux*)       OS_ARCH=Linux ;;
-        solaris*)     OS_ARCH=SunOS OS_RELEASE=5 ;;
-        mingw*)     OS_ARCH=WINNT ;;
-        darwin*)    OS_ARCH=Darwin ;;
+        linux*)         OS_ARCH=Linux ;;
+        solaris*)       OS_ARCH=SunOS OS_RELEASE=5 ;;
+        mingw*)         OS_ARCH=WINNT ;;
+        darwin*)        OS_ARCH=Darwin ;;   
+        beos* | haiku*) OS_ARCH=BeOS OS_TARGET=BeOS ;;
+dnl -------------------------------------------------------
+dnl Cross-compiling-haiku - This should probably be 
+dnl        beos*)        OS_ARCH=BeOS OS_TARGET=BeOS ;;
+dnl        haiku*)       OS_ARCH=Haiku OS_TARGET=Haiku ;;
+dnl -------------------------------------------------------
     esac
 else
     OS_ARCH=`uname -s | sed -e 's|/|_|g'`
@@ -733,7 +739,7 @@
     fi
     WIN_TOP_SRC=`cygpath -a -w $srcdir | sed -e 's|\\\\|/|g'`
     ;;
-*-beos*)
+*-beos*|*-haiku*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_BEOS -DBeOS -DBEOS -D_POSIX_SOURCE"
     ;;
 *os2*)
@@ -812,24 +818,40 @@
     RESOLVE_LINK_SYMBOLS=1
     ;;
         
-*-beos*)
+*-beos*|*-haiku*)
     AC_DEFINE(XP_BEOS)
     AC_DEFINE(BeOS)
     AC_DEFINE(BEOS)
     AC_DEFINE(_POSIX_SOURCE)
+    DSO_CFLAGS=-fPIC
     DSO_LDOPTS=-nostart
+    
     MDCPUCFG_H=_beos.cfg
     USE_BTHREADS=1
     PR_MD_ARCH_DIR=beos
     RESOLVE_LINK_SYMBOLS=1
+	
+	_HAIKU_BUILD=0
+	AC_CHECK_LIB(network,main,_HAIKU_BUILD=1)
+	if test "$_HAIKU_BUILD" = 1; then
+      OS_LIBS="$OS_LIBS -lnetwork"
+      AC_DEFINE(XP_HAIKU)
+      AC_DEFINE(BONE_VERSION)
+    else
+      AC_CHECK_LIB(bind, gethostbyaddr, [OS_LIBS="$OS_LIBS -lbind -lsocket"])
+    fi 
+
     case "${target_cpu}" in
     i*86)
-        _OPTIMIZE_FLAGS=-O2
-        _DEBUG_FLAGS='-gdwarf-2 -O0'
-        MKSHLIB='$(CCC) $(DSO_LDOPTS) -o $@'
-        AC_CHECK_LIB(bind, gethostbyaddr, [OS_LIBS="$OS_LIBS -lbind -lsocket"])
+        AC_DEFINE(i586)
+        CPU_ARCH=i586
+        _OPTIMIZE_FLAGS=-O3
+        _DEBUG_FLAGS='-gdwarf-2 -O0 -fno-inline'
+        MKSHLIB='$(CXX) $(DSO_LDOPTS) -o $@'
         ;;
     powerpc)
+        AC_DEFINE(ppc)
+        CPU_ARCH=ppc
         CC=mwcc
         CCC=mwcc
         LD=mwld
@@ -2095,7 +2117,7 @@
 	  fi])
 
 case "$target" in
-*-beos*)
+*-beos*|*-haiku*)
     AC_ARG_WITH(bthreads,
     [  --with-bthreads         Use system bthreads library as thread subsystem (BeOS only)],
     [	if test "$withval" = "yes"; then
Index: nsprpub/configure.in
===================================================================
--- nsprpub/configure.in	(revision 7)
+++ nsprpub/configure.in	(working copy)
@@ -381,7 +381,7 @@
     DEFINES="$DEFINES -UNDEBUG"
 
     case "${target_os}" in
-    beos*)
+    beos*|haiku*)
         DEFINES="$DEFINES -DDEBUG_${USER}"
         ;;
     msvc*|mks*|cygwin*|mingw*|os2*)
@@ -652,6 +652,12 @@
         solaris*)     OS_ARCH=SunOS OS_RELEASE=5 ;;
         mingw*)       OS_ARCH=WINNT ;;
         darwin*)      OS_ARCH=Darwin ;;
+        beos*|haiku*) OS_ARCH=BeOS OS_TARGET=BeOS OS_RELEASE=R5 ;;
+dnl -------------------------------------------------------
+dnl Cross-compiling-haiku - This should probably be 
+dnl        beos*)        OS_ARCH=BeOS OS_TARGET=BeOS OS_RELEASE=R5 ;;
+dnl        haiku*)       OS_ARCH=Haiku OS_TARGET=Haiku OS_RELEASE=R1 ;;
+dnl -------------------------------------------------------
     esac
 else
     OS_ARCH=`uname -s | sed -e 's|/|_|g'`
@@ -823,7 +829,7 @@
         CYGWIN_WRAPPER='sh $(topsrcdir)/build/cygwin-wrapper'
     fi
     ;;
-*-beos*)
+*-beos*|*-haiku*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_BEOS -DBeOS -DBEOS -D_POSIX_SOURCE"
     ;;
 *os2*)
@@ -898,24 +904,40 @@
     RESOLVE_LINK_SYMBOLS=1
     ;;
         
-*-beos*)
+*-beos*|*-haiku*)
     AC_DEFINE(XP_BEOS)
     AC_DEFINE(BeOS)
     AC_DEFINE(BEOS)
     AC_DEFINE(_POSIX_SOURCE)
-    DSO_LDOPTS=-nostart
+    DSO_CFLAGS=-fPIC
+    DSO_LDOPTS='-nostart'
+
     MDCPUCFG_H=_beos.cfg
     USE_BTHREADS=1
     PR_MD_ARCH_DIR=beos
     RESOLVE_LINK_SYMBOLS=1
+	
+	_HAIKU_BUILD=0
+	AC_CHECK_LIB(network,main,_HAIKU_BUILD=1)
+	if test "$_HAIKU_BUILD" = 1; then
+      OS_LIBS="$OS_LIBS -lnetwork"
+      AC_DEFINE(XP_HAIKU)
+      AC_DEFINE(BONE_VERSION)
+    else
+      AC_CHECK_LIB(bind, gethostbyaddr, [OS_LIBS="$OS_LIBS -lbind -lsocket"])
+    fi
+	
     case "${target_cpu}" in
     i*86)
-        _OPTIMIZE_FLAGS=-O2
-        _DEBUG_FLAGS='-gdwarf-2 -O0'
-        MKSHLIB='$(CCC) $(DSO_LDOPTS) -o $@'
-        AC_CHECK_LIB(bind, gethostbyaddr, [OS_LIBS="$OS_LIBS -lbind -lsocket"])
+        AC_DEFINE(i586)
+        CPU_ARCH=i586
+        _OPTIMIZE_FLAGS=-O3
+        _DEBUG_FLAGS='-gdwarf-2 -O0 -fno-inline'
+        MKSHLIB='$(CXX) $(DSO_LDOPTS) -o $@'
         ;;
     powerpc)
+        AC_DEFINE(ppc)
+        CPU_ARCH=ppc
         CC=mwcc
         CCC=mwcc
         LD=mwld
@@ -2245,11 +2267,13 @@
 
 dnl We don't want anything to link with libdl even if it's present on OS X, 
 dnl since it's not used and not part of the default installation.
-dnl The same goes for BeOS.
+dnl The same goes for BeOS and Haiku.
 
 case $target in
-*-darwin*|*-beos*)
+*-darwin*)
     ;;
+*-beos*|*-haiku*)
+    ;;
 *)
     AC_CHECK_LIB(dl, dlopen,
         AC_CHECK_HEADER(dlfcn.h,
@@ -2438,7 +2462,7 @@
 	  fi])
 
 case "$target" in
-*-beos*)
+*-beos*|*-haiku*)
     AC_ARG_WITH(bthreads,
     [  --with-bthreads         Use system bthreads library as thread subsystem
                           (BeOS only)],
