Index: config/config.mk
===================================================================
--- config/config.mk	(revision 7)
+++ config/config.mk	(working copy)
@@ -110,9 +110,6 @@
 LD		:= $(CC)
 endif
 endif
-ifeq ($(OS_ARCH),BeOS)
-BEOS_ADDON_WORKAROUND	= 1
-endif
 
 #
 # Strip off the excessively long version numbers on these platforms,
Index: config/rules.mk
===================================================================
--- config/rules.mk	(revision 7)
+++ config/rules.mk	(working copy)
@@ -517,7 +517,7 @@
 #
 # BeOS: add -Bsymbolic flag for components
 # 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter BeOS Haiku,$(OS_ARCH)))
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS += -Wl,-Bsymbolic
 endif
@@ -827,7 +827,7 @@
 ifdef MOZ_POST_PROGRAM_COMMAND
        $(MOZ_POST_PROGRAM_COMMAND) $@
 endif
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 ifdef BEOS_PROGRAM_RESOURCE
        xres -o $@ $(BEOS_PROGRAM_RESOURCE)
        mimeset $@
Index: directory/c-sdk/build.mk
===================================================================
--- directory/c-sdk/build.mk	(revision 7)
+++ directory/c-sdk/build.mk	(working copy)
@@ -442,7 +442,7 @@
 SO_FILES_TO_REMOVE=so_locations
 endif
 
-ifneq (,$(filter BeOS Darwin NetBSD,$(OS_ARCH)))
+ifneq (,$(filter BeOS Haiku Darwin NetBSD,$(OS_ARCH)))
 LINK_DLL	= $(MKSHLIB) $(OBJS)
 endif
 
Index: security/coreconf/config.mk
===================================================================
--- security/coreconf/config.mk	(revision 7)
+++ security/coreconf/config.mk	(working copy)
@@ -68,8 +68,12 @@
 ifeq (,$(filter-out $(TARGET_OSES),$(OS_TARGET)))
 include $(CORE_DEPTH)/coreconf/$(OS_TARGET).mk
 else
+ifeq ($(OS_ARCH), Haiku)
+include $(CORE_DEPTH)/coreconf/BeOS.mk
+else
 include $(CORE_DEPTH)/coreconf/$(OS_TARGET)$(OS_RELEASE).mk
 endif
+endif
 
 #######################################################################
 # [4.0] Master "Core Components" source and release <platform> tags   #
Index: security/nss/lib/ssl/config.mk
===================================================================
--- security/nss/lib/ssl/config.mk	(revision 7)
+++ security/nss/lib/ssl/config.mk	(working copy)
@@ -111,7 +111,7 @@
 	-lnspr4 \
 	$(NULL)
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_SHARED_LIBS += -lbe
 endif
 
Index: toolkit/mozapps/installer/package-name.mk
===================================================================
--- toolkit/mozapps/installer/package-name.mk	(revision 7)
+++ toolkit/mozapps/installer/package-name.mk	(working copy)
@@ -59,6 +59,9 @@
 ifeq ($(OS_ARCH),OS2)
 MOZ_PKG_PLATFORM := os2
 endif
+ifeq ($(OS_ARCH), Haiku)
+MOZ_PKG_PLATFORM := Haiku
+endif
 ifeq ($(OS_ARCH), BeOS)
 ifeq (,$(filter-out 6.%, $(OS_RELEASE)))
 MOZ_PKG_PLATFORM := Zeta
Index: toolkit/mozapps/installer/packager.mk
===================================================================
--- toolkit/mozapps/installer/packager.mk	(revision 7)
+++ toolkit/mozapps/installer/packager.mk	(working copy)
@@ -48,7 +48,7 @@
 ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
 MOZ_PKG_FORMAT  = DMG
 else
-ifeq (,$(filter-out OS2 WINNT BeOS, $(OS_ARCH)))
+ifeq (,$(filter-out OS2 WINNT BeOS Haiku, $(OS_ARCH)))
 MOZ_PKG_FORMAT  = ZIP
 ifeq ($(OS_ARCH),OS2)
 INSTALLER_DIR   = os2
@@ -251,7 +251,7 @@
 ifeq ($(OS_ARCH),IRIX)
 STRIP_FLAGS	= -f
 endif
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 STRIP_FLAGS	= -g
 PLATFORM_EXCLUDE_LIST = ! -name "*.stub" ! -name "$(MOZ_PKG_APPNAME)-bin"
 endif
