Index: mozilla/browser/app/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/browser/app/Makefile.in,v
retrieving revision 1.85.2.12
diff -u -8 -p -w -r1.85.2.12 Makefile.in
--- mozilla/browser/app/Makefile.in	12 May 2008 18:25:50 -0000	1.85.2.12
+++ mozilla/browser/app/Makefile.in	8 Jul 2009 21:49:35 -0000
@@ -146,17 +150,20 @@ endif
 ifdef BUILD_STATIC_LIBS
 RCFLAGS += -DMOZ_STATIC_BUILD
 endif
 ifdef DEBUG
 RCFLAGS += -DDEBUG
 endif
 endif
 
 ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
+ifeq ($(OS_ARCH),Haiku)
+BEOS_PROGRAM_RDEF = $(srcdir)/apprunner-beos.rdef
+endif
 BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
 endif
 endif
 
 ifeq ($(OS_ARCH),OS2)
 RESFILE=splashos2.res
Index: mozilla/calendar/sunbird/app/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/calendar/sunbird/app/Makefile.in,v
retrieving revision 1.21.2.23
diff -u -8 -p -w -r1.21.2.23 Makefile.in
--- mozilla/calendar/sunbird/app/Makefile.in	4 May 2008 01:32:07 -0000	1.21.2.23
+++ mozilla/calendar/sunbird/app/Makefile.in	8 Jul 2009 21:50:01 -0000
@@ -189,18 +193,21 @@ endif
 ifdef BUILD_STATIC_LIBS
 RCFLAGS += -DMOZ_STATIC_BUILD
 endif
 ifdef DEBUG
 RCFLAGS += -DDEBUG
 endif
 endif
 
 ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
+ifeq ($(OS_ARCH),Haiku)
+BEOS_PROGRAM_RDEF = $(srcdir)/apprunner-beos.rdef
+endif
 BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
 endif
 endif
 
 ifeq ($(OS_ARCH),OS2)
 RESFILE=splashos2.res
 RCFLAGS += -DMOZ_SUNBIRD
Index: mozilla/config/rules.mk
===================================================================
RCS file: /cvsroot/mozilla/config/rules.mk,v
retrieving revision 3.487.2.9
diff -u -8 -p -w -r3.487.2.9 rules.mk
--- mozilla/config/rules.mk	18 Dec 2007 21:33:08 -0000	3.487.2.9
+++ mozilla/config/rules.mk	8 Jul 2009 21:50:19 -0000
@@ -822,17 +826,20 @@ endif # OS2
 endif # WINCE
 
 ifdef ENABLE_STRIP
 	$(STRIP) $@
 endif
 ifdef MOZ_POST_PROGRAM_COMMAND
 	$(MOZ_POST_PROGRAM_COMMAND) $@
 endif
 ifneq (,$(filter BeOS Haiku,$(OS_ARCH)))
+ifdef BEOS_PROGRAM_RDEF
+	rc -o $(BEOS_PROGRAM_RESOURCE) $(BEOS_PROGRAM_RDEF)
+endif
 ifdef BEOS_PROGRAM_RESOURCE
 	xres -o $@ $(BEOS_PROGRAM_RESOURCE)
 	mimeset $@
 endif
 endif # BeOS
 
 $(HOST_PROGRAM): $(HOST_PROGOBJS) $(HOST_LIBS_DEPS) $(HOST_EXTRA_DEPS) Makefile Makefile.in
 ifeq ($(MOZ_OS2_TOOLS),VACPP)
Index: mozilla/mail/app/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/mail/app/Makefile.in,v
retrieving revision 1.46.2.10
diff -u -8 -p -w -r1.46.2.10 Makefile.in
--- mozilla/mail/app/Makefile.in	22 Oct 2008 21:04:53 -0000	1.46.2.10
+++ mozilla/mail/app/Makefile.in	8 Jul 2009 21:52:18 -0000
@@ -142,17 +146,20 @@ RCFLAGS += -DMOZ_THUNDERBIRD -I$(srcdir)
 else
 RCFLAGS += -DMOZ_THUNDERBIRD --include-dir $(srcdir)
 endif
 ifdef BUILD_STATIC_LIBS
 RCFLAGS += -DMOZ_STATIC_BUILD
 endif
 endif
 
 ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
+ifeq ($(OS_ARCH),Haiku)
+BEOS_PROGRAM_RDEF = $(srcdir)/apprunner-beos.rdef
+endif
 BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
 endif
 endif
 
 ifeq ($(OS_ARCH),OS2)
 RESFILE=splashos2.res
Index: mozilla/xpfe/bootstrap/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/xpfe/bootstrap/Attic/Makefile.in,v
retrieving revision 1.283.2.6
diff -u -8 -p -w -r1.283.2.6 Makefile.in
--- mozilla/xpfe/bootstrap/Makefile.in	29 Sep 2007 20:15:11 -0000	1.283.2.6
+++ mozilla/xpfe/bootstrap/Makefile.in	8 Jul 2009 21:54:55 -0000
@@ -215,16 +219,19 @@ ifdef GC_LEAK_DETECTOR
 LIBS += -lboehm
 endif
 
 ifdef WRAP_SYSTEM_INCLUDES
 DEFINES		+= -DWRAP_SYSTEM_INCLUDES
 endif
 
 ifeq ($(MOZ_WIDGET_TOOLKIT),beos)
+ifeq ($(OS_ARCH),Haiku)
+BEOS_PROGRAM_RDEF = $(srcdir)/apprunner-beos.rdef
+endif
 BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 CPPSRCS += nsNativeAppSupportBeOS.cpp
 CPPSRCS += nsNativeAppSupportBase.cpp
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
 endif
 endif
 
