Index: security/nss/cmd/lib/secutil.h
===================================================================
--- security/nss/cmd/lib/secutil.h	(revision 7)
+++ security/nss/cmd/lib/secutil.h	(working copy)
@@ -450,7 +450,7 @@
 
 void printflags(char *trusts, unsigned int flags);
 
-#if !defined(XP_UNIX) && !defined(XP_OS2)
+#if !defined(XP_UNIX) && !defined(XP_OS2) && !defined(XP_BEOS)
 extern int ffs(unsigned int i);
 #endif
 
Index: security/nss/lib/dev/ckhelper.c
===================================================================
--- security/nss/lib/dev/ckhelper.c	(revision 7)
+++ security/nss/lib/dev/ckhelper.c	(working copy)
@@ -230,13 +230,13 @@
   PRStatus *rvStatus
 )
 {
-    CK_BBOOL bool;
+    CK_BBOOL boolval;
     CK_ATTRIBUTE_PTR attr;
     CK_ATTRIBUTE atemplate = { 0, NULL, 0 };
     CK_RV ckrv;
     void *epv = nssSlot_GetCryptokiEPV(slot);
     attr = &atemplate;
-    NSS_CK_SET_ATTRIBUTE_VAR(attr, attribute, bool);
+    NSS_CK_SET_ATTRIBUTE_VAR(attr, attribute, boolval);
     nssSession_EnterMonitor(session);
     ckrv = CKAPI(epv)->C_GetAttributeValue(session->handle, object, 
                                            &atemplate, 1);
@@ -246,7 +246,7 @@
 	return PR_FALSE;
     }
     *rvStatus = PR_SUCCESS;
-    return (PRBool)(bool == CK_TRUE);
+    return (PRBool)(boolval == CK_TRUE);
 }
 
 NSS_IMPLEMENT PRStatus 
Index: security/nss/lib/freebl/unix_rand.c
===================================================================
--- security/nss/lib/freebl/unix_rand.c	(revision 7)
+++ security/nss/lib/freebl/unix_rand.c	(working copy)
@@ -669,7 +669,7 @@
 #endif /* VMS */
 
 #ifdef BEOS
-#include <be/kernel/OS.h>
+#include <kernel/OS.h>
 
 static size_t
 GetHighResClock(void *buf, size_t maxbytes)
