diff -urN cairo-1.8.4/configure.ac cairo-1.8.4-haiku/configure.ac
--- cairo-1.8.4/configure.ac	2008-12-19 00:44:28.000000000 -0800
+++ cairo-1.8.4-haiku/configure.ac	2008-12-19 00:29:17.000000000 -0800
@@ -142,7 +142,7 @@
 
 CAIRO_ENABLE_SURFACE_BACKEND(beos, BeOS/Zeta, no, [
   case "$host" in
-    *-*-beos)
+    *-*-beos| *-*-haiku*)
       beos_LIBS=""
       dnl Add libbe and libzeta if available
       AC_CHECK_LIB(be,main,beos_LIBS="$beos_LIBS -lbe")
