diff -up make-3.82/arscan.c.orig make-3.82/arscan.c
--- make-3.82/arscan.c.orig	2010-09-20 13:33:16.283901952 -0600
+++ make-3.82/arscan.c	2010-09-20 13:34:10.106168320 -0600
@@ -253,7 +253,7 @@ ar_scan (char *archive, long int (*funct
 #endif
 
 #ifndef WINDOWS32
-# ifndef __BEOS__
+# if !defined(__BEOS__) && !defined(__HAIKU__)
 #  include <ar.h>
 # else
    /* BeOS 5 doesn't have <ar.h> but has archives in the same format
