diff -ruw flex-2.5.35/configure.in flex-2.5.35-haiku/configure.in
--- flex-2.5.35/configure.in	2008-02-09 16:46:28.027525120 +0100
+++ flex-2.5.35-haiku/configure.in	2011-03-05 18:05:15.222822400 +0100
@@ -45,12 +45,12 @@
 AC_PROG_AWK
 AC_PROG_INSTALL
 
-AC_PATH_PROG(BISON, bison,bison)
+AC_CHECK_PROG(BISON, bison,bison)
 AC_PATH_PROG(HELP2MAN, help2man, help2man)
 
 # Check for a GNU m4 that supports --prefix-builtins
 
-AC_PATH_PROGS(M4, gm4 gnum4 m4, m4)
+AC_CHECK_PROGS(M4, gm4 gnum4 m4, m4)
 
 if test x"$M4" != x; then
   AC_MSG_CHECKING([for GNU m4])
diff -ruw flex-2.5.35/tests/test-pthread/Makefile.am flex-2.5.35-haiku/tests/test-pthread/Makefile.am
--- flex-2.5.35/tests/test-pthread/Makefile.am	2006-11-09 21:14:49.001310720 +0100
+++ flex-2.5.35-haiku/tests/test-pthread/Makefile.am	2011-03-05 18:04:53.553648128 +0100
@@ -27,7 +27,8 @@
 OBJS = scanner.o # parser.o
 
 AM_CPPFLAGS = -I$(srcdir) -I$(top_srcdir) -I$(top_builddir)
-LDFLAGS = -lpthread
+
+#LDFLAGS = -lpthread
 #LFLAGS = --header="scanner.h"
 #YFLAGS = --defines --output=parser.c
 
