From acd6cab76425fed45a300822cd41f30548948d44 Mon Sep 17 00:00:00 2001
From: Stephen Hutchinson <qyot27@gmail.com>
Date: Mon, 18 Jan 2021 04:41:52 -0500
Subject: [PATCH 2/2] avisynth_conf.h.in: use mounted system libdir on Haiku

---
 avs_core/avisynth_conf.h.in | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/avs_core/avisynth_conf.h.in b/avs_core/avisynth_conf.h.in
index 57bc107f..cdf96f19 100644
--- a/avs_core/avisynth_conf.h.in
+++ b/avs_core/avisynth_conf.h.in
@@ -5,7 +5,15 @@
 
 #ifdef AVS_POSIX
 #define user_avs_plugindir_configurable "@USER_AVS_PLUGINDIR_LOCATION@"
-#define system_avs_plugindir "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/avisynth"
+    #ifdef AVS_HAIKU
+        #ifdef X86_32
+            #define system_avs_plugindir "/system/lib/x86/avisynth"
+        #else
+            #define system_avs_plugindir "/system/lib/avisynth"
+        #endif
+    #else
+        #define system_avs_plugindir "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@/avisynth"
+    #endif
 #endif
 
 #endif // _AVS_CONF_H_
-- 
2.30.0

