diff -urN dumb-0.9.3/Makefile dumb-0.9.3-haiku/Makefile
--- dumb-0.9.3/Makefile	2005-08-08 00:18:41.049283072 +0000
+++ dumb-0.9.3-haiku/Makefile	2010-09-04 07:41:53.857473024 +0000
@@ -64,11 +64,11 @@
 -include make/config.txt
 
 
-ifeq "$(OSTYPE)" "beos"
+ifeq "$(OSTYPE)" "haiku"
 
-INCLUDE_INSTALL_PATH := /boot/develop/headers
-LIB_INSTALL_PATH := /boot/develop/lib/x86
-BIN_INSTALL_PATH := /boot/home/config/bin
+INCLUDE_INSTALL_PATH := /boot/common/include
+LIB_INSTALL_PATH := /boot/common/lib
+BIN_INSTALL_PATH := /boot/common/bin
 # DEFAULT_PREFIX is not set, so config.sh will not prompt for PREFIX.
 LINK_MATH :=
 
@@ -80,9 +80,9 @@
 DEFAULT_PREFIX := /usr/local
 endif
 export DEFAULT_PREFIX
-INCLUDE_INSTALL_PATH := $(PREFIX)/include
-LIB_INSTALL_PATH := $(PREFIX)/lib
-BIN_INSTALL_PATH := $(PREFIX)/bin
+INCLUDE_INSTALL_PATH := $(DESTDIR)$(PREFIX)/include
+LIB_INSTALL_PATH := $(DESTDIR)$(PREFIX)/lib
+BIN_INSTALL_PATH := $(DESTDIR)$(PREFIX)/bin
 
 endif
 
diff -urN dumb-0.9.3/make/dumbask.c dumb-0.9.3-haiku/make/dumbask.c
--- dumb-0.9.3/make/dumbask.c	2005-08-08 00:18:40.048758784 +0000
+++ dumb-0.9.3-haiku/make/dumbask.c	2010-09-04 07:20:26.941621248 +0000
@@ -21,10 +21,10 @@
 	printf("%s", argv[1]);
 
 	for (;;) {
+		int i;
 		char c = getchar();
 		if (c == EOF) return 0;
 		c = toupper(c);
-		int i;
 		for (i = 0; options[i]; i++)
 			if (c == toupper(options[i]))
 				return i;
diff -urN dumb-0.9.3/make/unix.inc dumb-0.9.3-haiku/make/unix.inc
--- dumb-0.9.3/make/unix.inc	2005-08-08 00:18:40.048496640 +0000
+++ dumb-0.9.3-haiku/make/unix.inc	2010-09-04 07:29:01.080216064 +0000
@@ -15,7 +15,7 @@
 
 EXE_SUFFIX :=
 
-LINK_MATH := -lm
+LINK_MATH :=
 LINK_ALLEGRO := `allegro-config --libs`
 
 # PREFIX is set by config.sh.
