diff -Naur SDL2-2.0.0/include/begin_code.h SDL2-2.0.0-haiku/include/begin_code.h
--- SDL2-2.0.0/include/begin_code.h	2013-08-11 21:57:54.011796480 -0500
+++ SDL2-2.0.0-haiku/include/begin_code.h	2013-08-18 21:04:46.580386816 -0500
@@ -36,7 +36,9 @@
 /* Some compilers use a special export keyword */
 #ifndef DECLSPEC
 # if defined(__BEOS__) || defined(__HAIKU__)
-#  if defined(__GNUC__)
+#  if defined(__GNUC__) && __GNUC__ >= 4
+#   define DECLSPEC __attribute__ ((visibility("default")))
+#  elif defined(__GNUC__) && __GNUC__ >= 2
 #   define DECLSPEC __declspec(dllexport)
 #  else
 #   define DECLSPEC __declspec(export)
diff -Naur SDL2-2.0.0/src/main/beos/SDL_BApp.h SDL2-2.0.0-haiku/src/main/beos/SDL_BApp.h
--- SDL2-2.0.0/src/main/beos/SDL_BApp.h	2013-08-11 21:57:54.058982400 -0500
+++ SDL2-2.0.0-haiku/src/main/beos/SDL_BApp.h	2013-08-18 21:05:04.364642304 -0500
@@ -371,7 +371,7 @@
 
 
     /* Members */
-    vector<SDL_Window*> _window_map; /* Keeps track of SDL_Windows by index-id*/
+    std::vector<SDL_Window*> _window_map; /* Keeps track of SDL_Windows by index-id*/
 
     display_mode *_saved_mode;
     BGLView      *_current_context;
