--- a/Makefile	2017-12-22 12:07:37.019660800 +0000
+++ b/Makefile	2018-03-25 14:32:28.139722752 +0000
@@ -210,7 +210,7 @@
 #-----------------------------------------------------------------------------
 # make install is validated only for the following targets
 #-----------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
+ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS Haiku))
 
 DESTDIR     ?=
 # directory variables : GNU conventions prefer lowercase
