Index: quilt-0.48/changes2changelog
===================================================================
--- quilt-0.48.orig/changes2changelog
+++ quilt-0.48/changes2changelog
@@ -1,4 +1,4 @@
-#! /usr/bin/perl
+#!perl
 
 $ENV{"LC_ALL"} = "C";
 
Index: quilt-0.48/quilt.quiltrc
===================================================================
--- quilt-0.48.orig/quilt.quiltrc
+++ quilt-0.48/quilt.quiltrc
@@ -12,8 +12,8 @@ QUILT_PATCH_OPTS="--unified"
 #QUILT_DIFFSTAT_OPTS="-f0"
 
 # Options to pass to commands (QUILT_${COMMAND}_ARGS)
-QUILT_PUSH_ARGS="--color=auto"
-QUILT_DIFF_ARGS="--no-timestamps --color=auto"
+QUILT_PUSH_ARGS=""
+QUILT_DIFF_ARGS="--no-timestamps"
 QUILT_REFRESH_ARGS="--no-timestamps --backup"
 
 # (Add "-p ab" to QUILT_DIFF_ARGS and QUILT_REFRESH_ARGS to get
@@ -28,4 +28,4 @@ QUILT_PATCHES_PREFIX=yes
 
 # Use a specific editor for quilt (defaults to the value of $EDITOR before
 # sourcing this configuration file, or vi if $EDITOR wasn't set).
-#EDITOR=nedit
+EDITOR=nano
Index: quilt-0.48/configure.ac
===================================================================
--- quilt-0.48.orig/configure.ac
+++ quilt-0.48/configure.ac
@@ -69,17 +69,6 @@ You can get this patch at ftp://ftp.gnu.
 fi
 
 QUILT_COMPAT_PROG_PATH(CP, cp, [gcp cp])
-AC_MSG_CHECKING(whether $CP -l works)
-touch conftest.1
-if $CP -l conftest.1 conftest.2 2>/dev/null; then
-	AC_MSG_RESULT(yes)
-else
-	AC_MSG_ERROR([no
-
-You appear to have a `cp' that does not support hard links.
-You can download GNU fileutils from ftp.gnu.org
-])
-fi
 
 QUILT_COMPAT_PROG_PATH(DATE, date, [gdate date])
 
Index: quilt-0.48/quilt/scripts/dependency-graph.in
===================================================================
--- quilt-0.48.orig/quilt/scripts/dependency-graph.in
+++ quilt-0.48/quilt/scripts/dependency-graph.in
@@ -438,8 +438,7 @@ if ($reduce) {
     $out = new FileHandle("| tred")
 	or die "tred: $!\n";
 } else {
-    $out = new FileHandle("> /dev/stdout")
-	or die "$!\n";
+    $out = *STDOUT;
 }
 
 # Write graph
Index: quilt-0.48/lib/backup-files.c
===================================================================
--- quilt-0.48.orig/lib/backup-files.c
+++ quilt-0.48/lib/backup-files.c
@@ -224,7 +224,8 @@ link_or_copy_file(const char *from, cons
 	if (link(from, to) == 0)
 		return 0;
 	if (errno != EXDEV && errno != EPERM &&
-	    errno != EMLINK && errno != ENOSYS) {
+	    errno != EMLINK && errno != ENOSYS &&
+	    errno != EOPNOTSUPP) {
 		fprintf(stderr, "Could not link file `%s' to `%s': %s\n",
 		       from, to, strerror(errno));
 		return 1;
