From 006dc02277d977d13a67f31a1c5fbb3f2e155cb9 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Sun, 12 Feb 2017 15:40:59 -0500
Subject: [PATCH] sckaddr: Fix build on Haiku.

---
 src/common/sckaddr.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/common/sckaddr.cpp b/src/common/sckaddr.cpp
index 0388132..49a6481 100644
--- a/src/common/sckaddr.cpp
+++ b/src/common/sckaddr.cpp
@@ -109,6 +109,11 @@ wxIMPLEMENT_DYNAMIC_CLASS(wxUNIXaddress, wxSockAddress);
     #endif
 #endif // __WINDOWS__
 
+#ifdef __HAIKU__
+    #define HAVE_GETHOSTBYNAME
+    #define HAVE_GETSERVBYNAME
+#endif
+
 // we assume that we have gethostbyaddr_r() if and only if we have
 // gethostbyname_r() and that it uses the similar conventions to it (see
 // comment in configure)
-- 
2.7.0

