Index: browser/app/Makefile.in
===================================================================
--- browser/app/Makefile.in	(revision 7)
+++ browser/app/Makefile.in	(working copy)
@@ -151,7 +151,7 @@
 endif
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
Index: browser/components/migration/src/Makefile.in
===================================================================
--- browser/components/migration/src/Makefile.in	(revision 7)
+++ browser/components/migration/src/Makefile.in	(working copy)
@@ -78,7 +78,7 @@
            nsPhoenixProfileMigrator.cpp \
            $(NULL)
 
-ifneq ($(OS_ARCH),BeOS)
+ifneq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 CPPSRCS += nsDogbertProfileMigrator.cpp
 endif
 
Index: calendar/libical/src/libical/Makefile.in
===================================================================
--- calendar/libical/src/libical/Makefile.in	(revision 7)
+++ calendar/libical/src/libical/Makefile.in	(working copy)
@@ -89,7 +89,7 @@
 
 DEFINES += -DHAVE_CONFIG_H
 
-ifneq (,$(filter BeOS Darwin OpenVMS NetBSD FreeBSD,$(OS_ARCH)))
+ifneq (,$(filter BeOS Haiku Darwin OpenVMS NetBSD FreeBSD,$(OS_ARCH)))
 DEFINES	+= -D__USE_BSD=1
 endif
 
Index: calendar/sunbird/app/Makefile.in
===================================================================
--- calendar/sunbird/app/Makefile.in	(revision 7)
+++ calendar/sunbird/app/Makefile.in	(working copy)
@@ -194,7 +194,8 @@
 endif
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
+BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
 endif
Index: directory/c-sdk/ldap/libraries/libldap/Makefile.in
===================================================================
--- directory/c-sdk/ldap/libraries/libldap/Makefile.in	(revision 7)
+++ directory/c-sdk/ldap/libraries/libldap/Makefile.in	(working copy)
@@ -237,7 +237,7 @@
 SHARED_LIBRARY_LIBS = $(dist_libdir)/lib$(LBER_LIBNAME).a
 endif
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_LIBS = -L$(dist_libdir) -l$(LBER_LIBNAME) -lbe
 endif
 
@@ -302,7 +302,7 @@
 ifdef SO_FILES_TO_REMOVE
 	-$(RM) $(SO_FILES_TO_REMOVE)
 endif
-ifneq (,$(filter AIX Linux HP-UX Darwin BeOS QNX NetBSD OSF1 OpenBSD, $(OS_ARCH)))
+ifneq (,$(filter AIX Linux HP-UX Darwin BeOS Haiku QNX NetBSD OSF1 OpenBSD, $(OS_ARCH)))
 	$(LINK_LIB) ../liblber/$(OBJDIR_NAME)/*.a
 else
 ifeq ($(OS_ARCH),OS2)
Index: directory/c-sdk/ldap/libraries/libprldap/Makefile.in
===================================================================
--- directory/c-sdk/ldap/libraries/libprldap/Makefile.in	(revision 7)
+++ directory/c-sdk/ldap/libraries/libprldap/Makefile.in	(working copy)
@@ -165,7 +165,7 @@
 EXTRA_LIBS += -L$(dist_libdir) $(DYNAMICNSPR)
 endif
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_LIBS = -L$(dist_libdir) -l$(LDAP_LIBNAME)
 EXTRA_LIBS += -L$(dist_libdir) $(DYNAMICNSPR)
 endif
Index: directory/c-sdk/ldap/libraries/libssldap/Makefile
===================================================================
--- directory/c-sdk/ldap/libraries/libssldap/Makefile	(revision 7)
+++ directory/c-sdk/ldap/libraries/libssldap/Makefile	(working copy)
@@ -167,7 +167,7 @@
 EXTRA_LIBS += $(OS_LIBS)
 endif
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_LIBS = -L$(dist_libdir) -l$(LDAP_LIBNAME) -l$(PRLDAP_LIBNAME)
 ifeq ($(NSS_DYNAMIC_SOFTOKN),1)
 EXTRA_LIBS += -l$(SOFTOKN_LIBNAME)
Index: directory/c-sdk/ldap/libraries/libssldap/Makefile.in
===================================================================
--- directory/c-sdk/ldap/libraries/libssldap/Makefile.in	(revision 7)
+++ directory/c-sdk/ldap/libraries/libssldap/Makefile.in	(working copy)
@@ -182,7 +182,7 @@
 EXTRA_LIBS += $(OS_LIBS)
 endif
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_LIBS = -L$(dist_libdir) -l$(LDAP_LIBNAME) -l$(PRLDAP_LIBNAME)
 ifeq ($(NSS_DYNAMIC_SOFTOKN),1)
 EXTRA_LIBS += -l$(SOFTOKN_LIBNAME)
Index: embedding/componentlib/Makefile.in
===================================================================
--- embedding/componentlib/Makefile.in	(revision 7)
+++ embedding/componentlib/Makefile.in	(working copy)
@@ -152,7 +152,7 @@
 OS_LIBS	+= ole32.lib shell32.lib
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 OS_LIBS += -ltracker
 endif
 
Index: intl/uconv/src/Makefile.in
===================================================================
--- intl/uconv/src/Makefile.in	(revision 7)
+++ intl/uconv/src/Makefile.in	(working copy)
@@ -119,7 +119,7 @@
 CPPSRCS			+= nsMacCharset.cpp
 EXPORT_RESOURCE		+= maccharset.properties
 else
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 CPPSRCS			+= nsBeOSCharset.cpp
 else
 CPPSRCS			+= nsUNIXCharset.cpp
Index: ipc/ipcd/client/src/Makefile.in
===================================================================
--- ipc/ipcd/client/src/Makefile.in	(revision 7)
+++ ipc/ipcd/client/src/Makefile.in	(working copy)
@@ -64,7 +64,7 @@
 ifeq ($(OS_ARCH),WINNT)
 CPPSRCS +=  ipcConnectionWin.cpp
 else
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 CPPSRCS +=  ipcConnectionStub.cpp
 else
 CPPSRCS +=  ipcConnectionUnix.cpp
Index: ipc/ipcd/daemon/src/Makefile.in
===================================================================
--- ipc/ipcd/daemon/src/Makefile.in	(revision 7)
+++ ipc/ipcd/daemon/src/Makefile.in	(working copy)
@@ -59,7 +59,7 @@
 ifeq ($(OS_ARCH),WINNT)
 CPPSRCS +=  ipcdWin.cpp
 else
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 CPPSRCS +=  ipcdStub.cpp
 else
 CPPSRCS +=  ipcdUnix.cpp
Index: js/src/Makefile.in
===================================================================
--- js/src/Makefile.in	(revision 7)
+++ js/src/Makefile.in	(working copy)
@@ -246,7 +246,7 @@
 LDFLAGS		+= $(pathsubst -l%,$(NSPR_STATIC_PATH)/%.a,$(NSPR_LIBS))
 
 # BeOS and HP-UX do not require the extra linking of "-lm"
-ifeq (,$(filter BeOS HP-UX WINNT WINCE OpenVMS,$(OS_ARCH)))
+ifeq (,$(filter BeOS Haiku HP-UX WINNT WINCE OpenVMS,$(OS_ARCH)))
 LDFLAGS		+= -lm
 endif
 
Index: mail/app/Makefile.in
===================================================================
--- mail/app/Makefile.in	(revision 7)
+++ mail/app/Makefile.in	(working copy)
@@ -149,7 +149,7 @@
 endif
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 BEOS_PROGRAM_RESOURCE = $(srcdir)/apprunner-beos.rsrc
 ifdef BUILD_STATIC_LIBS
 OS_LIBS += -ltracker -lgame
Index: modules/libpr0n/decoders/Makefile.in
===================================================================
--- modules/libpr0n/decoders/Makefile.in	(revision 7)
+++ modules/libpr0n/decoders/Makefile.in	(working copy)
@@ -52,7 +52,7 @@
 ifeq ($(OS_ARCH),OS2)
 DIRS = icon/os2 icon
 endif
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 DIRS = icon/beos icon
 endif
 ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
Index: modules/libpr0n/decoders/icon/Makefile.in
===================================================================
--- modules/libpr0n/decoders/icon/Makefile.in	(revision 7)
+++ modules/libpr0n/decoders/icon/Makefile.in	(working copy)
@@ -82,7 +82,7 @@
 PLATFORM = mac
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 PLATFORM = beos
 endif
 
@@ -102,7 +102,7 @@
 ifdef MOZ_ENABLE_GNOMEUI
 USE_ICON_DECODER = 1
 endif
-ifeq (,$(filter-out Darwin OS2 BeOS,$(OS_ARCH)))
+ifeq (,$(filter-out Darwin OS2 BeOS Haiku,$(OS_ARCH)))
 USE_ICON_DECODER = 1
 endif
 
Index: modules/plugin/base/src/Makefile.in
===================================================================
--- modules/plugin/base/src/Makefile.in	(revision 7)
+++ modules/plugin/base/src/Makefile.in	(working copy)
@@ -96,7 +96,7 @@
 		nsJSNPRuntime.cpp \
 		$(NULL)
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 	CPPSRCS += nsPluginsDirBeOS.cpp
 	CPPSRCS += nsPluginNativeWindow.cpp
 else
Index: modules/libreg/src/Makefile.in
===================================================================
--- modules/libreg/src/Makefile.in	(revision 7)
+++ modules/libreg/src/Makefile.in	(working copy)
@@ -66,7 +66,7 @@
 
 DEFINES         += -DUSE_BUFFERED_REGISTRY_IO	
 # Memory mapped files are not supported under QNX, Neutrino, HP-UX and BeOS
-#ifeq (,$(filter BeOS HP-UX QNX,$(OS_ARCH)))
+#ifeq (,$(filter BeOS Haiku HP-UX QNX,$(OS_ARCH)))
 #CSRCS		+= mmapio.c
 #DEFINES		+= -DUSE_MMAP_REGISTRY_IO
 #endif
Index: nsprpub/lib/prstreams/Makefile.in
===================================================================
--- nsprpub/lib/prstreams/Makefile.in	(revision 7)
+++ nsprpub/lib/prstreams/Makefile.in	(working copy)
@@ -69,7 +69,7 @@
     endif
 endif
 
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
     CFLAGS += -frtti -fexceptions
 endif
 
@@ -118,7 +118,7 @@
     endif
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
     OS_LIBS = -lstdc++.r4
 endif
 
Index: toolkit/library/Makefile.in
===================================================================
--- toolkit/library/Makefile.in	(revision 7)
+++ toolkit/library/Makefile.in	(working copy)
@@ -364,7 +364,7 @@
 EXTRA_DSO_LDOPTS += $(MOZ_PANGO_LIBS)
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_DSO_LDOPTS += -lbe
 endif
 
Index: xpcom/build/Makefile.in
===================================================================
--- xpcom/build/Makefile.in	(revision 7)
+++ xpcom/build/Makefile.in	(working copy)
@@ -173,7 +173,7 @@
 EXTRA_DSO_LDOPTS += $(TK_LIBS)
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_DSO_LDOPTS += -lbe
 endif
 
Index: xpcom/obsolete/Makefile.in
===================================================================
--- xpcom/obsolete/Makefile.in	(revision 7)
+++ xpcom/obsolete/Makefile.in	(working copy)
@@ -120,7 +120,7 @@
 EXTRA_DSO_LDOPTS += $(TK_LIBS)
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 EXTRA_DSO_LDOPTS += -lbe
 endif
 
Index: xpcom/reflect/xptcall/src/md/unix/Makefile.in
===================================================================
--- xpcom/reflect/xptcall/src/md/unix/Makefile.in	(revision 7)
+++ xpcom/reflect/xptcall/src/md/unix/Makefile.in	(working copy)
@@ -95,9 +95,10 @@
 endif
 endif
 #
-# BeOS/Intel (uses the same unixish_x86 code)
+# BeOS/Intel (uses the same as Linux, works well. Migrate others!)
 #
-ifeq ($(OS_ARCH)$(OS_TEST),BeOSBePC)
+#ifeq ($(OS_ARCH)$(OS_TEST),BeOSBePC)
+ifeq (,$(filter-out BeOSBePC HaikuBePC, $(OS_ARCH)$(OS_TEST)))
 CPPSRCS		:= xptcinvoke_gcc_x86_unix.cpp xptcstubs_gcc_x86_unix.cpp
 endif
 #
Index: xpinstall/packager/Makefile.in
===================================================================
--- xpinstall/packager/Makefile.in	(revision 7)
+++ xpinstall/packager/Makefile.in	(working copy)
@@ -48,7 +48,7 @@
 ifeq ($(OS_ARCH),IRIX)
 STRIP_FLAGS	= -f
 endif
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 STRIP_FLAGS	= -g
 PLATFORM_EXCLUDE_LIST = ! -name "*.stub" ! -name "$(MOZ_PKG_APPNAME)-bin"
 endif
Index: xulrunner/app/Makefile.in
===================================================================
--- xulrunner/app/Makefile.in	(revision 7)
+++ xulrunner/app/Makefile.in	(working copy)
@@ -59,7 +59,7 @@
 ifeq ($(USE_SHORT_LIBNAME), 1)
 PROGRAM = xulrunner$(BIN_SUFFIX)
 else
-ifeq ($(OS_ARCH), BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 PROGRAM = xulrunner$(BIN_SUFFIX)
 else
 PROGRAM = xulrunner-bin$(BIN_SUFFIX)
@@ -85,7 +85,7 @@
 ifeq ($(OS_ARCH),WINNT)
 CPPSRCS += nsRegisterGREWin.cpp
 else
-ifneq (,$(filter OS2 Darwin,$(OS_ARCH)))
+ifneq (,$(filter OS2 Darwin BeOS Haiku,$(OS_ARCH)))
 CPPSRCS += nsRegisterGREVoid.cpp
 else
 CPPSRCS += nsRegisterGREUnix.cpp
@@ -174,7 +174,7 @@
 endif
 endif
 
-ifeq ($(OS_ARCH),BeOS)
+ifeq (,$(filter-out BeOS Haiku, $(OS_ARCH)))
 OS_LIBS += -ltracker
 endif
 
