Index: calendar/base/content/preferences/preferences.xul
===================================================================
--- calendar/base/content/preferences/preferences.xul	(revision 7)
+++ calendar/base/content/preferences/preferences.xul	(working copy)
@@ -71,8 +71,13 @@
             style="&prefWindow.styleGNOME;">
 #endif
 #endif
+#ifdef XP_BEOS
+            title="&prefWindow.titleGNOME;"
+            style="&prefWindow.styleGNOME;">
 #endif
+#endif
 
+
     <prefpane id="paneGeneral"
               label="&paneGeneral.title;"
               onpaneload="gCalendarGeneralPane.init();"
