Index: browser/app/firefox-branding.js
===================================================================
--- browser/app/firefox-branding.js	(revision 7)
+++ browser/app/firefox-branding.js	(working copy)
@@ -1,5 +1,4 @@
 pref("startup.homepage_override_url","http://www.mozilla.org/projects/%APP%/%VERSION%/whatsnew/");
-pref("startup.homepage_welcome_url","http://www.mozilla.org/projects/%APP%/%VERSION%/firstrun/");
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
 pref("app.update.url.manual", "http://www.mozilla.org/products/%APP%/");
@@ -8,7 +7,7 @@
 pref("app.update.url.details", "http://www.mozilla.org/projects/%APP%/");
 
 // Release notes URL
-pref("app.releaseNotesURL", "http://www.mozilla.org/projects/%APP%/%VERSION%/releasenotes/");
+pref("app.releaseNotesURL", "about:buildconfig");
 
 // Search codes belong only in builds with official branding
 pref("browser.search.param.yahoo-fr", "");
