diff --git a/src/core/config.c b/src/core/config.c
index cd173dfb..9135e0fc 100644
--- a/src/core/config.c
+++ b/src/core/config.c
@@ -27,6 +27,10 @@
 #include <mgba-util/platform/3ds/3ds-vfs.h>
 #endif
 
+#ifdef __HAIKU__
+#include <FindDirectory.h>
+#endif
+
 #define SECTION_NAME_MAX 128
 
 struct mCoreConfigEnumerateData {
@@ -227,6 +231,12 @@ void mCoreConfigDirectory(char* out, size_t outLength) {
 	UNUSED(portable);
 	snprintf(out, outLength, "/%s", projectName);
 	FSUSER_CreateDirectory(sdmcArchive, fsMakePath(PATH_ASCII, out), 0);
+#elif defined(__HAIKU__)
+	UNUSED(portable);
+	char path[B_PATH_NAME_LENGTH];
+	find_directory(B_USER_SETTINGS_DIRECTORY, 0, false, path, B_PATH_NAME_LENGTH);
+	snprintf(out, outLength, "%s/%s", path, binaryName);
+	mkdir(out, 0755);
 #else
 	getcwd(out, outLength);
 	strncat(out, PATH_SEP "portable.ini", outLength - strlen(out));
diff --git a/src/platform/qt/InputController.cpp b/src/platform/qt/InputController.cpp
index c6a598ce..f19130dd 100644
--- a/src/platform/qt/InputController.cpp
+++ b/src/platform/qt/InputController.cpp
@@ -272,8 +272,8 @@ void InputController::setPreferredGamepad(uint32_t type, int index) {
 		return;
 	}
 #ifdef BUILD_SDL
-	char name[34] = {0};
 #if SDL_VERSION_ATLEAST(2, 0, 0)
+	char name[34] = {0};
 	SDL_JoystickGetGUIDString(SDL_JoystickGetGUID(SDL_JoystickListGetPointer(&s_sdlEvents.joysticks, index)->joystick), name, sizeof(name));
 #else
 	const char* name = SDL_JoystickName(SDL_JoystickIndex(SDL_JoystickListGetPointer(&s_sdlEvents.joysticks, index)->joystick));
