From 3b6eb2569b1cb515a8c48974bb54f6fa2a4ca834 Mon Sep 17 00:00:00 2001
From: Kostadin Damyanov <maxmight@gmail.com>
Date: Tue, 13 Oct 2015 00:55:20 +0300
Subject: [PATCH] Makefile fix.

---
 Makefile | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 0718bbb..e447b09 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 #
 # Original Author:
 # Copyright 2010 Vincent Driessen. All rights reserved.
-# 
+#
 # Redistribution and use in source and binary forms, with or without
 # modification, are permitted provided that the following conditions are met:
 #
@@ -28,8 +28,8 @@
 
 prefix=/usr/local
 
-datarootdir=$(prefix)/share
-docdir=$(datarootdir)/doc/gitflow
+datarootdir=$(prefix)/git-flow
+docdir=$(datarootdir)/doc/
 # files that need mode 755
 EXEC_FILES=git-flow
 
@@ -53,10 +53,9 @@ all:
 	@echo "       make uninstall"
 
 install:
-	install -d -m 0755 $(prefix)/bin
 	install -d -m 0755 $(docdir)/hooks
-	install -m 0755 $(EXEC_FILES) $(prefix)/bin
-	install -m 0644 $(SCRIPT_FILES) $(prefix)/bin
+	install -m 0755 $(EXEC_FILES) $(libdir)/git-core
+	install -m 0644 $(SCRIPT_FILES) $(libdir)/git-core
 	install -m 0644 $(HOOK_FILES) $(docdir)/hooks
 
 uninstall:
-- 
2.2.2

