diff -urN readline-6.1/configure.in readline-6.1-haiku/configure.in
--- readline-6.1/configure.in	2009-10-12 07:12:15.054788096 -0700
+++ readline-6.1-haiku/configure.in	2010-10-26 10:21:29.204210176 -0700
@@ -85,6 +85,9 @@
     i[[3456]]86-*-beos*)
         cross_cache=${srcdir}/cross-build/x86-beos.cache
         ;;
+    i[[3456]]86-*-haiku*)
+        cross_cache=${srcdir}/cross-build/x86-haiku.cache
+        ;;        
     *)  echo "configure: cross-compiling for $host is not supported" >&2
         ;;
     esac
diff -urN readline-6.1/rltty.c readline-6.1-haiku/rltty.c
--- readline-6.1/rltty.c	2009-11-19 06:42:02.059768832 -0800
+++ readline-6.1-haiku/rltty.c	2010-10-26 10:25:50.489160704 -0700
@@ -238,7 +238,10 @@
      TIOTYPE oldtio, *tiop;
 {
   _rl_echoing_p = (oldtio.sgttyb.sg_flags & ECHO);
+
+#if defined (ECHOCTL)
   _rl_echoctl = (oldtio.sgttyb.sg_flags & ECHOCTL);
+#endif
 
   /* Copy the original settings to the structure we're going to use for
      our settings. */
