diff -ur orig/readline-6.3/support/shlib-install readline-6.3/support/shlib-install
--- orig/readline-6.3/support/shlib-install	2009-10-28 14:30:18.057409536 +0100
+++ readline-6.3/support/shlib-install	2013-05-03 01:12:28.695468032 +0200
@@ -117,7 +117,7 @@
 # Create symlinks to the installed library.  This section is incomplete.
 #
 case "$host_os-$host_vendor" in
-*linux*|freebsd*-gentoo)
+*linux*|freebsd*-gentoo|haiku*)
 	# libname.so.M -> libname.so.M.N
 	${echo} ${RM} ${INSTALLDIR}/$LINK2
 	if [ -z "$uninstall" ]; then
diff -ur orig/readline-6.3/support/shobj-conf readline-6.3/support/shobj-conf
--- orig/readline-6.3/support/shobj-conf	2009-10-28 14:20:21.057147392 +0100
+++ readline-6.3/support/shobj-conf	2013-05-03 01:12:38.631242752 +0200
@@ -109,7 +109,7 @@
 	;;
 
 # All versions of Linux (including Gentoo/FreeBSD) or the semi-mythical GNU Hurd.
-linux*-*|gnu*-*|k*bsd*-gnu-*|freebsd*-gentoo)
+linux*-*|gnu*-*|k*bsd*-gnu-*|freebsd*-gentoo|haiku*)
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
 	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
