diff -Naur ladspa_sdk-1.13/src/makefile ladspa_sdk-1.13-haiku/src/makefile
--- ladspa_sdk/src/makefile	2007-11-06 10:42:45.008388608 +0000
+++ ladspa_sdk-haiku/src/makefile	2012-12-17 19:04:36.860094464 +0000
@@ -4,9 +4,9 @@
 #
 # Change these if you want to install somewhere else.
 
-INSTALL_PLUGINS_DIR	=	/usr/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/bin/
+INSTALL_PLUGINS_DIR	=	${DESTDIR}/`finddir B_COMMON_LIB_DIRECTORY`
+INSTALL_INCLUDE_DIR	=	${DESTDIR}/`finddir B_COMMON_HEADERS_DIRECTORY`
+INSTALL_BINARY_DIR	=	${DESTDIR}/`finddir B_COMMON_BIN_DIRECTORY`
 
 ###############################################################################
 #
@@ -14,7 +14,6 @@
 #
 
 INCLUDES	=	-I.
-LIBRARIES	=	-ldl -lm
 CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
@@ -46,22 +45,10 @@
 # TARGETS
 #
 
-test:	/tmp/test.wav ../snd/noise.wav always
-	@echo ---------------------------------------------
-	@echo First listen to the white noise input signal:
-	@echo ---------------------------------------------
-	-sndfile-play ../snd/noise.wav
-	@echo -------------------------
-	@echo Compare to plugin output.
-	@echo -------------------------
-	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
-	-sndfile-play /tmp/test.wav
-	@echo Test complete.
-
 install:	targets
-	-mkdirhier $(INSTALL_PLUGINS_DIR)
-	-mkdirhier $(INSTALL_INCLUDE_DIR)
-	-mkdirhier $(INSTALL_BINARY_DIR)
+	mkdir -p $(INSTALL_PLUGINS_DIR)
+	mkdir -p $(INSTALL_INCLUDE_DIR)
+	mkdir -p $(INSTALL_BINARY_DIR)
 	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
 	cp ladspa.h $(INSTALL_INCLUDE_DIR)
 	cp ../bin/* $(INSTALL_BINARY_DIR)
