diff -urN libdvdnav-4.1.3/configure.ac libdvdnav-4.1.3-haiku/configure.ac
--- libdvdnav-4.1.3/configure.ac	2009-02-19 07:41:01.000000000 +0000
+++ libdvdnav-4.1.3-haiku/configure.ac	2009-02-19 07:33:27.000000000 +0000
@@ -145,6 +145,8 @@
     ;;
   *mingw32* | *cygwin*)
     ;;
+  *haiku*)
+    ;; 
   *)
     AC_CHECK_LIB(pthread, pthread_create,
 	[THREAD_LIBS="-lpthread"],
@@ -165,6 +167,8 @@
   *cygwin*)
     LDFLAGS="-no-undefined $LDFLAGS"
     ;;
+  *haiku*)
+    ;;
   *)
     AC_CHECK_LIB(c, dlopen,
 	         DYNAMIC_LD_LIBS="",
diff -urN libdvdnav-4.1.3/src/remap.c libdvdnav-4.1.3-haiku/src/remap.c
--- libdvdnav-4.1.3/src/remap.c	2009-02-19 07:41:01.000000000 +0000
+++ libdvdnav-4.1.3-haiku/src/remap.c	2009-02-19 08:35:48.000000000 +0000
@@ -28,7 +28,7 @@
 
 #ifndef _MSC_VER
 #include <sys/param.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #else
 #ifndef MAXPATHLEN
 #define MAXPATHLEN 255
