diff -Naur slang-2.2.2/src/slang.h slang-2.2.2-haiku/src/slang.h
--- slang-2.2.2/src/slang.h	2009-12-05 17:03:51.043253760 -0200
+++ slang-2.2.2-haiku/src/slang.h	2010-09-21 08:33:49.000000000 -0300
@@ -1377,7 +1377,7 @@
 
 /*{{{ SLang getkey interface Functions */
 
-#ifdef REAL_UNIX_SYSTEM
+#if defined(REAL_UNIX_SYSTEM) || defined(__HAIKU__)
 SL_EXTERN int SLang_TT_Baud_Rate;
 SL_EXTERN int SLang_TT_Read_FD;
 #else
@@ -1614,7 +1614,7 @@
 #endif
 
 #ifndef __GO32__
-#if defined(VMS) || defined(REAL_UNIX_SYSTEM)
+#if defined(VMS) || defined(REAL_UNIX_SYSTEM) || defined(__HAIKU__)
 SL_EXTERN int SLtt_Blink_Mode;
 SL_EXTERN int SLtt_Use_Blink_For_ACS;
 SL_EXTERN int SLtt_Newline_Ok;
