diff -urN slang-2.1.3/autoconf/aclocal.m4 slang-2.1.3-haiku/autoconf/aclocal.m4
--- slang-2.1.3/autoconf/aclocal.m4	2007-10-22 17:49:55.000000000 +0000
+++ slang-2.1.3-haiku/autoconf/aclocal.m4	2008-08-24 21:38:53.000000000 +0000
@@ -580,7 +580,7 @@
     ELF_CC="\$(CC)"
     ELF_CFLAGS="\$(CFLAGS) -fPIC"
     ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-O1 -Wl,--version-script,\$(VERSION_SCRIPT) -Wl,-soname,\$(ELFLIB_MAJOR)"
-    ELF_DEP_LIBS="\$(DL_LIB) -lm -lc"
+    ELF_DEP_LIBS="\$(DL_LIB) $(M_LIB) -lc"
     CC_SHARED="\$(CC) \$(CFLAGS) -shared -fPIC"
     ;;
   *solaris* )
@@ -590,14 +590,14 @@
       ELF_CC="\$(CC)"
       ELF_CFLAGS="\$(CFLAGS) -fPIC"
       ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-ztext -Wl,-h,\$(ELFLIB_MAJOR)"
-      ELF_DEP_LIBS="\$(DL_LIB) -lm -lc"
+      ELF_DEP_LIBS="\$(DL_LIB) $(M_LIB) -lc"
       CC_SHARED="\$(CC) \$(CFLAGS) -G -fPIC"
     else
       DYNAMIC_LINK_FLAGS=""
       ELF_CC="\$(CC)"
       ELF_CFLAGS="\$(CFLAGS) -K PIC"
       ELF_LINK="\$(CC) \$(LDFLAGS) -G -h\$(ELFLIB_MAJOR)"
-      ELF_DEP_LIBS="\$(DL_LIB) -lm -lc"
+      ELF_DEP_LIBS="\$(DL_LIB) $(M_LIB) -lc"
       CC_SHARED="\$(CC) \$(CFLAGS) -G -K PIC"
     fi
     ;;
@@ -663,7 +663,15 @@
     #  ELF_LINK="ld -Bshareable -x"
     #fi
     ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-soname,\$(ELFLIB_MAJOR)"
-    ELF_DEP_LIBS="\$(DL_LIB) -lm"
+    ELF_DEP_LIBS="\$(DL_LIB) $(M_LIB)"
+    CC_SHARED="\$(CC) \$(CFLAGS) -shared -fPIC"
+    ;;
+  *haiku* )
+    DYNAMIC_LINK_FLAGS="-Wl,-export-dynamic"
+    ELF_CC="\$(CC)"
+    ELF_CFLAGS="\$(CFLAGS) -fPIC"
+    ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-O1 -Wl,--version-script,\$(VERSION_SCRIPT) -Wl,-soname,\$(ELFLIB_MAJOR)"
+    ELF_DEP_LIBS="\$(DL_LIB)"
     CC_SHARED="\$(CC) \$(CFLAGS) -shared -fPIC"
     ;;
   *cygwin* )
@@ -674,7 +682,7 @@
     DLL_IMPLIB_NAME="lib\$(THIS_LIB)\$(ELFLIB_MAJOR_VERSION).dll.a"
     #ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-O1 -Wl,--version-script,\$(VERSION_SCRIPT) -Wl,-soname,\$(ELFLIB_MAJOR) -Wl,--out-implib=\$(DLL_IMPLIB_NAME) -Wl,-export-all-symbols -Wl,-enable-auto-import"
     ELF_LINK="\$(CC) \$(LDFLAGS) -shared -Wl,-O1 -Wl,--version-script,\$(VERSION_SCRIPT) -Wl,-soname,\$(ELFLIB_MAJOR) -Wl,--out-implib=\$(DLL_IMPLIB_NAME)"
-    ELF_DEP_LIBS="\$(DL_LIB) -lm"
+    ELF_DEP_LIBS="\$(DL_LIB) $(M_LIB)"
     CC_SHARED="\$(CC) \$(CFLAGS) -shared -DSLANG_DLL=1"
     dnl# CYGWIN prohibits undefined symbols when linking shared libs
     SLANG_LIB_FOR_MODULES="-L\$(ELFDIR) -lslang"
@@ -691,7 +699,7 @@
     ELF_CC="\$(CC)"
     ELF_CFLAGS="\$(CFLAGS) -fPIC"
     ELF_LINK="\$(CC) \$(LDFLAGS) -shared"
-    ELF_DEP_LIBS="\$(DL_LIB) -lm -lc"
+    ELF_DEP_LIBS="\$(DL_LIB) $(M_LIB) -lc"
     CC_SHARED="\$(CC) \$(CFLAGS) -shared -fPIC"
 esac
 
diff -urN slang-2.1.3/demo/Makefile.in slang-2.1.3-haiku/demo/Makefile.in
--- slang-2.1.3/demo/Makefile.in	2006-05-12 19:09:10.000000000 +0000
+++ slang-2.1.3-haiku/demo/Makefile.in	2008-08-24 21:40:28.000000000 +0000
@@ -29,7 +29,7 @@
 ALL_CFLAGS = $(CFLAGS) -Dunix -I$(SLANG_INCLUDE)
 
 COMPILE_CMD = $(CC) -c $(ALL_CFLAGS)
-EXECLIBS = -L$(SLANG_LIB) -lslang -lm $(TCAPLIB)
+EXECLIBS = -L$(SLANG_LIB) -lslang $(M_LIB) $(TCAPLIB)
 EXECDEPS = $(SLANG_LIB)/libslang.a
 DOT_O_DEPS = demolib.c
 
diff -urN slang-2.1.3/doc/tm/tools/Makefile slang-2.1.3-haiku/doc/tm/tools/Makefile
--- slang-2.1.3/doc/tm/tools/Makefile	2006-05-12 19:09:10.000000000 +0000
+++ slang-2.1.3-haiku/doc/tm/tools/Makefile	2008-08-24 21:40:28.000000000 +0000
@@ -7,7 +7,7 @@
 OBJDIR=$(ARCH)objs#
 
 SLANGDIR=../../../../src
-EXECLIBS = -L$(SLANGDIR)/$(ARCH)$(ELF)objs -lslang -lm
+EXECLIBS = -L$(SLANGDIR)/$(ARCH)$(ELF)objs -lslang $(M_LIB)
 EXECINC = -I$(SLANGDIR)
 
 COMPILE_CMD=$(CC) -c $(CFLAGS) $(EXECINC)
diff -urN slang-2.1.3/modules/Makefile.in slang-2.1.3-haiku/modules/Makefile.in
--- slang-2.1.3/modules/Makefile.in	2007-09-08 02:47:29.000000000 +0000
+++ slang-2.1.3-haiku/modules/Makefile.in	2008-08-24 21:40:28.000000000 +0000
@@ -69,7 +69,7 @@
 #---------------------------------------------------------------------------
 RPATH = @RPATH@
 
-LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) -lm
+LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) $(M_LIB)
 #LIBS = $(RPATH) $(DL_LIB) -lm
 INCS = $(SLANG_INC)
 
diff -urN slang-2.1.3/slsh/Makefile.in slang-2.1.3-haiku/slsh/Makefile.in
--- slang-2.1.3/slsh/Makefile.in	2007-09-09 01:39:18.000000000 +0000
+++ slang-2.1.3-haiku/slsh/Makefile.in	2008-08-24 21:40:28.000000000 +0000
@@ -14,7 +14,7 @@
 SLANG_SRCINC	= -I@SRCDIR@
 SLANG_SRCLIB	= -L@ELFDIR@
 #---------------------------------------------------------------------------
-OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ -lm
+OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@
 RPATH		= @RPATH@
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
@@ -64,11 +64,11 @@
 #----------------------------------------------------------------------------
 @SET_MAKE@
 SHELL = /bin/sh
-INST_LIBS = $(RPATH) $(SLANG_INST_LIB) $(DEST_LIB_DIR) -lslang $(READLINE_LIB) $(OTHER_LIBS)
+INST_LIBS = $(RPATH) $(SLANG_INST_LIB) $(DEST_LIB_DIR) -lslang $(READLINE_LIB) $(OTHER_LIBS) $(M_LIB)
 DEFS = -DSLSH_CONF_DIR='"$(SLSH_CONF_DIR)"' -DSLSH_PATH='"$(SLSH_LIB_DIR)"' \
  -DSLSH_CONF_DIR_ENV='$(SLSH_CONF_DIR_ENV)' -DSLSH_LIB_DIR_ENV='$(SLSH_LIB_DIR_ENV)' \
  -DSLSH_PATH_ENV='$(SLSH_PATH_ENV)' 
-SRC_LIBS = $(SLANG_SRCLIB) -lslang $(READLINE_LIB) $(OTHER_LIBS)
+SRC_LIBS = $(SLANG_SRCLIB) -lslang $(READLINE_LIB) $(OTHER_LIBS) $(M_LIB)
 #
 all: slsh_exe
 slsh_exe: slsh.o readline.o
diff -urN slang-2.1.3/src/Makefile.in slang-2.1.3-haiku/src/Makefile.in
--- slang-2.1.3/src/Makefile.in	2007-09-08 02:47:29.000000000 +0000
+++ slang-2.1.3-haiku/src/Makefile.in	2008-08-24 21:40:28.000000000 +0000
@@ -36,7 +36,7 @@
 #---------------------------------------------------------------------------
 # Misc Libraries
 MISC_TERMINFO_DIRS = @MISC_TERMINFO_DIRS@
-OTHERLIBS = @TERMCAP@ @LIBS@ -lm
+OTHERLIBS = @TERMCAP@ @LIBS@
 #---------------------------------------------------------------------------
 # Directory where library is going to go when installed
 #---------------------------------------------------------------------------
@@ -107,7 +107,7 @@
 ELFDIR_ELF_LIB = $(ELFDIR)/$(ELFLIB_BUILD_NAME)#
 ELFDIR_DLL_LIB = $(ELFDIR)/$(DLL_IMPLIB_NAME)#
 
-EXECLIBS = -L$(OBJDIR) -lslang $(OTHERLIBS) $(DL_LIB)
+EXECLIBS = -L$(OBJDIR) -lslang $(OTHERLIBS) $(DL_LIB) $(M_LIB)
 EXECDEPS = $(OBJDIR_NORMAL_LIB)
 
 OFILES = @PROGRAM_OFILES@
diff -urN slang-2.1.3/src/curses/Makefile slang-2.1.3-haiku/src/curses/Makefile
--- slang-2.1.3/src/curses/Makefile	2006-05-12 19:09:10.000000000 +0000
+++ slang-2.1.3-haiku/src/curses/Makefile	2008-08-24 21:40:28.000000000 +0000
@@ -29,23 +29,23 @@
 firework: $(CURSES_H) firework.c
 	$(COMPILE) $@.c -o $@ $(LFLAGS)
 tclock: $(CURSES_H) tclock.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 worm: worm.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 knight: knight.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 xmas: xmas.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 newdemo: newdemo.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 testcurs: testcurs.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 lrtest: lrtest.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 t: t.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 key: key.c
-	$(COMPILE) $@.c -o $@ $(LFLAGS) -lm
+	$(COMPILE) $@.c -o $@ $(LFLAGS) $(M_LIB)
 
 clean:
 	/bin/rm $(EXECS)
diff -urN slang-2.1.3/src/test/Makefile slang-2.1.3-haiku/src/test/Makefile
--- slang-2.1.3/src/test/Makefile	2007-07-31 16:11:38.000000000 +0000
+++ slang-2.1.3-haiku/src/test/Makefile	2008-08-24 21:40:28.000000000 +0000
@@ -13,7 +13,7 @@
 RUN_TEST_PGM = ./$(TEST_PGM)
 SLANGINC = ..
 SLANGLIB = ../$(ARCH)objs
-OTHERLIBS = -lm
+OTHERLIBS = $(M_LIB)
 
 run_tests: $(TEST_PGM)
 	@echo ""
diff -urN slang-2.1.3/src/util/Makefile slang-2.1.3-haiku/src/util/Makefile
--- slang-2.1.3/src/util/Makefile	2006-05-12 19:09:10.000000000 +0000
+++ slang-2.1.3-haiku/src/util/Makefile	2008-08-24 21:40:28.000000000 +0000
@@ -1,5 +1,5 @@
 INCS = -I..
-LIBS = -L../$(ARCH)objs -lslang -lm
+LIBS = -L../$(ARCH)objs -lslang $(M_LIB)
 
 bcdump: bcdump.c ../$(ARCH)objs/libslang.a
 	$(CC) $(CFLAGS) $(INCS) bcdump.c -o bcdump $(LIBS)
