diff -urN slang-2.1.3/src/sldisply.c slang-2.1.3-haiku/src/sldisply.c
--- slang-2.1.3/src/sldisply.c	2007-05-16 02:47:04.000000000 +0000
+++ slang-2.1.3-haiku/src/sldisply.c	2008-08-24 04:35:29.000000000 +0000
@@ -26,13 +26,13 @@
 
 #if !defined(VMS) || (__VMS_VER >= 70000000)
 # include <sys/time.h>
-# ifdef __QNX__
+# if defined __QNX__ || defined __HAIKU__
 #  include <sys/select.h>
 # endif
 # include <sys/types.h>
 #endif
 
-#ifdef __BEOS__
+#if defined __BEOS__ && !defined __HAIKU__
 /* Prototype for select */
 # include <net/socket.h>
 #endif
diff -urN slang-2.1.3/src/slutty.c slang-2.1.3-haiku/src/slutty.c
--- slang-2.1.3/src/slutty.c	2007-04-23 06:36:37.000000000 +0000
+++ slang-2.1.3-haiku/src/slutty.c	2008-08-24 04:37:32.000000000 +0000
@@ -43,7 +43,7 @@
 # endif
 #endif
 
-#ifdef __BEOS__
+#if defined  __BEOS__ && !defined __HAIKU__
 /* Prototype for select */
 # include <net/socket.h>
 #endif
@@ -54,7 +54,7 @@
 # include <sys/ioctl.h>
 #endif
 
-#ifdef __QNX__
+#if defined __QNX__ || defined __HAIKU__
 # include <sys/select.h>
 #endif
 
