diff -Naur faac-1.28/configure.in faac-1.28-haiku/configure.in
--- faac-1.28/configure.in	2009-02-04 18:55:38.033816576 -0600
+++ faac-1.28-haiku/configure.in	2013-11-26 17:01:55.148373504 -0600
@@ -3,7 +3,7 @@
 AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_ARG_WITH( mp4v2,  [  --with-mp4v2            compile libmp4v2],WITHMP4V2=$withval, WITHMP4V2=yes)
 AC_ARG_ENABLE( drm,  [  --enable-drm            Digital Radio Mondiale support], enable_drm=$enableval, enable_drm=no)
@@ -31,7 +31,7 @@
 
 AC_CHECK_DECLS([MP4Create, MP4MetadataDelete],
                AC_CHECK_LIB(mp4v2, MP4MetadataDelete, external_mp4v2=yes,
-                            external_mp4v2=no, -lstdc++),
+                            external_mp4v2=no),
                external_mp4v2=no, [#include <mp4.h>])
 
 if test x$external_mp4v2 = xyes; then
diff -Naur faac-1.28/frontend/Makefile.am faac-1.28-haiku/frontend/Makefile.am
--- faac-1.28/frontend/Makefile.am	2008-12-15 18:56:00.034603008 -0600
+++ faac-1.28-haiku/frontend/Makefile.am	2013-11-26 17:06:24.569114624 -0600
@@ -5,7 +5,7 @@
 
 if WITH_MP4V2
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/common/mp4v2
-LDADD = $(top_builddir)/libfaac/libfaac.la $(top_srcdir)/common/mp4v2/libmp4v2.a -lm -lstdc++
+LDADD = $(top_builddir)/libfaac/libfaac.la $(top_srcdir)/common/mp4v2/libmp4v2.a -lm
 else
 INCLUDES = -I$(top_srcdir)/include 
 LDADD = $(top_builddir)/libfaac/libfaac.la -lm
