--- speex_types.h-old	2006-05-25 11:24:43.000000000 +0000
+++ speex_types.h	2008-05-15 01:19:23.000000000 +0000
@@ -64,7 +64,7 @@
    typedef int32_t spx_int32_t;
    typedef u_int32_t spx_uint32_t;
 
-#elif defined(__BEOS__)
+#elif defined(__BEOS__) && !defined(__HAIKU__)
 
    /* Be */
 #  include <inttypes.h>
@@ -73,6 +73,15 @@
    typedef int32_t spx_int32_t;
    typedef u_int32_t spx_uint32_t;
 
+#elif defined(__HAIKU__)
+
+   /* Haiku */
+#  include <sys/types.h>
+   typedef short spx_int16_t;
+   typedef unsigned short spx_uint16_t;
+   typedef int spx_int32_t;
+   typedef unsigned int spx_uint32_t;
+
 #elif defined (__EMX__)
 
    /* OS/2 GCC */
