Index: Vision/Makefile
===================================================================
--- Vision/Makefile	(revision 908)
+++ Vision/Makefile	(working copy)
@@ -21,7 +21,7 @@
 BUILD_TYPE := Release
 
 # system libs
-REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot
+REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot -lstdc++
 
 # lua
 LUA_LIBS := -L./src/lua/lib -llua -llualib
@@ -71,7 +71,7 @@
 CFLAGS += -DBUILD_DATE=\"$(BUILDDATE)\"
 CFLAGS += -fno-pic
 CFLAGS += -ffast-math
-CFLAGS += -march=pentium -mcpu=pentiumpro
+CFLAGS += -march=pentium
 CFLAGS += -Wall -W -Wno-multichar -Wpointer-arith
 CFLAGS += -Wwrite-strings -Woverloaded-virtual
 CFLAGS += -Wconversion -Wpointer-arith
