From 596b889024d5022d976898a3a97ea72ff58a68e3 Mon Sep 17 00:00:00 2001
From: Rene Gollent <rene@gollent.com>
Date: Mon, 13 Apr 2015 22:06:12 -0400
Subject: applying patch vision-0.9.7-gcc4.patch


diff --git a/Makefile b/Makefile
index 8e6a39a..e5efefb 100644
--- a/Makefile
+++ b/Makefile
@@ -21,7 +21,7 @@ BUILDDATE := $(shell date +%b_%d_%Y)
 BUILD_TYPE := Release
 
 # system libs
-REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot
+REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot -lstdc++ -lsupc++
 
 # lua
 LUA_LIBS := -L./src/lua/lib -llua -llualib
@@ -82,7 +82,7 @@ CFLAGS += $(VISION_ADDL_BUILD_FLAGS)
 
 LD := gcc
 
-LDFLAGS := $(REG_LIBS) -nodefaultlibs
+LDFLAGS := $(REG_LIBS)
 
 CFLAGS += -DUSE_INFOPOPPER=1
 
-- 
1.8.3.4

