Index: Makefile
===================================================================
--- Vision/Makefile	(revision 946)
+++ Vision/Makefile	(working copy)
@@ -21,7 +21,7 @@
 BUILD_TYPE := Release

 # system libs
-REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot
+REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot -lstdc++

 # lua
 LUA_LIBS := -L./src/lua/lib -llua -llualib
@@ -84,10 +84,7 @@

 LDFLAGS := $(REG_LIBS) -nodefaultlibs

-ifeq ($(wildcard /boot/common/include/infopopper/InfoPopper.h), )
-else
-  CFLAGS += -DUSE_INFOPOPPER=1
-endif
+CFLAGS += -DUSE_INFOPOPPER=1

 DEP := gcc -MM -DBEOS_BUILD

