Index: Vision/Makefile
===================================================================
--- Vision/Makefile	(revision 945)
+++ Vision/Makefile	(working copy)
@@ -21,7 +21,7 @@
 BUILD_TYPE := Release

 # system libs
-REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot
+REG_LIBS := -lbe -ltranslation -ltextencoding -ltracker -lnetwork -lroot -lstdc++ -lsupc++

 # lua
 LUA_LIBS := -L./src/lua/lib -llua -llualib
