diff -urN xvidcore/build/generic/configure.in xvidcore-haiku/build/generic/configure.in
--- xvidcore/build/generic/configure.in	2009-02-19 01:51:37.000000000 +0000
+++ xvidcore-haiku/build/generic/configure.in	2009-02-19 01:56:16.000000000 +0000
@@ -221,7 +221,7 @@
 STATIC_EXTENSION=""
 OBJECT_EXTENSION=""
 case "$target_os" in
-     *bsd*|linux*|beos|irix*|solaris*)
+     *bsd*|linux*|beos|*haiku*|irix*|solaris*)
 	AC_MSG_RESULT([.so .a .o])
 	STATIC_EXTENSION="a"
 	SHARED_EXTENSION="so"
@@ -299,7 +299,7 @@
 	   SPECIFIC_LDFLAGS="-r -keep_private_externs -nostdlib && \$(CC) \$(LDFLAGS) \$(PRE_SHARED_LIB) -o libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR) -bundle -flat_namespace -undefined suppress"
 	fi
 	;;
-     beos)
+     beos | *haiku*)
 	AC_MSG_RESULT([ok])
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
 	SHARED_LIB="libxvidcore.\$(SHARED_EXTENSION)"
@@ -379,7 +379,7 @@
    if test "$found_nasm_comp_prog" = "yes" ; then
        AC_MSG_CHECKING([for asm object format])
        case "$target_os" in
-           *bsd*|linux*|beos|irix*|solaris*)
+           *bsd*|linux*|beos|irix*|solaris*|*haiku*)
                if test "$ARCHITECTURE" = "X86_64" ; then
                    AC_MSG_RESULT([elf64])
                    NASM_FORMAT="elf64"
