diff -urN taglib-1.5/taglib/toolkit/tbytevector.h taglib-1.5-haiku/taglib/toolkit/tbytevector.h
--- taglib-1.5/taglib/toolkit/tbytevector.h	2009-03-02 00:20:39.000000000 +0000
+++ taglib-1.5-haiku/taglib/toolkit/tbytevector.h	2009-03-02 00:04:02.000000000 +0000
@@ -30,7 +30,12 @@
 #include "taglib_export.h"
 
 #include <vector>
+#include <string>
+#ifdef __HAIKU__
+#include <iostream>
+#else
 #include <ostream>
+#endif
 
 namespace TagLib {
 
diff -urN taglib-1.5/taglib/toolkit/tstring.h taglib-1.5-haiku/taglib/toolkit/tstring.h
--- taglib-1.5/taglib/toolkit/tstring.h	2009-03-02 00:20:39.000000000 +0000
+++ taglib-1.5-haiku/taglib/toolkit/tstring.h	2009-03-02 00:03:03.000000000 +0000
@@ -31,7 +31,11 @@
 #include "tbytevector.h"
 
 #include <string>
+#ifdef __HAIKU__
+#include <iostream>
+#else
 #include <ostream>
+#endif
 
 /*!
  * \relates TagLib::String
