Index: taglib/toolkit/tstring.h
===================================================================
--- taglib/toolkit/tstring.h	(revision 989741)
+++ taglib/toolkit/tstring.h	(working copy)
@@ -31,7 +31,11 @@
 #include "tbytevector.h"
 
 #include <string>
+#ifdef __HAIKU__
+#include <iostream>
+#else
 #include <ostream>
+#endif
 
 /*!
  * \relates TagLib::String
Index: taglib/toolkit/tbytevector.h
===================================================================
--- taglib/toolkit/tbytevector.h	(revision 989741)
+++ taglib/toolkit/tbytevector.h	(working copy)
@@ -30,7 +30,11 @@
 #include "taglib_export.h"
 
 #include <vector>
+#ifdef __HAIKU__
+#include <iostream>
+#else
 #include <ostream>
+#endif
 
 namespace TagLib {
 
