diff -ur rhapsody-0.28b.org/configure rhapsody-0.28b/configure
--- rhapsody-0.28b.org/configure	2006-02-24 07:46:19.017301504 +0100
+++ rhapsody-0.28b/configure	2012-11-22 20:29:15.727973888 +0100
@@ -168,6 +168,25 @@

 	STDDOCSPATH="/usr/share/doc/rhapsody"

+elif [ "$OS" = "Haiku" ] ; then
+	CCPATH="/boot/develop/tools/gnupro/bin /boot/develop/bin /bin"
+	CURSESPATH="/boot/develop/headers/3rdparty /boot/develop/headers /boot/develop/headers/gnu"
+
+	GCCBASELIBS="-lncurses -lnetwork"
+	GCCBASELIBPATHS=""
+	GCCBASEINCPATHS=""
+	GCCBASEFLAGS="-g -Wall -Wno-unused"
+
+	CCBASELIBS=""
+	CCBASELIBPATHS=""
+	CCBASEINCPATHS=""
+	CCBASEFLAGS=""
+
+	STDDOCSPATH="`finddir B_COMMON_DOCUMENTATION_DIRECTORY`/rhapsody"
+	if [ -z "$INSTALLPATH" ] ; then
+		INSTALLPATH="`finddir B_COMMON_BIN_DIRECTORY`"
+	fi
+
 elif [ "$OS" = "HP-UX" ] ; then
 	CCPATH="/bin /usr/bin /usr/local/bin"
 	CURSESPATH="/usr/include /usr/local/include /usr/include/ncurses /usr/local/include/ncurses"
@@ -235,7 +254,7 @@
 # Set install, docs, header and lib path if not provided  #
 ###########################################################

-if [ $USER = "root" ] ; then
+if [ $USER = "root" -o "$UID" = "0" ] ; then
 	echo "Running as root."

 	if [ -z "$INSTALLPATH" ] ; then
