diff -urN freetype-2.4.9/include/freetype/config/ftoption.h freetype-2.4.9-enable-subpixel-rendering/include/freetype/config/ftoption.h
--- freetype-2.4.9/include/freetype/config/ftoption.h	2011-07-20 04:46:15.035913728 +0000
+++ freetype-2.4.9-enable-subpixel-rendering/include/freetype/config/ftoption.h	2012-09-05 16:30:50.453509120 +0000
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
