From 157ebccb093d91ac1cb46f764241b435bbea829e Mon Sep 17 00:00:00 2001
From: Chris Roberts <cpr420@gmail.com>
Date: Thu, 12 May 2022 17:55:51 -0600
Subject: [PATCH] disable translations on 32 bit

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 288e3ba..7cbc6ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -241,7 +241,7 @@ if(WIN32)
         DEPENDS ${QM_FILES}
         COMMENT "If you don't have the .po, you need to run make getSvnTranslations first then re-run cmake"
     )
-else()
+elseif(NOT HAIKU)
     add_custom_target(BuildTranslations ALL
         DEPENDS ${QM_FILES}
         COMMENT "If you don't have the .po, you need to run make getSvnTranslations first then re-run cmake"
-- 
2.30.2

