diff --git a/configure.ac b/configure.ac
index c63adcb..0d981c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -28,7 +28,7 @@ AC_INIT(libXau, 1.0.5, [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg]
 AM_INIT_AUTOMAKE([dist-bzip2])
 AM_MAINTAINER_MODE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 
 # Require xorg-macros: XORG_CWARNFLAGS, XORG_CHANGELOG, XORG_WITH_LINT
 m4_ifndef([XORG_MACROS_VERSION], [AC_FATAL([must install xorg-macros 1.2 or later before running autoconf/autogen])])
@@ -58,6 +58,9 @@ if test "x$xthreads" = "xyes" ; then
     if test "x$mtsafe" = "x" ; then
 	AC_CHECK_LIB(nsl, gethostbyname_r, [mtsafe=yes])
     fi
+    if test "x$mtsafe" = "x" ; then
+	AC_CHECK_LIB(network, gethostbyname_r, [mtsafe=yes])
+    fi
     if test "x$mtsafe" = "xyes" ; then
 	AC_DEFINE(XUSE_MTSAFE_API, 1, 
 		  [Whether libXau needs to use MT safe API's])
