From cb03f4d4b90ab2a437ac7269afb80caf7c085d2e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Revol?= <revol@free.fr>
Date: Tue, 29 Jul 2014 19:23:51 +0200
Subject: [PATCH] Add check for libnetwork (Haiku)

---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 7f914ea..6139fbc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -61,6 +61,9 @@ if test "x$xthreads" = "xyes" ; then
     if test "x$mtsafe" = "x" ; then
 	AC_CHECK_LIB(nsl, gethostbyname_r, [mtsafe=yes])
     fi
+    if test "x$mtsafe" = "x" ; then
+	AC_CHECK_LIB(network, gethostbyname_r, [mtsafe=yes])
+    fi
     if test "x$mtsafe" = "xyes" ; then
 	AC_DEFINE(XUSE_MTSAFE_API, 1,
 		  [Whether libXau needs to use MT safe API's])
-- 
1.8.3.4

