diff -Naur libXau-1.0.5/configure.ac libXau-1.0.5-haiku/configure.ac
--- libXau-1.0.5/configure.ac	2009-08-28 04:34:23.023855104 +0000
+++ libXau-1.0.5-haiku/configure.ac	2010-04-26 23:21:53.573308928 +0000
@@ -58,6 +58,9 @@
     if test "x$mtsafe" = "x" ; then
 	AC_CHECK_LIB(nsl, gethostbyname_r, [mtsafe=yes])
     fi
+    if test "x$mtsafe" = "x" ; then
+	AC_CHECK_LIB(network, gethostbyname_r, [mtsafe=yes])
+    fi
     if test "x$mtsafe" = "xyes" ; then
 	AC_DEFINE(XUSE_MTSAFE_API, 1, 
 		  [Whether libXau needs to use MT safe API's])
