Only in xmoto-0.5.11-patch/: autom4te.cache
diff -ur xmoto-0.5.11/configure xmoto-0.5.11-patch/configure
--- xmoto-0.5.11/configure	2014-03-29 10:38:54.037748736 +0000
+++ xmoto-0.5.11-patch/configure	2014-12-11 21:25:12.893386752 +0000
@@ -7565,13 +7565,12 @@
 
 
 # m
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for floor in -lm" >&5
-$as_echo_n "checking for floor in -lm... " >&6; }
-if ${ac_cv_lib_m_floor+:} false; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing floor" >&5
+$as_echo_n "checking for library containing floor... " >&6; }
+if ${ac_cv_search_floor+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lm  $LIBS"
+  ac_func_search_save_LIBS=$LIBS
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -7590,23 +7589,35 @@
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_m_floor=yes
-else
-  ac_cv_lib_m_floor=no
+for ac_lib in '' ; do
+  if test -z "$ac_lib"; then
+    ac_res="none required"
+  else
+    ac_res=-l$ac_lib
+    LIBS="-l$ac_lib  $ac_func_search_save_LIBS"
+  fi
+  if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_search_floor=$ac_res
 fi
 rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
+    conftest$ac_exeext
+  if ${ac_cv_search_floor+:} false; then :
+  break
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_m_floor" >&5
-$as_echo "$ac_cv_lib_m_floor" >&6; }
-if test "x$ac_cv_lib_m_floor" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBM 1
-_ACEOF
+done
+if ${ac_cv_search_floor+:} false; then :
 
-  LIBS="-lm $LIBS"
+else
+  ac_cv_search_floor=no
+fi
+rm conftest.$ac_ext
+LIBS=$ac_func_search_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_search_floor" >&5
+$as_echo "$ac_cv_search_floor" >&6; }
+ac_res=$ac_cv_search_floor
+if test "$ac_res" != no; then :
+  test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
 else
   as_fn_error $? "No math library found" "$LINENO" 5
diff -ur xmoto-0.5.11/configure.in xmoto-0.5.11-patch/configure.in
--- xmoto-0.5.11/configure.in	2014-03-29 10:35:24.036700160 +0000
+++ xmoto-0.5.11-patch/configure.in	2014-12-11 21:40:34.446955520 +0000
@@ -155,7 +155,7 @@
 AM_CONDITIONAL([USE_SDLGFX], test "$USE_SDLGFX" = "1")
 
 # m
-AC_CHECK_LIB(m, floor, , AC_MSG_ERROR(No math library found))
+AC_SEARCH_LIBS(floor, , , AC_MSG_ERROR(No math library found))
 
 AC_SEARCH_LIBS(lua_pushboolean,lua lua5.1 lua50, [], [AC_MSG_ERROR(liblua5.1, liblua50, or liblua required)])
 AC_SEARCH_LIBS(luaopen_math,lualib lualib5.1 lualib50, [], [AC_MSG_ERROR(liblualib5.1, liblualib50, or liblualib required)])
diff -ur xmoto-0.5.11/src/GameInit.cpp xmoto-0.5.11-patch/src/GameInit.cpp
--- xmoto-0.5.11/src/GameInit.cpp	2011-10-11 20:18:35.013369344 +0000
+++ xmoto-0.5.11-patch/src/GameInit.cpp	2014-12-11 05:17:55.756023296 +0000
@@ -269,7 +269,7 @@
     struct sigaction v_act;
 
     v_act.sa_handler  = xmexit_term;
-#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#if !defined(__APPLE__) && !defined(__FreeBSD__) && !defined(__HAIKU__)
     v_act.sa_restorer = NULL;
 #endif
     sigemptyset(&v_act.sa_mask);
diff -ur xmoto-0.5.11/src/net/extSDL_net.cpp xmoto-0.5.11-patch/src/net/extSDL_net.cpp
--- xmoto-0.5.11/src/net/extSDL_net.cpp	2011-10-11 20:18:28.058720256 +0000
+++ xmoto-0.5.11-patch/src/net/extSDL_net.cpp	2014-12-11 02:57:51.884998144 +0000
@@ -22,7 +22,7 @@
 #include <errno.h>
 
 // read the .h to understand why i redefine SDLNet_TCP_Send
-#if !defined(WIN32) && !defined(__APPLE__)
+#if !defined(WIN32) && !defined(__APPLE__) && !defined(__HAIKU__)
 #include <sys/socket.h>
 
 #define SOCKET	int
diff -ur xmoto-0.5.11/src/VFileIO.cpp xmoto-0.5.11-patch/src/VFileIO.cpp
--- xmoto-0.5.11/src/VFileIO.cpp	2011-10-11 20:18:35.002883584 +0000
+++ xmoto-0.5.11-patch/src/VFileIO.cpp	2014-12-11 05:09:19.959709184 +0000
@@ -106,7 +106,7 @@
 #endif
 }
 
-#if !defined(WIN32) && !defined(__MORPHOS__) && !defined(__amigaos4__)
+#if !defined(WIN32) && !defined(__MORPHOS__) && !defined(__amigaos4__) && !defined(__HAIKU__)
 void strlwr(char *pc) {
   for(unsigned int i=0; i<strlen(pc); i++) pc[i] = tolower(pc[i]);
 }
