diff -Naur libelf-0.8.13/aclocal.m4 libelf-0.8.13-haiku/aclocal.m4
--- libelf-0.8.13/aclocal.m4	2008-05-23 08:17:56.050593792 +0000
+++ libelf-0.8.13-haiku/aclocal.m4	2012-08-21 14:55:07.622854144 +0000
@@ -235,7 +235,7 @@
     AC_REQUIRE([AC_PROG_CC])
     AC_PATH_PROG(LD, ld, ld)
     case "$host" in
-      *-linux*|*-gnu*)
+      *-linux*|*-gnu*|*-haiku*)
 	if test "$GCC" = yes; then
 	  mr_TARGET_ELF
 	  if test "$mr_cv_target_elf" = yes; then
@@ -248,7 +248,11 @@
 	    SONAME_SFX='.so.$(MAJOR)'
 	    LINK_SHLIB='$(CC) -shared -Wl,-soname,$(SONAME)'
 	    INSTALL_SHLIB='$(INSTALL_PROGRAM)'
-	    DEPSHLIBS='-lc'
+	      case "$host" in
+		    *-linux*|*-gnu*)
+	          DEPSHLIBS='-lc'
+	      ;;
+	      esac
 	  else
 	    AC_MSG_WARN([shared libraries not supported for $host])
 	    mr_enable_shared=no
