From 5abba7f0505e75fceabd4dd8992a7e02bb149d64 Mon Sep 17 00:00:00 2001
From: Alexander Ivash <Alexander.Ivash@StealthBits.com>
Date: Tue, 24 Dec 2019 16:04:25 +0300
Subject: [PATCH] add include

---
 qml/SGVideoNode.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qml/SGVideoNode.cpp b/qml/SGVideoNode.cpp
index b12089d8e..3f2a07d72 100644
--- a/qml/SGVideoNode.cpp
+++ b/qml/SGVideoNode.cpp
@@ -25,6 +25,7 @@
 #include "QtAV/VideoFrame.h"
 #include <QtCore/QScopedPointer>
 #include <QtGui/QOpenGLFunctions>
+#include <QtQuick/QSGMaterial>
 #include <QtQuick/QSGMaterialShader>
 
 // all in QSGRenderThread
