diff -Naur ggz-base-libs-snapshot-0.99.5/ggzcore/netxml.c ggz-base-libs-snapshot-0.99.5-haiku/ggzcore/netxml.c
--- ggz-base-libs-snapshot-0.99.5/ggzcore/netxml.c	2009-01-02 19:32:30.017825792 +0000
+++ ggz-base-libs-snapshot-0.99.5/ggzcore/netxml.c	2011-06-08 19:17:54.251396096 +0000
@@ -37,6 +37,11 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#ifdef (__HAIKU__)
+#include <sys/select.h>
+#endif
+
+
 #ifdef HAVE_WINSOCK2_H
 # include <winsock2.h>
 #endif

--- ggz-base-libs-snapshot-0.99.5/m4/ggz/platform.m4	2012-12-22 15:35:37.236191744 +0000
+++ ggz-base-libs-snapshot-0.99.5/m4/ggz/platform.m4	2012-12-22 15:26:06.672661504 +0000
@@ -251,7 +251,9 @@
 # On some systems additional libraries may be needed.  Simply include them
 # if they're there.
 AC_CHECK_LIB(socket, socket, [LIBS="$LIBS -lsocket"])
+AC_CHECK_LIB(network, socket, [LIBS="$LIBS -lnetwork"])
 AC_CHECK_LIB(nsl, gethostbyname, [LIBS="$LIBS -lnsl"])
+AC_CHECK_LIB(network, gethostbyname, [LIBS="$LIBS -lnetwork"])
 ])
 
 dnl ------------------------------------------------------------------------
