diff -urN vorbis-tools-1.2.0/configure.ac vorbis-tools-1.2.0-haiku/configure.ac
--- vorbis-tools-1.2.0/configure.ac	2009-02-21 06:39:35.000000000 +0000
+++ vorbis-tools-1.2.0-haiku/configure.ac	2009-02-21 06:28:52.000000000 +0000
@@ -155,6 +155,7 @@
 
 SOCKET_LIBS=
 AC_CHECK_LIB(socket, socket, SOCKET_LIBS="-lsocket")
+AC_CHECK_LIB(network, socket, SOCKET_LIBS="-lnetwork")
 AC_CHECK_LIB(nsl, gethostbyname, SOCKET_LIBS="-lnsl $SOCKET_LIBS")
 
 
