diff -urN vorbis-tools-1.2.0/configure.ac vorbis-tools-1.2.0-haiku/configure.ac
--- vorbis-tools-1.2.0/configure.ac	2008-03-03 05:37:31.000000000 +0000
+++ vorbis-tools-1.2.0-haiku/configure.ac	2009-09-15 06:49:02.000000000 +0000
@@ -155,6 +155,7 @@
 
 SOCKET_LIBS=
 AC_CHECK_LIB(socket, socket, SOCKET_LIBS="-lsocket")
+AC_CHECK_LIB(network, socket, SOCKET_LIBS="-lnetwork")
 AC_CHECK_LIB(nsl, gethostbyname, SOCKET_LIBS="-lnsl $SOCKET_LIBS")
 
 
diff -urN vorbis-tools-1.2.0/oggenc/oggenc.c vorbis-tools-1.2.0-haiku/oggenc/oggenc.c
--- vorbis-tools-1.2.0/oggenc/oggenc.c	2008-03-03 05:37:27.000000000 +0000
+++ vorbis-tools-1.2.0-haiku/oggenc/oggenc.c	2009-09-15 06:55:48.000000000 +0000
@@ -262,12 +262,12 @@
             {
                 /* Create a filename from existing filename, replacing extension with .ogg or .oga */
                 char *start, *end;
-
+                char *extension = (opt.with_skeleton) ? ".oga" : ".ogg";
                 start = infiles[i];
+
                 end = strrchr(infiles[i], '.');
                 end = end?end:(start + strlen(infiles[i])+1);
 
-                char *extension = (opt.with_skeleton) ? ".oga" : ".ogg";
                 out_fn = malloc(end - start + 5);
                 strncpy(out_fn, start, end-start);
                 out_fn[end-start] = 0;
