diff -Naur allegro-4.4.1.1/src/beos/bdwindow.cpp allegro-4.4.1.1-haiku/src/beos/bdwindow.cpp
--- allegro-4.4.1.1/src/beos/bdwindow.cpp	2008-10-21 01:03:43.042467328 +0000
+++ allegro-4.4.1.1-haiku/src/beos/bdwindow.cpp	2010-09-20 02:39:19.000000000 +0000
@@ -413,9 +413,9 @@
    }
 
 #ifdef ALLEGRO_NO_ASM
-   bmp->read_bank = _be_gfx_bwindow_read_write_bank;
-   bmp->write_bank = _be_gfx_bwindow_read_write_bank;
-   _screen_vtable.unwrite_bank = _be_gfx_bwindow_unwrite_bank;
+    bmp->read_bank = (void *)_be_gfx_bwindow_read_write_bank;
+    bmp->write_bank = (void *)_be_gfx_bwindow_read_write_bank;
+    _screen_vtable.unwrite_bank = (void *)_be_gfx_bwindow_unwrite_bank;
 #else
    bmp->read_bank = _be_gfx_bwindow_read_write_bank_asm;
    bmp->write_bank = _be_gfx_bwindow_read_write_bank_asm;
diff -Naur allegro-4.4.1.1/src/beos/bwindow.cpp allegro-4.4.1.1-haiku/src/beos/bwindow.cpp
--- allegro-4.4.1.1/src/beos/bwindow.cpp	2008-10-20 08:27:27.042467328 +0000
+++ allegro-4.4.1.1-haiku/src/beos/bwindow.cpp	2010-09-20 02:40:37.000000000 +0000
@@ -347,9 +347,9 @@
    }
    
 #ifdef ALLEGRO_NO_ASM
-   bmp->read_bank = _be_gfx_bwindow_read_write_bank;
-   bmp->write_bank = _be_gfx_bwindow_read_write_bank;
-   _screen_vtable.unwrite_bank = _be_gfx_bwindow_unwrite_bank;
+   bmp->read_bank = (void *)_be_gfx_bwindow_read_write_bank;
+   bmp->write_bank = (void *)_be_gfx_bwindow_read_write_bank;
+   _screen_vtable.unwrite_bank = (void *)_be_gfx_bwindow_unwrite_bank;
 #else
    bmp->read_bank = _be_gfx_bwindow_read_write_bank_asm;
    bmp->write_bank = _be_gfx_bwindow_read_write_bank_asm;
diff -Naur allegro-4.4.1.1/src/beos/bwscreen.cpp allegro-4.4.1.1-haiku/src/beos/bwscreen.cpp
--- allegro-4.4.1.1/src/beos/bwscreen.cpp	2008-10-20 08:27:27.042729472 +0000
+++ allegro-4.4.1.1-haiku/src/beos/bwscreen.cpp	2010-09-20 02:40:07.000000000 +0000
@@ -363,9 +363,9 @@
 
 #ifdef ALLEGRO_NO_ASM
    if (gfx_capabilities) {
-      bmp->write_bank = be_gfx_bwindowscreen_read_write_bank;
-      bmp->read_bank  = be_gfx_bwindowscreen_read_write_bank;
-      _screen_vtable.unwrite_bank = be_gfx_bwindowscreen_unwrite_bank;
+      bmp->write_bank = (void *)be_gfx_bwindowscreen_read_write_bank;
+      bmp->read_bank  = (void *)be_gfx_bwindowscreen_read_write_bank;
+      _screen_vtable.unwrite_bank = (void *)be_gfx_bwindowscreen_unwrite_bank;
    }
 #else
    if (gfx_capabilities) {
