diff --git a/CMakeLists.txt b/CMakeLists.txt
index 245be10..6ab68c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,7 +291,7 @@ if(APPLE)
     endif(COMPILER_GCC)
 endif(APPLE)
 
-if(BEOS)
+if(HAIKU)
     if(CMAKE_SYSTEM_NAME STREQUAL Haiku)
         set(ALLEGRO_HAIKU 1)
     else(CMAKE_SYSTEM_NAME STREQUAL Haiku)
@@ -300,9 +300,9 @@ if(BEOS)
     set(WFLAGS "-W -Wall -Wno-unused -Wno-multichar")
     set(WFLAGS_C_ONLY "")
     set(WFLAGS_CXX_ONLY "-Wno-ctor-dtor-privacy")
-endif(BEOS)
+endif(HAIKU)
 
-if(UNIX AND NOT APPLE AND NOT BEOS)
+if(UNIX AND NOT APPLE AND NOT HAIKU)
     set(ALLEGRO_UNIX 1)
 endif()
 
@@ -628,10 +628,10 @@ endif(WIN32)
 
 # -- Haiku/BeOS --
 
-if(BEOS)
+if(HAIKU)
     list(APPEND PLATFORM_SOURCES ${ALLEGRO_SRC_BEOS_FILES})
     list(APPEND PLATFORM_LIBS game midi device)
-endif(BEOS)
+endif(HAIKU)
 
 # -- Mac OS X --
 
@@ -727,9 +727,9 @@ endif(WIN32)
 if(ALLEGRO_WITH_XWINDOWS)
     add_headers("" include/xalleg.h)
 endif(ALLEGRO_WITH_XWINDOWS)
-if(BEOS)
+if(HAIKU)
     add_headers("" include/bealleg.h)
-endif(BEOS)
+endif(HAIKU)
 
 #-----------------------------------------------------------------------------#
 #
