From c1c5b8d4e142a48abb28f2c9bb81f2e170e3f564 Mon Sep 17 00:00:00 2001
From: Augustin Cavalier <waddlesplash@gmail.com>
Date: Mon, 19 Jan 2015 17:49:37 -0500
Subject: [PATCH] Make xmlif compatible with POSIX.

---
 xmlif/xmlif.l | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/xmlif/xmlif.l b/xmlif/xmlif.l
index 0a11cdf..a6efe97 100644
--- a/xmlif/xmlif.l
+++ b/xmlif/xmlif.l
@@ -46,22 +46,22 @@ static int nselections;		/* number of selections */
 static ifsense;			/* sense of last `if' or unless seen */
 static char *attribute;		/* last attribute scanned */
 
-struct stack_t {
+struct stack_type {
     int			matched;	/* matched at current level */
     int			suppressed;    /* suppressed branch? */
-    struct stack_t	*up;
+    struct stack_type	*up;
 };
-static struct stack_t head, *end = &head;
+static struct stack_type head, *end = &head;
 
 static void push_level(void)
 /* create activation record for the current level */
 {
-    struct stack_t *newelt;
+    struct stack_type *newelt;
 
 #ifdef DEBUG
     fprintf(stderr, "{push_level()}");
 #endif /* DEBUG */
-    newelt = (struct stack_t *)malloc(sizeof(struct stack_t));
+    newelt = (struct stack_type *)malloc(sizeof(struct stack_type));
     newelt->up = end;
     end = newelt;
 
@@ -72,7 +72,7 @@ static void push_level(void)
 static void pop_level(void)
 /* delete activation record for the current level */
 {
-    struct stack_t *up = end->up;
+    struct stack_type *up = end->up;
 
 #ifdef DEBUG
     fprintf(stderr, "{pop_level()}");
@@ -96,7 +96,7 @@ static void stash_attribute(char *attr)
 static void end_attribute(void)
 /* we've seen all the attributes of a conditional, process them now */
 {
-    struct stack_t	*up;
+    struct stack_type	*up;
 
     if (attribute)
 	free(attribute);
-- 
1.8.3.4

