--- strigi-0.7.8-orig/strigidaemon/CMakeLists.txt	2013-01-25 17:49:35.005767168 +0000
+++ strigi-0.7.8/strigidaemon/CMakeLists.txt	2013-02-13 00:55:35.019660800 +0000
@@ -126,6 +126,6 @@
         DESTINATION ${LIB_DESTINATION}/cmake/LibSearchClient)
 file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/dummy.cpp "")
 add_executable(strigidaemon dummy.cpp)
-target_link_libraries(strigidaemon libstrigidaemon)
+target_link_libraries(strigidaemon libstrigidaemon network)
 install(TARGETS strigidaemon
         RUNTIME DESTINATION bin)

--- strigi-0.7.8-orig/strigidaemon/bin/daemon/strigithread.cpp	2013-01-25 17:49:35.020971520 +0000
+++ strigi-0.7.8/strigidaemon/bin/daemon/strigithread.cpp	2013-02-13 00:52:47.447741952 +0000
@@ -30,7 +30,9 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/resource.h>
+#ifndef __HAIKU__
 #include <sys/syscall.h>
+#endif
 
 // define two enums and a constant for use of ioprio
 enum {
@@ -108,6 +110,7 @@
     StrigiThread* thread = static_cast<StrigiThread*>(d);
 
 #ifndef __APPLE__
+#ifndef __HAIKU__
     if (thread->getPriority() > 0) {
         // renice the thread
         int r  = setpriority(PRIO_PROCESS, 0, thread->getPriority());
@@ -139,7 +142,7 @@
 #endif
     }
 #endif
-
+#endif
     // start the actual work
     thread->run(0);
     STRIGI_LOG_DEBUG(string("strigi.daemon.") + thread->name + ".threadstarter", "end of thread");
