diff -urN xinetd-2.3.14/configure.in xinetd-2.3.14-haiku/configure.in
--- xinetd-2.3.14/configure.in	2009-03-01 10:18:52.000000000 +0000
+++ xinetd-2.3.14-haiku/configure.in	2009-03-01 10:02:23.000000000 +0000
@@ -7,9 +7,9 @@
 AC_PROG_RANLIB
 
 AC_CHECK_LIB(compat, main)
-AC_SEARCH_LIBS(gethostbyname, nsl socket resolv)
-AC_SEARCH_LIBS(getnameinfo, nsl socket resolv)
-AC_SEARCH_LIBS(inet_ntop, nsl socket resolv)
+AC_SEARCH_LIBS(gethostbyname, nsl socket resolv network)
+AC_SEARCH_LIBS(getnameinfo, nsl socket resolv network)
+AC_SEARCH_LIBS(inet_ntop, nsl socket resolv network)
 AC_CHECK_FUNCS(getnameinfo inet_ntop)
 AC_CHECK_FUNCS(difftime, ,
    AC_CHECK_LIB(m, difftime))
@@ -27,8 +27,9 @@
       AC_CHECK_LIB(nsl, socket)))
 AC_CHECK_FUNC(inet_aton, ,
 	AC_CHECK_LIB(nsl, inet_aton, ,
+           AC_CHECK_LIB(network, inet_aton, ,
 		AC_CHECK_LIB(socket, inet_aton, ,
-			AC_CHECK_LIB(resolv, inet_aton, ,[AC_DEFINE(NO_INET_ATON)]))))
+			AC_CHECK_LIB(resolv, inet_aton, ,[AC_DEFINE(NO_INET_ATON)])))))
 AC_CHECK_FUNC(setenv,[AC_DEFINE(HAVE_SETENV)])
 AC_CHECK_FUNC(strsignal, [AC_DEFINE(HAVE_STRSIGNAL)])
 AC_CHECK_LIB(c, sys_siglist, [AC_DEFINE(HAVE_SYS_SIGLIST)])
