diff -urN gperf-3.0.4/tests/test2.c gperf-3.0.4-haiku/tests/test2.c
--- gperf-3.0.4/tests/test2.c	2009-06-25 12:10:46.000000000 +0000
+++ gperf-3.0.4-haiku/tests/test2.c	2009-06-25 12:08:34.000000000 +0000
@@ -11,7 +11,7 @@
 #if !defined O_BINARY && defined _O_BINARY
 # define O_BINARY _O_BINARY
 #endif
-#ifdef __BEOS__
+#if defined __BEOS__ || defined __HAIKU__
 # undef O_BINARY
 #endif
 #if O_BINARY
