diff -Naur enet-1.2.2/configure.ac enet-1.2.2-haiku/configure.ac
--- enet-1.2.2/configure.ac	2010-05-20 13:55:27.056623104 -0300
+++ enet-1.2.2-haiku/configure.ac	2011-04-24 01:31:41.000000000 -0300
@@ -16,6 +16,8 @@
 
 AC_CHECK_MEMBER(struct msghdr.msg_flags, [AC_DEFINE(HAS_MSGHDR_FLAGS)], , [#include <sys/socket.h>])
 
+AC_CHECK_LIB(network, connect)
+
 AC_CHECK_TYPE(socklen_t, [AC_DEFINE(HAS_SOCKLEN_T)], , 
               #include <sys/types.h>
               #include <sys/socket.h>
