--- a/src/gui/kernel/qclipboard_haiku_old.cpp	2015-11-21 18:27:00.272891904 +0000
+++ b/src/gui/kernel/qclipboard_haiku.cpp	2015-11-21 18:21:37.567803904 +0000
@@ -37,7 +37,7 @@
 
 				for ( int i = 0; msg->GetInfo(B_MIME_TYPE, i, &name, &type, &count) == B_OK; i++ ) {
 					const void *data;
-					int32 dataLen = 0;
+					ssize_t dataLen = 0;
 					qDebug() << "mimeData " << name;
 					status_t stat = msg->FindData(name,B_MIME_TYPE,&data,&dataLen);
 					if(dataLen && stat==B_OK)	{
