--- libdvdread-4.2.0/configure.ac	2011-10-07 20:20:59.008912896 +0000
+++ libdvdread-4.2.0-haiku/configure.ac	2012-04-09 16:22:42.155713536 +0000
@@ -148,6 +148,8 @@
   *os2*)
     LDFLAGS="-no-undefined -Zbin-files $LDFLAGS"
     ;;
+  *haiku*)
+  	;;
   *)
     AC_CHECK_LIB(c, dlopen,
                  DYNAMIC_LD_LIBS="",
--- libdvdread-4.2.0/src/bswap.h	2008-10-09 22:04:24.005242880 +0000
+++ libdvdread-4.2.0-haiku/src/bswap.h	2012-04-09 16:24:41.829947904 +0000
@@ -73,7 +73,7 @@
  * functionality!
  */
 
-#elif defined(__FreeBSD__) || defined(__sun) || defined(__bsdi__) || defined(WIN32) || defined(__CYGWIN__) || defined(__BEOS__) || defined(__OS2__)
+#elif defined(__FreeBSD__) || defined(__sun) || defined(__bsdi__) || defined(WIN32) || defined(__CYGWIN__) || defined(__BEOS__) || defined(__OS2__)|| defined(__HAIKU__)
 #define B2N_16(x)                             \
  x = ((((x) & 0xff00) >> 8) |                 \
       (((x) & 0x00ff) << 8))
