diff --git a/configure.ac b/configure.ac
index 743fc1f..30ca197 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,7 +10,7 @@ AC_SUBST(ILMBASE_VERSION_API, ${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR})
 
 AC_CANONICAL_HOST
 AC_CONFIG_SRCDIR(ImathTest/main.cpp)
-AM_CONFIG_HEADER(config/IlmBaseConfig.h)
+AC_CONFIG_HEADER(config/IlmBaseConfig.h)
 AM_INIT_AUTOMAKE(1.6.3)  dnl Require automake 1.6.3 or better
 AM_MAINTAINER_MODE
 
