diff -Naur scons-1.0.1/engine/SCons/Platform/__init__.py scons-1.0.1-haiku-1/engine/SCons/Platform/__init__.py
--- scons-1.0.1/engine/SCons/Platform/__init__.py	2008-09-06 16:34:17.000000000 +0200
+++ scons-1.0.1-haiku-1/engine/SCons/Platform/__init__.py	2008-09-28 19:06:30.000000000 +0200
@@ -76,6 +76,8 @@
             return 'aix'
         elif string.find(sys.platform, 'darwin') != -1:
             return 'darwin'
+        elif string.find(sys.platform, 'haiku') != -1:
+            return 'haiku'
         else:
             return 'posix'
     elif os.name == 'os2':
diff -Naur scons-1.0.1/engine/SCons/Platform/haiku.py scons-1.0.1-haiku-1/engine/SCons/Platform/haiku.py
--- scons-1.0.1/engine/SCons/Platform/haiku.py	1970-01-01 01:00:00.000000000 +0100
+++ scons-1.0.1-haiku-1/engine/SCons/Platform/haiku.py	2008-10-05 13:27:12.000000000 +0200
@@ -0,0 +1,41 @@
+"""SCons.Platform.haiku
+
+Platform-specific initialization for Haiku systems.
+
+There normally shouldn't be any need to import this module directly.  It
+will usually be imported through the generic SCons.Platform.Platform()
+selection method.
+"""
+
+#
+# Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008 The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+__revision__ = ""
+
+import posix
+
+def generate(env):
+    posix.generate(env)
+    env['ENV']['PATH'] = '.:/boot/home/config/bin:/boot/common/bin:/boot/system/bin:/boot/develop/tools/gnupro/bin'
+    # help the linker find the startfiles
+    env['ENV']['BELIBRARIES'] = '/boot/develop/lib/x86'
diff -Naur scons-1.0.1/script/scons scons-1.0.1-haiku-1/script/scons
--- scons-1.0.1/script/scons	2008-09-06 16:34:17.000000000 +0200
+++ scons-1.0.1-haiku-1/script/scons	2008-10-05 13:20:42.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /bin/env python
 #
 # SCons - a Software Constructor
 #
diff -Naur scons-1.0.1/script/scons-time scons-1.0.1-haiku-1/script/scons-time
--- scons-1.0.1/script/scons-time	2008-09-06 16:34:17.000000000 +0200
+++ scons-1.0.1-haiku-1/script/scons-time	2008-10-05 13:20:49.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/bin/env python
 #
 # scons-time - run SCons timings and collect statistics
 #
diff -Naur scons-1.0.1/script/sconsign scons-1.0.1-haiku-1/script/sconsign
--- scons-1.0.1/script/sconsign	2008-09-06 16:34:17.000000000 +0200
+++ scons-1.0.1-haiku-1/script/sconsign	2008-10-05 13:20:59.000000000 +0200
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /bin/env python
 #
 # SCons - a Software Constructor
 #
