diff -Naur qt4pas-V2.5_Qt4.5.3/Qt4Pas.pro qt4pas-V2.5_Qt4.5.3-haiku/Qt4Pas.pro
--- qt4pas-V2.5_Qt4.5.3/Qt4Pas.pro	2011-12-24 23:53:12.013107200 +0000
+++ qt4pas-V2.5_Qt4.5.3-haiku/Qt4Pas.pro	2017-09-13 09:26:58.148373504 +0000
@@ -31,6 +31,11 @@
   }   
 }  
 
+haiku: {
+     message("HAIKU: Added stack alignment options to CXXFLAGS in support of SSE on arch:" $$QMAKE_HOST.arch)
+     QMAKE_CXXFLAGS += -mstackrealign -mincoming-stack-boundary=2
+}
+
 !equals(QT_MAJOR_VERSION,4)|!equals(QT_MINOR_VERSION,5) {
   message("Note: This binding version was generated for Qt 4.5.3. Current Qt is" $$QT_VERSION)
   message("Qt documents binary compatibility in each Version Change Note: http://qt.nokia.com/developer/changes")
