--- karlb-bumprace-034aca498e89/src/bumprace.c.org	2016-04-30 10:18:57.736100352 +0200
+++ karlb-bumprace-034aca498e89/src/bumprace.c	2016-04-30 10:18:42.094371840 +0200
@@ -53,7 +53,7 @@
     //the players
        player user[MAX_PLAYER_NUM]; int pl=0,playernum=2;
     //general setting
-       int final=1,fullscreen=1,pageflip=0,sound=1,precision=10,fps=0,particle=1,mode,dofadeout=1,levelset=0;
+       int final=1,fullscreen=0,pageflip=0,sound=1,precision=10,fps=0,particle=1,mode,dofadeout=1,levelset=0;
        const int NUMBER_OF_LEVELS[2]={21,9};
     //level settings
        float gravity=0,turbo=5,laser_switch;
