diff -ur cJSON-1.7.15/library_config/libcjson_utils.pc.in cJSON-1.7.15-pc/library_config/libcjson_utils.pc.in
--- cJSON-1.7.15/library_config/libcjson_utils.pc.in	2021-08-25 11:15:09.062390272 +0000
+++ cJSON-1.7.15-pc/library_config/libcjson_utils.pc.in	2023-01-16 21:01:48.867958784 +0000
@@ -1,3 +1,4 @@
+prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
diff -ur cJSON-1.7.15/library_config/libcjson.pc.in cJSON-1.7.15-pc/library_config/libcjson.pc.in
--- cJSON-1.7.15/library_config/libcjson.pc.in	2021-08-25 11:15:09.062390272 +0000
+++ cJSON-1.7.15-pc/library_config/libcjson.pc.in	2023-01-16 21:01:50.018612224 +0000
@@ -1,3 +1,4 @@
+prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
