diff -urN libdvdread-4.1.3/configure.ac libdvdread-4.1.3-haiku/configure.ac
--- libdvdread-4.1.3/configure.ac	2009-02-19 08:11:30.000000000 +0000
+++ libdvdread-4.1.3-haiku/configure.ac	2009-02-19 08:03:55.000000000 +0000
@@ -145,6 +145,8 @@
   *cygwin*)
     LDFLAGS="-no-undefined $LDFLAGS"
     ;;
+  *haiku*)
+    ;;
   *)
     AC_CHECK_LIB(c, dlopen,
 	         DYNAMIC_LD_LIBS="",
diff -urN libdvdread-4.1.3/src/bswap.h libdvdread-4.1.3-haiku/src/bswap.h
--- libdvdread-4.1.3/src/bswap.h	2009-02-19 08:11:31.000000000 +0000
+++ libdvdread-4.1.3-haiku/src/bswap.h	2009-02-19 08:09:28.000000000 +0000
@@ -71,7 +71,7 @@
  * functionality!
  */
 
-#elif defined(__FreeBSD__) || defined(__sun) || defined(__bsdi__) || defined(WIN32) || defined(__CYGWIN__) || defined(__BEOS__)
+#elif defined(__FreeBSD__) || defined(__sun) || defined(__bsdi__) || defined(WIN32) || defined(__CYGWIN__) || defined(__BEOS__) || defined(__HAIKU__)
 #define B2N_16(x) \
  x = ((((x) & 0xff00) >> 8) | \
       (((x) & 0x00ff) << 8))
